// @firestone-hs/simulate-bgs-battle/dist/cards/card.interface.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.hasPlayedBloodGemsOnAny = exports.hasPlayedBloodGemsOnMe = exports.hasOnSpawnFail = exports.hasOnAfterMagnetize = exports.hasOnAfterMagnetizeSelf = exports.hasOnAfterMagnetizeToSelf = exports.hasOnBeforeMagnetizeSelf = exports.hasOnBeforeMagnetize = exports.hasOnMinionKilled = exports.hasOnAfterDeath = exports.hasOnDeath = exports.hasAfterDealDamage = exports.hasOnDamaged = exports.hasAfterHeroDamaged = exports.hasOnStatsChanged = exports.hasOnWindfuryUpdated = exports.hasOnVenomousUpdated = exports.hasOnStealthUpdated = exports.hasOnRebornUpdated = exports.hasOnTauntUpdated = exports.hasOnDivineShieldUpdated = exports.hasEndOfTurn = exports.hasOnCardAddedToHand = exports.hasDeathrattleSpawnEnchantment = exports.hasOnDeathrattleTriggered = exports.hasDefaultScriptDataNum = exports.hasAvenge = exports.hasOnBattlecryTriggered = exports.hasRebornSelfEffect = exports.hasRebornEffect = exports.hasBattlecry = exports.hasDeathrattleSpawn = exports.hasOnDespawned = exports.hasAfterOtherSpawned = exports.hasOnOtherSpawned = exports.hasOnOtherAuraSpawned = exports.hasOnSpawned = exports.hasOnMinionAttacked = exports.hasOnWheneverAnotherMinionAttacks = exports.hasRally = exports.hasAfterTavernSpellCast = exports.hasOnTavernSpellCast = exports.hasCastTavernSpell = exports.hasStartOfCombatFromHand = exports.hasStartOfCombat = exports.hasDefaultCharges = void 0;
const hasDefaultCharges = (card) => (card === null || card === void 0 ? void 0 : card.defaultCharges) !== undefined;
exports.hasDefaultCharges = hasDefaultCharges;
const hasStartOfCombat = (card) => (card === null || card === void 0 ? void 0 : card.startOfCombat) !== undefined;
exports.hasStartOfCombat = hasStartOfCombat;
const hasStartOfCombatFromHand = (card) => (card === null || card === void 0 ? void 0 : card.startOfCombatFromHand) !== undefined;
exports.hasStartOfCombatFromHand = hasStartOfCombatFromHand;
const hasCastTavernSpell = (card) => (card === null || card === void 0 ? void 0 : card.castTavernSpell) !== undefined;
exports.hasCastTavernSpell = hasCastTavernSpell;
const hasOnTavernSpellCast = (card) => (card === null || card === void 0 ? void 0 : card.onTavernSpellCast) !== undefined;
exports.hasOnTavernSpellCast = hasOnTavernSpellCast;
const hasAfterTavernSpellCast = (card) => (card === null || card === void 0 ? void 0 : card.afterTavernSpellCast) !== undefined;
exports.hasAfterTavernSpellCast = hasAfterTavernSpellCast;
const hasRally = (card) => (card === null || card === void 0 ? void 0 : card.rally) !== undefined;
exports.hasRally = hasRally;
const hasOnWheneverAnotherMinionAttacks = (card) => (card === null || card === void 0 ? void 0 : card.onWheneverAnotherMinionAttacks) !== undefined;
exports.hasOnWheneverAnotherMinionAttacks = hasOnWheneverAnotherMinionAttacks;
const hasOnMinionAttacked = (card) => (card === null || card === void 0 ? void 0 : card.onAttacked) !== undefined;
exports.hasOnMinionAttacked = hasOnMinionAttacked;
const hasOnSpawned = (card) => (card === null || card === void 0 ? void 0 : card.onSpawned) !== undefined;
exports.hasOnSpawned = hasOnSpawned;
const hasOnOtherAuraSpawned = (card) => (card === null || card === void 0 ? void 0 : card.onOtherSpawnedAura) !== undefined;
exports.hasOnOtherAuraSpawned = hasOnOtherAuraSpawned;
const hasOnOtherSpawned = (card) => (card === null || card === void 0 ? void 0 : card.onOtherSpawned) !== undefined;
exports.hasOnOtherSpawned = hasOnOtherSpawned;
const hasAfterOtherSpawned = (card) => (card === null || card === void 0 ? void 0 : card.afterOtherSpawned) !== undefined;
exports.hasAfterOtherSpawned = hasAfterOtherSpawned;
const hasOnDespawned = (card) => (card === null || card === void 0 ? void 0 : card.onDespawned) !== undefined;
exports.hasOnDespawned = hasOnDespawned;
const hasDeathrattleSpawn = (card) => (card === null || card === void 0 ? void 0 : card.deathrattleSpawn) !== undefined;
exports.hasDeathrattleSpawn = hasDeathrattleSpawn;
const hasBattlecry = (card) => (card === null || card === void 0 ? void 0 : card.battlecry) !== undefined;
exports.hasBattlecry = hasBattlecry;
const hasRebornEffect = (card) => (card === null || card === void 0 ? void 0 : card.rebornEffect) !== undefined;
exports.hasRebornEffect = hasRebornEffect;
const hasRebornSelfEffect = (card) => (card === null || card === void 0 ? void 0 : card.rebornSelfEffect) !== undefined;
exports.hasRebornSelfEffect = hasRebornSelfEffect;
const hasOnBattlecryTriggered = (card) => (card === null || card === void 0 ? void 0 : card.onBattlecryTriggered) !== undefined;
exports.hasOnBattlecryTriggered = hasOnBattlecryTriggered;
const hasAvenge = (card) => (card === null || card === void 0 ? void 0 : card.avenge) !== undefined;
exports.hasAvenge = hasAvenge;
const hasDefaultScriptDataNum = (card) => (card === null || card === void 0 ? void 0 : card.defaultScriptDataNum) !== undefined;
exports.hasDefaultScriptDataNum = hasDefaultScriptDataNum;
const hasOnDeathrattleTriggered = (card) => (card === null || card === void 0 ? void 0 : card.onDeathrattleTriggered) !== undefined;
exports.hasOnDeathrattleTriggered = hasOnDeathrattleTriggered;
const hasDeathrattleSpawnEnchantment = (card) => (card === null || card === void 0 ? void 0 : card.deathrattleSpawnEnchantmentEffect) !== undefined;
exports.hasDeathrattleSpawnEnchantment = hasDeathrattleSpawnEnchantment;
const hasOnCardAddedToHand = (card) => (card === null || card === void 0 ? void 0 : card.onCardAddedToHand) !== undefined;
exports.hasOnCardAddedToHand = hasOnCardAddedToHand;
const hasEndOfTurn = (card) => (card === null || card === void 0 ? void 0 : card.endOfTurn) !== undefined;
exports.hasEndOfTurn = hasEndOfTurn;
const hasOnDivineShieldUpdated = (card) => (card === null || card === void 0 ? void 0 : card.onDivineShieldUpdated) !== undefined;
exports.hasOnDivineShieldUpdated = hasOnDivineShieldUpdated;
const hasOnTauntUpdated = (card) => (card === null || card === void 0 ? void 0 : card.onTauntUpdated) !== undefined;
exports.hasOnTauntUpdated = hasOnTauntUpdated;
const hasOnRebornUpdated = (card) => (card === null || card === void 0 ? void 0 : card.onRebornUpdated) !== undefined;
exports.hasOnRebornUpdated = hasOnRebornUpdated;
const hasOnStealthUpdated = (card) => (card === null || card === void 0 ? void 0 : card.onStealthUpdated) !== undefined;
exports.hasOnStealthUpdated = hasOnStealthUpdated;
const hasOnVenomousUpdated = (card) => (card === null || card === void 0 ? void 0 : card.onVenomousUpdated) !== undefined;
exports.hasOnVenomousUpdated = hasOnVenomousUpdated;
const hasOnWindfuryUpdated = (card) => (card === null || card === void 0 ? void 0 : card.onWindfuryUpdated) !== undefined;
exports.hasOnWindfuryUpdated = hasOnWindfuryUpdated;
const hasOnStatsChanged = (card) => (card === null || card === void 0 ? void 0 : card.onStatsChanged) !== undefined;
exports.hasOnStatsChanged = hasOnStatsChanged;
const hasAfterHeroDamaged = (card) => (card === null || card === void 0 ? void 0 : card.afterHeroDamaged) !== undefined;
exports.hasAfterHeroDamaged = hasAfterHeroDamaged;
const hasOnDamaged = (card) => (card === null || card === void 0 ? void 0 : card.onDamaged) !== undefined;
exports.hasOnDamaged = hasOnDamaged;
const hasAfterDealDamage = (card) => (card === null || card === void 0 ? void 0 : card.afterDealDamage) !== undefined;
exports.hasAfterDealDamage = hasAfterDealDamage;
const hasOnDeath = (card) => (card === null || card === void 0 ? void 0 : card.onDeath) !== undefined;
exports.hasOnDeath = hasOnDeath;
const hasOnAfterDeath = (card) => (card === null || card === void 0 ? void 0 : card.onAfterDeath) !== undefined;
exports.hasOnAfterDeath = hasOnAfterDeath;
const hasOnMinionKilled = (card) => (card === null || card === void 0 ? void 0 : card.onMinionKilled) !== undefined;
exports.hasOnMinionKilled = hasOnMinionKilled;
const hasOnBeforeMagnetize = (card) => (card === null || card === void 0 ? void 0 : card.onBeforeMagnetize) !== undefined;
exports.hasOnBeforeMagnetize = hasOnBeforeMagnetize;
const hasOnBeforeMagnetizeSelf = (card) => (card === null || card === void 0 ? void 0 : card.onBeforeMagnetizeSelf) !== undefined;
exports.hasOnBeforeMagnetizeSelf = hasOnBeforeMagnetizeSelf;
const hasOnAfterMagnetizeToSelf = (card) => (card === null || card === void 0 ? void 0 : card.onAfterMagnetizeToSelf) !== undefined;
exports.hasOnAfterMagnetizeToSelf = hasOnAfterMagnetizeToSelf;
const hasOnAfterMagnetizeSelf = (card) => (card === null || card === void 0 ? void 0 : card.onAfterMagnetizeSelf) !== undefined;
exports.hasOnAfterMagnetizeSelf = hasOnAfterMagnetizeSelf;
const hasOnAfterMagnetize = (card) => (card === null || card === void 0 ? void 0 : card.onAfterMagnetize) !== undefined;
exports.hasOnAfterMagnetize = hasOnAfterMagnetize;
const hasOnSpawnFail = (card) => (card === null || card === void 0 ? void 0 : card.onSpawnFail) !== undefined;
exports.hasOnSpawnFail = hasOnSpawnFail;
const hasPlayedBloodGemsOnMe = (card) => (card === null || card === void 0 ? void 0 : card.playedBloodGemsOnMe) !== undefined;
exports.hasPlayedBloodGemsOnMe = hasPlayedBloodGemsOnMe;
const hasPlayedBloodGemsOnAny = (card) => (card === null || card === void 0 ? void 0 : card.playedBloodGemsOnAny) !== undefined;
exports.hasPlayedBloodGemsOnAny = hasPlayedBloodGemsOnAny;
