// @firestone-hs/simulate-bgs-battle/dist/simulation/damage-to-hero.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dealDamageToHero = void 0;
const card_interface_1 = require("../cards/card.interface");
const _card_mappings_1 = require("../cards/impl/_card-mappings");
const dealDamageToHero = (source, hero, board, damage, gameState) => {
    for (const entity of board) {
        const afterHeroDamagedImpl = _card_mappings_1.cardMappings[entity.cardId];
        if ((0, card_interface_1.hasAfterHeroDamaged)(afterHeroDamagedImpl)) {
            afterHeroDamagedImpl.afterHeroDamaged(entity, {
                damage: damage,
                board: board,
                hero: hero,
                gameState,
            });
        }
        const afterDealDamageImpl = _card_mappings_1.cardMappings[entity.cardId];
        if ((0, card_interface_1.hasAfterDealDamage)(afterDealDamageImpl)) {
            afterDealDamageImpl.afterDealDamage(entity, {
                damagedEntity: hero,
                damageDealer: source,
                damage: damage,
                board: board,
                hero: hero,
                gameState,
            });
        }
    }
};
exports.dealDamageToHero = dealDamageToHero;
