// @firestone-hs/simulate-bgs-battle/dist/simulation/minion-kill.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.onMinionKill = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const card_interface_1 = require("../cards/card.interface");
const _card_mappings_1 = require("../cards/impl/_card-mappings");
const utils_1 = require("../services/utils");
const stats_1 = require("./stats");
const onMinionKill = (killer, killerIsAttacking, victim, killerBoard, killerHero, victimBoard, victimHero, defenderNeighbours, gameState) => {
    if (!(killer === null || killer === void 0 ? void 0 : killer.cardId)) {
        return { dmgDoneByAttacker: 0, dmgDoneByDefender: 0 };
    }
    let damageDoneByAttacker = 0;
    let damageDoneByDefender = 0;
    for (const boardEntity of killerBoard) {
        const onMinionKilledImpl = _card_mappings_1.cardMappings[boardEntity.cardId];
        if ((0, card_interface_1.hasOnMinionKilled)(onMinionKilledImpl)) {
            const { dmgDoneByAttacker, dmgDoneByDefender } = onMinionKilledImpl.onMinionKilled(boardEntity, {
                killer: killer,
                killerIsAttacking: killerIsAttacking,
                minionKilled: victim,
                attackingHero: killerHero,
                attackingBoard: killerBoard,
                defendingHero: victimHero,
                defendingBoard: victimBoard,
                defenderNeighbours: defenderNeighbours,
                gameState,
                playerIsFriendly: killerHero.friendly,
            });
            damageDoneByAttacker += dmgDoneByAttacker;
            damageDoneByDefender += dmgDoneByDefender;
        }
    }
    switch (killer.cardId) {
        case "BG27_507":
        case "BG27_507_G":
            if (killer.health > 0 && !killer.definitelyDead && killer.abiityChargesLeft > 0) {
                (0, stats_1.modifyStats)(killer, killer, victim.attack, victim.maxHealth, killerBoard, killerHero, gameState);
                gameState.spectator.registerPowerTarget(killer, killer, killerBoard, killerHero, victimHero);
                killer.abiityChargesLeft--;
            }
            break;
        case "BG27_513":
        case "BG27_513_G":
            if (gameState.sharedState.currentAttackerEntityId != null &&
                gameState.sharedState.currentAttackerEntityId === killer.entityId) {
                const tideOracleMorgleTarget = (0, utils_1.pickRandom)(killerHero.hand
                    .filter((e) => !!(e === null || e === void 0 ? void 0 : e.cardId))
                    .filter((e) => { var _a; return ((_a = gameState.allCards.getCard(e.cardId).type) === null || _a === void 0 ? void 0 : _a.toUpperCase()) === reference_data_1.CardType[reference_data_1.CardType.MINION]; }));
                if (tideOracleMorgleTarget) {
                    const tideOracleMorgleMultiplier = killer.cardId === "BG27_513" ? 1 : 2;
                    (0, stats_1.modifyStats)(tideOracleMorgleTarget, killer, tideOracleMorgleMultiplier * victim.attack, tideOracleMorgleMultiplier * victim.maxHealth, killerBoard, killerHero, gameState);
                    gameState.spectator.registerPowerTarget(killer, tideOracleMorgleTarget, killerBoard, killerHero, victimHero);
                }
            }
    }
    return { dmgDoneByAttacker: damageDoneByAttacker, dmgDoneByDefender: damageDoneByDefender };
};
exports.onMinionKill = onMinionKill;
