// @firestone-hs/simulate-bgs-battle/dist/simulation/on-attack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.applyOnAttackEffects = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const card_interface_1 = require("../cards/card.interface");
const _card_mappings_1 = require("../cards/impl/_card-mappings");
const reborn_1 = require("../keywords/reborn");
const rally_1 = require("../mechanics/rally");
const utils_1 = require("../utils");
const stats_1 = require("./stats");
const applyOnAttackEffects = (attacker, attackingBoard, attackingBoardHero, defendingEntity, defendingBoard, defendingBoardHero, gameState) => {
    var _a;
    let damageDoneByAttacker = 0;
    let damageDoneByDefender = 0;
    for (const trinket of (_a = attackingBoardHero.trinkets) !== null && _a !== void 0 ? _a : []) {
        const onAttackImpl = _card_mappings_1.cardMappings[trinket.cardId];
        if ((0, card_interface_1.hasOnWheneverAnotherMinionAttacks)(onAttackImpl)) {
            const { dmgDoneByAttacker, dmgDoneByDefender } = onAttackImpl.onWheneverAnotherMinionAttacks(trinket, {
                attacker: attacker,
                attackingHero: attackingBoardHero,
                attackingBoard: attackingBoard,
                defendingEntity: defendingEntity,
                defendingHero: defendingBoardHero,
                defendingBoard: defendingBoard,
                gameState,
                playerIsFriendly: attackingBoardHero.friendly,
            });
            damageDoneByAttacker += dmgDoneByAttacker;
            damageDoneByDefender += dmgDoneByDefender;
        }
    }
    const attackingBoardOtherEntities = attackingBoard.filter((e) => e !== attacker);
    for (const boardEntity of attackingBoardOtherEntities) {
        const onAttackImpl = _card_mappings_1.cardMappings[boardEntity.cardId];
        if ((0, card_interface_1.hasOnWheneverAnotherMinionAttacks)(onAttackImpl)) {
            const { dmgDoneByAttacker, dmgDoneByDefender } = onAttackImpl.onWheneverAnotherMinionAttacks(boardEntity, {
                attacker: attacker,
                attackingHero: attackingBoardHero,
                attackingBoard: attackingBoard,
                defendingEntity: defendingEntity,
                defendingHero: defendingBoardHero,
                defendingBoard: defendingBoard,
                gameState,
                playerIsFriendly: attackingBoardHero.friendly,
            });
            damageDoneByAttacker += dmgDoneByAttacker;
            damageDoneByDefender += dmgDoneByDefender;
        }
        const enchantments = boardEntity.enchantments;
        for (const enchantment of enchantments) {
            const onAttackImpl = _card_mappings_1.cardMappings[enchantment.cardId];
            if ((0, card_interface_1.hasOnWheneverAnotherMinionAttacks)(onAttackImpl)) {
                const { dmgDoneByAttacker, dmgDoneByDefender } = onAttackImpl.onWheneverAnotherMinionAttacks(enchantment, {
                    attacker: attacker,
                    attackingHero: attackingBoardHero,
                    attackingBoard: attackingBoard,
                    defendingEntity: defendingEntity,
                    defendingHero: defendingBoardHero,
                    defendingBoard: defendingBoard,
                    gameState,
                    playerIsFriendly: attackingBoardHero.friendly,
                });
                damageDoneByAttacker += dmgDoneByAttacker;
                damageDoneByDefender += dmgDoneByDefender;
            }
        }
    }
    const { dmgDoneByAttacker: dmgDoneByRally, dmgDoneByDefender: dmgDoneByRallyDefender } = (0, rally_1.triggerRally)(attackingBoard, attackingBoardHero, attacker, defendingBoard, defendingBoardHero, defendingEntity, gameState);
    damageDoneByAttacker += dmgDoneByRally;
    damageDoneByDefender += dmgDoneByRallyDefender;
    if (attacker.cardId === "BG29_888" || attacker.cardId === "BG29_888_G") {
        const multiplier = attacker.cardId === "BG29_888_G" ? 2 : 1;
        (0, stats_1.modifyStats)(attacker, attacker, 2 * multiplier, 0, attackingBoard, attackingBoardHero, gameState);
    }
    else if (attacker.cardId === "BG24_708" ||
        attacker.cardId === "BG24_708_G") {
        attackingBoard
            .filter((e) => (0, utils_1.hasCorrectTribe)(e, attackingBoardHero, reference_data_1.Race.PIRATE, gameState.anomalies, gameState.allCards))
            .forEach((e) => {
            (0, stats_1.modifyStats)(e, attacker, attacker.cardId === "BG24_708_G" ? 4 : 2, attacker.cardId === "BG24_708_G" ? 4 : 2, attackingBoard, attackingBoardHero, gameState);
        });
    }
    else if (attacker.cardId === "BG29_120" || attacker.cardId === "BG29_120_G") {
        const loops = attacker.cardId === "BG29_120_G" ? 2 : 1;
        const attackerIndex = attackingBoard.indexOf(attacker);
        for (let i = 1; i <= loops; i++) {
            const target = attackingBoard[attackerIndex + i];
            if (!!target) {
                (0, reborn_1.updateReborn)(target, true, attackingBoard, attackingBoardHero, defendingBoardHero, gameState);
                gameState.spectator.registerPowerTarget(attacker, target, attackingBoard, attackingBoardHero, defendingBoardHero);
            }
        }
    }
    const eclipsion = attackingBoard.find((e) => [
        "TB_BaconShop_HERO_08_Buddy",
        "TB_BaconShop_HERO_08_Buddy_G",
    ].includes(e.cardId) && e.abiityChargesLeft > 0);
    if (!!eclipsion && attacker.immuneWhenAttackCharges == null) {
        attacker.immuneWhenAttackCharges = 1;
        attackingBoard
            .filter((e) => [
            "TB_BaconShop_HERO_08_Buddy",
            "TB_BaconShop_HERO_08_Buddy_G",
        ].includes(e.cardId))
            .forEach((e) => {
            e.abiityChargesLeft--;
        });
    }
    for (const trinket of attackingBoardHero.trinkets) {
        switch (trinket.cardId) {
            case "BG30_MagicItem_925":
                (0, stats_1.modifyStats)(attacker, trinket, 4, 0, attackingBoard, attackingBoardHero, gameState);
                break;
        }
    }
    return { damageDoneByAttacker, damageDoneByDefender };
};
exports.applyOnAttackEffects = applyOnAttackEffects;
