// @firestone-hs/simulate-bgs-battle/dist/simulation/on-being-attacked.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.applyOnBeingAttackedBuffs = void 0;
const card_interface_1 = require("../cards/card.interface");
const _card_mappings_1 = require("../cards/impl/_card-mappings");
const divine_shield_1 = require("../keywords/divine-shield");
const reborn_1 = require("../keywords/reborn");
const venomous_1 = require("../keywords/venomous");
const utils_1 = require("../utils");
const secrets_1 = require("./secrets");
const stats_1 = require("./stats");
const applyOnBeingAttackedBuffs = (attackerEntity, attackerBoard, attackerHero, defendingEntity, defendingBoard, defendingPlayerEntity, gameState) => {
    var _a;
    for (const secret of ((_a = defendingPlayerEntity.secrets) !== null && _a !== void 0 ? _a : []).filter((s) => !s.triggered)) {
        switch (secret.cardId) {
            case "TB_Bacon_Secrets_07":
            case "TB_Bacon_Secrets_07b":
                if (!defendingEntity.divineShield) {
                    secret.triggered = true;
                    (0, divine_shield_1.updateDivineShield)(defendingEntity, defendingBoard, defendingPlayerEntity, attackerHero, true, gameState);
                    if (secret.cardId === "TB_Bacon_Secrets_07b") {
                        defendingEntity.strongDivineShield = true;
                    }
                }
                break;
            case "TB_Bacon_Secrets_04":
                if (defendingBoard.length < 7) {
                    secret.triggered = true;
                    (0, secrets_1.handleSplittingImage)(defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState);
                }
                break;
            case "TB_Bacon_Secrets_15":
            case "TB_Bacon_Secrets_15b":
                if (defendingBoard.length < 7) {
                    secret.triggered = true;
                    (0, secrets_1.handlePackTactics)(defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState, secret.cardId);
                }
                break;
            case "TB_Bacon_Secrets_02":
                if (defendingBoard.length < 7) {
                    secret.triggered = true;
                    (0, secrets_1.handleSnakeTrap)(defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState);
                }
                break;
            case "TB_Bacon_Secrets_01":
            case "TB_Bacon_Secrets_01b":
                if (defendingBoard.length < 7) {
                    secret.triggered = true;
                    const spawns = (0, secrets_1.handleVenomstrikeTrap)(defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState);
                    if (secret.cardId === "TB_Bacon_Secrets_01b") {
                        for (const spawn of spawns) {
                            (0, reborn_1.updateReborn)(spawn, true, defendingBoard, defendingPlayerEntity, attackerHero, gameState);
                        }
                    }
                }
                break;
        }
    }
    for (const entity of defendingBoard) {
        const onBeingAttackedImpl = _card_mappings_1.cardMappings[entity.cardId];
        if ((0, card_interface_1.hasOnMinionAttacked)(onBeingAttackedImpl)) {
            onBeingAttackedImpl.onAttacked(entity, {
                defendingEntity: defendingEntity,
                attacker: attackerEntity,
                attackingHero: attackerHero,
                attackingBoard: attackerBoard,
                defendingHero: defendingPlayerEntity,
                defendingBoard: defendingBoard,
                gameState,
            });
        }
    }
    if (defendingEntity.taunt) {
        defendingBoard
            .filter((e) => e.cardId === "TB_BaconShop_HERO_95_Buddy" ||
            e.cardId === "TB_BaconShop_HERO_95_Buddy_G")
            .forEach((entity) => {
            const mult = entity.cardId === "TB_BaconShop_HERO_95_Buddy" ? 1 : 2;
            (0, utils_1.addStatsToBoard)(entity, defendingBoard, defendingPlayerEntity, 1 * mult, 1 * mult, gameState);
        });
    }
    if (defendingEntity.cardId === "BG24_300" ||
        defendingEntity.cardId === "BG24_300_G") {
        (0, stats_1.modifyStats)(defendingEntity, defendingEntity, defendingEntity.cardId === "BG24_300_G" ? 2 : 1, 0, defendingBoard, defendingPlayerEntity, gameState);
    }
    else if (["BG27_022", "BG27_022_G"].includes(defendingEntity.cardId)) {
        const totalStats = defendingEntity.attack + defendingEntity.health;
        const attackerAttack = attackerEntity.attack;
        if (defendingEntity.divineShield) {
            defendingEntity.health = 1;
            defendingEntity.attack = totalStats - defendingEntity.health;
        }
        else if (attackerEntity.venomous || attackerEntity.poisonous) {
        }
        else if (attackerAttack < totalStats) {
            defendingEntity.health = attackerAttack + 1;
            defendingEntity.attack = totalStats - defendingEntity.health;
        }
    }
    else if (["BG28_406", "BG28_406_G"].includes(defendingEntity.cardId)) {
        (0, venomous_1.updateVenomous)(defendingEntity, true, defendingBoard, defendingPlayerEntity, attackerHero, gameState);
    }
};
exports.applyOnBeingAttackedBuffs = applyOnBeingAttackedBuffs;
