// @firestone-hs/simulate-bgs-battle/dist/simulation/quest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.onQuestProgressUpdated = void 0;
const onQuestProgressUpdated = (playerEntity, quest, board, gameState) => {
    quest.ProgressCurrent++;
    if (quest.ProgressCurrent < quest.ProgressTotal) {
        return;
    }
    const rewardCardId = gameState.allCards.getCard(quest.RewardDbfId).id;
    playerEntity.questRewards.push(rewardCardId);
    playerEntity.questRewardEntities.push({
        cardId: rewardCardId,
        entityId: gameState.sharedState.currentEntityId++,
        avengeDefault: gameState.cardsData.avengeValue(rewardCardId),
        avengeCurrent: gameState.cardsData.avengeValue(rewardCardId),
        scriptDataNum1: gameState.cardsData.defaultScriptDataNum(rewardCardId),
    });
    playerEntity.questEntities = playerEntity.questEntities.filter((q) => q.CardId !== quest.CardId);
    updateStateAfterQuestCreated(rewardCardId, board, playerEntity, gameState);
};
exports.onQuestProgressUpdated = onQuestProgressUpdated;
const updateStateAfterQuestCreated = (rewardCardId, board, playerEntity, gameState) => {
    switch (rewardCardId) {
        case "BG24_Reward_125":
            board.forEach((e) => {
                e.attack += 4;
            });
            break;
        default:
            break;
    }
};
