// @firestone-hs/simulate-bgs-battle/dist/simulation/secrets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleVenomstrikeTrap = exports.handleSnakeTrap = exports.handlePackTactics = exports.handleSplittingImage = void 0;
const utils_1 = require("../utils");
const add_minion_to_board_1 = require("./add-minion-to-board");
const deathrattle_spawns_1 = require("./deathrattle-spawns");
const spawns_1 = require("./spawns");
const handleSplittingImage = (defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState) => {
    const copy = {
        ...defendingEntity,
        attack: 3,
        health: 3,
        maxHealth: 3,
    };
    const candidateEntities = (0, deathrattle_spawns_1.spawnEntities)(defendingEntity.cardId, 1, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState, defendingEntity.friendly, false, false, true, copy);
    const indexFromRight = defendingBoard.length - (defendingBoard.indexOf(defendingEntity) + 1);
    (0, spawns_1.performEntitySpawns)(candidateEntities, defendingBoard, defendingPlayerEntity, defendingEntity, indexFromRight, attackerBoard, attackerHero, gameState);
};
exports.handleSplittingImage = handleSplittingImage;
const handlePackTactics = (defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState, secretCardId) => {
    const copy = (0, utils_1.copyEntity)(defendingEntity);
    (0, add_minion_to_board_1.removeAurasFromSelf)(copy, defendingBoard, defendingPlayerEntity, gameState);
    const candidateEntities = (0, deathrattle_spawns_1.spawnEntities)(copy.cardId, 1, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState, defendingEntity.friendly, false, false, true, copy);
    if (secretCardId === "TB_Bacon_Secrets_15") {
        candidateEntities.forEach((e) => {
            e.health = 3;
            e.maxHealth = 3;
            e.attack = 3;
            e.maxAttack = 3;
        });
    }
    const indexFromRight = defendingBoard.length - (defendingBoard.indexOf(defendingEntity) + 1);
    const spawned = (0, spawns_1.performEntitySpawns)(candidateEntities, defendingBoard, defendingPlayerEntity, defendingEntity, indexFromRight, attackerBoard, attackerHero, gameState);
    spawned.forEach((e) => {
        e.hasAttacked = defendingEntity.hasAttacked;
    });
};
exports.handlePackTactics = handlePackTactics;
const handleSnakeTrap = (defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState) => {
    const candidateEntities = (0, deathrattle_spawns_1.spawnEntities)("EX1_554t", 3, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState, defendingEntity.friendly, false);
    (0, spawns_1.performEntitySpawns)(candidateEntities, defendingBoard, defendingPlayerEntity, defendingEntity, 0, attackerBoard, attackerHero, gameState);
};
exports.handleSnakeTrap = handleSnakeTrap;
const handleVenomstrikeTrap = (defendingEntity, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState) => {
    const candidateEntities = (0, deathrattle_spawns_1.spawnEntities)("BG_EX1_170", 1, defendingBoard, defendingPlayerEntity, attackerBoard, attackerHero, gameState, defendingEntity.friendly, false);
    const spawns = (0, spawns_1.performEntitySpawns)(candidateEntities, defendingBoard, defendingPlayerEntity, defendingEntity, 0, attackerBoard, attackerHero, gameState);
    return spawns;
};
exports.handleVenomstrikeTrap = handleVenomstrikeTrap;
