// @firestone-hs/simulate-bgs-battle/dist/debug-state.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.debugState = void 0;
exports.debugState = {
    active: false,
    forcedCurrentAttacker: null,
    forcedFaceOff: [],
    forcedFaceOffBase: [],
    forcedRandomPicks: [],
    isCorrectEntity: (proposedEntity, actualEntity) => {
        if (proposedEntity.entityId) {
            return proposedEntity.entityId === actualEntity.entityId;
        }
        if (proposedEntity.cardId) {
            const cardMatches = proposedEntity.cardId === actualEntity.cardId;
            if (proposedEntity.attack != null && proposedEntity.health != null) {
                return cardMatches && proposedEntity.attack === actualEntity.attack && proposedEntity.health === actualEntity.health;
            }
            return cardMatches;
        }
        return (proposedEntity.attack === actualEntity.attack &&
            proposedEntity.health === actualEntity.health);
    },
    forcedRandomPicksBase: [],
    onBattleStart: () => {
        exports.debugState.forcedFaceOff = [...exports.debugState.forcedFaceOffBase];
        exports.debugState.forcedRandomPicks = [...exports.debugState.forcedRandomPicksBase];
    },
};
