// @firestone-hs/reference-data/dist/enums/arena-client-state-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArenaClientStateType = void 0;
var ArenaClientStateType;
(function (ArenaClientStateType) {
    ArenaClientStateType[ArenaClientStateType["None"] = 0] = "None";
    ArenaClientStateType[ArenaClientStateType["Normal_Landing"] = 1] = "Normal_Landing";
    ArenaClientStateType[ArenaClientStateType["Normal_Draft"] = 2] = "Normal_Draft";
    ArenaClientStateType[ArenaClientStateType["Normal_Redraft"] = 3] = "Normal_Redraft";
    ArenaClientStateType[ArenaClientStateType["Normal_Ready"] = 4] = "Normal_Ready";
    ArenaClientStateType[ArenaClientStateType["Normal_DeckEdit"] = 5] = "Normal_DeckEdit";
    ArenaClientStateType[ArenaClientStateType["Normal_DeckCollection"] = 6] = "Normal_DeckCollection";
    ArenaClientStateType[ArenaClientStateType["Normal_Rewards"] = 7] = "Normal_Rewards";
    ArenaClientStateType[ArenaClientStateType["Underground_Landing"] = 8] = "Underground_Landing";
    ArenaClientStateType[ArenaClientStateType["Underground_Draft"] = 9] = "Underground_Draft";
    ArenaClientStateType[ArenaClientStateType["Underground_Redraft"] = 10] = "Underground_Redraft";
    ArenaClientStateType[ArenaClientStateType["Underground_Ready"] = 11] = "Underground_Ready";
    ArenaClientStateType[ArenaClientStateType["Underground_DeckEdit"] = 12] = "Underground_DeckEdit";
    ArenaClientStateType[ArenaClientStateType["Underground_DeckCollection"] = 13] = "Underground_DeckCollection";
    ArenaClientStateType[ArenaClientStateType["Underground_Rewards"] = 14] = "Underground_Rewards";
})(ArenaClientStateType = exports.ArenaClientStateType || (exports.ArenaClientStateType = {}));
