// @firestone-hs/reference-data/dist/enums/board.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Board = void 0;
var Board;
(function (Board) {
    Board[Board["RANDOM"] = 0] = "RANDOM";
    Board[Board["STORMWIND"] = 1] = "STORMWIND";
    Board[Board["ORGRIMMAR"] = 2] = "ORGRIMMAR";
    Board[Board["PANDARIA"] = 3] = "PANDARIA";
    Board[Board["STRANGLETHORN"] = 4] = "STRANGLETHORN";
    Board[Board["NAXXRAMAS"] = 5] = "NAXXRAMAS";
    Board[Board["GOBLINS_VS_GNOMES"] = 6] = "GOBLINS_VS_GNOMES";
    Board[Board["BLACKROCK_MOUNTAIN"] = 7] = "BLACKROCK_MOUNTAIN";
    Board[Board["THE_GRAND_TOURNAMENT"] = 8] = "THE_GRAND_TOURNAMENT";
    Board[Board["THE_MUSEUM"] = 10] = "THE_MUSEUM";
    Board[Board["EXCAVATION_SITE"] = 11] = "EXCAVATION_SITE";
    Board[Board["STORMWIND_OLD_GODS"] = 12] = "STORMWIND_OLD_GODS";
    Board[Board["KARAZHAN"] = 13] = "KARAZHAN";
    Board[Board["MEAN_STREETS_OF_GADGETZAN"] = 14] = "MEAN_STREETS_OF_GADGETZAN";
    Board[Board["UNGORO"] = 26] = "UNGORO";
    Board[Board["FROZEN_THRONE"] = 54] = "FROZEN_THRONE";
    Board[Board["KOBOLDS_AND_CATACOMBS"] = 61] = "KOBOLDS_AND_CATACOMBS";
    Board[Board["THE_WITCHWOOD"] = 64] = "THE_WITCHWOOD";
    Board[Board["BOOMSDAY"] = 65] = "BOOMSDAY";
    Board[Board["RASTAKHANS_RUMBLE"] = 68] = "RASTAKHANS_RUMBLE";
    Board[Board["NON_COMBAT_ENCOUNTER"] = 69] = "NON_COMBAT_ENCOUNTER";
    Board[Board["DALARAN"] = 70] = "DALARAN";
    Board[Board["BACON"] = 71] = "BACON";
    Board[Board["SAVIORS_OF_ULDUM"] = 73] = "SAVIORS_OF_ULDUM";
    Board[Board["SAVIORS_OF_ULDUM_ALT"] = 75] = "SAVIORS_OF_ULDUM_ALT";
    Board[Board["DRAGONS"] = 78] = "DRAGONS";
    Board[Board["SAVIORS_OF_ULDUM_TB"] = 79] = "SAVIORS_OF_ULDUM_TB";
    Board[Board["DRAGONS_ADVENTURE_EVIL_4_2"] = 81] = "DRAGONS_ADVENTURE_EVIL_4_2";
    Board[Board["ASHES_OF_OUTLANDS"] = 84] = "ASHES_OF_OUTLANDS";
    Board[Board["GREEN_SCREEN_BOARD"] = 85] = "GREEN_SCREEN_BOARD";
    Board[Board["SCHOLOMANCE"] = 86] = "SCHOLOMANCE";
    Board[Board["BOOK_OF_HEROES"] = 87] = "BOOK_OF_HEROES";
    Board[Board["LETTUCE"] = 88] = "LETTUCE";
    Board[Board["DARKMOON_FAIRE"] = 89] = "DARKMOON_FAIRE";
    Board[Board["BARRENS"] = 91] = "BARRENS";
    Board[Board["MERCENARIES_BARRENS"] = 102] = "MERCENARIES_BARRENS";
    Board[Board["MERCENARIES_FELWOOD"] = 103] = "MERCENARIES_FELWOOD";
    Board[Board["MERCENARIES_WINTERSPRING"] = 104] = "MERCENARIES_WINTERSPRING";
    Board[Board["MERCENARIES_BLACKROCK_MOUNTAIN"] = 105] = "MERCENARIES_BLACKROCK_MOUNTAIN";
    Board[Board["STORMWIND_2"] = 106] = "STORMWIND_2";
    Board[Board["ALTERAC"] = 118] = "ALTERAC";
    Board[Board["MERCENARIES_ALTERAC_VALLEY"] = 119] = "MERCENARIES_ALTERAC_VALLEY";
    Board[Board["VOYAGE_TO_SUNKEN_CITY"] = 120] = "VOYAGE_TO_SUNKEN_CITY";
    Board[Board["MERCENARIES_ONYXIA"] = 121] = "MERCENARIES_ONYXIA";
    Board[Board["MERCENARIES_DARKSHORE"] = 123] = "MERCENARIES_DARKSHORE";
    Board[Board["MERCENARIES_VOYAGE_TO_THE_SUNKEN_CITY"] = 124] = "MERCENARIES_VOYAGE_TO_THE_SUNKEN_CITY";
    Board[Board["MERCENARIES_THE_DARKMOON_TROUPE"] = 125] = "MERCENARIES_THE_DARKMOON_TROUPE";
    Board[Board["REVENDRETH"] = 126] = "REVENDRETH";
    Board[Board["MARCH_OF_THE_LICH_KING"] = 129] = "MARCH_OF_THE_LICH_KING";
    Board[Board["BATTLE_OF_THE_BANDS"] = 130] = "BATTLE_OF_THE_BANDS";
    Board[Board["TITANS"] = 131] = "TITANS";
    Board[Board["WW"] = 133] = "WW";
    Board[Board["TOY"] = 134] = "TOY";
    Board[Board["RAPTOR"] = 140] = "RAPTOR";
    Board[Board["TLC"] = 141] = "TLC";
    Board[Board["TIME"] = 142] = "TIME";
    Board[Board["CATA"] = 143] = "CATA";
})(Board = exports.Board || (exports.Board = {}));
