// @firestone-hs/reference-data/dist/enums/booster-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BoosterType = void 0;
var BoosterType;
(function (BoosterType) {
    BoosterType[BoosterType["CLASSIC"] = 1] = "CLASSIC";
    BoosterType[BoosterType["DRAFT_HERO"] = 3] = "DRAFT_HERO";
    BoosterType[BoosterType["DRAFT_COMMON"] = 5] = "DRAFT_COMMON";
    BoosterType[BoosterType["DRAFT_RARE"] = 6] = "DRAFT_RARE";
    BoosterType[BoosterType["GOBLINS_VS_GNOMES"] = 9] = "GOBLINS_VS_GNOMES";
    BoosterType[BoosterType["THE_GRAND_TOURNAMENT"] = 10] = "THE_GRAND_TOURNAMENT";
    BoosterType[BoosterType["OLD_GODS"] = 11] = "OLD_GODS";
    BoosterType[BoosterType["WELCOME_BUNDLE"] = 17] = "WELCOME_BUNDLE";
    BoosterType[BoosterType["CLASSIC_1"] = 18] = "CLASSIC_1";
    BoosterType[BoosterType["MEAN_STREETS"] = 19] = "MEAN_STREETS";
    BoosterType[BoosterType["UNGORO"] = 20] = "UNGORO";
    BoosterType[BoosterType["FROZEN_THRONE"] = 21] = "FROZEN_THRONE";
    BoosterType[BoosterType["GOLDEN_CLASSIC"] = 23] = "GOLDEN_CLASSIC";
    BoosterType[BoosterType["KOBOLDS_AND_CATACOMBS"] = 30] = "KOBOLDS_AND_CATACOMBS";
    BoosterType[BoosterType["WITCHWOOD"] = 31] = "WITCHWOOD";
    BoosterType[BoosterType["DRAFT_HERO_POWER"] = 37] = "DRAFT_HERO_POWER";
    BoosterType[BoosterType["THE_BOOMSDAY_PROJECT"] = 38] = "THE_BOOMSDAY_PROJECT";
    BoosterType[BoosterType["RASTAKHANS_RUMBLE"] = 40] = "RASTAKHANS_RUMBLE";
    BoosterType[BoosterType["MAMMOTH_BUNDLE"] = 41] = "MAMMOTH_BUNDLE";
    BoosterType[BoosterType["DALARAN"] = 49] = "DALARAN";
    BoosterType[BoosterType["ULDUM"] = 128] = "ULDUM";
    BoosterType[BoosterType["WELCOME_BUNDLE_1"] = 181] = "WELCOME_BUNDLE_1";
    BoosterType[BoosterType["DRAGONS"] = 347] = "DRAGONS";
    BoosterType[BoosterType["BLACK_TEMPLE"] = 423] = "BLACK_TEMPLE";
    BoosterType[BoosterType["SCHOLOMANCE"] = 468] = "SCHOLOMANCE";
    BoosterType[BoosterType["STANDARD_HUNTER"] = 470] = "STANDARD_HUNTER";
    BoosterType[BoosterType["YEAR_OF_DRAGON"] = 498] = "YEAR_OF_DRAGON";
    BoosterType[BoosterType["STANDARD_MAGE"] = 545] = "STANDARD_MAGE";
    BoosterType[BoosterType["THE_BARRENS"] = 553] = "THE_BARRENS";
    BoosterType[BoosterType["WAILING_CAVERNS"] = 583] = "WAILING_CAVERNS";
    BoosterType[BoosterType["STORMWIND"] = 602] = "STORMWIND";
    BoosterType[BoosterType["GOLDEN_SCHOLOMANCE"] = 603] = "GOLDEN_SCHOLOMANCE";
    BoosterType[BoosterType["DARKMOON_FAIRE"] = 616] = "DARKMOON_FAIRE";
    BoosterType[BoosterType["MERCENARIES"] = 629] = "MERCENARIES";
    BoosterType[BoosterType["STANDARD_DRUID"] = 631] = "STANDARD_DRUID";
    BoosterType[BoosterType["STANDARD_PALADIN"] = 632] = "STANDARD_PALADIN";
    BoosterType[BoosterType["STANDARD_WARRIOR"] = 633] = "STANDARD_WARRIOR";
    BoosterType[BoosterType["STANDARD_PRIEST"] = 634] = "STANDARD_PRIEST";
    BoosterType[BoosterType["STANDARD_ROGUE"] = 635] = "STANDARD_ROGUE";
    BoosterType[BoosterType["STANDARD_SHAMAN"] = 636] = "STANDARD_SHAMAN";
    BoosterType[BoosterType["STANDARD_WARLOCK"] = 637] = "STANDARD_WARLOCK";
    BoosterType[BoosterType["STANDARD_DEMONHUNTER"] = 638] = "STANDARD_DEMONHUNTER";
    BoosterType[BoosterType["GOLDEN_DARKMOON_FAIRE"] = 643] = "GOLDEN_DARKMOON_FAIRE";
    BoosterType[BoosterType["ALTERAC_VALLEY"] = 665] = "ALTERAC_VALLEY";
    BoosterType[BoosterType["GOLDEN_THE_BARRENS"] = 686] = "GOLDEN_THE_BARRENS";
    BoosterType[BoosterType["YEAR_OF_PHOENIX"] = 688] = "YEAR_OF_PHOENIX";
    BoosterType[BoosterType["THE_SUNKEN_CITY"] = 694] = "THE_SUNKEN_CITY";
    BoosterType[BoosterType["STANDARD_PACK"] = 713] = "STANDARD_PACK";
    BoosterType[BoosterType["WILD_PACK"] = 714] = "WILD_PACK";
    BoosterType[BoosterType["GOLDEN_STANDARD_PACK"] = 716] = "GOLDEN_STANDARD_PACK";
    BoosterType[BoosterType["EPIC_PACK"] = 722] = "EPIC_PACK";
    BoosterType[BoosterType["REVENDRETH"] = 729] = "REVENDRETH";
    BoosterType[BoosterType["STORMWIND_GOLDEN"] = 737] = "STORMWIND_GOLDEN";
    BoosterType[BoosterType["TITANS"] = 819] = "TITANS";
    BoosterType[BoosterType["RETURN_OF_THE_LICH_KING"] = 821] = "RETURN_OF_THE_LICH_KING";
    BoosterType[BoosterType["ALTERAC_VALLEY_GOLDEN"] = 841] = "ALTERAC_VALLEY_GOLDEN";
    BoosterType[BoosterType["GOLDEN_THE_SUNKEN_CITY"] = 850] = "GOLDEN_THE_SUNKEN_CITY";
    BoosterType[BoosterType["BATTLE_OF_THE_BANDS"] = 854] = "BATTLE_OF_THE_BANDS";
    BoosterType[BoosterType["GOLDEN_REVENDRETH"] = 874] = "GOLDEN_REVENDRETH";
    BoosterType[BoosterType["CAVERNS_OF_TIME"] = 894] = "CAVERNS_OF_TIME";
    BoosterType[BoosterType["PATH_OF_ARTHAS"] = 903] = "PATH_OF_ARTHAS";
    BoosterType[BoosterType["GOLDEN_WILD_PACK"] = 904] = "GOLDEN_WILD_PACK";
    BoosterType[BoosterType["STANDARD_DEATHKNIGHT"] = 918] = "STANDARD_DEATHKNIGHT";
    BoosterType[BoosterType["GOLDEN_RETURN_OF_THE_LICH_KING"] = 921] = "GOLDEN_RETURN_OF_THE_LICH_KING";
    BoosterType[BoosterType["WILD_WEST"] = 922] = "WILD_WEST";
    BoosterType[BoosterType["GOLDEN_BATTLE_OF_THE_BANDS"] = 932] = "GOLDEN_BATTLE_OF_THE_BANDS";
    BoosterType[BoosterType["WHIZBANGS_WORKSHOP"] = 933] = "WHIZBANGS_WORKSHOP";
    BoosterType[BoosterType["GOLDEN_TITANS"] = 937] = "GOLDEN_TITANS";
    BoosterType[BoosterType["GOLDEN_YEAR_OF_THE_PHOENIX"] = 938] = "GOLDEN_YEAR_OF_THE_PHOENIX";
    BoosterType[BoosterType["GOLDEN_ASHES_OF_OUTLAND"] = 939] = "GOLDEN_ASHES_OF_OUTLAND";
    BoosterType[BoosterType["ISLAND_VACATION"] = 941] = "ISLAND_VACATION";
    BoosterType[BoosterType["WILD_WEST2"] = 944] = "WILD_WEST2";
    BoosterType[BoosterType["GOLDEN_CAVERNS_OF_TIME"] = 945] = "GOLDEN_CAVERNS_OF_TIME";
    BoosterType[BoosterType["GOLDEN_WILD_WEST"] = 952] = "GOLDEN_WILD_WEST";
    BoosterType[BoosterType["THE_GREAT_DARK_BEYOND"] = 965] = "THE_GREAT_DARK_BEYOND";
    BoosterType[BoosterType["GOLDEN_WHIZBANGS_WORKSHOP"] = 970] = "GOLDEN_WHIZBANGS_WORKSHOP";
    BoosterType[BoosterType["WHIZBANG_CATCH_UP"] = 971] = "WHIZBANG_CATCH_UP";
    BoosterType[BoosterType["INTO_THE_EMERALD_DREAM"] = 975] = "INTO_THE_EMERALD_DREAM";
    BoosterType[BoosterType["GOLDEN_ISLAND_VACATION"] = 977] = "GOLDEN_ISLAND_VACATION";
    BoosterType[BoosterType["ISLAND_VACATION_CATCH_UP"] = 978] = "ISLAND_VACATION_CATCH_UP";
    BoosterType[BoosterType["THE_LOST_CITY"] = 982] = "THE_LOST_CITY";
    BoosterType[BoosterType["YEAR_OF_THE_PEGASUS"] = 984] = "YEAR_OF_THE_PEGASUS";
    BoosterType[BoosterType["GOLDEN_YEAR_OF_THE_PEGASUS"] = 985] = "GOLDEN_YEAR_OF_THE_PEGASUS";
    BoosterType[BoosterType["GOLDEN_THE_GREAT_DARK_BEYOND"] = 986] = "GOLDEN_THE_GREAT_DARK_BEYOND";
    BoosterType[BoosterType["THE_GREAT_DARK_BEYOND_CATCH_UP"] = 987] = "THE_GREAT_DARK_BEYOND_CATCH_UP";
    BoosterType[BoosterType["ACROSS_THE_TIMEWAYS"] = 989] = "ACROSS_THE_TIMEWAYS";
    BoosterType[BoosterType["GOLDEN_INTO_THE_EMERALD_DREAM"] = 990] = "GOLDEN_INTO_THE_EMERALD_DREAM";
    BoosterType[BoosterType["CATACLYSM"] = 1030] = "CATACLYSM";
    BoosterType[BoosterType["GOLDEN_CATACLYSM"] = 1031] = "GOLDEN_CATACLYSM";
    BoosterType[BoosterType["INTO_THE_EMERALD_DREAM_CATCH_UP"] = 1033] = "INTO_THE_EMERALD_DREAM_CATCH_UP";
    BoosterType[BoosterType["GOLDEN_THE_LOST_CITY"] = 1040] = "GOLDEN_THE_LOST_CITY";
    BoosterType[BoosterType["YEAR_OF_THE_RAPTOR"] = 1044] = "YEAR_OF_THE_RAPTOR";
    BoosterType[BoosterType["GOLDEN_YEAR_OF_THE_RAPTOR"] = 1045] = "GOLDEN_YEAR_OF_THE_RAPTOR";
    BoosterType[BoosterType["THE_LOST_CITY_CATCH_UP"] = 1046] = "THE_LOST_CITY_CATCH_UP";
    BoosterType[BoosterType["GOLDEN_ACROSS_THE_TIMEWAYS"] = 1055] = "GOLDEN_ACROSS_THE_TIMEWAYS";
    BoosterType[BoosterType["ACROSS_THE_TIMEWAYS_CATCH_UP"] = 1056] = "ACROSS_THE_TIMEWAYS_CATCH_UP";
    BoosterType[BoosterType["CATACLYSM_CATCH_UP"] = 1063] = "CATACLYSM_CATCH_UP";
})(BoosterType = exports.BoosterType || (exports.BoosterType = {}));
