// @firestone-hs/reference-data/dist/enums/card-text-builder-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardTextBuilderType = void 0;
var CardTextBuilderType;
(function (CardTextBuilderType) {
    CardTextBuilderType[CardTextBuilderType["DEFAULT"] = 0] = "DEFAULT";
    CardTextBuilderType[CardTextBuilderType["JADE_GOLEM"] = 1] = "JADE_GOLEM";
    CardTextBuilderType[CardTextBuilderType["JADE_GOLEM_TRIGGER"] = 2] = "JADE_GOLEM_TRIGGER";
    CardTextBuilderType[CardTextBuilderType["MODULAR_ENTITY"] = 3] = "MODULAR_ENTITY";
    CardTextBuilderType[CardTextBuilderType["KAZAKUS_POTION_EFFECT"] = 4] = "KAZAKUS_POTION_EFFECT";
    CardTextBuilderType[CardTextBuilderType["PRIMORDIAL_WAND"] = 5] = "PRIMORDIAL_WAND";
    CardTextBuilderType[CardTextBuilderType["ALTERNATE_CARD_TEXT"] = 6] = "ALTERNATE_CARD_TEXT";
    CardTextBuilderType[CardTextBuilderType["SCRIPT_DATA_NUM_1"] = 7] = "SCRIPT_DATA_NUM_1";
    CardTextBuilderType[CardTextBuilderType["GALAKROND_COUNTER"] = 8] = "GALAKROND_COUNTER";
    CardTextBuilderType[CardTextBuilderType["DECORATE"] = 9] = "DECORATE";
    CardTextBuilderType[CardTextBuilderType["PLAYER_TAG_THRESHOLD"] = 10] = "PLAYER_TAG_THRESHOLD";
    CardTextBuilderType[CardTextBuilderType["ENTITY_TAG_THRESHOLD"] = 11] = "ENTITY_TAG_THRESHOLD";
    CardTextBuilderType[CardTextBuilderType["MULTIPLE_ENTITY_NAMES"] = 12] = "MULTIPLE_ENTITY_NAMES";
    CardTextBuilderType[CardTextBuilderType["GAMEPLAY_STRING"] = 13] = "GAMEPLAY_STRING";
    CardTextBuilderType[CardTextBuilderType["ZOMBEAST"] = 14] = "ZOMBEAST";
    CardTextBuilderType[CardTextBuilderType["ZOMBEAST_ENCHANTMENT"] = 15] = "ZOMBEAST_ENCHANTMENT";
    CardTextBuilderType[CardTextBuilderType["HIDDEN_CHOICE"] = 16] = "HIDDEN_CHOICE";
    CardTextBuilderType[CardTextBuilderType["INVESTIGATE"] = 17] = "INVESTIGATE";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_CREATOR_ENTITY"] = 18] = "REFERENCE_CREATOR_ENTITY";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_1_ENTITY"] = 19] = "REFERENCE_SCRIPT_DATA_NUM_1_ENTITY";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_1_NUM_2_ENTITY"] = 20] = "REFERENCE_SCRIPT_DATA_NUM_1_NUM_2_ENTITY";
    CardTextBuilderType[CardTextBuilderType["UNDATAKAH_ENCHANT"] = 21] = "UNDATAKAH_ENCHANT";
    CardTextBuilderType[CardTextBuilderType["SPELL_DAMAGE_ONLY"] = 22] = "SPELL_DAMAGE_ONLY";
    CardTextBuilderType[CardTextBuilderType["DRUSTVAR_HORROR"] = 23] = "DRUSTVAR_HORROR";
    CardTextBuilderType[CardTextBuilderType["HIDDEN_ENTITY"] = 24] = "HIDDEN_ENTITY";
    CardTextBuilderType[CardTextBuilderType["SCORE_VALUE_COUNT_DOWN"] = 25] = "SCORE_VALUE_COUNT_DOWN";
    CardTextBuilderType[CardTextBuilderType["SCRIPT_DATA_NUM_1_NUM_2"] = 26] = "SCRIPT_DATA_NUM_1_NUM_2";
    CardTextBuilderType[CardTextBuilderType["POWERED_UP"] = 27] = "POWERED_UP";
    CardTextBuilderType[CardTextBuilderType["MULTIPLE_ALT_TEXT_SCRIPT_DATA_NUMS"] = 28] = "MULTIPLE_ALT_TEXT_SCRIPT_DATA_NUMS";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_1_ENTITY_POWER"] = 29] = "REFERENCE_SCRIPT_DATA_NUM_1_ENTITY_POWER";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_1_CARD_DBID"] = 30] = "REFERENCE_SCRIPT_DATA_NUM_1_CARD_DBID";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_CARD_RACE"] = 31] = "REFERENCE_SCRIPT_DATA_NUM_CARD_RACE";
    CardTextBuilderType[CardTextBuilderType["BG_QUEST"] = 32] = "BG_QUEST";
    CardTextBuilderType[CardTextBuilderType["MULTIPLE_ALT_TEXT_SCRIPT_DATA_NUMS_REF_SDN6_CARD_DBID"] = 33] = "MULTIPLE_ALT_TEXT_SCRIPT_DATA_NUMS_REF_SDN6_CARD_DBID";
    CardTextBuilderType[CardTextBuilderType["ZILLIAX_DELUXE_3000"] = 34] = "ZILLIAX_DELUXE_3000";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_1_NUM_2_ENTITY_POWER"] = 35] = "REFERENCE_SCRIPT_DATA_NUM_1_NUM_2_ENTITY_POWER";
    CardTextBuilderType[CardTextBuilderType["BATTLEGROUNDS_ZILLIAX"] = 36] = "BATTLEGROUNDS_ZILLIAX";
    CardTextBuilderType[CardTextBuilderType["SPELL_ABSORB"] = 37] = "SPELL_ABSORB";
    CardTextBuilderType[CardTextBuilderType["ALT_TEXT_REFERENCE_SCRIPT_DATA_NUM_1_NUM_2_ENTITY_POWER"] = 38] = "ALT_TEXT_REFERENCE_SCRIPT_DATA_NUM_1_NUM_2_ENTITY_POWER";
    CardTextBuilderType[CardTextBuilderType["REWIND_MECHANIC_CARD_TEXT_BUILDER"] = 39] = "REWIND_MECHANIC_CARD_TEXT_BUILDER";
    CardTextBuilderType[CardTextBuilderType["BATTLEGROUNDS_TAVERN_SPELL"] = 40] = "BATTLEGROUNDS_TAVERN_SPELL";
    CardTextBuilderType[CardTextBuilderType["DYNAMIC_KEYWORD"] = 41] = "DYNAMIC_KEYWORD";
    CardTextBuilderType[CardTextBuilderType["REFERENCE_SCRIPT_DATA_NUM_1_CLASS"] = 42] = "REFERENCE_SCRIPT_DATA_NUM_1_CLASS";
    CardTextBuilderType[CardTextBuilderType["HERALD"] = 43] = "HERALD";
    CardTextBuilderType[CardTextBuilderType["ALTERNATE_CARD_TEXT_WITH_SCRIPT_DATA"] = 44] = "ALTERNATE_CARD_TEXT_WITH_SCRIPT_DATA";
    CardTextBuilderType[CardTextBuilderType["BATTLEGROUNDS_DEEP_BLUES_SPELL"] = 45] = "BATTLEGROUNDS_DEEP_BLUES_SPELL";
})(CardTextBuilderType = exports.CardTextBuilderType || (exports.CardTextBuilderType = {}));
