// @firestone-hs/reference-data/dist/enums/card-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardType = void 0;
var CardType;
(function (CardType) {
    CardType[CardType["INVALID"] = 0] = "INVALID";
    CardType[CardType["GAME"] = 1] = "GAME";
    CardType[CardType["PLAYER"] = 2] = "PLAYER";
    CardType[CardType["HERO"] = 3] = "HERO";
    CardType[CardType["MINION"] = 4] = "MINION";
    CardType[CardType["SPELL"] = 5] = "SPELL";
    CardType[CardType["ENCHANTMENT"] = 6] = "ENCHANTMENT";
    CardType[CardType["WEAPON"] = 7] = "WEAPON";
    CardType[CardType["ITEM"] = 8] = "ITEM";
    CardType[CardType["TOKEN"] = 9] = "TOKEN";
    CardType[CardType["HERO_POWER"] = 10] = "HERO_POWER";
    CardType[CardType["BLANK"] = 11] = "BLANK";
    CardType[CardType["GAME_MODE_BUTTON"] = 12] = "GAME_MODE_BUTTON";
    CardType[CardType["MOVE_MINION_HOVER_TARGET"] = 22] = "MOVE_MINION_HOVER_TARGET";
    CardType[CardType["LETTUCE_ABILITY"] = 23] = "LETTUCE_ABILITY";
    CardType[CardType["BATTLEGROUND_HERO_BUDDY"] = 24] = "BATTLEGROUND_HERO_BUDDY";
    CardType[CardType["LOCATION"] = 39] = "LOCATION";
    CardType[CardType["BATTLEGROUND_QUEST_REWARD"] = 40] = "BATTLEGROUND_QUEST_REWARD";
    CardType[CardType["BATTLEGROUND_SPELL"] = 42] = "BATTLEGROUND_SPELL";
    CardType[CardType["BATTLEGROUND_ANOMALY"] = 43] = "BATTLEGROUND_ANOMALY";
    CardType[CardType["BATTLEGROUND_TRINKET"] = 44] = "BATTLEGROUND_TRINKET";
    CardType[CardType["PET"] = 45] = "PET";
})(CardType = exports.CardType || (exports.CardType = {}));
