// @firestone-hs/reference-data/dist/enums/card-class.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardClass = void 0;
var CardClass;
(function (CardClass) {
    CardClass[CardClass["INVALID"] = 0] = "INVALID";
    CardClass[CardClass["DEATHKNIGHT"] = 1] = "DEATHKNIGHT";
    CardClass[CardClass["DRUID"] = 2] = "DRUID";
    CardClass[CardClass["HUNTER"] = 3] = "HUNTER";
    CardClass[CardClass["MAGE"] = 4] = "MAGE";
    CardClass[CardClass["PALADIN"] = 5] = "PALADIN";
    CardClass[CardClass["PRIEST"] = 6] = "PRIEST";
    CardClass[CardClass["ROGUE"] = 7] = "ROGUE";
    CardClass[CardClass["SHAMAN"] = 8] = "SHAMAN";
    CardClass[CardClass["WARLOCK"] = 9] = "WARLOCK";
    CardClass[CardClass["WARRIOR"] = 10] = "WARRIOR";
    CardClass[CardClass["DREAM"] = 11] = "DREAM";
    CardClass[CardClass["NEUTRAL"] = 12] = "NEUTRAL";
    CardClass[CardClass["WHIZBANG"] = 13] = "WHIZBANG";
    CardClass[CardClass["DEMONHUNTER"] = 14] = "DEMONHUNTER";
})(CardClass = exports.CardClass || (exports.CardClass = {}));
