// @firestone-hs/reference-data/dist/enums/card-set.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardSet = void 0;
var CardSet;
(function (CardSet) {
    CardSet[CardSet["INVALID"] = 0] = "INVALID";
    CardSet[CardSet["TEST_TEMPORARY"] = 1] = "TEST_TEMPORARY";
    CardSet[CardSet["BASIC"] = 2] = "BASIC";
    CardSet[CardSet["EXPERT1"] = 3] = "EXPERT1";
    CardSet[CardSet["REWARD"] = 4] = "REWARD";
    CardSet[CardSet["MISSIONS"] = 5] = "MISSIONS";
    CardSet[CardSet["DEMO"] = 6] = "DEMO";
    CardSet[CardSet["NONE"] = 7] = "NONE";
    CardSet[CardSet["CHEAT"] = 8] = "CHEAT";
    CardSet[CardSet["BLANK"] = 9] = "BLANK";
    CardSet[CardSet["DEBUG_SP"] = 10] = "DEBUG_SP";
    CardSet[CardSet["PROMO"] = 11] = "PROMO";
    CardSet[CardSet["NAXX"] = 12] = "NAXX";
    CardSet[CardSet["GVG"] = 13] = "GVG";
    CardSet[CardSet["BRM"] = 14] = "BRM";
    CardSet[CardSet["TGT"] = 15] = "TGT";
    CardSet[CardSet["CREDITS"] = 16] = "CREDITS";
    CardSet[CardSet["HERO_SKINS"] = 17] = "HERO_SKINS";
    CardSet[CardSet["TB"] = 18] = "TB";
    CardSet[CardSet["SLUSH"] = 19] = "SLUSH";
    CardSet[CardSet["LOE"] = 20] = "LOE";
    CardSet[CardSet["OG"] = 21] = "OG";
    CardSet[CardSet["OG_RESERVE"] = 22] = "OG_RESERVE";
    CardSet[CardSet["KARA"] = 23] = "KARA";
    CardSet[CardSet["KARA_RESERVE"] = 24] = "KARA_RESERVE";
    CardSet[CardSet["GANGS"] = 25] = "GANGS";
    CardSet[CardSet["GANGS_RESERVE"] = 26] = "GANGS_RESERVE";
    CardSet[CardSet["UNGORO"] = 27] = "UNGORO";
    CardSet[CardSet["ICECROWN"] = 1001] = "ICECROWN";
    CardSet[CardSet["LOOTAPALOOZA"] = 1004] = "LOOTAPALOOZA";
    CardSet[CardSet["GILNEAS"] = 1125] = "GILNEAS";
    CardSet[CardSet["BOOMSDAY"] = 1127] = "BOOMSDAY";
    CardSet[CardSet["TROLL"] = 1129] = "TROLL";
    CardSet[CardSet["DALARAN"] = 1130] = "DALARAN";
    CardSet[CardSet["TAVERNS_OF_TIME"] = 1143] = "TAVERNS_OF_TIME";
    CardSet[CardSet["ULDUM"] = 1158] = "ULDUM";
    CardSet[CardSet["DRAGONS"] = 1347] = "DRAGONS";
    CardSet[CardSet["YOD"] = 1403] = "YOD";
    CardSet[CardSet["BLACK_TEMPLE"] = 1414] = "BLACK_TEMPLE";
    CardSet[CardSet["WILD_EVENT"] = 1439] = "WILD_EVENT";
    CardSet[CardSet["SCHOLOMANCE"] = 1443] = "SCHOLOMANCE";
    CardSet[CardSet["BATTLEGROUNDS"] = 1453] = "BATTLEGROUNDS";
    CardSet[CardSet["DEMON_HUNTER_INITIATE"] = 1463] = "DEMON_HUNTER_INITIATE";
    CardSet[CardSet["DARKMOON_FAIRE"] = 1466] = "DARKMOON_FAIRE";
    CardSet[CardSet["THE_BARRENS"] = 1525] = "THE_BARRENS";
    CardSet[CardSet["WAILING_CAVERNS"] = 1559] = "WAILING_CAVERNS";
    CardSet[CardSet["STORMWIND"] = 1578] = "STORMWIND";
    CardSet[CardSet["LETTUCE"] = 1586] = "LETTUCE";
    CardSet[CardSet["ALTERAC_VALLEY"] = 1626] = "ALTERAC_VALLEY";
    CardSet[CardSet["LEGACY"] = 1635] = "LEGACY";
    CardSet[CardSet["CORE"] = 1637] = "CORE";
    CardSet[CardSet["VANILLA"] = 1646] = "VANILLA";
    CardSet[CardSet["THE_SUNKEN_CITY"] = 1658] = "THE_SUNKEN_CITY";
    CardSet[CardSet["REVENDRETH"] = 1691] = "REVENDRETH";
    CardSet[CardSet["MERCENARIES_DEV"] = 1705] = "MERCENARIES_DEV";
    CardSet[CardSet["OLD_CORE_230"] = 1810] = "OLD_CORE_230";
    CardSet[CardSet["RETURN_OF_THE_LICH_KING"] = 1776] = "RETURN_OF_THE_LICH_KING";
    CardSet[CardSet["BATTLE_OF_THE_BANDS"] = 1809] = "BATTLE_OF_THE_BANDS";
    CardSet[CardSet["PATH_OF_ARTHAS"] = 1869] = "PATH_OF_ARTHAS";
    CardSet[CardSet["TITANS"] = 1858] = "TITANS";
    CardSet[CardSet["WILD_WEST"] = 1892] = "WILD_WEST";
    CardSet[CardSet["WHIZBANGS_WORKSHOP"] = 1897] = "WHIZBANGS_WORKSHOP";
    CardSet[CardSet["CAVERNS_OF_TIME"] = 1898] = "CAVERNS_OF_TIME";
    CardSet[CardSet["TUTORIAL"] = 1904] = "TUTORIAL";
    CardSet[CardSet["GREAT_DARK_BEYOND"] = 1935] = "GREAT_DARK_BEYOND";
    CardSet[CardSet["GIFT"] = 1941] = "GIFT";
    CardSet[CardSet["PERILS_IN_PARADISE"] = 1905] = "PERILS_IN_PARADISE";
    CardSet[CardSet["INTO_THE_EMERALD_DREAM"] = 1946] = "INTO_THE_EMERALD_DREAM";
    CardSet[CardSet["LOST_CITY_OF_UNGORO"] = 1952] = "LOST_CITY_OF_UNGORO";
    CardSet[CardSet["ACROSS_THE_TIMEWAYS"] = 1957] = "ACROSS_THE_TIMEWAYS";
    CardSet[CardSet["COSMETICS"] = 1961] = "COSMETICS";
    CardSet[CardSet["CATACLYSM"] = 1980] = "CATACLYSM";
})(CardSet = exports.CardSet || (exports.CardSet = {}));
