// @firestone-hs/reference-data/dist/enums/game-tags.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GameTag = void 0;
var GameTag;
(function (GameTag) {
    GameTag[GameTag["TAG_NOT_SET"] = 0] = "TAG_NOT_SET";
    GameTag[GameTag["TAG_SCRIPT_DATA_NUM_1"] = 2] = "TAG_SCRIPT_DATA_NUM_1";
    GameTag[GameTag["TAG_SCRIPT_DATA_NUM_2"] = 3] = "TAG_SCRIPT_DATA_NUM_2";
    GameTag[GameTag["TAG_SCRIPT_DATA_ENT_1"] = 4] = "TAG_SCRIPT_DATA_ENT_1";
    GameTag[GameTag["TAG_SCRIPT_DATA_ENT_2"] = 5] = "TAG_SCRIPT_DATA_ENT_2";
    GameTag[GameTag["MISSION_EVENT"] = 6] = "MISSION_EVENT";
    GameTag[GameTag["TIMEOUT"] = 7] = "TIMEOUT";
    GameTag[GameTag["TURN_START"] = 8] = "TURN_START";
    GameTag[GameTag["TURN_TIMER_SLUSH"] = 9] = "TURN_TIMER_SLUSH";
    GameTag[GameTag["PREMIUM"] = 12] = "PREMIUM";
    GameTag[GameTag["PLAYSTATE"] = 17] = "PLAYSTATE";
    GameTag[GameTag["LAST_AFFECTED_BY"] = 18] = "LAST_AFFECTED_BY";
    GameTag[GameTag["STEP"] = 19] = "STEP";
    GameTag[GameTag["TURN"] = 20] = "TURN";
    GameTag[GameTag["FATIGUE"] = 22] = "FATIGUE";
    GameTag[GameTag["CURRENT_PLAYER"] = 23] = "CURRENT_PLAYER";
    GameTag[GameTag["FIRST_PLAYER"] = 24] = "FIRST_PLAYER";
    GameTag[GameTag["RESOURCES_USED"] = 25] = "RESOURCES_USED";
    GameTag[GameTag["RESOURCES"] = 26] = "RESOURCES";
    GameTag[GameTag["HERO_ENTITY"] = 27] = "HERO_ENTITY";
    GameTag[GameTag["MAXHANDSIZE"] = 28] = "MAXHANDSIZE";
    GameTag[GameTag["STARTHANDSIZE"] = 29] = "STARTHANDSIZE";
    GameTag[GameTag["PLAYER_ID"] = 30] = "PLAYER_ID";
    GameTag[GameTag["TEAM_ID"] = 31] = "TEAM_ID";
    GameTag[GameTag["TRIGGER_VISUAL"] = 32] = "TRIGGER_VISUAL";
    GameTag[GameTag["RECENTLY_ARRIVED"] = 33] = "RECENTLY_ARRIVED";
    GameTag[GameTag["PROTECTING"] = 34] = "PROTECTING";
    GameTag[GameTag["PROTECTED"] = 35] = "PROTECTED";
    GameTag[GameTag["DEFENDING"] = 36] = "DEFENDING";
    GameTag[GameTag["PROPOSED_DEFENDER"] = 37] = "PROPOSED_DEFENDER";
    GameTag[GameTag["ATTACKING"] = 38] = "ATTACKING";
    GameTag[GameTag["PROPOSED_ATTACKER"] = 39] = "PROPOSED_ATTACKER";
    GameTag[GameTag["ATTACHED"] = 40] = "ATTACHED";
    GameTag[GameTag["EXHAUSTED"] = 43] = "EXHAUSTED";
    GameTag[GameTag["DAMAGE"] = 44] = "DAMAGE";
    GameTag[GameTag["HEALTH"] = 45] = "HEALTH";
    GameTag[GameTag["ATK"] = 47] = "ATK";
    GameTag[GameTag["COST"] = 48] = "COST";
    GameTag[GameTag["ZONE"] = 49] = "ZONE";
    GameTag[GameTag["CONTROLLER"] = 50] = "CONTROLLER";
    GameTag[GameTag["OWNER"] = 51] = "OWNER";
    GameTag[GameTag["DEFINITION"] = 52] = "DEFINITION";
    GameTag[GameTag["ENTITY_ID"] = 53] = "ENTITY_ID";
    GameTag[GameTag["HISTORY_PROXY"] = 54] = "HISTORY_PROXY";
    GameTag[GameTag["ELITE"] = 114] = "ELITE";
    GameTag[GameTag["MAXRESOURCES"] = 176] = "MAXRESOURCES";
    GameTag[GameTag["CARD_SET"] = 183] = "CARD_SET";
    GameTag[GameTag["CARDTEXT"] = 184] = "CARDTEXT";
    GameTag[GameTag["DURABILITY_DEPRECATED"] = 187] = "DURABILITY_DEPRECATED";
    GameTag[GameTag["SILENCED"] = 188] = "SILENCED";
    GameTag[GameTag["WINDFURY"] = 189] = "WINDFURY";
    GameTag[GameTag["TAUNT"] = 190] = "TAUNT";
    GameTag[GameTag["STEALTH"] = 191] = "STEALTH";
    GameTag[GameTag["SPELLPOWER"] = 192] = "SPELLPOWER";
    GameTag[GameTag["DIVINE_SHIELD"] = 194] = "DIVINE_SHIELD";
    GameTag[GameTag["CHARGE"] = 197] = "CHARGE";
    GameTag[GameTag["NEXT_STEP"] = 198] = "NEXT_STEP";
    GameTag[GameTag["CLASS"] = 199] = "CLASS";
    GameTag[GameTag["CARDRACE"] = 200] = "CARDRACE";
    GameTag[GameTag["FACTION"] = 201] = "FACTION";
    GameTag[GameTag["CARDTYPE"] = 202] = "CARDTYPE";
    GameTag[GameTag["RARITY"] = 203] = "RARITY";
    GameTag[GameTag["STATE"] = 204] = "STATE";
    GameTag[GameTag["SUMMONED"] = 205] = "SUMMONED";
    GameTag[GameTag["FREEZE"] = 208] = "FREEZE";
    GameTag[GameTag["ENRAGED"] = 212] = "ENRAGED";
    GameTag[GameTag["OVERLOAD"] = 215] = "OVERLOAD";
    GameTag[GameTag["LOYALTY"] = 216] = "LOYALTY";
    GameTag[GameTag["DEATHRATTLE"] = 217] = "DEATHRATTLE";
    GameTag[GameTag["BATTLECRY"] = 218] = "BATTLECRY";
    GameTag[GameTag["SECRET"] = 219] = "SECRET";
    GameTag[GameTag["COMBO"] = 220] = "COMBO";
    GameTag[GameTag["CANT_HEAL"] = 221] = "CANT_HEAL";
    GameTag[GameTag["CANT_DAMAGE"] = 222] = "CANT_DAMAGE";
    GameTag[GameTag["CANT_SET_ASIDE"] = 223] = "CANT_SET_ASIDE";
    GameTag[GameTag["CANT_REMOVE_FROM_GAME"] = 224] = "CANT_REMOVE_FROM_GAME";
    GameTag[GameTag["CANT_READY"] = 225] = "CANT_READY";
    GameTag[GameTag["CANT_ATTACK"] = 227] = "CANT_ATTACK";
    GameTag[GameTag["CANT_DISCARD"] = 230] = "CANT_DISCARD";
    GameTag[GameTag["CANT_PLAY"] = 231] = "CANT_PLAY";
    GameTag[GameTag["CANT_DRAW"] = 232] = "CANT_DRAW";
    GameTag[GameTag["CANT_BE_HEALED"] = 239] = "CANT_BE_HEALED";
    GameTag[GameTag["IMMUNE"] = 240] = "IMMUNE";
    GameTag[GameTag["CANT_BE_SET_ASIDE"] = 241] = "CANT_BE_SET_ASIDE";
    GameTag[GameTag["CANT_BE_REMOVED_FROM_GAME"] = 242] = "CANT_BE_REMOVED_FROM_GAME";
    GameTag[GameTag["CANT_BE_READIED"] = 243] = "CANT_BE_READIED";
    GameTag[GameTag["CANT_BE_ATTACKED"] = 245] = "CANT_BE_ATTACKED";
    GameTag[GameTag["CANT_BE_TARGETED"] = 246] = "CANT_BE_TARGETED";
    GameTag[GameTag["CANT_BE_DESTROYED"] = 247] = "CANT_BE_DESTROYED";
    GameTag[GameTag["CANT_BE_SUMMONING_SICK"] = 253] = "CANT_BE_SUMMONING_SICK";
    GameTag[GameTag["FROZEN"] = 260] = "FROZEN";
    GameTag[GameTag["JUST_PLAYED"] = 261] = "JUST_PLAYED";
    GameTag[GameTag["LINKED_ENTITY"] = 262] = "LINKED_ENTITY";
    GameTag[GameTag["ZONE_POSITION"] = 263] = "ZONE_POSITION";
    GameTag[GameTag["CANT_BE_FROZEN"] = 264] = "CANT_BE_FROZEN";
    GameTag[GameTag["COMBO_ACTIVE"] = 266] = "COMBO_ACTIVE";
    GameTag[GameTag["CARD_TARGET"] = 267] = "CARD_TARGET";
    GameTag[GameTag["NUM_CARDS_PLAYED_THIS_TURN"] = 269] = "NUM_CARDS_PLAYED_THIS_TURN";
    GameTag[GameTag["CANT_BE_TARGETED_BY_OPPONENTS"] = 270] = "CANT_BE_TARGETED_BY_OPPONENTS";
    GameTag[GameTag["NUM_TURNS_IN_PLAY"] = 271] = "NUM_TURNS_IN_PLAY";
    GameTag[GameTag["NUM_TURNS_LEFT"] = 272] = "NUM_TURNS_LEFT";
    GameTag[GameTag["NUM_TURNS_IN_HAND"] = 273] = "NUM_TURNS_IN_HAND";
    GameTag[GameTag["CURRENT_SPELLPOWER_BASE"] = 291] = "CURRENT_SPELLPOWER_BASE";
    GameTag[GameTag["ARMOR"] = 292] = "ARMOR";
    GameTag[GameTag["MORPH"] = 293] = "MORPH";
    GameTag[GameTag["IS_MORPHED"] = 294] = "IS_MORPHED";
    GameTag[GameTag["TEMP_RESOURCES"] = 295] = "TEMP_RESOURCES";
    GameTag[GameTag["OVERLOAD_OWED"] = 296] = "OVERLOAD_OWED";
    GameTag[GameTag["NUM_ATTACKS_THIS_TURN"] = 297] = "NUM_ATTACKS_THIS_TURN";
    GameTag[GameTag["NEXT_ALLY_BUFF"] = 302] = "NEXT_ALLY_BUFF";
    GameTag[GameTag["MAGNET"] = 303] = "MAGNET";
    GameTag[GameTag["FIRST_CARD_PLAYED_THIS_TURN"] = 304] = "FIRST_CARD_PLAYED_THIS_TURN";
    GameTag[GameTag["MULLIGAN_STATE"] = 305] = "MULLIGAN_STATE";
    GameTag[GameTag["TAUNT_READY"] = 306] = "TAUNT_READY";
    GameTag[GameTag["STEALTH_READY"] = 307] = "STEALTH_READY";
    GameTag[GameTag["CHARGE_READY"] = 308] = "CHARGE_READY";
    GameTag[GameTag["CANT_BE_TARGETED_BY_SPELLS"] = 311] = "CANT_BE_TARGETED_BY_SPELLS";
    GameTag[GameTag["SHOULDEXITCOMBAT"] = 312] = "SHOULDEXITCOMBAT";
    GameTag[GameTag["CREATOR"] = 313] = "CREATOR";
    GameTag[GameTag["CANT_BE_SILENCED"] = 314] = "CANT_BE_SILENCED";
    GameTag[GameTag["PARENT_CARD"] = 316] = "PARENT_CARD";
    GameTag[GameTag["NUM_MINIONS_PLAYED_THIS_TURN"] = 317] = "NUM_MINIONS_PLAYED_THIS_TURN";
    GameTag[GameTag["PREDAMAGE"] = 318] = "PREDAMAGE";
    GameTag[GameTag["COLLECTIBLE"] = 321] = "COLLECTIBLE";
    GameTag[GameTag["TARGETING_ARROW_TEXT"] = 325] = "TARGETING_ARROW_TEXT";
    GameTag[GameTag["HEALING_DOES_DAMAGE"] = 326] = "HEALING_DOES_DAMAGE";
    GameTag[GameTag["DATABASE_ID"] = 327] = "DATABASE_ID";
    GameTag[GameTag["ENCHANTMENT_BIRTH_VISUAL"] = 330] = "ENCHANTMENT_BIRTH_VISUAL";
    GameTag[GameTag["ENCHANTMENT_IDLE_VISUAL"] = 331] = "ENCHANTMENT_IDLE_VISUAL";
    GameTag[GameTag["CANT_BE_TARGETED_BY_HERO_POWERS"] = 332] = "CANT_BE_TARGETED_BY_HERO_POWERS";
    GameTag[GameTag["_333"] = 333] = "_333";
    GameTag[GameTag["MAIN_HAND_WEAPON_ENTITY"] = 334] = "MAIN_HAND_WEAPON_ENTITY";
    GameTag[GameTag["HEALTH_MINIMUM"] = 337] = "HEALTH_MINIMUM";
    GameTag[GameTag["TAG_ONE_TURN_EFFECT"] = 338] = "TAG_ONE_TURN_EFFECT";
    GameTag[GameTag["SILENCE"] = 339] = "SILENCE";
    GameTag[GameTag["COUNTER"] = 340] = "COUNTER";
    GameTag[GameTag["ZONES_REVEALED"] = 348] = "ZONES_REVEALED";
    GameTag[GameTag["ADJACENT_BUFF"] = 350] = "ADJACENT_BUFF";
    GameTag[GameTag["FORCED_PLAY"] = 352] = "FORCED_PLAY";
    GameTag[GameTag["LOW_HEALTH_THRESHOLD"] = 353] = "LOW_HEALTH_THRESHOLD";
    GameTag[GameTag["SPELLPOWER_DOUBLE"] = 356] = "SPELLPOWER_DOUBLE";
    GameTag[GameTag["SPELL_HEALING_DOUBLE"] = 357] = "SPELL_HEALING_DOUBLE";
    GameTag[GameTag["NUM_OPTIONS_PLAYED_THIS_TURN"] = 358] = "NUM_OPTIONS_PLAYED_THIS_TURN";
    GameTag[GameTag["TO_BE_DESTROYED"] = 360] = "TO_BE_DESTROYED";
    GameTag[GameTag["AURA"] = 362] = "AURA";
    GameTag[GameTag["POISONOUS"] = 363] = "POISONOUS";
    GameTag[GameTag["HERO_POWER_DOUBLE"] = 366] = "HERO_POWER_DOUBLE";
    GameTag[GameTag["AI_MUST_PLAY"] = 367] = "AI_MUST_PLAY";
    GameTag[GameTag["NUM_MINIONS_PLAYER_KILLED_THIS_TURN"] = 368] = "NUM_MINIONS_PLAYER_KILLED_THIS_TURN";
    GameTag[GameTag["NUM_MINIONS_KILLED_THIS_TURN"] = 369] = "NUM_MINIONS_KILLED_THIS_TURN";
    GameTag[GameTag["AFFECTED_BY_SPELL_POWER"] = 370] = "AFFECTED_BY_SPELL_POWER";
    GameTag[GameTag["EXTRA_MINION_DEATHRATTLES_BASE"] = 371] = "EXTRA_MINION_DEATHRATTLES_BASE";
    GameTag[GameTag["START_WITH_1_HEALTH"] = 372] = "START_WITH_1_HEALTH";
    GameTag[GameTag["IMMUNE_WHILE_ATTACKING"] = 373] = "IMMUNE_WHILE_ATTACKING";
    GameTag[GameTag["MULTIPLY_HERO_DAMAGE"] = 374] = "MULTIPLY_HERO_DAMAGE";
    GameTag[GameTag["MULTIPLY_BUFF_VALUE"] = 375] = "MULTIPLY_BUFF_VALUE";
    GameTag[GameTag["CUSTOM_KEYWORD_EFFECT"] = 376] = "CUSTOM_KEYWORD_EFFECT";
    GameTag[GameTag["TOPDECK"] = 377] = "TOPDECK";
    GameTag[GameTag["CANT_BE_TARGETED_BY_BATTLECRIES"] = 379] = "CANT_BE_TARGETED_BY_BATTLECRIES";
    GameTag[GameTag["HERO_POWER"] = 380] = "HERO_POWER";
    GameTag[GameTag["DEATHRATTLE_RETURN_ZONE"] = 382] = "DEATHRATTLE_RETURN_ZONE";
    GameTag[GameTag["STEADY_SHOT_CAN_TARGET"] = 383] = "STEADY_SHOT_CAN_TARGET";
    GameTag[GameTag["DISPLAYED_CREATOR"] = 385] = "DISPLAYED_CREATOR";
    GameTag[GameTag["POWERED_UP"] = 386] = "POWERED_UP";
    GameTag[GameTag["SPARE_PART"] = 388] = "SPARE_PART";
    GameTag[GameTag["FORGETFUL"] = 389] = "FORGETFUL";
    GameTag[GameTag["CAN_SUMMON_MAXPLUSONE_MINION"] = 390] = "CAN_SUMMON_MAXPLUSONE_MINION";
    GameTag[GameTag["OBFUSCATED"] = 391] = "OBFUSCATED";
    GameTag[GameTag["BURNING"] = 392] = "BURNING";
    GameTag[GameTag["OVERLOAD_LOCKED"] = 393] = "OVERLOAD_LOCKED";
    GameTag[GameTag["NUM_TIMES_HERO_POWER_USED_THIS_GAME"] = 394] = "NUM_TIMES_HERO_POWER_USED_THIS_GAME";
    GameTag[GameTag["CURRENT_HEROPOWER_DAMAGE_BONUS"] = 395] = "CURRENT_HEROPOWER_DAMAGE_BONUS";
    GameTag[GameTag["HEROPOWER_DAMAGE"] = 396] = "HEROPOWER_DAMAGE";
    GameTag[GameTag["NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN"] = 398] = "NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN";
    GameTag[GameTag["NUM_CARDS_DRAWN_THIS_TURN"] = 399] = "NUM_CARDS_DRAWN_THIS_TURN";
    GameTag[GameTag["AI_ONE_SHOT_KILL"] = 400] = "AI_ONE_SHOT_KILL";
    GameTag[GameTag["EVIL_GLOW"] = 401] = "EVIL_GLOW";
    GameTag[GameTag["HIDE_STATS"] = 402] = "HIDE_STATS";
    GameTag[GameTag["INSPIRE"] = 403] = "INSPIRE";
    GameTag[GameTag["RECEIVES_DOUBLE_SPELLDAMAGE_BONUS"] = 404] = "RECEIVES_DOUBLE_SPELLDAMAGE_BONUS";
    GameTag[GameTag["HEROPOWER_ADDITIONAL_ACTIVATIONS"] = 405] = "HEROPOWER_ADDITIONAL_ACTIVATIONS";
    GameTag[GameTag["HEROPOWER_ACTIVATIONS_THIS_TURN"] = 406] = "HEROPOWER_ACTIVATIONS_THIS_TURN";
    GameTag[GameTag["REVEALED"] = 410] = "REVEALED";
    GameTag[GameTag["EXTRA_BATTLECRIES_BASE"] = 411] = "EXTRA_BATTLECRIES_BASE";
    GameTag[GameTag["NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME"] = 412] = "NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME";
    GameTag[GameTag["CANNOT_ATTACK_HEROES"] = 413] = "CANNOT_ATTACK_HEROES";
    GameTag[GameTag["LOCK_AND_LOAD"] = 414] = "LOCK_AND_LOAD";
    GameTag[GameTag["DISCOVER"] = 415] = "DISCOVER";
    GameTag[GameTag["SHADOWFORM"] = 416] = "SHADOWFORM";
    GameTag[GameTag["NUM_FRIENDLY_MINIONS_THAT_ATTACKED_THIS_TURN"] = 417] = "NUM_FRIENDLY_MINIONS_THAT_ATTACKED_THIS_TURN";
    GameTag[GameTag["NUM_RESOURCES_SPENT_THIS_GAME"] = 418] = "NUM_RESOURCES_SPENT_THIS_GAME";
    GameTag[GameTag["CHOOSE_BOTH"] = 419] = "CHOOSE_BOTH";
    GameTag[GameTag["ELECTRIC_CHARGE_LEVEL"] = 420] = "ELECTRIC_CHARGE_LEVEL";
    GameTag[GameTag["HEAVILY_ARMORED"] = 421] = "HEAVILY_ARMORED";
    GameTag[GameTag["DONT_SHOW_IMMUNE"] = 422] = "DONT_SHOW_IMMUNE";
    GameTag[GameTag["PREHEALING"] = 425] = "PREHEALING";
    GameTag[GameTag["APPEAR_FUNCTIONALLY_DEAD"] = 426] = "APPEAR_FUNCTIONALLY_DEAD";
    GameTag[GameTag["OVERLOAD_THIS_GAME"] = 427] = "OVERLOAD_THIS_GAME";
    GameTag[GameTag["SPELLS_COST_HEALTH"] = 431] = "SPELLS_COST_HEALTH";
    GameTag[GameTag["HISTORY_PROXY_NO_BIG_CARD"] = 432] = "HISTORY_PROXY_NO_BIG_CARD";
    GameTag[GameTag["IGNORE_TAUNT"] = 433] = "IGNORE_TAUNT";
    GameTag[GameTag["TRANSFORMED_FROM_CARD"] = 435] = "TRANSFORMED_FROM_CARD";
    GameTag[GameTag["CTHUN"] = 436] = "CTHUN";
    GameTag[GameTag["CAST_RANDOM_SPELLS"] = 437] = "CAST_RANDOM_SPELLS";
    GameTag[GameTag["SHIFTING"] = 438] = "SHIFTING";
    GameTag[GameTag["JADE_GOLEM"] = 441] = "JADE_GOLEM";
    GameTag[GameTag["EMBRACE_THE_SHADOW"] = 442] = "EMBRACE_THE_SHADOW";
    GameTag[GameTag["CHOOSE_ONE"] = 443] = "CHOOSE_ONE";
    GameTag[GameTag["EXTRA_ATTACKS_THIS_TURN"] = 444] = "EXTRA_ATTACKS_THIS_TURN";
    GameTag[GameTag["SEEN_CTHUN"] = 445] = "SEEN_CTHUN";
    GameTag[GameTag["MINION_TYPE_REFERENCE"] = 447] = "MINION_TYPE_REFERENCE";
    GameTag[GameTag["UNTOUCHABLE"] = 448] = "UNTOUCHABLE";
    GameTag[GameTag["RED_MANA_CRYSTALS"] = 449] = "RED_MANA_CRYSTALS";
    GameTag[GameTag["RED_MANA_GEM"] = 449] = "RED_MANA_GEM";
    GameTag[GameTag["SCORE_LABELID_1"] = 450] = "SCORE_LABELID_1";
    GameTag[GameTag["SCORE_VALUE_1"] = 451] = "SCORE_VALUE_1";
    GameTag[GameTag["SCORE_LABELID_2"] = 452] = "SCORE_LABELID_2";
    GameTag[GameTag["SCORE_VALUE_2"] = 453] = "SCORE_VALUE_2";
    GameTag[GameTag["SCORE_LABELID_3"] = 454] = "SCORE_LABELID_3";
    GameTag[GameTag["SCORE_VALUE_3"] = 455] = "SCORE_VALUE_3";
    GameTag[GameTag["CANT_BE_FATIGUED"] = 456] = "CANT_BE_FATIGUED";
    GameTag[GameTag["AUTO_ATTACK"] = 457] = "AUTO_ATTACK";
    GameTag[GameTag["ARMS_DEALING"] = 458] = "ARMS_DEALING";
    GameTag[GameTag["QUEST"] = 462] = "QUEST";
    GameTag[GameTag["TAG_LAST_KNOWN_COST_IN_HAND"] = 466] = "TAG_LAST_KNOWN_COST_IN_HAND";
    GameTag[GameTag["DEFINING_ENCHANTMENT"] = 469] = "DEFINING_ENCHANTMENT";
    GameTag[GameTag["FINISH_ATTACK_SPELL_ON_DAMAGE"] = 470] = "FINISH_ATTACK_SPELL_ON_DAMAGE";
    GameTag[GameTag["MODULAR_ENTITY_PART_1"] = 471] = "MODULAR_ENTITY_PART_1";
    GameTag[GameTag["MODULAR_ENTITY_PART_2"] = 472] = "MODULAR_ENTITY_PART_2";
    GameTag[GameTag["MODIFY_DEFINITION_ATTACK"] = 473] = "MODIFY_DEFINITION_ATTACK";
    GameTag[GameTag["MODIFY_DEFINITION_HEALTH"] = 474] = "MODIFY_DEFINITION_HEALTH";
    GameTag[GameTag["MODIFY_DEFINITION_COST"] = 475] = "MODIFY_DEFINITION_COST";
    GameTag[GameTag["MULTIPLE_CLASSES"] = 476] = "MULTIPLE_CLASSES";
    GameTag[GameTag["ALL_TARGETS_RANDOM"] = 477] = "ALL_TARGETS_RANDOM";
    GameTag[GameTag["MULTI_CLASS_GROUP"] = 480] = "MULTI_CLASS_GROUP";
    GameTag[GameTag["GRIMY_GOONS"] = 482] = "GRIMY_GOONS";
    GameTag[GameTag["JADE_LOTUS"] = 483] = "JADE_LOTUS";
    GameTag[GameTag["KABAL"] = 484] = "KABAL";
    GameTag[GameTag["ADDITIONAL_PLAY_REQS_1"] = 515] = "ADDITIONAL_PLAY_REQS_1";
    GameTag[GameTag["ADDITIONAL_PLAY_REQS_2"] = 516] = "ADDITIONAL_PLAY_REQS_2";
    GameTag[GameTag["ELEMENTAL_POWERED_UP"] = 532] = "ELEMENTAL_POWERED_UP";
    GameTag[GameTag["QUEST_PROGRESS"] = 534] = "QUEST_PROGRESS";
    GameTag[GameTag["QUEST_PROGRESS_TOTAL"] = 535] = "QUEST_PROGRESS_TOTAL";
    GameTag[GameTag["QUEST_CONTRIBUTOR"] = 541] = "QUEST_CONTRIBUTOR";
    GameTag[GameTag["ADAPT"] = 546] = "ADAPT";
    GameTag[GameTag["IS_CURRENT_TURN_AN_EXTRA_TURN"] = 547] = "IS_CURRENT_TURN_AN_EXTRA_TURN";
    GameTag[GameTag["EXTRA_TURNS_TAKEN_THIS_GAME"] = 548] = "EXTRA_TURNS_TAKEN_THIS_GAME";
    GameTag[GameTag["SHIFTING_MINION"] = 549] = "SHIFTING_MINION";
    GameTag[GameTag["SHIFTING_WEAPON"] = 550] = "SHIFTING_WEAPON";
    GameTag[GameTag["DEATH_KNIGHT"] = 554] = "DEATH_KNIGHT";
    GameTag[GameTag["BOSS"] = 556] = "BOSS";
    GameTag[GameTag["TREASURE"] = 557] = "TREASURE";
    GameTag[GameTag["TREASURE_DEFINTIONAL_ATTACK"] = 558] = "TREASURE_DEFINTIONAL_ATTACK";
    GameTag[GameTag["TREASURE_DEFINTIONAL_COST"] = 559] = "TREASURE_DEFINTIONAL_COST";
    GameTag[GameTag["TREASURE_DEFINTIONAL_HEALTH"] = 560] = "TREASURE_DEFINTIONAL_HEALTH";
    GameTag[GameTag["ACTS_LIKE_A_SPELL"] = 561] = "ACTS_LIKE_A_SPELL";
    GameTag[GameTag["STAMPEDE"] = 564] = "STAMPEDE";
    GameTag[GameTag["EMPOWERED_TREASURE"] = 646] = "EMPOWERED_TREASURE";
    GameTag[GameTag["ONE_SIDED_GHOSTLY"] = 648] = "ONE_SIDED_GHOSTLY";
    GameTag[GameTag["CURRENT_NEGATIVE_SPELLPOWER"] = 651] = "CURRENT_NEGATIVE_SPELLPOWER";
    GameTag[GameTag["DONT_PICK_FROM_SUBSETS"] = 676] = "DONT_PICK_FROM_SUBSETS";
    GameTag[GameTag["IS_VAMPIRE"] = 680] = "IS_VAMPIRE";
    GameTag[GameTag["CORRUPTED"] = 681] = "CORRUPTED";
    GameTag[GameTag["HIDE_HEALTH"] = 682] = "HIDE_HEALTH";
    GameTag[GameTag["HIDE_ATTACK"] = 683] = "HIDE_ATTACK";
    GameTag[GameTag["HIDE_COST"] = 684] = "HIDE_COST";
    GameTag[GameTag["LIFESTEAL"] = 685] = "LIFESTEAL";
    GameTag[GameTag["OVERRIDE_EMOTE_0"] = 740] = "OVERRIDE_EMOTE_0";
    GameTag[GameTag["OVERRIDE_EMOTE_1"] = 741] = "OVERRIDE_EMOTE_1";
    GameTag[GameTag["OVERRIDE_EMOTE_2"] = 742] = "OVERRIDE_EMOTE_2";
    GameTag[GameTag["OVERRIDE_EMOTE_3"] = 743] = "OVERRIDE_EMOTE_3";
    GameTag[GameTag["OVERRIDE_EMOTE_4"] = 744] = "OVERRIDE_EMOTE_4";
    GameTag[GameTag["OVERRIDE_EMOTE_5"] = 745] = "OVERRIDE_EMOTE_5";
    GameTag[GameTag["SCORE_FOOTERID"] = 751] = "SCORE_FOOTERID";
    GameTag[GameTag["RECRUIT"] = 763] = "RECRUIT";
    GameTag[GameTag["LOOT_CARD_1"] = 764] = "LOOT_CARD_1";
    GameTag[GameTag["LOOT_CARD_2"] = 765] = "LOOT_CARD_2";
    GameTag[GameTag["LOOT_CARD_3"] = 766] = "LOOT_CARD_3";
    GameTag[GameTag["OFF_HAND_WEAPON_ENTITY"] = 775] = "OFF_HAND_WEAPON_ENTITY";
    GameTag[GameTag["HERO_POWER_DISABLED"] = 777] = "HERO_POWER_DISABLED";
    GameTag[GameTag["VALEERASHADOW"] = 779] = "VALEERASHADOW";
    GameTag[GameTag["OVERRIDECARDNAME"] = 781] = "OVERRIDECARDNAME";
    GameTag[GameTag["OVERRIDECARDTEXTBUILDER"] = 782] = "OVERRIDECARDTEXTBUILDER";
    GameTag[GameTag["DUNGEON_PASSIVE_BUFF"] = 783] = "DUNGEON_PASSIVE_BUFF";
    GameTag[GameTag["GHOSTLY"] = 785] = "GHOSTLY";
    GameTag[GameTag["DISGUISED_TWIN"] = 788] = "DISGUISED_TWIN";
    GameTag[GameTag["SECRET_DEATHRATTLE"] = 789] = "SECRET_DEATHRATTLE";
    GameTag[GameTag["RUSH"] = 791] = "RUSH";
    GameTag[GameTag["REVEAL_CHOICES"] = 792] = "REVEAL_CHOICES";
    GameTag[GameTag["HERO_DECK_ID"] = 793] = "HERO_DECK_ID";
    GameTag[GameTag["HIDDEN_CHOICE"] = 813] = "HIDDEN_CHOICE";
    GameTag[GameTag["ZOMBEAST"] = 823] = "ZOMBEAST";
    GameTag[GameTag["HERO_EMOTE_SILENCED"] = 832] = "HERO_EMOTE_SILENCED";
    GameTag[GameTag["MINION_IN_HAND_BUFF"] = 845] = "MINION_IN_HAND_BUFF";
    GameTag[GameTag["ECHO"] = 846] = "ECHO";
    GameTag[GameTag["MAGNETIC"] = 849] = "MAGNETIC";
    GameTag[GameTag["MODULAR"] = 849] = "MODULAR";
    GameTag[GameTag["IGNORE_HIDE_STATS_FOR_BIG_CARD"] = 857] = "IGNORE_HIDE_STATS_FOR_BIG_CARD";
    GameTag[GameTag["AI_OPPONENT_CARD_COPY"] = 858] = "AI_OPPONENT_CARD_COPY";
    GameTag[GameTag["REAL_TIME_TRANSFORM"] = 859] = "REAL_TIME_TRANSFORM";
    GameTag[GameTag["WAIT_FOR_PLAYER_RECONNECT_PERIOD"] = 860] = "WAIT_FOR_PLAYER_RECONNECT_PERIOD";
    GameTag[GameTag["ETHEREAL"] = 880] = "ETHEREAL";
    GameTag[GameTag["EXTRA_DEATHRATTLES_BASE"] = 882] = "EXTRA_DEATHRATTLES_BASE";
    GameTag[GameTag["PHASED_RESTART"] = 888] = "PHASED_RESTART";
    GameTag[GameTag["HEALTH_DISPLAY"] = 917] = "HEALTH_DISPLAY";
    GameTag[GameTag["ENABLE_HEALTH_DISPLAY"] = 920] = "ENABLE_HEALTH_DISPLAY";
    GameTag[GameTag["VOODOO_LINK"] = 921] = "VOODOO_LINK";
    GameTag[GameTag["OVERKILL"] = 923] = "OVERKILL";
    GameTag[GameTag["PROPHECY"] = 924] = "PROPHECY";
    GameTag[GameTag["ATTACKABLE_BY_RUSH"] = 930] = "ATTACKABLE_BY_RUSH";
    GameTag[GameTag["SHIFTING_SPELL"] = 936] = "SHIFTING_SPELL";
    GameTag[GameTag["USE_ALTERNATE_CARD_TEXT"] = 955] = "USE_ALTERNATE_CARD_TEXT";
    GameTag[GameTag["COLLECTIONMANAGER_FILTER_MANA_EVEN"] = 956] = "COLLECTIONMANAGER_FILTER_MANA_EVEN";
    GameTag[GameTag["COLLECTIONMANAGER_FILTER_MANA_ODD"] = 957] = "COLLECTIONMANAGER_FILTER_MANA_ODD";
    GameTag[GameTag["AMOUNT_HEALED_THIS_GAME"] = 958] = "AMOUNT_HEALED_THIS_GAME";
    GameTag[GameTag["SUPPRESS_DEATH_SOUND"] = 959] = "SUPPRESS_DEATH_SOUND";
    GameTag[GameTag["ECHOING_OOZE_SPELL"] = 963] = "ECHOING_OOZE_SPELL";
    GameTag[GameTag["ZOMBEAST_DEBUG_CURRENT_BEAST_DATABASE_ID"] = 964] = "ZOMBEAST_DEBUG_CURRENT_BEAST_DATABASE_ID";
    GameTag[GameTag["ZOMBEAST_DEBUG_CURRENT_ITERATION"] = 965] = "ZOMBEAST_DEBUG_CURRENT_ITERATION";
    GameTag[GameTag["ZOMBEAST_DEBUG_MAX_ITERATIONS"] = 966] = "ZOMBEAST_DEBUG_MAX_ITERATIONS";
    GameTag[GameTag["START_OF_GAME_KEYWORD"] = 968] = "START_OF_GAME_KEYWORD";
    GameTag[GameTag["ENCHANTMENT_INVISIBLE"] = 976] = "ENCHANTMENT_INVISIBLE";
    GameTag[GameTag["PUZZLE"] = 979] = "PUZZLE";
    GameTag[GameTag["PUZZLE_PROGRESS"] = 980] = "PUZZLE_PROGRESS";
    GameTag[GameTag["PUZZLE_PROGRESS_TOTAL"] = 981] = "PUZZLE_PROGRESS_TOTAL";
    GameTag[GameTag["PUZZLE_TYPE"] = 982] = "PUZZLE_TYPE";
    GameTag[GameTag["PUZZLE_COMPLETED"] = 984] = "PUZZLE_COMPLETED";
    GameTag[GameTag["CONCEDE_BUTTON_ALTERNATIVE_TEXT"] = 985] = "CONCEDE_BUTTON_ALTERNATIVE_TEXT";
    GameTag[GameTag["HIDE_RESTART_BUTTON"] = 990] = "HIDE_RESTART_BUTTON";
    GameTag[GameTag["WILD"] = 991] = "WILD";
    GameTag[GameTag["HALL_OF_FAME"] = 992] = "HALL_OF_FAME";
    GameTag[GameTag["MARK_OF_EVIL"] = 994] = "MARK_OF_EVIL";
    GameTag[GameTag["EVILZUG"] = 994] = "EVILZUG";
    GameTag[GameTag["DECK_RULE_MOD_DECK_SIZE"] = 997] = "DECK_RULE_MOD_DECK_SIZE";
    GameTag[GameTag["FAST_BATTLECRY"] = 998] = "FAST_BATTLECRY";
    GameTag[GameTag["END_TURN_BUTTON_ALTERNATIVE_APPEARANCE"] = 1000] = "END_TURN_BUTTON_ALTERNATIVE_APPEARANCE";
    GameTag[GameTag["WAND"] = 1015] = "WAND";
    GameTag[GameTag["TREAT_AS_PLAYED_HERO_CARD"] = 1016] = "TREAT_AS_PLAYED_HERO_CARD";
    GameTag[GameTag["LITERALLY_UNPLAYABLE"] = 1020] = "LITERALLY_UNPLAYABLE";
    GameTag[GameTag["NUM_HERO_POWER_DAMAGE_THIS_GAME"] = 1025] = "NUM_HERO_POWER_DAMAGE_THIS_GAME";
    GameTag[GameTag["PUZZLE_NAME"] = 1026] = "PUZZLE_NAME";
    GameTag[GameTag["TURN_INDICATOR_ALTERNATIVE_APPEARANCE"] = 1027] = "TURN_INDICATOR_ALTERNATIVE_APPEARANCE";
    GameTag[GameTag["PREVIOUS_PUZZLE_COMPLETED"] = 1042] = "PREVIOUS_PUZZLE_COMPLETED";
    GameTag[GameTag["GLORIOUSGLOOP"] = 1044] = "GLORIOUSGLOOP";
    GameTag[GameTag["HEALTH_DISPLAY_COLOR"] = 1046] = "HEALTH_DISPLAY_COLOR";
    GameTag[GameTag["HEALTH_DISPLAY_NEGATIVE"] = 1047] = "HEALTH_DISPLAY_NEGATIVE";
    GameTag[GameTag["WHIZBANG_DECK_ID"] = 1048] = "WHIZBANG_DECK_ID";
    GameTag[GameTag["HIDE_OUT_OF_CARDS_WARNING"] = 1050] = "HIDE_OUT_OF_CARDS_WARNING";
    GameTag[GameTag["GEARS"] = 1052] = "GEARS";
    GameTag[GameTag["LUNAHIGHLIGHTHINT"] = 1054] = "LUNAHIGHLIGHTHINT";
    GameTag[GameTag["SUPPRESS_JOBS_DONE_VO"] = 1055] = "SUPPRESS_JOBS_DONE_VO";
    GameTag[GameTag["SHRINE"] = 1057] = "SHRINE";
    GameTag[GameTag["ALL_HEALING_DOUBLE"] = 1058] = "ALL_HEALING_DOUBLE";
    GameTag[GameTag["BLOCK_ALL_INPUT"] = 1071] = "BLOCK_ALL_INPUT";
    GameTag[GameTag["PUZZLE_MODE"] = 1073] = "PUZZLE_MODE";
    GameTag[GameTag["CARD_DOES_NOTHING"] = 1075] = "CARD_DOES_NOTHING";
    GameTag[GameTag["CASTS_WHEN_DRAWN"] = 1077] = "CASTS_WHEN_DRAWN";
    GameTag[GameTag["DISPLAY_CARD_ON_MOUSEOVER"] = 1078] = "DISPLAY_CARD_ON_MOUSEOVER";
    GameTag[GameTag["DECK_POWER_UP"] = 1080] = "DECK_POWER_UP";
    GameTag[GameTag["SIDEKICK"] = 1081] = "SIDEKICK";
    GameTag[GameTag["SIDEKICK_HERO_POWER"] = 1082] = "SIDEKICK_HERO_POWER";
    GameTag[GameTag["REBORN"] = 1085] = "REBORN";
    GameTag[GameTag["UPGRADED_HERO_POWER"] = 1086] = "UPGRADED_HERO_POWER";
    GameTag[GameTag["SQUELCH_NON_GAME_TRIGGERS_AND_MODIFIERS"] = 1087] = "SQUELCH_NON_GAME_TRIGGERS_AND_MODIFIERS";
    GameTag[GameTag["QUEST_REWARD_DATABASE_ID"] = 1089] = "QUEST_REWARD_DATABASE_ID";
    GameTag[GameTag["DORMANT_VISUAL"] = 1090] = "DORMANT_VISUAL";
    GameTag[GameTag["CUSTOMTEXT1"] = 1093] = "CUSTOMTEXT1";
    GameTag[GameTag["CUSTOMTEXT2"] = 1094] = "CUSTOMTEXT2";
    GameTag[GameTag["CUSTOMTEXT3"] = 1095] = "CUSTOMTEXT3";
    GameTag[GameTag["FLOOPY"] = 1097] = "FLOOPY";
    GameTag[GameTag["PLAYER_BASE_SHRINE_DECK_ID"] = 1099] = "PLAYER_BASE_SHRINE_DECK_ID";
    GameTag[GameTag["HIDE_WATERMARK"] = 1107] = "HIDE_WATERMARK";
    GameTag[GameTag["EXTRA_MINION_BATTLECRIES_BASE"] = 1112] = "EXTRA_MINION_BATTLECRIES_BASE";
    GameTag[GameTag["RUN_PROGRESS"] = 1113] = "RUN_PROGRESS";
    GameTag[GameTag["NON_KEYWORD_ECHO"] = 1114] = "NON_KEYWORD_ECHO";
    GameTag[GameTag["PLAYER_TAG_THRESHOLD_TAG_ID"] = 1115] = "PLAYER_TAG_THRESHOLD_TAG_ID";
    GameTag[GameTag["PLAYER_TAG_THRESHOLD_VALUE"] = 1116] = "PLAYER_TAG_THRESHOLD_VALUE";
    GameTag[GameTag["HEALING_DOES_DAMAGE_HINT"] = 1117] = "HEALING_DOES_DAMAGE_HINT";
    GameTag[GameTag["AFFECTED_BY_HEALING_DOES_DAMAGE"] = 1118] = "AFFECTED_BY_HEALING_DOES_DAMAGE";
    GameTag[GameTag["DECK_LIST_SORT_ORDER"] = 1125] = "DECK_LIST_SORT_ORDER";
    GameTag[GameTag["EXTRA_BATTLECRIES_ADDITIONAL"] = 1126] = "EXTRA_BATTLECRIES_ADDITIONAL";
    GameTag[GameTag["EXTRA_DEATHRATTLES_ADDITIONAL"] = 1131] = "EXTRA_DEATHRATTLES_ADDITIONAL";
    GameTag[GameTag["ALTERNATE_MOUSE_OVER_CARD"] = 1132] = "ALTERNATE_MOUSE_OVER_CARD";
    GameTag[GameTag["ENCHANTMENT_BANNER_TEXT"] = 1135] = "ENCHANTMENT_BANNER_TEXT";
    GameTag[GameTag["MOUSE_OVER_CARD_APPEARANCE"] = 1142] = "MOUSE_OVER_CARD_APPEARANCE";
    GameTag[GameTag["IS_ADVENTURE_SCENARIO"] = 1172] = "IS_ADVENTURE_SCENARIO";
    GameTag[GameTag["DEPRECATED_TWINSPELL_COPY"] = 1186] = "DEPRECATED_TWINSPELL_COPY";
    GameTag[GameTag["PROXY_GALAKROND"] = 1190] = "PROXY_GALAKROND";
    GameTag[GameTag["SIDE_QUEST"] = 1192] = "SIDE_QUEST";
    GameTag[GameTag["TWINSPELL"] = 1193] = "TWINSPELL";
    GameTag[GameTag["GALAKROND_IN_PLAY"] = 1194] = "GALAKROND_IN_PLAY";
    GameTag[GameTag["COIN_MANA_GEM"] = 1199] = "COIN_MANA_GEM";
    GameTag[GameTag["NOT_A_SPELL_CAST"] = 1200] = "NOT_A_SPELL_CAST";
    GameTag[GameTag["MEGA_WINDFURY"] = 1207] = "MEGA_WINDFURY";
    GameTag[GameTag["ELUSIVE"] = 1211] = "ELUSIVE";
    GameTag[GameTag["EMPOWER"] = 1263] = "EMPOWER";
    GameTag[GameTag["EMPOWER_PRIEST"] = 1264] = "EMPOWER_PRIEST";
    GameTag[GameTag["EMPOWER_ROGUE"] = 1265] = "EMPOWER_ROGUE";
    GameTag[GameTag["EMPOWER_SHAMAN"] = 1266] = "EMPOWER_SHAMAN";
    GameTag[GameTag["EMPOWER_WARLOCK"] = 1267] = "EMPOWER_WARLOCK";
    GameTag[GameTag["EMPOWER_WARRIOR"] = 1268] = "EMPOWER_WARRIOR";
    GameTag[GameTag["TWINSPELLPENDING"] = 1269] = "TWINSPELLPENDING";
    GameTag[GameTag["DRUSTVAR_HORROR_DEBUG_CURRENT_SPELL_DATABASE_ID"] = 1280] = "DRUSTVAR_HORROR_DEBUG_CURRENT_SPELL_DATABASE_ID";
    GameTag[GameTag["DRUSTVAR_HORROR_DEBUG_CURRENT_ITERATION"] = 1281] = "DRUSTVAR_HORROR_DEBUG_CURRENT_ITERATION";
    GameTag[GameTag["HEROIC_HERO_POWER"] = 1282] = "HEROIC_HERO_POWER";
    GameTag[GameTag["DRUSTVAR_HORROR_DEBUG_MAX_ITERATIONS"] = 1283] = "DRUSTVAR_HORROR_DEBUG_MAX_ITERATIONS";
    GameTag[GameTag["CREATOR_DBID"] = 1284] = "CREATOR_DBID";
    GameTag[GameTag["FATIGUE_REFERENCE"] = 1290] = "FATIGUE_REFERENCE";
    GameTag[GameTag["HERO_FLYING"] = 1293] = "HERO_FLYING";
    GameTag[GameTag["UI_BUFF_HEALTH_UP"] = 1294] = "UI_BUFF_HEALTH_UP";
    GameTag[GameTag["UI_BUFF_SET_COST_ZERO"] = 1295] = "UI_BUFF_SET_COST_ZERO";
    GameTag[GameTag["UI_BUFF_COST_DOWN"] = 1296] = "UI_BUFF_COST_DOWN";
    GameTag[GameTag["UI_BUFF_ATK_UP"] = 1297] = "UI_BUFF_ATK_UP";
    GameTag[GameTag["UI_BUFF_COST_UP"] = 1298] = "UI_BUFF_COST_UP";
    GameTag[GameTag["DEBUG_DISPLAY_TAG_BOTTOM_RIGHT"] = 1313] = "DEBUG_DISPLAY_TAG_BOTTOM_RIGHT";
    GameTag[GameTag["DEBUG_DISPLAY_TAG_TOP_RIGHT"] = 1314] = "DEBUG_DISPLAY_TAG_TOP_RIGHT";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_ENTITY_1"] = 1318] = "SMART_DISCOVER_DEBUG_ENTITY_1";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_ENTITY_2"] = 1319] = "SMART_DISCOVER_DEBUG_ENTITY_2";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_ENTITY_3"] = 1320] = "SMART_DISCOVER_DEBUG_ENTITY_3";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_TEST_COMPLETE"] = 1324] = "SMART_DISCOVER_DEBUG_TEST_COMPLETE";
    GameTag[GameTag["COPIED_BY_KHADGAR"] = 1326] = "COPIED_BY_KHADGAR";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_PASSIVE_EVAL_RESULT_1"] = 1328] = "SMART_DISCOVER_DEBUG_PASSIVE_EVAL_RESULT_1";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_PASSIVE_EVAL_RESULT_2"] = 1329] = "SMART_DISCOVER_DEBUG_PASSIVE_EVAL_RESULT_2";
    GameTag[GameTag["SMART_DISCOVER_DEBUG_PASSIVE_EVAL_RESULT_3"] = 1330] = "SMART_DISCOVER_DEBUG_PASSIVE_EVAL_RESULT_3";
    GameTag[GameTag["OUTCAST"] = 1333] = "OUTCAST";
    GameTag[GameTag["ALTERNATE_CHAPTER_VO"] = 1334] = "ALTERNATE_CHAPTER_VO";
    GameTag[GameTag["AI_MAKES_DECISIONS_FOR_PLAYER"] = 1335] = "AI_MAKES_DECISIONS_FOR_PLAYER";
    GameTag[GameTag["HAS_BEEN_REBORN"] = 1336] = "HAS_BEEN_REBORN";
    GameTag[GameTag["USE_DISCOVER_VISUALS"] = 1342] = "USE_DISCOVER_VISUALS";
    GameTag[GameTag["DOUBLE_FATIGUE_DAMAGE"] = 1346] = "DOUBLE_FATIGUE_DAMAGE";
    GameTag[GameTag["BOARD_VISUAL_STATE"] = 1347] = "BOARD_VISUAL_STATE";
    GameTag[GameTag["BACON_DUMMY_PLAYER"] = 1349] = "BACON_DUMMY_PLAYER";
    GameTag[GameTag["SQUELCH_LIFETIME_EFFECTS"] = 1350] = "SQUELCH_LIFETIME_EFFECTS";
    GameTag[GameTag["ALLOW_MOVE_MINION"] = 1356] = "ALLOW_MOVE_MINION";
    GameTag[GameTag["TAG_TB_RANDOM_DECK_TIME_ID"] = 1358] = "TAG_TB_RANDOM_DECK_TIME_ID";
    GameTag[GameTag["NEXT_OPPONENT_PLAYER_ID"] = 1360] = "NEXT_OPPONENT_PLAYER_ID";
    GameTag[GameTag["MAIN_GALAKROND"] = 1361] = "MAIN_GALAKROND";
    GameTag[GameTag["GOOD_OL_GENERIC_FRIENDLY_DRAGON_DISCOVER_VISUALS"] = 1364] = "GOOD_OL_GENERIC_FRIENDLY_DRAGON_DISCOVER_VISUALS";
    GameTag[GameTag["GALAKROND_HERO_CARD"] = 1365] = "GALAKROND_HERO_CARD";
    GameTag[GameTag["INVOKE_COUNTER"] = 1366] = "INVOKE_COUNTER";
    GameTag[GameTag["PLAYER_LEADERBOARD_PLACE"] = 1373] = "PLAYER_LEADERBOARD_PLACE";
    GameTag[GameTag["PLAYER_TECH_LEVEL"] = 1377] = "PLAYER_TECH_LEVEL";
    GameTag[GameTag["BACON_HERO_POWER_ACTIVATED"] = 1398] = "BACON_HERO_POWER_ACTIVATED";
    GameTag[GameTag["USE_FAST_ACTOR_TRANSITION_ANIMATIONS"] = 1402] = "USE_FAST_ACTOR_TRANSITION_ANIMATIONS";
    GameTag[GameTag["STUDY"] = 1414] = "STUDY";
    GameTag[GameTag["BACON_ODD_PLAYER_OUT"] = 1415] = "BACON_ODD_PLAYER_OUT";
    GameTag[GameTag["BACON_WON_LAST_COMBAT"] = 1422] = "BACON_WON_LAST_COMBAT";
    GameTag[GameTag["BACON_IS_KEL_THUZAD"] = 1423] = "BACON_IS_KEL_THUZAD";
    GameTag[GameTag["HIGHLIGHT_ATTACKING_MINION_DURING_COMBAT"] = 1424] = "HIGHLIGHT_ATTACKING_MINION_DURING_COMBAT";
    GameTag[GameTag["SPELLBURST"] = 1427] = "SPELLBURST";
    GameTag[GameTag["BATTLEGROUNDS_PREMIUM_DBF_ID"] = 1429] = "BATTLEGROUNDS_PREMIUM_DBF_ID";
    GameTag[GameTag["RULEBOOK"] = 1430] = "RULEBOOK";
    GameTag[GameTag["FX_DATANUM_1"] = 1436] = "FX_DATANUM_1";
    GameTag[GameTag["BACON_ACTION_CARD"] = 1437] = "BACON_ACTION_CARD";
    GameTag[GameTag["GAME_MODE_BUTTON_SLOT"] = 1438] = "GAME_MODE_BUTTON_SLOT";
    GameTag[GameTag["TECH_LEVEL"] = 1440] = "TECH_LEVEL";
    GameTag[GameTag["TECH_LEVEL_MANA_GEM"] = 1442] = "TECH_LEVEL_MANA_GEM";
    GameTag[GameTag["UI_BUFF_DURABILITY_UP"] = 1443] = "UI_BUFF_DURABILITY_UP";
    GameTag[GameTag["PLAYER_TRIPLES"] = 1447] = "PLAYER_TRIPLES";
    GameTag[GameTag["DISABLE_TURN_INDICATORS"] = 1448] = "DISABLE_TURN_INDICATORS";
    GameTag[GameTag["COLLECTION_RELATED_CARD_DATABASE_ID"] = 1452] = "COLLECTION_RELATED_CARD_DATABASE_ID";
    GameTag[GameTag["SQUELCH_SUMMON_EFFECTS"] = 1453] = "SQUELCH_SUMMON_EFFECTS";
    GameTag[GameTag["IS_BACON_POOL_MINION"] = 1456] = "IS_BACON_POOL_MINION";
    GameTag[GameTag["SUPPRESS_ALL_SUMMON_VO"] = 1458] = "SUPPRESS_ALL_SUMMON_VO";
    GameTag[GameTag["BACON_TRIPLE_CANDIDATE"] = 1460] = "BACON_TRIPLE_CANDIDATE";
    GameTag[GameTag["BATTLEGROUNDS_PREMIUM_EMOTES"] = 1463] = "BATTLEGROUNDS_PREMIUM_EMOTES";
    GameTag[GameTag["MOVE_MINION_HOVER_TARGET_SLOT"] = 1464] = "MOVE_MINION_HOVER_TARGET_SLOT";
    GameTag[GameTag["BACON_COIN_ON_ENEMY_MINIONS"] = 1467] = "BACON_COIN_ON_ENEMY_MINIONS";
    GameTag[GameTag["BACON_TRIPLED_BASE_MINION_ID"] = 1471] = "BACON_TRIPLED_BASE_MINION_ID";
    GameTag[GameTag["ALWAYS_USE_FAST_ACTOR_TRIGGERS"] = 1473] = "ALWAYS_USE_FAST_ACTOR_TRIGGERS";
    GameTag[GameTag["ENCHANTMENT_AS_ENTITY"] = 1489] = "ENCHANTMENT_AS_ENTITY";
    GameTag[GameTag["BACON_HERO_CAN_BE_DRAFTED"] = 1491] = "BACON_HERO_CAN_BE_DRAFTED";
    GameTag[GameTag["TRANSIENT_ENTITY"] = 1493] = "TRANSIENT_ENTITY";
    GameTag[GameTag["BACON_MAX_PLAYER_TECH_LEVEL"] = 1494] = "BACON_MAX_PLAYER_TECH_LEVEL";
    GameTag[GameTag["CAN_TARGET_CARDS_IN_HAND"] = 1508] = "CAN_TARGET_CARDS_IN_HAND";
    GameTag[GameTag["DISABLE_NONHERO_GOLDEN_ANIMATIONS"] = 1514] = "DISABLE_NONHERO_GOLDEN_ANIMATIONS";
    GameTag[GameTag["WATERMARK_OVERRIDE_CARD_SET"] = 1517] = "WATERMARK_OVERRIDE_CARD_SET";
    GameTag[GameTag["DORMANT"] = 1518] = "DORMANT";
    GameTag[GameTag["DORMANT_AWAKEN_CONDITION_ENCHANT"] = 1519] = "DORMANT_AWAKEN_CONDITION_ENCHANT";
    GameTag[GameTag["SUPPRESS_SUMMON_VO_FOR_PLAYER"] = 1521] = "SUPPRESS_SUMMON_VO_FOR_PLAYER";
    GameTag[GameTag["CORRUPT"] = 1524] = "CORRUPT";
    GameTag[GameTag["ALLOW_GAME_SPEEDUP"] = 1526] = "ALLOW_GAME_SPEEDUP";
    GameTag[GameTag["POISONOUS_INSTANT"] = 1528] = "POISONOUS_INSTANT";
    GameTag[GameTag["FORCE_NO_CUSTOM_SPELLS"] = 1529] = "FORCE_NO_CUSTOM_SPELLS";
    GameTag[GameTag["START_OF_COMBAT"] = 1531] = "START_OF_COMBAT";
    GameTag[GameTag["DONT_PICK_FROM_RANDOM_SUBSETS"] = 1544] = "DONT_PICK_FROM_RANDOM_SUBSETS";
    GameTag[GameTag["LIBRAM"] = 1546] = "LIBRAM";
    GameTag[GameTag["CORRUPTED_CARD"] = 1551] = "CORRUPTED_CARD";
    GameTag[GameTag["BACON_HERO_EARLY_ACCESS"] = 1554] = "BACON_HERO_EARLY_ACCESS";
    GameTag[GameTag["SPAWN_TIME_COUNT"] = 1556] = "SPAWN_TIME_COUNT";
    GameTag[GameTag["SKIP_MULLIGAN"] = 1561] = "SKIP_MULLIGAN";
    GameTag[GameTag["COPIED_FROM_ENTITY_ID"] = 1565] = "COPIED_FROM_ENTITY_ID";
    GameTag[GameTag["BACON_SELL_VALUE"] = 1587] = "BACON_SELL_VALUE";
    GameTag[GameTag["BACON_ONLY_IN_DRAGON_LOBBY"] = 1591] = "BACON_ONLY_IN_DRAGON_LOBBY";
    GameTag[GameTag["BACON_ONLY_IN_MURLOC_LOBBY"] = 1592] = "BACON_ONLY_IN_MURLOC_LOBBY";
    GameTag[GameTag["BACON_ONLY_IN_DEMON_LOBBY"] = 1593] = "BACON_ONLY_IN_DEMON_LOBBY";
    GameTag[GameTag["BACON_ONLY_IN_BEAST_LOBBY"] = 1594] = "BACON_ONLY_IN_BEAST_LOBBY";
    GameTag[GameTag["BACON_ONLY_IN_MECH_LOBBY"] = 1595] = "BACON_ONLY_IN_MECH_LOBBY";
    GameTag[GameTag["BACON_ONLY_IN_PIRATE_LOBBY"] = 1596] = "BACON_ONLY_IN_PIRATE_LOBBY";
    GameTag[GameTag["BACON_VERDANTSPHERES"] = 1598] = "BACON_VERDANTSPHERES";
    GameTag[GameTag["OPPONENT_SIDE_GHOSTLY"] = 1609] = "OPPONENT_SIDE_GHOSTLY";
    GameTag[GameTag["FORCE_NO_CUSTOM_LIFETIME_SPELLS"] = 1613] = "FORCE_NO_CUSTOM_LIFETIME_SPELLS";
    GameTag[GameTag["FORCE_NO_CUSTOM_SUMMON_SPELLS"] = 1614] = "FORCE_NO_CUSTOM_SUMMON_SPELLS";
    GameTag[GameTag["FORCE_NO_CUSTOM_KEYWORD_SPELLS"] = 1615] = "FORCE_NO_CUSTOM_KEYWORD_SPELLS";
    GameTag[GameTag["USE_LEADERBOARD_AS_SPAWN_ORIGIN"] = 1628] = "USE_LEADERBOARD_AS_SPAWN_ORIGIN";
    GameTag[GameTag["BACON_MUKLA_BANANA_SPAWN_COUNT"] = 1629] = "BACON_MUKLA_BANANA_SPAWN_COUNT";
    GameTag[GameTag["REPLACEMENT_ENTITY"] = 1632] = "REPLACEMENT_ENTITY";
    GameTag[GameTag["SPELL_SCHOOL"] = 1635] = "SPELL_SCHOOL";
    GameTag[GameTag["FRENZY"] = 1637] = "FRENZY";
    GameTag[GameTag["COIN_MANA_GEM_FOR_CHOICE_CARDS"] = 1643] = "COIN_MANA_GEM_FOR_CHOICE_CARDS";
    GameTag[GameTag["METAMORPHOSIS"] = 1644] = "METAMORPHOSIS";
    GameTag[GameTag["HERO_POWER_ENTITY"] = 1646] = "HERO_POWER_ENTITY";
    GameTag[GameTag["BACON_PLAYER_RESULTS_HERO_OVERRIDE"] = 1649] = "BACON_PLAYER_RESULTS_HERO_OVERRIDE";
    GameTag[GameTag["DISCOVER_STUDIES_VISUAL"] = 1650] = "DISCOVER_STUDIES_VISUAL";
    GameTag[GameTag["LETTUCE_CONTROLLER"] = 1653] = "LETTUCE_CONTROLLER";
    GameTag[GameTag["LETTUCE_ABILITY_OWNER"] = 1654] = "LETTUCE_ABILITY_OWNER";
    GameTag[GameTag["LETTUCE_SELECTED_TARGET"] = 1657] = "LETTUCE_SELECTED_TARGET";
    GameTag[GameTag["LETTUCE_SELECTED_SUBCARD_INDEX"] = 1661] = "LETTUCE_SELECTED_SUBCARD_INDEX";
    GameTag[GameTag["LETTUCE_MERCENARY"] = 1665] = "LETTUCE_MERCENARY";
    GameTag[GameTag["LETTUCE_ROLE"] = 1666] = "LETTUCE_ROLE";
    GameTag[GameTag["LETTUCE_IS_COMBAT_ACTION_TAKEN"] = 1668] = "LETTUCE_IS_COMBAT_ACTION_TAKEN";
    GameTag[GameTag["LETTUCE_COOLDOWN_CONFIG"] = 1669] = "LETTUCE_COOLDOWN_CONFIG";
    GameTag[GameTag["LETTUCE_CURRENT_COOLDOWN"] = 1670] = "LETTUCE_CURRENT_COOLDOWN";
    GameTag[GameTag["LETTUCE_PASSIVE_ABILITY"] = 1671] = "LETTUCE_PASSIVE_ABILITY";
    GameTag[GameTag["LIFESTEAL_DAMAGES_OPPOSING_HERO"] = 1675] = "LIFESTEAL_DAMAGES_OPPOSING_HERO";
    GameTag[GameTag["LETTUCE_ABILITY_SUMMONED_MINION"] = 1676] = "LETTUCE_ABILITY_SUMMONED_MINION";
    GameTag[GameTag["SI_7"] = 1678] = "SI_7";
    GameTag[GameTag["LETTUCE_ABILITY"] = 1679] = "LETTUCE_ABILITY";
    GameTag[GameTag["SPELLS_CAST_TWICE"] = 1681] = "SPELLS_CAST_TWICE";
    GameTag[GameTag["CHOICE_NAME_DISPLAY_TYPE"] = 1687] = "CHOICE_NAME_DISPLAY_TYPE";
    GameTag[GameTag["BACON_ONLY_IN_ELEMENTAL_LOBBY"] = 1688] = "BACON_ONLY_IN_ELEMENTAL_LOBBY";
    GameTag[GameTag["CHOICE_ACTOR_TYPE"] = 1692] = "CHOICE_ACTOR_TYPE";
    GameTag[GameTag["FORCE_GREEN_GLOW_ACTIVE"] = 1693] = "FORCE_GREEN_GLOW_ACTIVE";
    GameTag[GameTag["SOURCE_OVERRIDE_FOR_MODIFIER_TEXT"] = 1694] = "SOURCE_OVERRIDE_FOR_MODIFIER_TEXT";
    GameTag[GameTag["LETTUCE_ABILITY_TILE_VISUAL_SELF_ONLY"] = 1697] = "LETTUCE_ABILITY_TILE_VISUAL_SELF_ONLY";
    GameTag[GameTag["LETTUCE_ABILITY_TILE_VISUAL_ALL_VISIBLE"] = 1698] = "LETTUCE_ABILITY_TILE_VISUAL_ALL_VISIBLE";
    GameTag[GameTag["ACTION_STEP_TYPE"] = 1700] = "ACTION_STEP_TYPE";
    GameTag[GameTag["FAKE_ZONE"] = 1702] = "FAKE_ZONE";
    GameTag[GameTag["FAKE_ZONE_POSITION"] = 1703] = "FAKE_ZONE_POSITION";
    GameTag[GameTag["LETTUCE_MAX_IN_PLAY_MERCENARIES"] = 1704] = "LETTUCE_MAX_IN_PLAY_MERCENARIES";
    GameTag[GameTag["LETTUCE_MERCENARIES_TO_NOMINATE"] = 1705] = "LETTUCE_MERCENARIES_TO_NOMINATE";
    GameTag[GameTag["LETTUCE_COOLDOWN_WHILE_BENCHED"] = 1708] = "LETTUCE_COOLDOWN_WHILE_BENCHED";
    GameTag[GameTag["LETTUCE_COMBAT_FROM_HIGH_TO_LOW"] = 1712] = "LETTUCE_COMBAT_FROM_HIGH_TO_LOW";
    GameTag[GameTag["PENDING_TRANSFORM_TO_CARD"] = 1716] = "PENDING_TRANSFORM_TO_CARD";
    GameTag[GameTag["TRANSFORMED_FROM_CARD_VISUAL_TYPE"] = 1719] = "TRANSFORMED_FROM_CARD_VISUAL_TYPE";
    GameTag[GameTag["TRADEABLE"] = 1720] = "TRADEABLE";
    GameTag[GameTag["TOOL"] = 1722] = "TOOL";
    GameTag[GameTag["TRIGGER_WHEREVER_THIS_IS"] = 1724] = "TRIGGER_WHEREVER_THIS_IS";
    GameTag[GameTag["QUESTLINE"] = 1725] = "QUESTLINE";
    GameTag[GameTag["LETTUCE_MERCENARY_RESERVE"] = 1731] = "LETTUCE_MERCENARY_RESERVE";
    GameTag[GameTag["LETTUCE_SKIP_MERCENARY_RESERVE"] = 1732] = "LETTUCE_SKIP_MERCENARY_RESERVE";
    GameTag[GameTag["IS_DARKMOON_PRIZE"] = 1735] = "IS_DARKMOON_PRIZE";
    GameTag[GameTag["PLAYER_ID_LOOKUP"] = 1740] = "PLAYER_ID_LOOKUP";
    GameTag[GameTag["DECK_ACTION_COST"] = 1743] = "DECK_ACTION_COST";
    GameTag[GameTag["BACON_AVALANCHE"] = 1744] = "BACON_AVALANCHE";
    GameTag[GameTag["SIGIL"] = 1749] = "SIGIL";
    GameTag[GameTag["LETTUCE_DISABLE_AUTO_SELECT_NEXT_MERC"] = 1753] = "LETTUCE_DISABLE_AUTO_SELECT_NEXT_MERC";
    GameTag[GameTag["PLAYED_CTHUN_EYE"] = 1764] = "PLAYED_CTHUN_EYE";
    GameTag[GameTag["PLAYED_CTHUN_BODY"] = 1765] = "PLAYED_CTHUN_BODY";
    GameTag[GameTag["PLAYED_CTHUN_MAW"] = 1766] = "PLAYED_CTHUN_MAW";
    GameTag[GameTag["PLAYED_CTHUN_HEART"] = 1767] = "PLAYED_CTHUN_HEART";
    GameTag[GameTag["PROXY_CTHUN_SHATTERED"] = 1768] = "PROXY_CTHUN_SHATTERED";
    GameTag[GameTag["PROGRESSBAR_TOTAL"] = 1769] = "PROGRESSBAR_TOTAL";
    GameTag[GameTag["PROGRESSBAR_PROGRESS"] = 1770] = "PROGRESSBAR_PROGRESS";
    GameTag[GameTag["PROGRESSBAR_CARDID"] = 1771] = "PROGRESSBAR_CARDID";
    GameTag[GameTag["PROGRESSBAR_SHOW"] = 1772] = "PROGRESSBAR_SHOW";
    GameTag[GameTag["PROGRESSBAR_TEXT"] = 1773] = "PROGRESSBAR_TEXT";
    GameTag[GameTag["LIFESTEAL_DOES_DAMAGE_HINT"] = 1774] = "LIFESTEAL_DOES_DAMAGE_HINT";
    GameTag[GameTag["DARKMOON_TICKET"] = 1776] = "DARKMOON_TICKET";
    GameTag[GameTag["NUM_SPELLS_PLAYED_THIS_GAME"] = 1780] = "NUM_SPELLS_PLAYED_THIS_GAME";
    GameTag[GameTag["BACON_COMEONECOMEALL"] = 1789] = "BACON_COMEONECOMEALL";
    GameTag[GameTag["LETTUCE_ABILITY_USED_LAST_TURN"] = 1807] = "LETTUCE_ABILITY_USED_LAST_TURN";
    GameTag[GameTag["LETTUCE_NODE_TYPE"] = 1808] = "LETTUCE_NODE_TYPE";
    GameTag[GameTag["SHOW_DISCOVER_FROM_DECK"] = 1816] = "SHOW_DISCOVER_FROM_DECK";
    GameTag[GameTag["MINI_SET"] = 1824] = "MINI_SET";
    GameTag[GameTag["ARMOR_GAINED_THIS_GAME"] = 1828] = "ARMOR_GAINED_THIS_GAME";
    GameTag[GameTag["CANT_TRIGGER_DEATHRATTLE"] = 1831] = "CANT_TRIGGER_DEATHRATTLE";
    GameTag[GameTag["BACON_BLOODGEMBUFFATKVALUE"] = 1844] = "BACON_BLOODGEMBUFFATKVALUE";
    GameTag[GameTag["BACON_ONLY_IN_QUILBOAR_LOBBY"] = 1845] = "BACON_ONLY_IN_QUILBOAR_LOBBY";
    GameTag[GameTag["CANT_MOVE_MINION"] = 1848] = "CANT_MOVE_MINION";
    GameTag[GameTag["LETTUCE_MERCENARY_EXPERIENCE"] = 1852] = "LETTUCE_MERCENARY_EXPERIENCE";
    GameTag[GameTag["LETTUCE_IS_EQUPIMENT"] = 1855] = "LETTUCE_IS_EQUPIMENT";
    GameTag[GameTag["LETTUCE_EQUIPMENT_ID"] = 1856] = "LETTUCE_EQUIPMENT_ID";
    GameTag[GameTag["DARKMOON_FAIRE_PRIZES_ACTIVE"] = 1895] = "DARKMOON_FAIRE_PRIZES_ACTIVE";
    GameTag[GameTag["IGNORE_DECK_RULESET"] = 1896] = "IGNORE_DECK_RULESET";
    GameTag[GameTag["HONORABLE_KILL"] = 1920] = "HONORABLE_KILL";
    GameTag[GameTag["HAS_DIAMOND_QUALITY"] = 1932] = "HAS_DIAMOND_QUALITY";
    GameTag[GameTag["CURRENT_SPELLPOWER_ARCANE"] = 1936] = "CURRENT_SPELLPOWER_ARCANE";
    GameTag[GameTag["CURRENT_SPELLPOWER_FIRE"] = 1937] = "CURRENT_SPELLPOWER_FIRE";
    GameTag[GameTag["CURRENT_SPELLPOWER_FROST"] = 1938] = "CURRENT_SPELLPOWER_FROST";
    GameTag[GameTag["CURRENT_SPELLPOWER_NATURE"] = 1939] = "CURRENT_SPELLPOWER_NATURE";
    GameTag[GameTag["CURRENT_SPELLPOWER_HOLY"] = 1940] = "CURRENT_SPELLPOWER_HOLY";
    GameTag[GameTag["CURRENT_SPELLPOWER_SHADOW"] = 1941] = "CURRENT_SPELLPOWER_SHADOW";
    GameTag[GameTag["CURRENT_SPELLPOWER_FEL"] = 1942] = "CURRENT_SPELLPOWER_FEL";
    GameTag[GameTag["CURRENT_SPELLPOWER_PHYSICAL"] = 1943] = "CURRENT_SPELLPOWER_PHYSICAL";
    GameTag[GameTag["NON_KEYWORD_POISONOUS"] = 1944] = "NON_KEYWORD_POISONOUS";
    GameTag[GameTag["SPELLPOWER_ARCANE"] = 1945] = "SPELLPOWER_ARCANE";
    GameTag[GameTag["SPELLPOWER_FIRE"] = 1946] = "SPELLPOWER_FIRE";
    GameTag[GameTag["SPELLPOWER_FROST"] = 1947] = "SPELLPOWER_FROST";
    GameTag[GameTag["SPELLPOWER_NATURE"] = 1948] = "SPELLPOWER_NATURE";
    GameTag[GameTag["SPELLPOWER_HOLY"] = 1949] = "SPELLPOWER_HOLY";
    GameTag[GameTag["SPELLPOWER_SHADOW"] = 1950] = "SPELLPOWER_SHADOW";
    GameTag[GameTag["SPELLPOWER_FEL"] = 1951] = "SPELLPOWER_FEL";
    GameTag[GameTag["SPELLPOWER_PHYSICAL"] = 1952] = "SPELLPOWER_PHYSICAL";
    GameTag[GameTag["ENRAGE_TOOLTIP"] = 1954] = "ENRAGE_TOOLTIP";
    GameTag[GameTag["IMP"] = 1965] = "IMP";
    GameTag[GameTag["BACON_BLOOD_GEM_TOOLTIP"] = 1966] = "BACON_BLOOD_GEM_TOOLTIP";
    GameTag[GameTag["LETTUCE_HAS_MANUALLY_SELECTED_ABILITY"] = 1967] = "LETTUCE_HAS_MANUALLY_SELECTED_ABILITY";
    GameTag[GameTag["LETTUCE_KEEP_LAST_STANDING_MINION_ACTOR"] = 1976] = "LETTUCE_KEEP_LAST_STANDING_MINION_ACTOR";
    GameTag[GameTag["GOLDSPARKLES_HINT"] = 1984] = "GOLDSPARKLES_HINT";
    GameTag[GameTag["LETTUCE_USE_DETERMINISTIC_TEAM_ABILITY_QUEUING"] = 1990] = "LETTUCE_USE_DETERMINISTIC_TEAM_ABILITY_QUEUING";
    GameTag[GameTag["LETTUCE_SELECTED_ABILITY_QUEUE_ORDER"] = 1991] = "LETTUCE_SELECTED_ABILITY_QUEUE_ORDER";
    GameTag[GameTag["QUESTLINE_FINAL_REWARD_DATABASE_ID"] = 1992] = "QUESTLINE_FINAL_REWARD_DATABASE_ID";
    GameTag[GameTag["QUESTLINE_PART"] = 1993] = "QUESTLINE_PART";
    GameTag[GameTag["QUESTLINE_REQUIREMENT_MET_1"] = 1994] = "QUESTLINE_REQUIREMENT_MET_1";
    GameTag[GameTag["QUESTLINE_REQUIREMENT_MET_2"] = 1995] = "QUESTLINE_REQUIREMENT_MET_2";
    GameTag[GameTag["QUESTLINE_REQUIREMENT_MET_3"] = 1996] = "QUESTLINE_REQUIREMENT_MET_3";
    GameTag[GameTag["DONT_SHOW_IN_HISTORY"] = 2015] = "DONT_SHOW_IN_HISTORY";
    GameTag[GameTag["MAX_SLOTS_PER_PLAYER_OVERRIDE"] = 2017] = "MAX_SLOTS_PER_PLAYER_OVERRIDE";
    GameTag[GameTag["BG_BATTLE_STARTING"] = 2022] = "BG_BATTLE_STARTING";
    GameTag[GameTag["STATS_GAINS_GET_DOUBLED"] = 2028] = "STATS_GAINS_GET_DOUBLED";
    GameTag[GameTag["FAKE_CONTROLLER"] = 2032] = "FAKE_CONTROLLER";
    GameTag[GameTag["BACON_SKIN"] = 2038] = "BACON_SKIN";
    GameTag[GameTag["BACON_SKIN_PARENT_ID"] = 2039] = "BACON_SKIN_PARENT_ID";
    GameTag[GameTag["GAME_SEED"] = 2042] = "GAME_SEED";
    GameTag[GameTag["IS_USING_TRADE_OPTION"] = 2045] = "IS_USING_TRADE_OPTION";
    GameTag[GameTag["BACON_BOB_SKIN"] = 2049] = "BACON_BOB_SKIN";
    GameTag[GameTag["BACON_BARTENDER_CARD_ID"] = 2053] = "BACON_BARTENDER_CARD_ID";
    GameTag[GameTag["COIN_CARD"] = 2088] = "COIN_CARD";
    GameTag[GameTag["BACON_COMBAT_DAMAGE_CAP"] = 2089] = "BACON_COMBAT_DAMAGE_CAP";
    GameTag[GameTag["BACON_REFRESH_TOOLTIP"] = 2104] = "BACON_REFRESH_TOOLTIP";
    GameTag[GameTag["TARGETING_ARROW_TYPE"] = 2108] = "TARGETING_ARROW_TYPE";
    GameTag[GameTag["LETTUCE_CURRENT_BOUNTY_ID"] = 2120] = "LETTUCE_CURRENT_BOUNTY_ID";
    GameTag[GameTag["LETTUCE_OVERTIME"] = 2123] = "LETTUCE_OVERTIME";
    GameTag[GameTag["AVENGE"] = 2129] = "AVENGE";
    GameTag[GameTag["BACON_COMPANION_ID"] = 2130] = "BACON_COMPANION_ID";
    GameTag[GameTag["SPELL_RESISTANCE_ARCANE"] = 2138] = "SPELL_RESISTANCE_ARCANE";
    GameTag[GameTag["SPELL_RESISTANCE_FIRE"] = 2139] = "SPELL_RESISTANCE_FIRE";
    GameTag[GameTag["SPELL_RESISTANCE_FROST"] = 2140] = "SPELL_RESISTANCE_FROST";
    GameTag[GameTag["SPELL_RESISTANCE_NATURE"] = 2141] = "SPELL_RESISTANCE_NATURE";
    GameTag[GameTag["SPELL_RESISTANCE_HOLY"] = 2142] = "SPELL_RESISTANCE_HOLY";
    GameTag[GameTag["SPELL_RESISTANCE_SHADOW"] = 2143] = "SPELL_RESISTANCE_SHADOW";
    GameTag[GameTag["SPELL_RESISTANCE_FEL"] = 2144] = "SPELL_RESISTANCE_FEL";
    GameTag[GameTag["SPELL_WEAKNESS_ARCANE"] = 2145] = "SPELL_WEAKNESS_ARCANE";
    GameTag[GameTag["SPELL_WEAKNESS_FIRE"] = 2146] = "SPELL_WEAKNESS_FIRE";
    GameTag[GameTag["SPELL_WEAKNESS_FROST"] = 2147] = "SPELL_WEAKNESS_FROST";
    GameTag[GameTag["SPELL_WEAKNESS_NATURE"] = 2148] = "SPELL_WEAKNESS_NATURE";
    GameTag[GameTag["SPELL_WEAKNESS_HOLY"] = 2149] = "SPELL_WEAKNESS_HOLY";
    GameTag[GameTag["SPELL_WEAKNESS_SHADOW"] = 2150] = "SPELL_WEAKNESS_SHADOW";
    GameTag[GameTag["SPELL_WEAKNESS_FEL"] = 2151] = "SPELL_WEAKNESS_FEL";
    GameTag[GameTag["BACON_BUDDY"] = 2154] = "BACON_BUDDY";
    GameTag[GameTag["BACON_STARSTOBOUNCEOFF"] = 2155] = "BACON_STARSTOBOUNCEOFF";
    GameTag[GameTag["LETTUCE_KEYWORD_ATTACK"] = 2159] = "LETTUCE_KEYWORD_ATTACK";
    GameTag[GameTag["LETTUCE_KEYWORD_SPELL_COMBO"] = 2160] = "LETTUCE_KEYWORD_SPELL_COMBO";
    GameTag[GameTag["LETTUCE_BOUNTY_BOSS"] = 2168] = "LETTUCE_BOUNTY_BOSS";
    GameTag[GameTag["LETTUCE_IS_TREASURE_CARD"] = 2170] = "LETTUCE_IS_TREASURE_CARD";
    GameTag[GameTag["LETTUCE_SPELLDAMAGEARCANE"] = 2171] = "LETTUCE_SPELLDAMAGEARCANE";
    GameTag[GameTag["LETTUCE_SPELLDAMAGEFEL"] = 2172] = "LETTUCE_SPELLDAMAGEFEL";
    GameTag[GameTag["LETTUCE_SPELLDAMAGEFIRE"] = 2173] = "LETTUCE_SPELLDAMAGEFIRE";
    GameTag[GameTag["LETTUCE_SPELLDAMAGEFROST"] = 2174] = "LETTUCE_SPELLDAMAGEFROST";
    GameTag[GameTag["LETTUCE_SPELLDAMAGEHOLY"] = 2175] = "LETTUCE_SPELLDAMAGEHOLY";
    GameTag[GameTag["LETTUCE_SPELLDAMAGENATURE"] = 2176] = "LETTUCE_SPELLDAMAGENATURE";
    GameTag[GameTag["LETTUCE_SPELLDAMAGESHADOW"] = 2177] = "LETTUCE_SPELLDAMAGESHADOW";
    GameTag[GameTag["ROOTED"] = 2179] = "ROOTED";
    GameTag[GameTag["VULNERABLE"] = 2180] = "VULNERABLE";
    GameTag[GameTag["DEATHBLOW"] = 2185] = "DEATHBLOW";
    GameTag[GameTag["CORPSES"] = 2186] = "CORPSES";
    GameTag[GameTag["MAXRESOURCES_BLOOD"] = 2188] = "MAXRESOURCES_BLOOD";
    GameTag[GameTag["MAXRESOURCES_FROST"] = 2189] = "MAXRESOURCES_FROST";
    GameTag[GameTag["MAXRESOURCES_UNHOLY"] = 2190] = "MAXRESOURCES_UNHOLY";
    GameTag[GameTag["MAXRESOURCES_DEATH"] = 2191] = "MAXRESOURCES_DEATH";
    GameTag[GameTag["RESOURCES_BLOOD"] = 2192] = "RESOURCES_BLOOD";
    GameTag[GameTag["RESOURCES_FROST"] = 2193] = "RESOURCES_FROST";
    GameTag[GameTag["RESOURCES_UNHOLY"] = 2194] = "RESOURCES_UNHOLY";
    GameTag[GameTag["RESOURCES_DEATH"] = 2195] = "RESOURCES_DEATH";
    GameTag[GameTag["COST_BLOOD"] = 2196] = "COST_BLOOD";
    GameTag[GameTag["COST_FROST"] = 2197] = "COST_FROST";
    GameTag[GameTag["COST_UNHOLY"] = 2198] = "COST_UNHOLY";
    GameTag[GameTag["COST_DEATH"] = 2199] = "COST_DEATH";
    GameTag[GameTag["HAS_BLOOD_PLAGUE"] = 2211] = "HAS_BLOOD_PLAGUE";
    GameTag[GameTag["HAS_FROST_PLAGUE"] = 2212] = "HAS_FROST_PLAGUE";
    GameTag[GameTag["HAS_UNHOLY_PLAGUE"] = 2213] = "HAS_UNHOLY_PLAGUE";
    GameTag[GameTag["LETTUCE_BLEED"] = 2214] = "LETTUCE_BLEED";
    GameTag[GameTag["LETTUCE_KEYWORD_CRITICAL_DAMAGE"] = 2219] = "LETTUCE_KEYWORD_CRITICAL_DAMAGE";
    GameTag[GameTag["LETTUCE_KEYWORD_ROOT"] = 2220] = "LETTUCE_KEYWORD_ROOT";
    GameTag[GameTag["LETTUCE_SHOW_OPPOSING_FAKE_HAND"] = 2224] = "LETTUCE_SHOW_OPPOSING_FAKE_HAND";
    GameTag[GameTag["BACON_DIABLO_FIGHT_DIABLO_PLAYER_ID"] = 2226] = "BACON_DIABLO_FIGHT_DIABLO_PLAYER_ID";
    GameTag[GameTag["LETTUCE_VERSUS_SPELL_STATE"] = 2228] = "LETTUCE_VERSUS_SPELL_STATE";
    GameTag[GameTag["LETTUCE_START_OF_GAME_ABILITY"] = 2241] = "LETTUCE_START_OF_GAME_ABILITY";
    GameTag[GameTag["COLOSSAL"] = 2247] = "COLOSSAL";
    GameTag[GameTag["COLOSSAL_LIMB"] = 2248] = "COLOSSAL_LIMB";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_ARCANE"] = 2250] = "CURRENT_TEMP_SPELLPOWER_ARCANE";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_FEL"] = 2251] = "CURRENT_TEMP_SPELLPOWER_FEL";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_FIRE"] = 2252] = "CURRENT_TEMP_SPELLPOWER_FIRE";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_FROST"] = 2253] = "CURRENT_TEMP_SPELLPOWER_FROST";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_NATURE"] = 2254] = "CURRENT_TEMP_SPELLPOWER_NATURE";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_HOLY"] = 2255] = "CURRENT_TEMP_SPELLPOWER_HOLY";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_PHYSICAL"] = 2256] = "CURRENT_TEMP_SPELLPOWER_PHYSICAL";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_SHADOW"] = 2257] = "CURRENT_TEMP_SPELLPOWER_SHADOW";
    GameTag[GameTag["CURRENT_TEMP_SPELLPOWER_BASE"] = 2258] = "CURRENT_TEMP_SPELLPOWER_BASE";
    GameTag[GameTag["BACON_CHOSEN_BOARD_SKIN_ID"] = 2264] = "BACON_CHOSEN_BOARD_SKIN_ID";
    GameTag[GameTag["BACON_ONLY_IN_NAGA_LOBBY"] = 2272] = "BACON_ONLY_IN_NAGA_LOBBY";
    GameTag[GameTag["LETTUCE_ALLIANCE"] = 2279] = "LETTUCE_ALLIANCE";
    GameTag[GameTag["LETTUCE_HORDE"] = 2280] = "LETTUCE_HORDE";
    GameTag[GameTag["OBJECTIVE"] = 2311] = "OBJECTIVE";
    GameTag[GameTag["LETTUCE_REFRESH"] = 2312] = "LETTUCE_REFRESH";
    GameTag[GameTag["TIMES_BEEN_TRANSFORMED"] = 2321] = "TIMES_BEEN_TRANSFORMED";
    GameTag[GameTag["LETTUCE_ELVES"] = 2322] = "LETTUCE_ELVES";
    GameTag[GameTag["OBJECTIVE_AURA"] = 2329] = "OBJECTIVE_AURA";
    GameTag[GameTag["DREDGE"] = 2332] = "DREDGE";
    GameTag[GameTag["CURRENT_HEALING_POWER"] = 2333] = "CURRENT_HEALING_POWER";
    GameTag[GameTag["EARLY_CONCEDE_POPUP_AVAILABLE"] = 2340] = "EARLY_CONCEDE_POPUP_AVAILABLE";
    GameTag[GameTag["BACON_PLAYER_NUM_HERO_BUDDIES_GAINED"] = 2346] = "BACON_PLAYER_NUM_HERO_BUDDIES_GAINED";
    GameTag[GameTag["BACON_ONLY_IN_UNDEAD_LOBBY"] = 2347] = "BACON_ONLY_IN_UNDEAD_LOBBY";
    GameTag[GameTag["BATTLEGROUNDS_FAVORITE_FINISHER"] = 2348] = "BATTLEGROUNDS_FAVORITE_FINISHER";
    GameTag[GameTag["DAMAGE_DEALT_TO_HERO_LAST_TURN"] = 2349] = "DAMAGE_DEALT_TO_HERO_LAST_TURN";
    GameTag[GameTag["LOCATION_ACTION_COST"] = 2352] = "LOCATION_ACTION_COST";
    GameTag[GameTag["LOCATION_ACTION_COOLDOWN"] = 2353] = "LOCATION_ACTION_COOLDOWN";
    GameTag[GameTag["WHELP"] = 2355] = "WHELP";
    GameTag[GameTag["BACON_PIRATES_PLAYED_THIS_GAME"] = 2358] = "BACON_PIRATES_PLAYED_THIS_GAME";
    GameTag[GameTag["BACON_SPELLCRAFT_ID"] = 2359] = "BACON_SPELLCRAFT_ID";
    GameTag[GameTag["BACON_HERO_BUDDY_PROGRESS"] = 2364] = "BACON_HERO_BUDDY_PROGRESS";
    GameTag[GameTag["REVIVE"] = 2369] = "REVIVE";
    GameTag[GameTag["SUPPRESS_STAT_GAIN_EVENTS"] = 2374] = "SUPPRESS_STAT_GAIN_EVENTS";
    GameTag[GameTag["BACON_HEROPOWER_BASE_HERO_ID"] = 2376] = "BACON_HEROPOWER_BASE_HERO_ID";
    GameTag[GameTag["LETTUCE_CURSED_ABILITY_VISUAL"] = 2381] = "LETTUCE_CURSED_ABILITY_VISUAL";
    GameTag[GameTag["BACON_OMIT_WHEN_OUT_OF_ROTATION"] = 2387] = "BACON_OMIT_WHEN_OUT_OF_ROTATION";
    GameTag[GameTag["ALLIED"] = 2388] = "ALLIED";
    GameTag[GameTag["BACON_SPELLCRAFT"] = 2423] = "BACON_SPELLCRAFT";
    GameTag[GameTag["LETTUCE_KEYWORD_HEALING_POWER"] = 2434] = "LETTUCE_KEYWORD_HEALING_POWER";
    GameTag[GameTag["BACON_OVERRIDE_BG_COST"] = 2437] = "BACON_OVERRIDE_BG_COST";
    GameTag[GameTag["DONT_SUPPRESS_SUMMON_VO"] = 2440] = "DONT_SUPPRESS_SUMMON_VO";
    GameTag[GameTag["BACON_NUMBER_HERO_REFRESH_AVAILABLE"] = 2452] = "BACON_NUMBER_HERO_REFRESH_AVAILABLE";
    GameTag[GameTag["BACON_FREEZE_TOOLTIP"] = 2455] = "BACON_FREEZE_TOOLTIP";
    GameTag[GameTag["INFUSE"] = 2456] = "INFUSE";
    GameTag[GameTag["INFUSED"] = 2457] = "INFUSED";
    GameTag[GameTag["HAS_DRAG_TO_BUY"] = 2458] = "HAS_DRAG_TO_BUY";
    GameTag[GameTag["ENTITY_TAG_THRESHOLD_TAG_ID"] = 2459] = "ENTITY_TAG_THRESHOLD_TAG_ID";
    GameTag[GameTag["ENTITY_TAG_THRESHOLD_VALUE"] = 2460] = "ENTITY_TAG_THRESHOLD_VALUE";
    GameTag[GameTag["MERCENARIES_SPELL_WEAKNESS"] = 2464] = "MERCENARIES_SPELL_WEAKNESS";
    GameTag[GameTag["MERCENARIES_SPELL_RESISTANCE"] = 2465] = "MERCENARIES_SPELL_RESISTANCE";
    GameTag[GameTag["BACON_QUESTS_ACTIVE"] = 2468] = "BACON_QUESTS_ACTIVE";
    GameTag[GameTag["COLOSSAL_LIMB_ON_LEFT"] = 2469] = "COLOSSAL_LIMB_ON_LEFT";
    GameTag[GameTag["LETTUCE_ABILITY_TILE_VISUAL_PUBLIC_SPEED"] = 2470] = "LETTUCE_ABILITY_TILE_VISUAL_PUBLIC_SPEED";
    GameTag[GameTag["BACON_DIED_LAST_COMBAT"] = 2483] = "BACON_DIED_LAST_COMBAT";
    GameTag[GameTag["LETTUCE_ABILITY_TIER"] = 2493] = "LETTUCE_ABILITY_TIER";
    GameTag[GameTag["LETTUCE_EQUIPMENT_TIER"] = 2494] = "LETTUCE_EQUIPMENT_TIER";
    GameTag[GameTag["MANATHIRST"] = 2498] = "MANATHIRST";
    GameTag[GameTag["IMMOLATING"] = 2505] = "IMMOLATING";
    GameTag[GameTag["MERCS_EXPLORER"] = 2510] = "MERCS_EXPLORER";
    GameTag[GameTag["BACON_BUDDY_ENABLED"] = 2518] = "BACON_BUDDY_ENABLED";
    GameTag[GameTag["BACON_EVOLUTION_CARD_ID"] = 2519] = "BACON_EVOLUTION_CARD_ID";
    GameTag[GameTag["SPELLCRAFT_HINT"] = 2557] = "SPELLCRAFT_HINT";
    GameTag[GameTag["CORPSE"] = 2559] = "CORPSE";
    GameTag[GameTag["MERCS_BENCH"] = 2570] = "MERCS_BENCH";
    GameTag[GameTag["BACON_MINION_TYPE_REWARD"] = 2571] = "BACON_MINION_TYPE_REWARD";
    GameTag[GameTag["COPIED_HINT"] = 2572] = "COPIED_HINT";
    GameTag[GameTag["BLEEDING"] = 2575] = "BLEEDING";
    GameTag[GameTag["HAS_SIGNATURE_QUALITY"] = 2589] = "HAS_SIGNATURE_QUALITY";
    GameTag[GameTag["IMMOLATESTAGE"] = 2600] = "IMMOLATESTAGE";
    GameTag[GameTag["EVIL_TWIN_MUSTACHE"] = 2611] = "EVIL_TWIN_MUSTACHE";
    GameTag[GameTag["SINFUL_BRAND"] = 2613] = "SINFUL_BRAND";
    GameTag[GameTag["LETTUCE_KEYWORD_SILENCE"] = 2631] = "LETTUCE_KEYWORD_SILENCE";
    GameTag[GameTag["BACON_QUEST_COMPLETED"] = 2633] = "BACON_QUEST_COMPLETED";
    GameTag[GameTag["HAUNTED_SECRET"] = 2634] = "HAUNTED_SECRET";
    GameTag[GameTag["DONT_SUPPRESS_KEYWORD_VO"] = 2636] = "DONT_SUPPRESS_KEYWORD_VO";
    GameTag[GameTag["CARD_BACK_OVERRIDE"] = 2637] = "CARD_BACK_OVERRIDE";
    GameTag[GameTag["CORPSES_SPENT_THIS_GAME"] = 2639] = "CORPSES_SPENT_THIS_GAME";
    GameTag[GameTag["CARDTEXT_ENTITY_0"] = 2655] = "CARDTEXT_ENTITY_0";
    GameTag[GameTag["CARDTEXT_ENTITY_1"] = 2656] = "CARDTEXT_ENTITY_1";
    GameTag[GameTag["CARDTEXT_ENTITY_2"] = 2657] = "CARDTEXT_ENTITY_2";
    GameTag[GameTag["CARDTEXT_ENTITY_3"] = 2658] = "CARDTEXT_ENTITY_3";
    GameTag[GameTag["CARDTEXT_ENTITY_4"] = 2659] = "CARDTEXT_ENTITY_4";
    GameTag[GameTag["CARDTEXT_ENTITY_5"] = 2660] = "CARDTEXT_ENTITY_5";
    GameTag[GameTag["CARDTEXT_ENTITY_6"] = 2661] = "CARDTEXT_ENTITY_6";
    GameTag[GameTag["CARDTEXT_ENTITY_7"] = 2662] = "CARDTEXT_ENTITY_7";
    GameTag[GameTag["CARDTEXT_ENTITY_8"] = 2663] = "CARDTEXT_ENTITY_8";
    GameTag[GameTag["CARDTEXT_ENTITY_9"] = 2664] = "CARDTEXT_ENTITY_9";
    GameTag[GameTag["NON_KEYWORD_SPELLBURST"] = 2672] = "NON_KEYWORD_SPELLBURST";
    GameTag[GameTag["BACON_CARD_DBID_REWARD"] = 2673] = "BACON_CARD_DBID_REWARD";
    GameTag[GameTag["SECRET_LOCKED"] = 2676] = "SECRET_LOCKED";
    GameTag[GameTag["BANNED_IN_LOBBY_WITH_RACE"] = 2703] = "BANNED_IN_LOBBY_WITH_RACE";
    GameTag[GameTag["BACON_STEALTH_TOOLTIP"] = 2704] = "BACON_STEALTH_TOOLTIP";
    GameTag[GameTag["BACON_QUEST_TOOLTIP"] = 2705] = "BACON_QUEST_TOOLTIP";
    GameTag[GameTag["BACON_IS_HEROPOWER_QUESTREWARD"] = 2706] = "BACON_IS_HEROPOWER_QUESTREWARD";
    GameTag[GameTag["BACON_HERO_QUEST_REWARD_DATABASE_ID"] = 2713] = "BACON_HERO_QUEST_REWARD_DATABASE_ID";
    GameTag[GameTag["BACON_HERO_HEROPOWER_QUEST_REWARD_DATABASE_ID"] = 2714] = "BACON_HERO_HEROPOWER_QUEST_REWARD_DATABASE_ID";
    GameTag[GameTag["BACON_HERO_QUEST_REWARD_COMPLETED"] = 2715] = "BACON_HERO_QUEST_REWARD_COMPLETED";
    GameTag[GameTag["BACON_HERO_HEROPOWER_QUEST_REWARD_COMPLETED"] = 2716] = "BACON_HERO_HEROPOWER_QUEST_REWARD_COMPLETED";
    GameTag[GameTag["NUM_FRIENDLY_MINIONS_THAT_DIED_LAST_TURN"] = 2717] = "NUM_FRIENDLY_MINIONS_THAT_DIED_LAST_TURN";
    GameTag[GameTag["LETTUCE_FACTION"] = 2720] = "LETTUCE_FACTION";
    GameTag[GameTag["DEATH_SPELL_OVERRIDE"] = 2722] = "DEATH_SPELL_OVERRIDE";
    GameTag[GameTag["BACON_IS_BOB_QUEST"] = 2732] = "BACON_IS_BOB_QUEST";
    GameTag[GameTag["BACON_HERO_REWARD_CARD_DBID"] = 2748] = "BACON_HERO_REWARD_CARD_DBID";
    GameTag[GameTag["BACON_HERO_HEROPOWER_REWARD_CARD_DBID"] = 2749] = "BACON_HERO_HEROPOWER_REWARD_CARD_DBID";
    GameTag[GameTag["BACON_HERO_REWARD_MINION_TYPE"] = 2750] = "BACON_HERO_REWARD_MINION_TYPE";
    GameTag[GameTag["BACON_HERO_HEROPOWER_REWARD_MINION_TYPE"] = 2751] = "BACON_HERO_HEROPOWER_REWARD_MINION_TYPE";
    GameTag[GameTag["MERCENARIES_DISCOVER_SOURCE"] = 2752] = "MERCENARIES_DISCOVER_SOURCE";
    GameTag[GameTag["TITAN"] = 2772] = "TITAN";
    GameTag[GameTag["HERO_ATTACK_GIVEN_ADDITIONAL"] = 2776] = "HERO_ATTACK_GIVEN_ADDITIONAL";
    GameTag[GameTag["HERO_ARMOR_GIVEN_ADDITIONAL"] = 2778] = "HERO_ARMOR_GIVEN_ADDITIONAL";
    GameTag[GameTag["LETTUCE_CHARGE"] = 2779] = "LETTUCE_CHARGE";
    GameTag[GameTag["BACON_DIED_LAST_COMBAT_HINT"] = 2780] = "BACON_DIED_LAST_COMBAT_HINT";
    GameTag[GameTag["SHIFTING_LOCATION"] = 2783] = "SHIFTING_LOCATION";
    GameTag[GameTag["FORGE"] = 2785] = "FORGE";
    GameTag[GameTag["ENTITY_AS_ENCHANTMENT"] = 2787] = "ENTITY_AS_ENCHANTMENT";
    GameTag[GameTag["UNPLAYABLE_VISUALS"] = 2798] = "UNPLAYABLE_VISUALS";
    GameTag[GameTag["CARDTEXT_ENTITY_AS_NUMBERS"] = 2802] = "CARDTEXT_ENTITY_AS_NUMBERS";
    GameTag[GameTag["BACON_DOUBLE_QUEST_HERO_POWER"] = 2803] = "BACON_DOUBLE_QUEST_HERO_POWER";
    GameTag[GameTag["MERCENARIES_TREASURE_SCALE_LEVEL"] = 2810] = "MERCENARIES_TREASURE_SCALE_LEVEL";
    GameTag[GameTag["RIFF"] = 2819] = "RIFF";
    GameTag[GameTag["FINALE"] = 2820] = "FINALE";
    GameTag[GameTag["OVERHEAL"] = 2821] = "OVERHEAL";
    GameTag[GameTag["BACON_BLOODGEMBUFFHEALTHVALUE"] = 2827] = "BACON_BLOODGEMBUFFHEALTHVALUE";
    GameTag[GameTag["IS_TREANT"] = 2831] = "IS_TREANT";
    GameTag[GameTag["CARD_ALTERNATE_COST"] = 2837] = "CARD_ALTERNATE_COST";
    GameTag[GameTag["EMOTECHARACTER"] = 2839] = "EMOTECHARACTER";
    GameTag[GameTag["HAS_ACTIVATE_POWER"] = 2840] = "HAS_ACTIVATE_POWER";
    GameTag[GameTag["BACON_DEEP_BLUE"] = 2850] = "BACON_DEEP_BLUE";
    GameTag[GameTag["EMOTECLASS"] = 2851] = "EMOTECLASS";
    GameTag[GameTag["VENOMOUS"] = 2853] = "VENOMOUS";
    GameTag[GameTag["MAGNETIC_TO_RACE"] = 2859] = "MAGNETIC_TO_RACE";
    GameTag[GameTag["BACON_MAX_LEADERBOARD_ARMOR"] = 2867] = "BACON_MAX_LEADERBOARD_ARMOR";
    GameTag[GameTag["IS_USING_FORGE_OPTION"] = 2869] = "IS_USING_FORGE_OPTION";
    GameTag[GameTag["BACON_REBORN_TOOLTIP"] = 2870] = "BACON_REBORN_TOOLTIP";
    GameTag[GameTag["BACON_PUTRICIDES_CREATION_TOOLTIP"] = 2875] = "BACON_PUTRICIDES_CREATION_TOOLTIP";
    GameTag[GameTag["BACON_ELEMENTALS_PLAYED_THIS_GAME"] = 2878] = "BACON_ELEMENTALS_PLAYED_THIS_GAME";
    GameTag[GameTag["TAG_SCRIPT_DATA_NUM_3"] = 2889] = "TAG_SCRIPT_DATA_NUM_3";
    GameTag[GameTag["CARD_NAME_DATA_1"] = 2890] = "CARD_NAME_DATA_1";
    GameTag[GameTag["CONSTRUCTED_ANOMAY"] = 2895] = "CONSTRUCTED_ANOMAY";
    GameTag[GameTag["BACON_GLOBAL_ANOMALY_DBID"] = 2897] = "BACON_GLOBAL_ANOMALY_DBID";
    GameTag[GameTag["PLAGUE"] = 2899] = "PLAGUE";
    GameTag[GameTag["QUICKDRAW"] = 2905] = "QUICKDRAW";
    GameTag[GameTag["BACON_COSTS_HEALTH_TO_BUY"] = 2911] = "BACON_COSTS_HEALTH_TO_BUY";
    GameTag[GameTag["TAG_SCRIPT_DATA_NUM_4"] = 2919] = "TAG_SCRIPT_DATA_NUM_4";
    GameTag[GameTag["TAG_SCRIPT_DATA_NUM_5"] = 2920] = "TAG_SCRIPT_DATA_NUM_5";
    GameTag[GameTag["TAG_SCRIPT_DATA_NUM_6"] = 2921] = "TAG_SCRIPT_DATA_NUM_6";
    GameTag[GameTag["DECK_SWAP_ACTIVE"] = 2929] = "DECK_SWAP_ACTIVE";
    GameTag[GameTag["MAX_SIDEBOARD_CARDS"] = 2931] = "MAX_SIDEBOARD_CARDS";
    GameTag[GameTag["BONUS_EFFECTS"] = 2934] = "BONUS_EFFECTS";
    GameTag[GameTag["BACON_USE_COIN_BASED_BUDDY_METER"] = 2935] = "BACON_USE_COIN_BASED_BUDDY_METER";
    GameTag[GameTag["BACON_BUY_BUDDY"] = 2937] = "BACON_BUY_BUDDY";
    GameTag[GameTag["BACON_BUY_BUDDY_2"] = 2938] = "BACON_BUY_BUDDY_2";
    GameTag[GameTag["BACON_DUO_TEAMMATE_PLAYER_ID"] = 2939] = "BACON_DUO_TEAMMATE_PLAYER_ID";
    GameTag[GameTag["BACON_SHOW_HEROPOWER_BUDDY_AS_EVOLVING_BIG_CARD"] = 2943] = "BACON_SHOW_HEROPOWER_BUDDY_AS_EVOLVING_BIG_CARD";
    GameTag[GameTag["HIDDEN_CHOICE_OVERRIDE"] = 2946] = "HIDDEN_CHOICE_OVERRIDE";
    GameTag[GameTag["BACON_DUO_PLAYER_FIGHTS_FIRST_NEXT_COMBAT"] = 2975] = "BACON_DUO_PLAYER_FIGHTS_FIRST_NEXT_COMBAT";
    GameTag[GameTag["NEXT_OPPONENT_TEAMMATE_PLAYER_ID"] = 2988] = "NEXT_OPPONENT_TEAMMATE_PLAYER_ID";
    GameTag[GameTag["BACON_CURRENT_COMBAT_PLAYER_ID"] = 2989] = "BACON_CURRENT_COMBAT_PLAYER_ID";
    GameTag[GameTag["FORGED"] = 3011] = "FORGED";
    GameTag[GameTag["BUILDING_UP"] = 3016] = "BUILDING_UP";
    GameTag[GameTag["BACON_PAIR_CANDIDATE"] = 3031] = "BACON_PAIR_CANDIDATE";
    GameTag[GameTag["CTHUN_TAUNT_BUFF"] = 3034] = "CTHUN_TAUNT_BUFF";
    GameTag[GameTag["BACON_TRIGGER_UPBEAT"] = 3046] = "BACON_TRIGGER_UPBEAT";
    GameTag[GameTag["BACON_TRIGGER_XY"] = 3047] = "BACON_TRIGGER_XY";
    GameTag[GameTag["BACON_COMBAT_PHASE_HERO"] = 3048] = "BACON_COMBAT_PHASE_HERO";
    GameTag[GameTag["FAN_LINK"] = 3052] = "FAN_LINK";
    GameTag[GameTag["CTHUN_HEALTH_BUFF"] = 3053] = "CTHUN_HEALTH_BUFF";
    GameTag[GameTag["CTHUN_ATTACK_BUFF"] = 3054] = "CTHUN_ATTACK_BUFF";
    GameTag[GameTag["FORGE_REVEALED"] = 3070] = "FORGE_REVEALED";
    GameTag[GameTag["FORGES_INTO"] = 3074] = "FORGES_INTO";
    GameTag[GameTag["FX_DATANUM_2"] = 3077] = "FX_DATANUM_2";
    GameTag[GameTag["RITUALIST_MINION"] = 3078] = "RITUALIST_MINION";
    GameTag[GameTag["BACON_IS_SPELL_IN_POOL"] = 3081] = "BACON_IS_SPELL_IN_POOL";
    GameTag[GameTag["TAVERN_SPELLS_PLAYED_THIS_GAME"] = 3088] = "TAVERN_SPELLS_PLAYED_THIS_GAME";
    GameTag[GameTag["SUPPRES_ALL_SOUNDS_FOR_ENTITY"] = 3093] = "SUPPRES_ALL_SOUNDS_FOR_ENTITY";
    GameTag[GameTag["BACON_DUO_TEAM_ID"] = 3095] = "BACON_DUO_TEAM_ID";
    GameTag[GameTag["FX_DATANUM_3"] = 3109] = "FX_DATANUM_3";
    GameTag[GameTag["ALLOW_MOVE_BACON_SPELL"] = 3111] = "ALLOW_MOVE_BACON_SPELL";
    GameTag[GameTag["EXCAVATE"] = 3114] = "EXCAVATE";
    GameTag[GameTag["SUMMONED_WHEN_DRAWN"] = 3128] = "SUMMONED_WHEN_DRAWN";
    GameTag[GameTag["IS_ALTERNATE_HEROPOWER"] = 3130] = "IS_ALTERNATE_HEROPOWER";
    GameTag[GameTag["TITAN_ABILITY_USED_1"] = 3140] = "TITAN_ABILITY_USED_1";
    GameTag[GameTag["TITAN_ABILITY_USED_2"] = 3141] = "TITAN_ABILITY_USED_2";
    GameTag[GameTag["TITAN_ABILITY_USED_3"] = 3142] = "TITAN_ABILITY_USED_3";
    GameTag[GameTag["BACON_DUO_TRIPLE_CANDIDATE_TEAMMATE"] = 3145] = "BACON_DUO_TRIPLE_CANDIDATE_TEAMMATE";
    GameTag[GameTag["BACON_DUO_PAIR_CANDIDATE_TEAMMATE"] = 3146] = "BACON_DUO_PAIR_CANDIDATE_TEAMMATE";
    GameTag[GameTag["BACON_MAX_RESOURCES"] = 3148] = "BACON_MAX_RESOURCES";
    GameTag[GameTag["BG_DUO_EXCLUSIVE"] = 3166] = "BG_DUO_EXCLUSIVE";
    GameTag[GameTag["BG_SOLO_EXCLUSIVE"] = 3167] = "BG_SOLO_EXCLUSIVE";
    GameTag[GameTag["BACON_DUO_PASSABLE"] = 3178] = "BACON_DUO_PASSABLE";
    GameTag[GameTag["ANOMALY1"] = 3182] = "ANOMALY1";
    GameTag[GameTag["ANOMALY2"] = 3183] = "ANOMALY2";
    GameTag[GameTag["IS_USING_PASS_OPTION"] = 3185] = "IS_USING_PASS_OPTION";
    GameTag[GameTag["__BACON_WEIGHT_PICK_FOR_NEW_PLAYERS"] = 3190] = "__BACON_WEIGHT_PICK_FOR_NEW_PLAYERS";
    GameTag[GameTag["TUTORIAL_TARGET_OPPONENT_ANIM"] = 3192] = "TUTORIAL_TARGET_OPPONENT_ANIM";
    GameTag[GameTag["TUTORIAL_TARGET_MINION_ANIM"] = 3193] = "TUTORIAL_TARGET_MINION_ANIM";
    GameTag[GameTag["TUTORIAL_PLAY_MINION_ANIM"] = 3195] = "TUTORIAL_PLAY_MINION_ANIM";
    GameTag[GameTag["TUTORIAL_HERO_POWER_TARGET_MINION_ANIM"] = 3196] = "TUTORIAL_HERO_POWER_TARGET_MINION_ANIM";
    GameTag[GameTag["TUTORIAL_HERO_POWER_TARGET_OPPONENT_ANIM"] = 3197] = "TUTORIAL_HERO_POWER_TARGET_OPPONENT_ANIM";
    GameTag[GameTag["SUPPRESS_EVIL_TWIN_MUSTACHE_SOUND"] = 3198] = "SUPPRESS_EVIL_TWIN_MUSTACHE_SOUND";
    GameTag[GameTag["__BG_ARMOR_DUO"] = 3206] = "__BG_ARMOR_DUO";
    GameTag[GameTag["BACON_ANOMALY_ALL_HEROES_ARE_THIS_DBID"] = 3208] = "BACON_ANOMALY_ALL_HEROES_ARE_THIS_DBID";
    GameTag[GameTag["HERO_DOESNT_MOVE_ON_ATTACK"] = 3211] = "HERO_DOESNT_MOVE_ON_ATTACK";
    GameTag[GameTag["BACON_NO_TIER_UP_BUTTON"] = 3220] = "BACON_NO_TIER_UP_BUTTON";
    GameTag[GameTag["TOURIST"] = 3228] = "TOURIST";
    GameTag[GameTag["CURRENT_EXCAVATE_TIER"] = 3249] = "CURRENT_EXCAVATE_TIER";
    GameTag[GameTag["BACON_CONSUME_TOOLTIP"] = 3254] = "BACON_CONSUME_TOOLTIP";
    GameTag[GameTag["ALONE_RANGER"] = 3258] = "ALONE_RANGER";
    GameTag[GameTag["CUTSCENE_CARD_TYPE"] = 3265] = "CUTSCENE_CARD_TYPE";
    GameTag[GameTag["MINIATURIZE"] = 3318] = "MINIATURIZE";
    GameTag[GameTag["MINI"] = 3319] = "MINI";
    GameTag[GameTag["BACON_PASS_TOOLTIP"] = 3321] = "BACON_PASS_TOOLTIP";
    GameTag[GameTag["MAX_EXCAVATE_TIER"] = 3326] = "MAX_EXCAVATE_TIER";
    GameTag[GameTag["ZILLIAX_CUSTOMIZABLE_COSMETICMODULE"] = 3376] = "ZILLIAX_CUSTOMIZABLE_COSMETICMODULE";
    GameTag[GameTag["ZILLIAX_CUSTOMIZABLE_FUNCTIONALMODULE"] = 3377] = "ZILLIAX_CUSTOMIZABLE_FUNCTIONALMODULE";
    GameTag[GameTag["KEEP_HERO_CLASS"] = 3382] = "KEEP_HERO_CLASS";
    GameTag[GameTag["BACON_NEUTRAL_TRINKET"] = 3388] = "BACON_NEUTRAL_TRINKET";
    GameTag[GameTag["GIGANTIFY"] = 3399] = "GIGANTIFY";
    GameTag[GameTag["GIGANTIC"] = 3400] = "GIGANTIC";
    GameTag[GameTag["BACON_COMBAT_DAMAGE_CAP_ENABLED"] = 3403] = "BACON_COMBAT_DAMAGE_CAP_ENABLED";
    GameTag[GameTag["BACON_TRINKET"] = 3407] = "BACON_TRINKET";
    GameTag[GameTag["BONUS_KEYWORDS"] = 3423] = "BONUS_KEYWORDS";
    GameTag[GameTag["SIDEBOARD_TYPE"] = 3427] = "SIDEBOARD_TYPE";
    GameTag[GameTag["PALADIN_AURA"] = 3429] = "PALADIN_AURA";
    GameTag[GameTag["CREATED_BY_TWINSPELL"] = 3432] = "CREATED_BY_TWINSPELL";
    GameTag[GameTag["CREATED_BY_MINIATURIZE"] = 3433] = "CREATED_BY_MINIATURIZE";
    GameTag[GameTag["CREATED_BY_GIGANTIFY"] = 3434] = "CREATED_BY_GIGANTIFY";
    GameTag[GameTag["SUPPRESS_HERO_STANDARD_SUMMON_FX"] = 3438] = "SUPPRESS_HERO_STANDARD_SUMMON_FX";
    GameTag[GameTag["BACON_TRINKET_CAN_BE_DISCOUNTED"] = 3443] = "BACON_TRINKET_CAN_BE_DISCOUNTED";
    GameTag[GameTag["SILVER_HAND_RECRUIT"] = 3444] = "SILVER_HAND_RECRUIT";
    GameTag[GameTag["ZILLIAX_CUSTOMIZABLE_LINKED_COSMETICMOUDLE"] = 3450] = "ZILLIAX_CUSTOMIZABLE_LINKED_COSMETICMOUDLE";
    GameTag[GameTag["WHIZBANG_SPLENDIFEROUS_DECK_ID"] = 3451] = "WHIZBANG_SPLENDIFEROUS_DECK_ID";
    GameTag[GameTag["BACON_SHOW_COST_ON_DISCOVER"] = 3456] = "BACON_SHOW_COST_ON_DISCOVER";
    GameTag[GameTag["ZERG"] = 3457] = "ZERG";
    GameTag[GameTag["TERRAN"] = 3458] = "TERRAN";
    GameTag[GameTag["MIN_SIDEBOARD_CARDS"] = 3459] = "MIN_SIDEBOARD_CARDS";
    GameTag[GameTag["FORGETFUL_ATTACK_VISUAL"] = 3460] = "FORGETFUL_ATTACK_VISUAL";
    GameTag[GameTag["SHUDDERWOCKHIGHLIGHTHINT"] = 3463] = "SHUDDERWOCKHIGHLIGHTHINT";
    GameTag[GameTag["NUM_TURNS_LAST_AFFECTED_BY"] = 3464] = "NUM_TURNS_LAST_AFFECTED_BY";
    GameTag[GameTag["EXTRA_TURNS_SPELL_OVERRIDE"] = 3465] = "EXTRA_TURNS_SPELL_OVERRIDE";
    GameTag[GameTag["PROTOSS"] = 3469] = "PROTOSS";
    GameTag[GameTag["ZILLIAX_CUSTOMIZABLE_LINKED_FUNCTIONALMOUDLE"] = 3470] = "ZILLIAX_CUSTOMIZABLE_LINKED_FUNCTIONALMOUDLE";
    GameTag[GameTag["HIDE_HEALTH_NUMBER"] = 3471] = "HIDE_HEALTH_NUMBER";
    GameTag[GameTag["HIDE_ATTACK_NUMBER"] = 3472] = "HIDE_ATTACK_NUMBER";
    GameTag[GameTag["ZILLIAX_CUSTOMIZABLE_SAVED_VERSION"] = 3477] = "ZILLIAX_CUSTOMIZABLE_SAVED_VERSION";
    GameTag[GameTag["DUOS_QUEUED_NOT_ON_TEAM"] = 3478] = "DUOS_QUEUED_NOT_ON_TEAM";
    GameTag[GameTag["TAG_PLAYER_CONCEDED_OR_DISCONNECTED"] = 3479] = "TAG_PLAYER_CONCEDED_OR_DISCONNECTED";
    GameTag[GameTag["PLAYER_ABANDONED_BY_TEAMMATE"] = 3480] = "PLAYER_ABANDONED_BY_TEAMMATE";
    GameTag[GameTag["SUPPRESS_MILL_ANIMATION"] = 3481] = "SUPPRESS_MILL_ANIMATION";
    GameTag[GameTag["IGNORE_SUPPRESS_MILL_ANIMATION"] = 3482] = "IGNORE_SUPPRESS_MILL_ANIMATION";
    GameTag[GameTag["HERO_PASSIVE_ID"] = 3487] = "HERO_PASSIVE_ID";
    GameTag[GameTag["BACON_TEAMMATE_BONUS_MINION_DAMAGE_LAST_COMBAT"] = 3492] = "BACON_TEAMMATE_BONUS_MINION_DAMAGE_LAST_COMBAT";
    GameTag[GameTag["BACON_DUOS_PUNISH_LEAVERS"] = 3494] = "BACON_DUOS_PUNISH_LEAVERS";
    GameTag[GameTag["HERO_FRAME_TYPE"] = 3495] = "HERO_FRAME_TYPE";
    GameTag[GameTag["BACON_TRIPLED_BASE_MINION_ID2"] = 3499] = "BACON_TRIPLED_BASE_MINION_ID2";
    GameTag[GameTag["BACON_TRIPLED_BASE_MINION_ID3"] = 3500] = "BACON_TRIPLED_BASE_MINION_ID3";
    GameTag[GameTag["QUEST_HIDE_PROGRESS"] = 3523] = "QUEST_HIDE_PROGRESS";
    GameTag[GameTag["IMBUES_THIS_GAME"] = 3527] = "IMBUES_THIS_GAME";
    GameTag[GameTag["IGNORE_MODIFIER_HERO_POWER_CHECK"] = 3533] = "IGNORE_MODIFIER_HERO_POWER_CHECK";
    GameTag[GameTag["STARSHIP"] = 3555] = "STARSHIP";
    GameTag[GameTag["TRANSFORM"] = 3562] = "TRANSFORM";
    GameTag[GameTag["LAUNCHPAD"] = 3563] = "LAUNCHPAD";
    GameTag[GameTag["CORNER_REPLACEMENT_TYPE"] = 3564] = "CORNER_REPLACEMENT_TYPE";
    GameTag[GameTag["BACON_IS_MAGIC_ITEM_DISCOVER"] = 3565] = "BACON_IS_MAGIC_ITEM_DISCOVER";
    GameTag[GameTag["IS_NIGHTMARE_BONUS"] = 3567] = "IS_NIGHTMARE_BONUS";
    GameTag[GameTag["STARSHIP_PIECE"] = 3568] = "STARSHIP_PIECE";
    GameTag[GameTag["ROGUE_TOURIST"] = 3597] = "ROGUE_TOURIST";
    GameTag[GameTag["WARLOCK_TOURIST"] = 3598] = "WARLOCK_TOURIST";
    GameTag[GameTag["DEATH_KNIGHT_TOURIST"] = 3599] = "DEATH_KNIGHT_TOURIST";
    GameTag[GameTag["SHAMAN_TOURIST"] = 3600] = "SHAMAN_TOURIST";
    GameTag[GameTag["DEMON_HUNTER_TOURIST"] = 3601] = "DEMON_HUNTER_TOURIST";
    GameTag[GameTag["PRIEST_TOURIST"] = 3602] = "PRIEST_TOURIST";
    GameTag[GameTag["HUNTER_TOURIST"] = 3603] = "HUNTER_TOURIST";
    GameTag[GameTag["WARRIOR_TOURIST"] = 3604] = "WARRIOR_TOURIST";
    GameTag[GameTag["DRUID_TOURIST"] = 3605] = "DRUID_TOURIST";
    GameTag[GameTag["MAGE_TOURIST"] = 3606] = "MAGE_TOURIST";
    GameTag[GameTag["PALADIN_TOURIST"] = 3607] = "PALADIN_TOURIST";
    GameTag[GameTag["IS_GALAKROND_POOL"] = 3623] = "IS_GALAKROND_POOL";
    GameTag[GameTag["IMBUE"] = 3626] = "IMBUE";
    GameTag[GameTag["DARK_GIFT"] = 3627] = "DARK_GIFT";
    GameTag[GameTag["TAG_LAUNCHPAD_ABILITY"] = 3628] = "TAG_LAUNCHPAD_ABILITY";
    GameTag[GameTag["TEMPORARY"] = 3630] = "TEMPORARY";
    GameTag[GameTag["IS_CTHUN_POOL"] = 3633] = "IS_CTHUN_POOL";
    GameTag[GameTag["STARSHIP_LAUNCH_COST_DISCOUNT"] = 3640] = "STARSHIP_LAUNCH_COST_DISCOUNT";
    GameTag[GameTag["BACON_DONT_SHOW_PAIR_TRIPLE_DISCOVER_VFX"] = 3661] = "BACON_DONT_SHOW_PAIR_TRIPLE_DISCOVER_VFX";
    GameTag[GameTag["FALLBACK_ENCHANTMENT_PORTRAIT_DBID"] = 3664] = "FALLBACK_ENCHANTMENT_PORTRAIT_DBID";
    GameTag[GameTag["BACON_NUM_MAGNETIZE_THIS_GAME"] = 3670] = "BACON_NUM_MAGNETIZE_THIS_GAME";
    GameTag[GameTag["SHOW_SLEEP_ZZZ_OVERRIDE"] = 3672] = "SHOW_SLEEP_ZZZ_OVERRIDE";
    GameTag[GameTag["CLIENT_LIST_REPLACEMENTS_WHEN_PLAYED"] = 3677] = "CLIENT_LIST_REPLACEMENTS_WHEN_PLAYED";
    GameTag[GameTag["BACON_PIRATES_SUMMONED_THIS_GAME"] = 3685] = "BACON_PIRATES_SUMMONED_THIS_GAME";
    GameTag[GameTag["CARES_ABOUT_IMBUE_CARDS"] = 3692] = "CARES_ABOUT_IMBUE_CARDS";
    GameTag[GameTag["BACON_IS_POTENTIAL_TRINKET"] = 3705] = "BACON_IS_POTENTIAL_TRINKET";
    GameTag[GameTag["OUROBOSDEATHRATTLE"] = 3716] = "OUROBOSDEATHRATTLE";
    GameTag[GameTag["BACON_TURNS_LEFT_TO_DISCOVER_TRINKET"] = 3738] = "BACON_TURNS_LEFT_TO_DISCOVER_TRINKET";
    GameTag[GameTag["BACON_TRINKETS_ACTIVE"] = 3740] = "BACON_TRINKETS_ACTIVE";
    GameTag[GameTag["BACON_FIRST_TRINKET_DATABASE_ID"] = 3741] = "BACON_FIRST_TRINKET_DATABASE_ID";
    GameTag[GameTag["BACON_SECOND_TRINKET_DATABASE_ID"] = 3742] = "BACON_SECOND_TRINKET_DATABASE_ID";
    GameTag[GameTag["BACON_HEROPOWER_TRINKET_DATABASE_ID"] = 3743] = "BACON_HEROPOWER_TRINKET_DATABASE_ID";
    GameTag[GameTag["END_OF_TURN_TRIGGER"] = 3744] = "END_OF_TURN_TRIGGER";
    GameTag[GameTag["BACON_OVERRIDE_COST_COLOR"] = 3777] = "BACON_OVERRIDE_COST_COLOR";
    GameTag[GameTag["BACON_EVOLUTION_CARD_ID_2"] = 3778] = "BACON_EVOLUTION_CARD_ID_2";
    GameTag[GameTag["MINION_TYPE_MASK"] = 3802] = "MINION_TYPE_MASK";
    GameTag[GameTag["BACON_HERO_FIRST_TRINKET_LEADERBOARD_SDN1"] = 3805] = "BACON_HERO_FIRST_TRINKET_LEADERBOARD_SDN1";
    GameTag[GameTag["BACON_HERO_SECOND_TRINKET_LEADERBOARD_SDN1"] = 3806] = "BACON_HERO_SECOND_TRINKET_LEADERBOARD_SDN1";
    GameTag[GameTag["BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_SDN1"] = 3807] = "BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_SDN1";
    GameTag[GameTag["DEMON_PORTAL_DECK"] = 3808] = "DEMON_PORTAL_DECK";
    GameTag[GameTag["BACON_HERO_FIRST_TRINKET_LEADERBOARD_SDN2"] = 3812] = "BACON_HERO_FIRST_TRINKET_LEADERBOARD_SDN2";
    GameTag[GameTag["BACON_HERO_SECOND_TRINKET_LEADERBOARD_SDN2"] = 3813] = "BACON_HERO_SECOND_TRINKET_LEADERBOARD_SDN2";
    GameTag[GameTag["BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_SDN2"] = 3814] = "BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_SDN2";
    GameTag[GameTag["BACON_HERO_FIRST_TRINKET_LEADERBOARD_ALT_TEXT"] = 3815] = "BACON_HERO_FIRST_TRINKET_LEADERBOARD_ALT_TEXT";
    GameTag[GameTag["BACON_HERO_SECOND_TRINKET_LEADERBOARD_ALT_TEXT"] = 3816] = "BACON_HERO_SECOND_TRINKET_LEADERBOARD_ALT_TEXT";
    GameTag[GameTag["BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_ALT_TEXT"] = 3817] = "BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_ALT_TEXT";
    GameTag[GameTag["SKIP_ARMOR_ANIMATION"] = 3837] = "SKIP_ARMOR_ANIMATION";
    GameTag[GameTag["DRAW_SPELL_OVERRIDE"] = 3841] = "DRAW_SPELL_OVERRIDE";
    GameTag[GameTag["BACON_NUM_MULLIGAN_REFRESH_USED"] = 3842] = "BACON_NUM_MULLIGAN_REFRESH_USED";
    GameTag[GameTag["DRAENEI_TRIGGER_HINT"] = 3847] = "DRAENEI_TRIGGER_HINT";
    GameTag[GameTag["BACON_NUM_MAX_REROLL_PER_HERO"] = 3848] = "BACON_NUM_MAX_REROLL_PER_HERO";
    GameTag[GameTag["MILL_SPELL_OVERRIDE"] = 3854] = "MILL_SPELL_OVERRIDE";
    GameTag[GameTag["DIVINE_SHIELD_DAMAGE"] = 3859] = "DIVINE_SHIELD_DAMAGE";
    GameTag[GameTag["BATTLECRIES_TRIGGERED_THIS_GAME"] = 3873] = "BATTLECRIES_TRIGGERED_THIS_GAME";
    GameTag[GameTag["BACON_LOCKED_MULLIGAN_HERO"] = 3877] = "BACON_LOCKED_MULLIGAN_HERO";
    GameTag[GameTag["HAS_DARK_GIFT"] = 3880] = "HAS_DARK_GIFT";
    GameTag[GameTag["CREATED_AS_ON_PLAY_REPLACEMENT"] = 3882] = "CREATED_AS_ON_PLAY_REPLACEMENT";
    GameTag[GameTag["SHIFTING_TOP"] = 3894] = "SHIFTING_TOP";
    GameTag[GameTag["BACON_UNLOCK_MULLIGAN_HERO_ENABLED"] = 3896] = "BACON_UNLOCK_MULLIGAN_HERO_ENABLED";
    GameTag[GameTag["BACON_PREMIUM_FREE_REROLLS"] = 3901] = "BACON_PREMIUM_FREE_REROLLS";
    GameTag[GameTag["BACON_NUM_FREE_REROLLS_USED"] = 3904] = "BACON_NUM_FREE_REROLLS_USED";
    GameTag[GameTag["BACON_NUM_PAID_REROLLS_USED"] = 3905] = "BACON_NUM_PAID_REROLLS_USED";
    GameTag[GameTag["BACON_PLAYER_MULLIGAN_HERO_BEEN_REROLLED"] = 3906] = "BACON_PLAYER_MULLIGAN_HERO_BEEN_REROLLED";
    GameTag[GameTag["BG_COMBAT_SPEED_START_TIME"] = 3909] = "BG_COMBAT_SPEED_START_TIME";
    GameTag[GameTag["BG_COMBAT_SPEED_ACCELERATION"] = 3910] = "BG_COMBAT_SPEED_ACCELERATION";
    GameTag[GameTag["BG_COMBAT_SPEED_DECELERATION"] = 3911] = "BG_COMBAT_SPEED_DECELERATION";
    GameTag[GameTag["BG_COMBAT_SPEED_MAX_SPEED"] = 3912] = "BG_COMBAT_SPEED_MAX_SPEED";
    GameTag[GameTag["BG_COMBAT_SPEED_ATTACKS_REMAINING_BEFORE_SLOW_DOWN_MIN"] = 3913] = "BG_COMBAT_SPEED_ATTACKS_REMAINING_BEFORE_SLOW_DOWN_MIN";
    GameTag[GameTag["BACON_MULLIGAN_HERO_REROLL_ACTIVE"] = 3914] = "BACON_MULLIGAN_HERO_REROLL_ACTIVE";
    GameTag[GameTag["STABILIZED"] = 3916] = "STABILIZED";
    GameTag[GameTag["UNSTABLE"] = 3917] = "UNSTABLE";
    GameTag[GameTag["ADDITIONAL_HERO_POWER_INDEX"] = 3919] = "ADDITIONAL_HERO_POWER_INDEX";
    GameTag[GameTag["BACON_PORTAL_IN_SOLO"] = 3925] = "BACON_PORTAL_IN_SOLO";
    GameTag[GameTag["RAFAAM"] = 3928] = "RAFAAM";
    GameTag[GameTag["BACON_TRIGGER_XY_STAY"] = 3932] = "BACON_TRIGGER_XY_STAY";
    GameTag[GameTag["REWIND"] = 3934] = "REWIND";
    GameTag[GameTag["MYTHIC"] = 3935] = "MYTHIC";
    GameTag[GameTag["USED_REWIND"] = 3945] = "USED_REWIND";
    GameTag[GameTag["BACON_BEASTS_SUMMONED_THIS_GAME"] = 3962] = "BACON_BEASTS_SUMMONED_THIS_GAME";
    GameTag[GameTag["NON_KEYWORD_CHARGE"] = 3968] = "NON_KEYWORD_CHARGE";
    GameTag[GameTag["BACON_SHOW_OVERRIDEN_MINION_COST"] = 3973] = "BACON_SHOW_OVERRIDEN_MINION_COST";
    GameTag[GameTag["BACON_SHOW_REFRESH_LEFT_BANNER"] = 3982] = "BACON_SHOW_REFRESH_LEFT_BANNER";
    GameTag[GameTag["SUPPRESS_SPELL_POWER_IN_TEXT"] = 3986] = "SUPPRESS_SPELL_POWER_IN_TEXT";
    GameTag[GameTag["TAVERN_SPELL_ATTACK_INCREASE"] = 3989] = "TAVERN_SPELL_ATTACK_INCREASE";
    GameTag[GameTag["TAVERN_SPELL_HEALTH_INCREASE"] = 3990] = "TAVERN_SPELL_HEALTH_INCREASE";
    GameTag[GameTag["BACON_ELEMENTAL_BUFFHEALTHVALUE"] = 4001] = "BACON_ELEMENTAL_BUFFHEALTHVALUE";
    GameTag[GameTag["BACON_ELEMENTAL_BUFFATKVALUE"] = 4002] = "BACON_ELEMENTAL_BUFFATKVALUE";
    GameTag[GameTag["STARSHIP_LAUNCH_TRIGGER"] = 4013] = "STARSHIP_LAUNCH_TRIGGER";
    GameTag[GameTag["PET_ENTITY"] = 4017] = "PET_ENTITY";
    GameTag[GameTag["KINDRED"] = 4028] = "KINDRED";
    GameTag[GameTag["ADDITIONAL_HERO_POWER_ENTITY_1"] = 4029] = "ADDITIONAL_HERO_POWER_ENTITY_1";
    GameTag[GameTag["IS_RELAUNCHED_STARSHIP"] = 4030] = "IS_RELAUNCHED_STARSHIP";
    GameTag[GameTag["BG_COMBAT_SPEED_MIN_COMBAT_EVENTS_REMAINING_TO_START"] = 4035] = "BG_COMBAT_SPEED_MIN_COMBAT_EVENTS_REMAINING_TO_START";
    GameTag[GameTag["BACON_YAMATO_CANNON"] = 4036] = "BACON_YAMATO_CANNON";
    GameTag[GameTag["PET_VARIANT_ID"] = 4037] = "PET_VARIANT_ID";
    GameTag[GameTag["PET_EVENT_ID"] = 4038] = "PET_EVENT_ID";
    GameTag[GameTag["AGAMAGGAN_CURSE"] = 4056] = "AGAMAGGAN_CURSE";
    GameTag[GameTag["SPEND_CORPSE"] = 4058] = "SPEND_CORPSE";
    GameTag[GameTag["BACON_YAMATO_CANNON_TOOLTIP"] = 4061] = "BACON_YAMATO_CANNON_TOOLTIP";
    GameTag[GameTag["BACON_LIBERATOR_TOOLTIP"] = 4062] = "BACON_LIBERATOR_TOOLTIP";
    GameTag[GameTag["BACON_MEDIVAC_TOOLTIP"] = 4063] = "BACON_MEDIVAC_TOOLTIP";
    GameTag[GameTag["IMMUNE_TO_FIRE_SPELLS"] = 4066] = "IMMUNE_TO_FIRE_SPELLS";
    GameTag[GameTag["BACON_TURNS_TILL_ACTIVE"] = 4069] = "BACON_TURNS_TILL_ACTIVE";
    GameTag[GameTag["BG_COMBAT_SPEED_ATTACKS_REMAINING_BEFORE_SLOW_DOWN_MAX"] = 4072] = "BG_COMBAT_SPEED_ATTACKS_REMAINING_BEFORE_SLOW_DOWN_MAX";
    GameTag[GameTag["PET_SMALL_XP_TRIGGER_COUNT"] = 4076] = "PET_SMALL_XP_TRIGGER_COUNT";
    GameTag[GameTag["PET_MEDIUM_XP_TRIGGER_COUNT"] = 4077] = "PET_MEDIUM_XP_TRIGGER_COUNT";
    GameTag[GameTag["PET_LARGE_XP_TRIGGER_COUNT"] = 4078] = "PET_LARGE_XP_TRIGGER_COUNT";
    GameTag[GameTag["PET_ID"] = 4079] = "PET_ID";
    GameTag[GameTag["FABLED"] = 4085] = "FABLED";
    GameTag[GameTag["BACON_DONT_DISPLAY_HP_IN_LEADERBOARD_OR_STATS"] = 4087] = "BACON_DONT_DISPLAY_HP_IN_LEADERBOARD_OR_STATS";
    GameTag[GameTag["HEROPOWER_UNLIMITED_USES"] = 4088] = "HEROPOWER_UNLIMITED_USES";
    GameTag[GameTag["INTERACTABLE_OBJECT"] = 4089] = "INTERACTABLE_OBJECT";
    GameTag[GameTag["INTERACTABLE_OBJECT_COST"] = 4090] = "INTERACTABLE_OBJECT_COST";
    GameTag[GameTag["ARCANE_TICKET"] = 4091] = "ARCANE_TICKET";
    GameTag[GameTag["BACON_MAGICSHOP"] = 4097] = "BACON_MAGICSHOP";
    GameTag[GameTag["GOLDRINN_MULTIPLIER"] = 4118] = "GOLDRINN_MULTIPLIER";
    GameTag[GameTag["DYNAMIC_KEYWORD1"] = 4161] = "DYNAMIC_KEYWORD1";
    GameTag[GameTag["DYNAMIC_KEYWORD2"] = 4162] = "DYNAMIC_KEYWORD2";
    GameTag[GameTag["HAMUUL_ACTIVE"] = 4167] = "HAMUUL_ACTIVE";
    GameTag[GameTag["IMBUE_SUB_COUNTER"] = 4168] = "IMBUE_SUB_COUNTER";
    GameTag[GameTag["IS_EXTRA_TRIGGERED_POWER"] = 4169] = "IS_EXTRA_TRIGGERED_POWER";
    GameTag[GameTag["DONT_PLAY_VFX_FROM_EXTRA_TRIGGERED_POWER"] = 4174] = "DONT_PLAY_VFX_FROM_EXTRA_TRIGGERED_POWER";
    GameTag[GameTag["FABLED_PLUS"] = 4184] = "FABLED_PLUS";
    GameTag[GameTag["SPELL_SHIELD"] = 4195] = "SPELL_SHIELD";
    GameTag[GameTag["BACON_ELEMENTAL_TOOLTIP"] = 4197] = "BACON_ELEMENTAL_TOOLTIP";
    GameTag[GameTag["BACON_TAVERN_SPELL_TOOLTIP"] = 4198] = "BACON_TAVERN_SPELL_TOOLTIP";
    GameTag[GameTag["BACON_RALLY"] = 4204] = "BACON_RALLY";
    GameTag[GameTag["BACON_GOLD_SPENT_THIS_GAME"] = 4212] = "BACON_GOLD_SPENT_THIS_GAME";
    GameTag[GameTag["SUPPRESS_IMMOLATE_VISUAL_FOR_OPPONENT"] = 4213] = "SUPPRESS_IMMOLATE_VISUAL_FOR_OPPONENT";
    GameTag[GameTag["BACON_HERO_FIRST_TRINKET_LEADERBOARD_SDN3"] = 4218] = "BACON_HERO_FIRST_TRINKET_LEADERBOARD_SDN3";
    GameTag[GameTag["BACON_HERO_SECOND_TRINKET_LEADERBOARD_SDN3"] = 4219] = "BACON_HERO_SECOND_TRINKET_LEADERBOARD_SDN3";
    GameTag[GameTag["BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_SDN3"] = 4220] = "BACON_HERO_HEROPOWER_TRINKET_LEADERBOARD_SDN3";
    GameTag[GameTag["DISPLAY_ENTITY_IN_PLAY_ID"] = 4225] = "DISPLAY_ENTITY_IN_PLAY_ID";
    GameTag[GameTag["BACON_BOUNTY"] = 4231] = "BACON_BOUNTY";
    GameTag[GameTag["REPEATABLE"] = 4238] = "REPEATABLE";
    GameTag[GameTag["SHATTER"] = 4239] = "SHATTER";
    GameTag[GameTag["SHATTERED"] = 4240] = "SHATTERED";
    GameTag[GameTag["BACON_VOLUMIZER"] = 4243] = "BACON_VOLUMIZER";
    GameTag[GameTag["BACON_BUDDY_3_TIMES"] = 4248] = "BACON_BUDDY_3_TIMES";
    GameTag[GameTag["USES_CHARGES"] = 4257] = "USES_CHARGES";
    GameTag[GameTag["PET_TREATS_FED"] = 4280] = "PET_TREATS_FED";
    GameTag[GameTag["PET_TREATS_GENERATED"] = 4281] = "PET_TREATS_GENERATED";
    GameTag[GameTag["BACON_PLAYER_EXTRA_GOLD_NEXT_TURN"] = 4286] = "BACON_PLAYER_EXTRA_GOLD_NEXT_TURN";
    GameTag[GameTag["BACON_PLAYER_OVERDRAWN_GOLD_NEXT_TURN"] = 4287] = "BACON_PLAYER_OVERDRAWN_GOLD_NEXT_TURN";
    GameTag[GameTag["SUPPRESS_ALT_CARD_TEXT_FOR_OPPONENT"] = 4300] = "SUPPRESS_ALT_CARD_TEXT_FOR_OPPONENT";
    GameTag[GameTag["NUM_SHUFFLE_INTO_YOUR_DECK_ACTIONS"] = 4307] = "NUM_SHUFFLE_INTO_YOUR_DECK_ACTIONS";
    GameTag[GameTag["BACON_PLAYER_ENCHANTMENT_DISPLAY_ENABLED"] = 4308] = "BACON_PLAYER_ENCHANTMENT_DISPLAY_ENABLED";
    GameTag[GameTag["HERALD"] = 4309] = "HERALD";
    GameTag[GameTag["BACON_GUIDE_RELATED_CARD"] = 4314] = "BACON_GUIDE_RELATED_CARD";
    GameTag[GameTag["HERALD_COLOSSAL_AMOUNT"] = 4327] = "HERALD_COLOSSAL_AMOUNT";
    GameTag[GameTag["BACON_FACTION_BANNERS_ENABLED"] = 4383] = "BACON_FACTION_BANNERS_ENABLED";
    GameTag[GameTag["PLAYER_DEATHRATTLE"] = 4398] = "PLAYER_DEATHRATTLE";
    GameTag[GameTag["NOZDORMU_HINT"] = 4406] = "NOZDORMU_HINT";
    GameTag[GameTag["DISPLAY_ENTITY_IN_PLAY_HAS_BLUE_GLOW"] = 4407] = "DISPLAY_ENTITY_IN_PLAY_HAS_BLUE_GLOW";
    GameTag[GameTag["BACON_TIMEWARPED_POOL"] = 4408] = "BACON_TIMEWARPED_POOL";
    GameTag[GameTag["NOZDORMU_DORMANT_EFFECT_HELPER"] = 4409] = "NOZDORMU_DORMANT_EFFECT_HELPER";
    GameTag[GameTag["NOZDORMU_DORMANT_EFFECT_WATCHER"] = 4410] = "NOZDORMU_DORMANT_EFFECT_WATCHER";
    GameTag[GameTag["LOCK_VISUAL"] = 4414] = "LOCK_VISUAL";
    GameTag[GameTag["LOCK_VISUAL_STATE"] = 4415] = "LOCK_VISUAL_STATE";
    GameTag[GameTag["BACON_HERO_QUEST_REWARD_SDN1"] = 4423] = "BACON_HERO_QUEST_REWARD_SDN1";
    GameTag[GameTag["BACON_HERO_QUEST_REWARD_SDN2"] = 4424] = "BACON_HERO_QUEST_REWARD_SDN2";
    GameTag[GameTag["BACON_HERO_QUEST_REWARD_ALT_TEXT"] = 4425] = "BACON_HERO_QUEST_REWARD_ALT_TEXT";
    GameTag[GameTag["BACON_HERO_HEROPOWER_QUEST_REWARD_SDN1"] = 4426] = "BACON_HERO_HEROPOWER_QUEST_REWARD_SDN1";
    GameTag[GameTag["BACON_HERO_HEROPOWER_QUEST_REWARD_SDN2"] = 4427] = "BACON_HERO_HEROPOWER_QUEST_REWARD_SDN2";
    GameTag[GameTag["BACON_HERO_HEROPOWER_QUEST_REWARD_ALT_TEXT"] = 4428] = "BACON_HERO_HEROPOWER_QUEST_REWARD_ALT_TEXT";
    GameTag[GameTag["HERO_CANT_BE_DESTROYED"] = 4439] = "HERO_CANT_BE_DESTROYED";
    GameTag[GameTag["BACON_ALT_TAVERN_COIN"] = 4443] = "BACON_ALT_TAVERN_COIN";
    GameTag[GameTag["BACON_ALT_TAVERN_COIN_USED"] = 4444] = "BACON_ALT_TAVERN_COIN_USED";
    GameTag[GameTag["IS_FABLED_BUNDLE_CARD"] = 4445] = "IS_FABLED_BUNDLE_CARD";
    GameTag[GameTag["AZURE_ARCANE_SPELL_BUFF"] = 4448] = "AZURE_ARCANE_SPELL_BUFF";
    GameTag[GameTag["BACON_ALT_TAVERN_IN_PROGRESS"] = 4451] = "BACON_ALT_TAVERN_IN_PROGRESS";
    GameTag[GameTag["PET_LEVELOVERRIDE"] = 4452] = "PET_LEVELOVERRIDE";
    GameTag[GameTag["BACON_VOLUMIZER_ATTACK_BUFF"] = 4468] = "BACON_VOLUMIZER_ATTACK_BUFF";
    GameTag[GameTag["BACON_VOLUMIZER_HEALTH_BUFF"] = 4469] = "BACON_VOLUMIZER_HEALTH_BUFF";
    GameTag[GameTag["IS_SHOP_CHOICE"] = 4480] = "IS_SHOP_CHOICE";
    GameTag[GameTag["IS_A_SHOP_CHOICE_CARD"] = 4486] = "IS_A_SHOP_CHOICE_CARD";
    GameTag[GameTag["BACON_TIMEWARPED"] = 4503] = "BACON_TIMEWARPED";
    GameTag[GameTag["VISIBLE_PLAYER_ENCHANTMENT_CARD"] = 4508] = "VISIBLE_PLAYER_ENCHANTMENT_CARD";
    GameTag[GameTag["BACON_TIMES_VISITED_ALT_TAVERN"] = 4514] = "BACON_TIMES_VISITED_ALT_TAVERN";
    GameTag[GameTag["DISPLAY_ENTITY_ON_MOUSEOVER"] = 4516] = "DISPLAY_ENTITY_ON_MOUSEOVER";
    GameTag[GameTag["BACON_TIMEWARP_TOOLTIP"] = 4517] = "BACON_TIMEWARP_TOOLTIP";
    GameTag[GameTag["BACON_ALT_TAVERN_SYSTEM_ACTIVE"] = 4519] = "BACON_ALT_TAVERN_SYSTEM_ACTIVE";
    GameTag[GameTag["BACON_TURNS_UNTIL_ALT_TAVERN"] = 4529] = "BACON_TURNS_UNTIL_ALT_TAVERN";
    GameTag[GameTag["HERALD_COLOSSAL_CLASS"] = 4534] = "HERALD_COLOSSAL_CLASS";
    GameTag[GameTag["BACON_FREE_REFRESH_COUNT"] = 4536] = "BACON_FREE_REFRESH_COUNT";
    GameTag[GameTag["BACON_BLACK_MARKET_EXTRA_CHRONUM"] = 4542] = "BACON_BLACK_MARKET_EXTRA_CHRONUM";
    GameTag[GameTag["BIG_CARD_AS_TOOLTIP_USE_TOOKTIP_SYSTEM"] = 4544] = "BIG_CARD_AS_TOOLTIP_USE_TOOKTIP_SYSTEM";
    GameTag[GameTag["BACON_PIRATE_BUFFATKVALUE"] = 4567] = "BACON_PIRATE_BUFFATKVALUE";
    GameTag[GameTag["BACON_PIRATE_BUFFHEALTHVALUE"] = 4568] = "BACON_PIRATE_BUFFHEALTHVALUE";
    GameTag[GameTag["HAS_TIMEWARPED_TAVERN_ALT_TEXT"] = 4579] = "HAS_TIMEWARPED_TAVERN_ALT_TEXT";
    GameTag[GameTag["DECK_BUILDER_BAR_AMOUNT"] = 4601] = "DECK_BUILDER_BAR_AMOUNT";
    GameTag[GameTag["HERALD_COLOSSAL_PREMIUM"] = 4602] = "HERALD_COLOSSAL_PREMIUM";
    GameTag[GameTag["DORMANT_AWAKENED_THIS_TURN"] = 4606] = "DORMANT_AWAKENED_THIS_TURN";
    GameTag[GameTag["BACON_PIRATE_TOOLTIP"] = 4613] = "BACON_PIRATE_TOOLTIP";
    GameTag[GameTag["SHOULD_TRACK_DECK_COST"] = 4626] = "SHOULD_TRACK_DECK_COST";
    GameTag[GameTag["SKIP_HERO_TWEEN"] = 4628] = "SKIP_HERO_TWEEN";
    GameTag[GameTag["BACON_TIMEWARP_TURN1_TOOLTIP"] = 4635] = "BACON_TIMEWARP_TURN1_TOOLTIP";
    GameTag[GameTag["BACON_TIMEWARP_TURN2_TOOLTIP"] = 4636] = "BACON_TIMEWARP_TURN2_TOOLTIP";
    GameTag[GameTag["BACON_TOTAL_TIMEWARPS_FOR_GAME"] = 4637] = "BACON_TOTAL_TIMEWARPS_FOR_GAME";
    GameTag[GameTag["BACON_TIMEWARP_CHRONUM_TOOLTIP"] = 4638] = "BACON_TIMEWARP_CHRONUM_TOOLTIP";
    GameTag[GameTag["ALEXSTRASZA_HEALTHTRACKER"] = 4650] = "ALEXSTRASZA_HEALTHTRACKER";
    GameTag[GameTag["END_OF_TURN"] = 133790001] = "END_OF_TURN";
    GameTag[GameTag["SUMMON"] = 133790002] = "SUMMON";
    GameTag[GameTag["BG_SPELL"] = 133790003] = "BG_SPELL";
    GameTag[GameTag["DEAL_DAMAGE"] = 133790004] = "DEAL_DAMAGE";
    GameTag[GameTag["SHOW_ENTITY_START"] = 133790006] = "SHOW_ENTITY_START";
    GameTag[GameTag["SHOW_ENTITY_END"] = 133790007] = "SHOW_ENTITY_END";
    GameTag[GameTag["GENERATE_CORPSE"] = 133790008] = "GENERATE_CORPSE";
    GameTag[GameTag["BACON_BUFFS_TAVERN_SPELL"] = 133790009] = "BACON_BUFFS_TAVERN_SPELL";
    GameTag[GameTag["GIVES_ABYSSAL_CURSE"] = 133790010] = "GIVES_ABYSSAL_CURSE";
    GameTag[GameTag["BACON_TARGET_NO_TRIBE_MINION"] = 133790011] = "BACON_TARGET_NO_TRIBE_MINION";
})(GameTag = exports.GameTag || (exports.GameTag = {}));
