// @firestone-hs/reference-data/dist/enums/game-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GameType = void 0;
var GameType;
(function (GameType) {
    GameType[GameType["GT_UNKNOWN"] = 0] = "GT_UNKNOWN";
    GameType[GameType["GT_VS_AI"] = 1] = "GT_VS_AI";
    GameType[GameType["GT_VS_FRIEND"] = 2] = "GT_VS_FRIEND";
    GameType[GameType["GT_TUTORIAL"] = 4] = "GT_TUTORIAL";
    GameType[GameType["GT_ARENA"] = 5] = "GT_ARENA";
    GameType[GameType["GT_TEST_AI_VS_AI"] = 6] = "GT_TEST_AI_VS_AI";
    GameType[GameType["GT_RANKED"] = 7] = "GT_RANKED";
    GameType[GameType["GT_CASUAL"] = 8] = "GT_CASUAL";
    GameType[GameType["GT_TAVERNBRAWL"] = 16] = "GT_TAVERNBRAWL";
    GameType[GameType["GT_TB_1P_VS_AI"] = 17] = "GT_TB_1P_VS_AI";
    GameType[GameType["GT_TB_2P_COOP"] = 18] = "GT_TB_2P_COOP";
    GameType[GameType["GT_FSG_BRAWL_VS_FRIEND"] = 19] = "GT_FSG_BRAWL_VS_FRIEND";
    GameType[GameType["GT_FSG_BRAWL"] = 20] = "GT_FSG_BRAWL";
    GameType[GameType["GT_FSG_BRAWL_1P_VS_AI"] = 21] = "GT_FSG_BRAWL_1P_VS_AI";
    GameType[GameType["GT_FSG_BRAWL_2P_COOP"] = 22] = "GT_FSG_BRAWL_2P_COOP";
    GameType[GameType["GT_BATTLEGROUNDS"] = 23] = "GT_BATTLEGROUNDS";
    GameType[GameType["GT_BATTLEGROUNDS_FRIENDLY"] = 24] = "GT_BATTLEGROUNDS_FRIENDLY";
    GameType[GameType["GT_PVPDR_PAID"] = 28] = "GT_PVPDR_PAID";
    GameType[GameType["GT_PVPDR"] = 29] = "GT_PVPDR";
    GameType[GameType["GT_MERCENARIES_PVP"] = 30] = "GT_MERCENARIES_PVP";
    GameType[GameType["GT_MERCENARIES_PVE"] = 31] = "GT_MERCENARIES_PVE";
    GameType[GameType["GT_MERCENARIES_PVE_COOP"] = 32] = "GT_MERCENARIES_PVE_COOP";
    GameType[GameType["GT_MERCENARIES_AI_VS_AI"] = 33] = "GT_MERCENARIES_AI_VS_AI";
    GameType[GameType["GT_MERCENARIES_FRIENDLY"] = 34] = "GT_MERCENARIES_FRIENDLY";
    GameType[GameType["GT_BATTLEGROUNDS_AI_VS_AI"] = 35] = "GT_BATTLEGROUNDS_AI_VS_AI";
    GameType[GameType["GT_BATTLEGROUNDS_PLAYER_VS_AI"] = 36] = "GT_BATTLEGROUNDS_PLAYER_VS_AI";
    GameType[GameType["GT_BATTLEGROUNDS_DUO"] = 37] = "GT_BATTLEGROUNDS_DUO";
    GameType[GameType["GT_BATTLEGROUNDS_DUO_VS_AI"] = 38] = "GT_BATTLEGROUNDS_DUO_VS_AI";
    GameType[GameType["GT_BATTLEGROUNDS_DUO_FRIENDLY"] = 39] = "GT_BATTLEGROUNDS_DUO_FRIENDLY";
    GameType[GameType["GT_BATTLEGROUNDS_DUO_AI_VS_AI"] = 40] = "GT_BATTLEGROUNDS_DUO_AI_VS_AI";
    GameType[GameType["GT_BATTLEGROUNDS_DUO_1_PLAYER_VS_AI"] = 41] = "GT_BATTLEGROUNDS_DUO_1_PLAYER_VS_AI";
    GameType[GameType["GT_UNDERGROUND_ARENA"] = 42] = "GT_UNDERGROUND_ARENA";
    GameType[GameType["GT_ACHIEVEMENT_TEST"] = 43] = "GT_ACHIEVEMENT_TEST";
    GameType[GameType["GT_ARENA_PLAYER_VS_AI"] = 44] = "GT_ARENA_PLAYER_VS_AI";
    GameType[GameType["GT_UNDERGROUND_ARENA_PLAYER_VS_AI"] = 45] = "GT_UNDERGROUND_ARENA_PLAYER_VS_AI";
})(GameType = exports.GameType || (exports.GameType = {}));
