// @firestone-hs/reference-data/dist/enums/meta-tags.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetaTags = void 0;
var MetaTags;
(function (MetaTags) {
    MetaTags[MetaTags["TARGET"] = 0] = "TARGET";
    MetaTags[MetaTags["DAMAGE"] = 1] = "DAMAGE";
    MetaTags[MetaTags["HEALING"] = 2] = "HEALING";
    MetaTags[MetaTags["JOUST"] = 3] = "JOUST";
    MetaTags[MetaTags["SHOW_BIG_CARD"] = 5] = "SHOW_BIG_CARD";
    MetaTags[MetaTags["EFFECT_TIMING"] = 6] = "EFFECT_TIMING";
    MetaTags[MetaTags["HISTORY_TARGET"] = 7] = "HISTORY_TARGET";
    MetaTags[MetaTags["OVERRIDE_HISTORY"] = 8] = "OVERRIDE_HISTORY";
    MetaTags[MetaTags["HISTORY_TARGET_DONT_DUPLICATE_UNTIL_END"] = 9] = "HISTORY_TARGET_DONT_DUPLICATE_UNTIL_END";
    MetaTags[MetaTags["BEGIN_ARTIFICIAL_HISTORY_TILE"] = 10] = "BEGIN_ARTIFICIAL_HISTORY_TILE";
    MetaTags[MetaTags["BEGIN_ARTIFICIAL_HISTORY_TRIGGER_TILE"] = 11] = "BEGIN_ARTIFICIAL_HISTORY_TRIGGER_TILE";
    MetaTags[MetaTags["END_ARTIFICIAL_HISTORY_TILE"] = 12] = "END_ARTIFICIAL_HISTORY_TILE";
    MetaTags[MetaTags["START_DRAW"] = 13] = "START_DRAW";
    MetaTags[MetaTags["BURNED_CARD"] = 14] = "BURNED_CARD";
    MetaTags[MetaTags["EFFECT_SELECTION"] = 15] = "EFFECT_SELECTION";
    MetaTags[MetaTags["BEGIN_LISTENING_FOR_TURN_EVENTS"] = 16] = "BEGIN_LISTENING_FOR_TURN_EVENTS";
    MetaTags[MetaTags["HOLD_DRAWN_CARD"] = 17] = "HOLD_DRAWN_CARD";
    MetaTags[MetaTags["CONTROLLER_AND_ZONE_CHANGE"] = 18] = "CONTROLLER_AND_ZONE_CHANGE";
    MetaTags[MetaTags["ARTIFICIAL_PAUSE"] = 19] = "ARTIFICIAL_PAUSE";
    MetaTags[MetaTags["SLUSH_TIME"] = 20] = "SLUSH_TIME";
    MetaTags[MetaTags["ARTIFICIAL_HISTORY_INTERRUPT"] = 21] = "ARTIFICIAL_HISTORY_INTERRUPT";
    MetaTags[MetaTags["POISONOUS"] = 22] = "POISONOUS";
    MetaTags[MetaTags["CRITICAL_HIT"] = 23] = "CRITICAL_HIT";
    MetaTags[MetaTags["HISTORY_TRIGGER_SOURCE"] = 24] = "HISTORY_TRIGGER_SOURCE";
    MetaTags[MetaTags["HISTORY_SOURCE_OWNER"] = 25] = "HISTORY_SOURCE_OWNER";
    MetaTags[MetaTags["HISTORY_REMOVE_ENTITIES"] = 26] = "HISTORY_REMOVE_ENTITIES";
    MetaTags[MetaTags["SPEND_HEALTH"] = 27] = "SPEND_HEALTH";
    MetaTags[MetaTags["SPEND_ARMOR"] = 28] = "SPEND_ARMOR";
})(MetaTags = exports.MetaTags || (exports.MetaTags = {}));
