// @firestone-hs/reference-data/dist/enums/multi-class-group.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultiClassGroup = void 0;
var MultiClassGroup;
(function (MultiClassGroup) {
    MultiClassGroup[MultiClassGroup["INVALID"] = 0] = "INVALID";
    MultiClassGroup[MultiClassGroup["GRIMY_GOONS"] = 1] = "GRIMY_GOONS";
    MultiClassGroup[MultiClassGroup["JADE_LOTUS"] = 2] = "JADE_LOTUS";
    MultiClassGroup[MultiClassGroup["KABAL"] = 3] = "KABAL";
    MultiClassGroup[MultiClassGroup["PALADIN_PRIEST"] = 4] = "PALADIN_PRIEST";
    MultiClassGroup[MultiClassGroup["PRIEST_WARLOCK"] = 5] = "PRIEST_WARLOCK";
    MultiClassGroup[MultiClassGroup["WARLOCK_DEMONHUNTER"] = 6] = "WARLOCK_DEMONHUNTER";
    MultiClassGroup[MultiClassGroup["HUNTER_DEMONHUNTER"] = 7] = "HUNTER_DEMONHUNTER";
    MultiClassGroup[MultiClassGroup["DRUID_HUNTER"] = 8] = "DRUID_HUNTER";
    MultiClassGroup[MultiClassGroup["DRUID_SHAMAN"] = 9] = "DRUID_SHAMAN";
    MultiClassGroup[MultiClassGroup["MAGE_SHAMAN"] = 10] = "MAGE_SHAMAN";
    MultiClassGroup[MultiClassGroup["MAGE_ROGUE"] = 11] = "MAGE_ROGUE";
    MultiClassGroup[MultiClassGroup["ROGUE_WARRIOR"] = 12] = "ROGUE_WARRIOR";
    MultiClassGroup[MultiClassGroup["PALADIN_WARRIOR"] = 13] = "PALADIN_WARRIOR";
})(MultiClassGroup = exports.MultiClassGroup || (exports.MultiClassGroup = {}));
