// @firestone-hs/reference-data/dist/enums/race.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Race = void 0;
var Race;
(function (Race) {
    Race[Race["INVALID"] = 0] = "INVALID";
    Race[Race["BLOODELF"] = 1] = "BLOODELF";
    Race[Race["DRAENEI"] = 2] = "DRAENEI";
    Race[Race["DWARF"] = 3] = "DWARF";
    Race[Race["GNOME"] = 4] = "GNOME";
    Race[Race["GOBLIN"] = 5] = "GOBLIN";
    Race[Race["HUMAN"] = 6] = "HUMAN";
    Race[Race["NIGHTELF"] = 7] = "NIGHTELF";
    Race[Race["ORC"] = 8] = "ORC";
    Race[Race["TAUREN"] = 9] = "TAUREN";
    Race[Race["TROLL"] = 10] = "TROLL";
    Race[Race["UNDEAD"] = 11] = "UNDEAD";
    Race[Race["WORGEN"] = 12] = "WORGEN";
    Race[Race["GOBLIN2"] = 13] = "GOBLIN2";
    Race[Race["MURLOC"] = 14] = "MURLOC";
    Race[Race["DEMON"] = 15] = "DEMON";
    Race[Race["SCOURGE"] = 16] = "SCOURGE";
    Race[Race["MECH"] = 17] = "MECH";
    Race[Race["ELEMENTAL"] = 18] = "ELEMENTAL";
    Race[Race["OGRE"] = 19] = "OGRE";
    Race[Race["BEAST"] = 20] = "BEAST";
    Race[Race["TOTEM"] = 21] = "TOTEM";
    Race[Race["NERUBIAN"] = 22] = "NERUBIAN";
    Race[Race["PIRATE"] = 23] = "PIRATE";
    Race[Race["DRAGON"] = 24] = "DRAGON";
    Race[Race["BLANK"] = 25] = "BLANK";
    Race[Race["ALL"] = 26] = "ALL";
    Race[Race["EGG"] = 38] = "EGG";
    Race[Race["QUILBOAR"] = 43] = "QUILBOAR";
    Race[Race["CENTAUR"] = 80] = "CENTAUR";
    Race[Race["FURBOLG"] = 81] = "FURBOLG";
    Race[Race["HIGHELF"] = 83] = "HIGHELF";
    Race[Race["TREANT"] = 84] = "TREANT";
    Race[Race["OWLKIN"] = 85] = "OWLKIN";
    Race[Race["HALFORC"] = 88] = "HALFORC";
    Race[Race["LOCK"] = 89] = "LOCK";
    Race[Race["NAGA"] = 92] = "NAGA";
    Race[Race["OLDGOD"] = 93] = "OLDGOD";
    Race[Race["PANDAREN"] = 94] = "PANDAREN";
    Race[Race["GRONN"] = 95] = "GRONN";
    Race[Race["CELESTIAL"] = 96] = "CELESTIAL";
    Race[Race["GNOLL"] = 97] = "GNOLL";
    Race[Race["GOLEM"] = 98] = "GOLEM";
    Race[Race["HARPY"] = 99] = "HARPY";
    Race[Race["VULPERA"] = 100] = "VULPERA";
})(Race = exports.Race || (exports.Race = {}));
