// @firestone-hs/reference-data/dist/enums/reward-item-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RewardItemType = void 0;
var RewardItemType;
(function (RewardItemType) {
    RewardItemType[RewardItemType["NONE"] = 0] = "NONE";
    RewardItemType[RewardItemType["GOLD"] = 1] = "GOLD";
    RewardItemType[RewardItemType["DUST"] = 2] = "DUST";
    RewardItemType[RewardItemType["ARCANE_ORBS"] = 3] = "ARCANE_ORBS";
    RewardItemType[RewardItemType["BOOSTER"] = 4] = "BOOSTER";
    RewardItemType[RewardItemType["CARD"] = 6] = "CARD";
    RewardItemType[RewardItemType["RANDOM_CARD"] = 7] = "RANDOM_CARD";
    RewardItemType[RewardItemType["TAVERN_TICKET"] = 8] = "TAVERN_TICKET";
    RewardItemType[RewardItemType["CARD_BACK"] = 9] = "CARD_BACK";
    RewardItemType[RewardItemType["HERO_SKIN"] = 10] = "HERO_SKIN";
    RewardItemType[RewardItemType["CUSTOM_COIN"] = 11] = "CUSTOM_COIN";
    RewardItemType[RewardItemType["REWARD_TRACK_XP_BOOST"] = 12] = "REWARD_TRACK_XP_BOOST";
    RewardItemType[RewardItemType["CARD_SUBSET"] = 13] = "CARD_SUBSET";
    RewardItemType[RewardItemType["MERCENARY_CURRENCY"] = 14] = "MERCENARY_CURRENCY";
    RewardItemType[RewardItemType["MERCENARY_EQUIPMENT"] = 15] = "MERCENARY_EQUIPMENT";
    RewardItemType[RewardItemType["MERCENARY_XP"] = 16] = "MERCENARY_XP";
    RewardItemType[RewardItemType["MERCENARY"] = 18] = "MERCENARY";
    RewardItemType[RewardItemType["BATTLEGROUNDS_HERO_SKIN"] = 19] = "BATTLEGROUNDS_HERO_SKIN";
    RewardItemType[RewardItemType["BATTLEGROUNDS_GUIDE_SKIN"] = 20] = "BATTLEGROUNDS_GUIDE_SKIN";
    RewardItemType[RewardItemType["BATTLEGROUNDS_BOARD_SKIN"] = 21] = "BATTLEGROUNDS_BOARD_SKIN";
    RewardItemType[RewardItemType["BATTLEGROUNDS_FINISHER"] = 22] = "BATTLEGROUNDS_FINISHER";
    RewardItemType[RewardItemType["BATTLEGROUNDS_EMOTE"] = 23] = "BATTLEGROUNDS_EMOTE";
    RewardItemType[RewardItemType["RENOWN"] = 24] = "RENOWN";
    RewardItemType[RewardItemType["BATTLEGROUNDS_SEASON_BONUS"] = 25] = "BATTLEGROUNDS_SEASON_BONUS";
    RewardItemType[RewardItemType["HERO_CLASS"] = 26] = "HERO_CLASS";
    RewardItemType[RewardItemType["GAME_MODE"] = 27] = "GAME_MODE";
    RewardItemType[RewardItemType["DECK"] = 28] = "DECK";
    RewardItemType[RewardItemType["LOANER_DECKS"] = 29] = "LOANER_DECKS";
    RewardItemType[RewardItemType["BATTLEGROUNDS_TOKEN"] = 30] = "BATTLEGROUNDS_TOKEN";
    RewardItemType[RewardItemType["PET"] = 31] = "PET";
    RewardItemType[RewardItemType["PET_VARIANT"] = 32] = "PET_VARIANT";
})(RewardItemType = exports.RewardItemType || (exports.RewardItemType = {}));
