// @firestone-hs/reference-data/dist/enums/reward-type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RewardType = void 0;
var RewardType;
(function (RewardType) {
    RewardType[RewardType["NONE"] = -1] = "NONE";
    RewardType[RewardType["ARCANE_DUST"] = 0] = "ARCANE_DUST";
    RewardType[RewardType["BOOSTER_PACK"] = 1] = "BOOSTER_PACK";
    RewardType[RewardType["CARD"] = 2] = "CARD";
    RewardType[RewardType["CARD_BACK"] = 3] = "CARD_BACK";
    RewardType[RewardType["CRAFTABLE_CARD"] = 4] = "CRAFTABLE_CARD";
    RewardType[RewardType["FORGE_TICKET"] = 5] = "FORGE_TICKET";
    RewardType[RewardType["GOLD"] = 6] = "GOLD";
    RewardType[RewardType["MOUNT"] = 7] = "MOUNT";
    RewardType[RewardType["CLASS_CHALLENGE"] = 8] = "CLASS_CHALLENGE";
    RewardType[RewardType["EVENT"] = 9] = "EVENT";
    RewardType[RewardType["RANDOM_CARD"] = 10] = "RANDOM_CARD";
    RewardType[RewardType["BONUS_CHALLENGE"] = 11] = "BONUS_CHALLENGE";
    RewardType[RewardType["ADVENTURE_DECK"] = 12] = "ADVENTURE_DECK";
    RewardType[RewardType["ADVENTURE_HERO_POWER"] = 13] = "ADVENTURE_HERO_POWER";
    RewardType[RewardType["ARCANE_ORBS"] = 14] = "ARCANE_ORBS";
    RewardType[RewardType["DECK"] = 15] = "DECK";
    RewardType[RewardType["MINI_SET"] = 16] = "MINI_SET";
    RewardType[RewardType["MERCENARY_COIN"] = 17] = "MERCENARY_COIN";
    RewardType[RewardType["MERCENARY_EXP"] = 18] = "MERCENARY_EXP";
    RewardType[RewardType["MERCENARY_ABILITY_UNLOCK"] = 19] = "MERCENARY_ABILITY_UNLOCK";
    RewardType[RewardType["MERCENARY_EQUIPMENT"] = 20] = "MERCENARY_EQUIPMENT";
    RewardType[RewardType["REWARD_ITEM"] = 21] = "REWARD_ITEM";
    RewardType[RewardType["MERCENARY_BOOSTER"] = 22] = "MERCENARY_BOOSTER";
    RewardType[RewardType["MERCENARY_MERCENARY"] = 23] = "MERCENARY_MERCENARY";
    RewardType[RewardType["MERCENARY_RANDOM_MERCENARY"] = 24] = "MERCENARY_RANDOM_MERCENARY";
    RewardType[RewardType["MERCENARY_KNOCKOUT"] = 25] = "MERCENARY_KNOCKOUT";
    RewardType[RewardType["BATTLEGROUNDS_GUIDE_SKIN"] = 26] = "BATTLEGROUNDS_GUIDE_SKIN";
    RewardType[RewardType["BATTLEGROUNDS_HERO_SKIN"] = 27] = "BATTLEGROUNDS_HERO_SKIN";
    RewardType[RewardType["BATTLEGROUNDS_FINISHER"] = 28] = "BATTLEGROUNDS_FINISHER";
    RewardType[RewardType["BATTLEGROUNDS_BOARD_SKIN"] = 29] = "BATTLEGROUNDS_BOARD_SKIN";
    RewardType[RewardType["BATTLEGROUNDS_EMOTE"] = 30] = "BATTLEGROUNDS_EMOTE";
    RewardType[RewardType["MERCENARY_RENOWN"] = 31] = "MERCENARY_RENOWN";
    RewardType[RewardType["BATTLEGROUNDS_TOKEN"] = 32] = "BATTLEGROUNDS_TOKEN";
    RewardType[RewardType["PET"] = 33] = "PET";
    RewardType[RewardType["PET_VARIANT"] = 34] = "PET_VARIANT";
})(RewardType = exports.RewardType || (exports.RewardType = {}));
