// @firestone-hs/reference-data/dist/enums/scenario-id.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScenarioId = void 0;
var ScenarioId;
(function (ScenarioId) {
    ScenarioId[ScenarioId["SAMSUNG_PROMOTION_FAKE_ENTRY"] = 1] = "SAMSUNG_PROMOTION_FAKE_ENTRY";
    ScenarioId[ScenarioId["STANDARD_1_VS_1_GAME"] = 2] = "STANDARD_1_VS_1_GAME";
    ScenarioId[ScenarioId["TUTORIAL___HOGGER"] = 3] = "TUTORIAL___HOGGER";
    ScenarioId[ScenarioId["TUTORIAL___MILLHOUSE"] = 4] = "TUTORIAL___MILLHOUSE";
    ScenarioId[ScenarioId["TUTORIAL___MUKLA"] = 181] = "TUTORIAL___MUKLA";
    ScenarioId[ScenarioId["TUTORIAL___NESINGWARY"] = 201] = "TUTORIAL___NESINGWARY";
    ScenarioId[ScenarioId["TUTORIAL___ILLIDAN"] = 248] = "TUTORIAL___ILLIDAN";
    ScenarioId[ScenarioId["TUTORIAL___LOREMASTER_CHO"] = 249] = "TUTORIAL___LOREMASTER_CHO";
    ScenarioId[ScenarioId["AIDECK___NORMAL_MAGE"] = 252] = "AIDECK___NORMAL_MAGE";
    ScenarioId[ScenarioId["AIDECK___NORMAL_WARLOCK"] = 253] = "AIDECK___NORMAL_WARLOCK";
    ScenarioId[ScenarioId["AIDECK___NORMAL_HUNTER"] = 256] = "AIDECK___NORMAL_HUNTER";
    ScenarioId[ScenarioId["AIDECK___NORMAL_ROGUE"] = 257] = "AIDECK___NORMAL_ROGUE";
    ScenarioId[ScenarioId["AIDECK___NORMAL_PRIEST"] = 258] = "AIDECK___NORMAL_PRIEST";
    ScenarioId[ScenarioId["AIDECK___NORMAL_WARRIOR"] = 259] = "AIDECK___NORMAL_WARRIOR";
    ScenarioId[ScenarioId["AIDECK___EXPERT_MAGE"] = 260] = "AIDECK___EXPERT_MAGE";
    ScenarioId[ScenarioId["AIDECK___NORMAL_DRUID"] = 261] = "AIDECK___NORMAL_DRUID";
    ScenarioId[ScenarioId["AIDECK___NORMAL_PALADIN"] = 262] = "AIDECK___NORMAL_PALADIN";
    ScenarioId[ScenarioId["AIDECK___NORMAL_SHAMAN"] = 263] = "AIDECK___NORMAL_SHAMAN";
    ScenarioId[ScenarioId["AIDECK___EXPERT_WARRIOR"] = 264] = "AIDECK___EXPERT_WARRIOR";
    ScenarioId[ScenarioId["AIDECK___EXPERT_PRIEST"] = 265] = "AIDECK___EXPERT_PRIEST";
    ScenarioId[ScenarioId["AIDECK___EXPERT_WARLOCK"] = 266] = "AIDECK___EXPERT_WARLOCK";
    ScenarioId[ScenarioId["AIDECK___EXPERT_DRUID"] = 267] = "AIDECK___EXPERT_DRUID";
    ScenarioId[ScenarioId["AIDECK___EXPERT_ROGUE"] = 268] = "AIDECK___EXPERT_ROGUE";
    ScenarioId[ScenarioId["AIDECK___EXPERT_HUNTER"] = 269] = "AIDECK___EXPERT_HUNTER";
    ScenarioId[ScenarioId["AIDECK___EXPERT_PALADIN"] = 270] = "AIDECK___EXPERT_PALADIN";
    ScenarioId[ScenarioId["AIDECK___EXPERT_SHAMAN"] = 271] = "AIDECK___EXPERT_SHAMAN";
    ScenarioId[ScenarioId["NAX_01_ANUB"] = 273] = "NAX_01_ANUB";
    ScenarioId[ScenarioId["NAX_02_FAERLINA"] = 274] = "NAX_02_FAERLINA";
    ScenarioId[ScenarioId["NAX_04_NOTH"] = 275] = "NAX_04_NOTH";
    ScenarioId[ScenarioId["NAX_05_HEIGAN"] = 276] = "NAX_05_HEIGAN";
    ScenarioId[ScenarioId["NAX_06_LOATHEB"] = 277] = "NAX_06_LOATHEB";
    ScenarioId[ScenarioId["NAX_03_MAEXXNA"] = 278] = "NAX_03_MAEXXNA";
    ScenarioId[ScenarioId["NAX_07_RAZUVIOUS"] = 279] = "NAX_07_RAZUVIOUS";
    ScenarioId[ScenarioId["NAX_08_GOTHIK"] = 280] = "NAX_08_GOTHIK";
    ScenarioId[ScenarioId["NAX_09_HORSEMEN"] = 281] = "NAX_09_HORSEMEN";
    ScenarioId[ScenarioId["NAX_10_PATCHWERK"] = 282] = "NAX_10_PATCHWERK";
    ScenarioId[ScenarioId["NAX_11_GROBBULUS"] = 283] = "NAX_11_GROBBULUS";
    ScenarioId[ScenarioId["NAX_12_GLUTH"] = 284] = "NAX_12_GLUTH";
    ScenarioId[ScenarioId["NAX_13_THADDIUS"] = 285] = "NAX_13_THADDIUS";
    ScenarioId[ScenarioId["NAX_15_KELTHUZAD"] = 286] = "NAX_15_KELTHUZAD";
    ScenarioId[ScenarioId["NAX_14_SAPPHIRON"] = 287] = "NAX_14_SAPPHIRON";
    ScenarioId[ScenarioId["NAX_CHALLENGE_HUNTER"] = 288] = "NAX_CHALLENGE_HUNTER";
    ScenarioId[ScenarioId["NAX_CHALLENGE_WARRIOR"] = 292] = "NAX_CHALLENGE_WARRIOR";
    ScenarioId[ScenarioId["NAX_CHALLENGE_ROGUE"] = 293] = "NAX_CHALLENGE_ROGUE";
    ScenarioId[ScenarioId["NAX_CHALLENGE_DRUID"] = 294] = "NAX_CHALLENGE_DRUID";
    ScenarioId[ScenarioId["NAX_CHALLENGE_PRIEST"] = 295] = "NAX_CHALLENGE_PRIEST";
    ScenarioId[ScenarioId["NAX_CHALLENGE_SHAMAN"] = 296] = "NAX_CHALLENGE_SHAMAN";
    ScenarioId[ScenarioId["NAX_CHALLENGE_MAGE"] = 297] = "NAX_CHALLENGE_MAGE";
    ScenarioId[ScenarioId["NAX_CHALLENGE_PALADIN"] = 298] = "NAX_CHALLENGE_PALADIN";
    ScenarioId[ScenarioId["NAX_CHALLENGE_WARLOCK"] = 299] = "NAX_CHALLENGE_WARLOCK";
    ScenarioId[ScenarioId["NAX_01_ANUB_HEROIC"] = 300] = "NAX_01_ANUB_HEROIC";
    ScenarioId[ScenarioId["NAX_02_FAERLINA_HEROIC"] = 301] = "NAX_02_FAERLINA_HEROIC";
    ScenarioId[ScenarioId["NAX_03_MAEXXNA_HEROIC"] = 302] = "NAX_03_MAEXXNA_HEROIC";
    ScenarioId[ScenarioId["NAX_04_NOTH_HEROIC"] = 303] = "NAX_04_NOTH_HEROIC";
    ScenarioId[ScenarioId["NAX_05_HEIGAN_HEROIC"] = 304] = "NAX_05_HEIGAN_HEROIC";
    ScenarioId[ScenarioId["NAX_06_LOATHEB_HEROIC"] = 305] = "NAX_06_LOATHEB_HEROIC";
    ScenarioId[ScenarioId["NAX_07_RAZUVIOUS_HEROIC"] = 306] = "NAX_07_RAZUVIOUS_HEROIC";
    ScenarioId[ScenarioId["NAX_08_GOTHIK_HEROIC"] = 307] = "NAX_08_GOTHIK_HEROIC";
    ScenarioId[ScenarioId["NAX_09_HORSEMEN_HEROIC"] = 308] = "NAX_09_HORSEMEN_HEROIC";
    ScenarioId[ScenarioId["NAX_10_PATCHWERK_HEROIC"] = 309] = "NAX_10_PATCHWERK_HEROIC";
    ScenarioId[ScenarioId["NAX_11_GROBBULUS_HEROIC"] = 310] = "NAX_11_GROBBULUS_HEROIC";
    ScenarioId[ScenarioId["NAX_12_GLUTH_HEROIC"] = 311] = "NAX_12_GLUTH_HEROIC";
    ScenarioId[ScenarioId["NAX_13_THADDIUS_HEROIC"] = 312] = "NAX_13_THADDIUS_HEROIC";
    ScenarioId[ScenarioId["NAX_14_SAPPHIRON_HEROIC"] = 313] = "NAX_14_SAPPHIRON_HEROIC";
    ScenarioId[ScenarioId["NAX_15_KELTHUZAD_HEROIC"] = 314] = "NAX_15_KELTHUZAD_HEROIC";
    ScenarioId[ScenarioId["BRM_01_DIREBREW"] = 319] = "BRM_01_DIREBREW";
    ScenarioId[ScenarioId["BRM_02_GRIMSTONE"] = 320] = "BRM_02_GRIMSTONE";
    ScenarioId[ScenarioId["BRM_03_THAURISSAN"] = 321] = "BRM_03_THAURISSAN";
    ScenarioId[ScenarioId["BRM_04_GARR"] = 322] = "BRM_04_GARR";
    ScenarioId[ScenarioId["BRM_05_GEDDON"] = 323] = "BRM_05_GEDDON";
    ScenarioId[ScenarioId["BRM_06_MAJORDOMO"] = 324] = "BRM_06_MAJORDOMO";
    ScenarioId[ScenarioId["BRM_07_OMOKK"] = 325] = "BRM_07_OMOKK";
    ScenarioId[ScenarioId["BRM_08_DRAKKISATH"] = 326] = "BRM_08_DRAKKISATH";
    ScenarioId[ScenarioId["BRM_09_REND"] = 327] = "BRM_09_REND";
    ScenarioId[ScenarioId["BRM_10_RAZORGORE"] = 328] = "BRM_10_RAZORGORE";
    ScenarioId[ScenarioId["BRM_11_VAELASTRASZ"] = 329] = "BRM_11_VAELASTRASZ";
    ScenarioId[ScenarioId["BRM_12_CHROMAGGUS"] = 330] = "BRM_12_CHROMAGGUS";
    ScenarioId[ScenarioId["BRM_13_NEFARIAN"] = 331] = "BRM_13_NEFARIAN";
    ScenarioId[ScenarioId["BRM_14_OMNOTRON"] = 332] = "BRM_14_OMNOTRON";
    ScenarioId[ScenarioId["BRM_15_MALORIAK"] = 333] = "BRM_15_MALORIAK";
    ScenarioId[ScenarioId["BRM_16_ATRAMEDES"] = 334] = "BRM_16_ATRAMEDES";
    ScenarioId[ScenarioId["BRM_17_SUPERNEFARIAN"] = 335] = "BRM_17_SUPERNEFARIAN";
    ScenarioId[ScenarioId["BRM_CHALLENGE_PRIEST"] = 336] = "BRM_CHALLENGE_PRIEST";
    ScenarioId[ScenarioId["BRM_01_DIREBREW_HEROIC"] = 337] = "BRM_01_DIREBREW_HEROIC";
    ScenarioId[ScenarioId["BRM_02_GRIMSTONE_HEROIC"] = 338] = "BRM_02_GRIMSTONE_HEROIC";
    ScenarioId[ScenarioId["BRM_03_THAURISSAN_HEROIC"] = 339] = "BRM_03_THAURISSAN_HEROIC";
    ScenarioId[ScenarioId["BRM_04_GARR_HEROIC"] = 340] = "BRM_04_GARR_HEROIC";
    ScenarioId[ScenarioId["BRM_05_GEDDON_HEROIC"] = 341] = "BRM_05_GEDDON_HEROIC";
    ScenarioId[ScenarioId["BRM_06_MAJORDOMO_HEROIC"] = 343] = "BRM_06_MAJORDOMO_HEROIC";
    ScenarioId[ScenarioId["BRM_07_OMOKK_HEROIC"] = 344] = "BRM_07_OMOKK_HEROIC";
    ScenarioId[ScenarioId["BRM_08_DRAKKISATH_HEROIC"] = 345] = "BRM_08_DRAKKISATH_HEROIC";
    ScenarioId[ScenarioId["BRM_10_RAZORGORE_HEROIC"] = 346] = "BRM_10_RAZORGORE_HEROIC";
    ScenarioId[ScenarioId["BRM_11_VAELASTRASZ_HEROIC"] = 347] = "BRM_11_VAELASTRASZ_HEROIC";
    ScenarioId[ScenarioId["BRM_12_CHROMAGGUS_HEROIC"] = 348] = "BRM_12_CHROMAGGUS_HEROIC";
    ScenarioId[ScenarioId["BRM_13_NEFARIAN_HEROIC"] = 349] = "BRM_13_NEFARIAN_HEROIC";
    ScenarioId[ScenarioId["BRM_14_OMNOTRON_HEROIC"] = 350] = "BRM_14_OMNOTRON_HEROIC";
    ScenarioId[ScenarioId["BRM_15_MALORIAK_HEROIC"] = 351] = "BRM_15_MALORIAK_HEROIC";
    ScenarioId[ScenarioId["BRM_16_ATRAMEDES_HEROIC"] = 352] = "BRM_16_ATRAMEDES_HEROIC";
    ScenarioId[ScenarioId["BRM_CHALLENGE_PALADIN"] = 353] = "BRM_CHALLENGE_PALADIN";
    ScenarioId[ScenarioId["BRM_CHALLENGE_WARRIOR"] = 354] = "BRM_CHALLENGE_WARRIOR";
    ScenarioId[ScenarioId["BRM_CHALLENGE_HUNTER"] = 355] = "BRM_CHALLENGE_HUNTER";
    ScenarioId[ScenarioId["BRM_CHALLENGE_SHAMAN"] = 356] = "BRM_CHALLENGE_SHAMAN";
    ScenarioId[ScenarioId["BRM_CHALLENGE_DRUID"] = 357] = "BRM_CHALLENGE_DRUID";
    ScenarioId[ScenarioId["BRM_CHALLENGE_ROGUE"] = 358] = "BRM_CHALLENGE_ROGUE";
    ScenarioId[ScenarioId["BRM_CHALLENGE_WARLOCK"] = 359] = "BRM_CHALLENGE_WARLOCK";
    ScenarioId[ScenarioId["BRM_CHALLENGE_MAGE"] = 360] = "BRM_CHALLENGE_MAGE";
    ScenarioId[ScenarioId["BRM_09_REND_HEROIC"] = 361] = "BRM_09_REND_HEROIC";
    ScenarioId[ScenarioId["BRM_17_SUPERNEFARIAN_HEROIC"] = 362] = "BRM_17_SUPERNEFARIAN_HEROIC";
    ScenarioId[ScenarioId["BAXTER_TEST_SCENARIO_1"] = 404] = "BAXTER_TEST_SCENARIO_1";
    ScenarioId[ScenarioId["LOE_10_GIANTFIN"] = 1000] = "LOE_10_GIANTFIN";
    ScenarioId[ScenarioId["LOE_01_SUN_RAIDER_PHAERIX"] = 1040] = "LOE_01_SUN_RAIDER_PHAERIX";
    ScenarioId[ScenarioId["LOE_02_ZINAAR"] = 1041] = "LOE_02_ZINAAR";
    ScenarioId[ScenarioId["LOE_05_SCARVASH"] = 1060] = "LOE_05_SCARVASH";
    ScenarioId[ScenarioId["LOE_04_TEMPLE_ESCAPE"] = 1061] = "LOE_04_TEMPLE_ESCAPE";
    ScenarioId[ScenarioId["LOE07_MINECART"] = 1062] = "LOE07_MINECART";
    ScenarioId[ScenarioId["LOE_08_ARCHAEDAS"] = 1084] = "LOE_08_ARCHAEDAS";
    ScenarioId[ScenarioId["LOE_12_NAZJAR"] = 1086] = "LOE_12_NAZJAR";
    ScenarioId[ScenarioId["LOE_13_DEVILSAUR"] = 1143] = "LOE_13_DEVILSAUR";
    ScenarioId[ScenarioId["LOE_15_RAFAAM"] = 1146] = "LOE_15_RAFAAM";
    ScenarioId[ScenarioId["LOE_16_RAFAAM2"] = 1173] = "LOE_16_RAFAAM2";
    ScenarioId[ScenarioId["LOE_14_STEEL_SENTINEL"] = 1183] = "LOE_14_STEEL_SENTINEL";
    ScenarioId[ScenarioId["LOE_09_LORDSLITHERSPEAR"] = 1283] = "LOE_09_LORDSLITHERSPEAR";
    ScenarioId[ScenarioId["LOE_CHALLENGE_PALADIN"] = 1586] = "LOE_CHALLENGE_PALADIN";
    ScenarioId[ScenarioId["LOE_CHALLENGE_WARLOCK"] = 1603] = "LOE_CHALLENGE_WARLOCK";
    ScenarioId[ScenarioId["LOE_CHALLENGE_DRUID"] = 1624] = "LOE_CHALLENGE_DRUID";
    ScenarioId[ScenarioId["LOE_CHALLENGE_MAGE"] = 1625] = "LOE_CHALLENGE_MAGE";
    ScenarioId[ScenarioId["LOE_CHALLENGE_ROGUE"] = 1628] = "LOE_CHALLENGE_ROGUE";
    ScenarioId[ScenarioId["LOE_CHALLENGE_SHAMAN"] = 1637] = "LOE_CHALLENGE_SHAMAN";
    ScenarioId[ScenarioId["LOE_CHALLENGE_WARRIOR"] = 1638] = "LOE_CHALLENGE_WARRIOR";
    ScenarioId[ScenarioId["LOE_CHALLENGE_HUNTER"] = 1645] = "LOE_CHALLENGE_HUNTER";
    ScenarioId[ScenarioId["LOE_CHALLENGE_PRIEST"] = 1646] = "LOE_CHALLENGE_PRIEST";
    ScenarioId[ScenarioId["LOE_02_ZINAAR_HEROIC"] = 1650] = "LOE_02_ZINAAR_HEROIC";
    ScenarioId[ScenarioId["LOE_01_SUN_RAIDER_PHAERIX_HEROIC"] = 1651] = "LOE_01_SUN_RAIDER_PHAERIX_HEROIC";
    ScenarioId[ScenarioId["LOE_04_TEMPLE_ESCAPE_HEROIC"] = 1653] = "LOE_04_TEMPLE_ESCAPE_HEROIC";
    ScenarioId[ScenarioId["LOE_08_ARCHAEDAS_HEROIC"] = 1657] = "LOE_08_ARCHAEDAS_HEROIC";
    ScenarioId[ScenarioId["LOE_09_LORDSLITHERSPEAR_HEROIC"] = 1659] = "LOE_09_LORDSLITHERSPEAR_HEROIC";
    ScenarioId[ScenarioId["LOE_05_SCARVASH_HEROIC"] = 1660] = "LOE_05_SCARVASH_HEROIC";
    ScenarioId[ScenarioId["LOE_10_GIANTFIN_HEROIC"] = 1661] = "LOE_10_GIANTFIN_HEROIC";
    ScenarioId[ScenarioId["LOEA12_NAZJAR_HEROIC"] = 1666] = "LOEA12_NAZJAR_HEROIC";
    ScenarioId[ScenarioId["LOE07_MINECART_HEROIC"] = 1669] = "LOE07_MINECART_HEROIC";
    ScenarioId[ScenarioId["LOEA13_SKELESAURUS_HEROIC"] = 1670] = "LOEA13_SKELESAURUS_HEROIC";
    ScenarioId[ScenarioId["LOEA14_STEEL_SENTINEL_HEROIC"] = 1671] = "LOEA14_STEEL_SENTINEL_HEROIC";
    ScenarioId[ScenarioId["LOEA15_RAFAAM_HEROIC"] = 1672] = "LOEA15_RAFAAM_HEROIC";
    ScenarioId[ScenarioId["LOEA16_RAFAAM_HEROIC"] = 1673] = "LOEA16_RAFAAM_HEROIC";
    ScenarioId[ScenarioId["KARA_06_CRONE"] = 1723] = "KARA_06_CRONE";
    ScenarioId[ScenarioId["KARA_05_WOLF"] = 1724] = "KARA_05_WOLF";
    ScenarioId[ScenarioId["KARA_03_CHESS"] = 1749] = "KARA_03_CHESS";
    ScenarioId[ScenarioId["KARA_04_JULIANNE"] = 1753] = "KARA_04_JULIANNE";
    ScenarioId[ScenarioId["KARA_01_MIRROR"] = 1755] = "KARA_01_MIRROR";
    ScenarioId[ScenarioId["KARA_07_CURATOR"] = 1756] = "KARA_07_CURATOR";
    ScenarioId[ScenarioId["KARA_09_ILLHOOF"] = 1758] = "KARA_09_ILLHOOF";
    ScenarioId[ScenarioId["KARA_08_NIGHTBANE"] = 1759] = "KARA_08_NIGHTBANE";
    ScenarioId[ScenarioId["KARA_12_SHADE"] = 1760] = "KARA_12_SHADE";
    ScenarioId[ScenarioId["KARA_10_NETHERSPITE"] = 1761] = "KARA_10_NETHERSPITE";
    ScenarioId[ScenarioId["KARA_02_PANTRY"] = 1765] = "KARA_02_PANTRY";
    ScenarioId[ScenarioId["KARA_00_MALCHEZAAR"] = 1766] = "KARA_00_MALCHEZAAR";
    ScenarioId[ScenarioId["KARA_13_MEDIVH"] = 1767] = "KARA_13_MEDIVH";
    ScenarioId[ScenarioId["KARA_13_MEDIVH_2"] = 1783] = "KARA_13_MEDIVH_2";
    ScenarioId[ScenarioId["KARA_12_NETHERSPITE_HEROIC"] = 1815] = "KARA_12_NETHERSPITE_HEROIC";
    ScenarioId[ScenarioId["KARA_11_ARAN_HEROIC"] = 1816] = "KARA_11_ARAN_HEROIC";
    ScenarioId[ScenarioId["KARA_00_MALCHEZAAR_HEROIC"] = 1817] = "KARA_00_MALCHEZAAR_HEROIC";
    ScenarioId[ScenarioId["KARA_02_PANTRY_HEROIC"] = 1818] = "KARA_02_PANTRY_HEROIC";
    ScenarioId[ScenarioId["KARA_01_MIRROR_HEROIC"] = 1819] = "KARA_01_MIRROR_HEROIC";
    ScenarioId[ScenarioId["KARA_03_CHESS_HEROIC"] = 1820] = "KARA_03_CHESS_HEROIC";
    ScenarioId[ScenarioId["AIDECK___DEV_PALADIN"] = 1828] = "AIDECK___DEV_PALADIN";
    ScenarioId[ScenarioId["KARA_04_HEROIC"] = 1831] = "KARA_04_HEROIC";
    ScenarioId[ScenarioId["KARA_05_HEROIC"] = 1832] = "KARA_05_HEROIC";
    ScenarioId[ScenarioId["KARA_06_HEROIC"] = 1833] = "KARA_06_HEROIC";
    ScenarioId[ScenarioId["KARA_07_HEROIC"] = 1834] = "KARA_07_HEROIC";
    ScenarioId[ScenarioId["KARA_08_HEROIC"] = 1835] = "KARA_08_HEROIC";
    ScenarioId[ScenarioId["KARA_09_HEROIC"] = 1836] = "KARA_09_HEROIC";
    ScenarioId[ScenarioId["KARA_13_MEDIVH_HEROIC"] = 1837] = "KARA_13_MEDIVH_HEROIC";
    ScenarioId[ScenarioId["KARA_CHALLENGE_HUNTER"] = 1850] = "KARA_CHALLENGE_HUNTER";
    ScenarioId[ScenarioId["KARA_CHALLENGE_PALADIN"] = 1853] = "KARA_CHALLENGE_PALADIN";
    ScenarioId[ScenarioId["KARA_CHALLENGE_ROGUE"] = 1854] = "KARA_CHALLENGE_ROGUE";
    ScenarioId[ScenarioId["KARA_CHALLENGE_WARRIOR"] = 1856] = "KARA_CHALLENGE_WARRIOR";
    ScenarioId[ScenarioId["KARA_CHALLENGE_WARLOCK"] = 1857] = "KARA_CHALLENGE_WARLOCK";
    ScenarioId[ScenarioId["KARA_CHALLENGE_SHAMAN"] = 1860] = "KARA_CHALLENGE_SHAMAN";
    ScenarioId[ScenarioId["KARA_CHALLENGE_PRIEST"] = 1861] = "KARA_CHALLENGE_PRIEST";
    ScenarioId[ScenarioId["KARA_CHALLENGE_DRUID"] = 1862] = "KARA_CHALLENGE_DRUID";
    ScenarioId[ScenarioId["KARA_CHALLENGE_MAGE"] = 1863] = "KARA_CHALLENGE_MAGE";
    ScenarioId[ScenarioId["KARA_13_FINAL_HEROIC"] = 1880] = "KARA_13_FINAL_HEROIC";
    ScenarioId[ScenarioId["UNG_ZOO"] = 2009] = "UNG_ZOO";
    ScenarioId[ScenarioId["UNG_DRUID"] = 2010] = "UNG_DRUID";
    ScenarioId[ScenarioId["RETURNINGPLAYER_CHALLENGE_1OLD"] = 2097] = "RETURNINGPLAYER_CHALLENGE_1OLD";
    ScenarioId[ScenarioId["RETURNINGPLAYER_CHALLENGE_2OLD"] = 2098] = "RETURNINGPLAYER_CHALLENGE_2OLD";
    ScenarioId[ScenarioId["RETURNINGPLAYER_CHALLENGE_3OLD"] = 2099] = "RETURNINGPLAYER_CHALLENGE_3OLD";
    ScenarioId[ScenarioId["TAVERN_BRAWL_BRAWLISEUM"] = 2109] = "TAVERN_BRAWL_BRAWLISEUM";
    ScenarioId[ScenarioId["TAVERN_BRAWL_WILD_BRAWLISEUM"] = 2121] = "TAVERN_BRAWL_WILD_BRAWLISEUM";
    ScenarioId[ScenarioId["ICCA_01_LICHKINGPROLOGUE"] = 2199] = "ICCA_01_LICHKINGPROLOGUE";
    ScenarioId[ScenarioId["ICCA_04_SINDRAGOSA"] = 2212] = "ICCA_04_SINDRAGOSA";
    ScenarioId[ScenarioId["MAX_SANDBOX_MAGE"] = 2218] = "MAX_SANDBOX_MAGE";
    ScenarioId[ScenarioId["MAX_SANDBOX_SHAMAN"] = 2221] = "MAX_SANDBOX_SHAMAN";
    ScenarioId[ScenarioId["ICCA_05_LANATHEL"] = 2262] = "ICCA_05_LANATHEL";
    ScenarioId[ScenarioId["_ICCA_06_MARROWGAR"] = 2263] = "_ICCA_06_MARROWGAR";
    ScenarioId[ScenarioId["ICCA_07_PUTRICIDE"] = 2264] = "ICCA_07_PUTRICIDE";
    ScenarioId[ScenarioId["ICCA_08_LICHKING"] = 2265] = "ICCA_08_LICHKING";
    ScenarioId[ScenarioId["ICCA_09_DEATHBRINGER_SAURFANG"] = 2271] = "ICCA_09_DEATHBRINGER_SAURFANG";
    ScenarioId[ScenarioId["ICCA_10_DEATHWHISPER_VALITHRIA"] = 2495] = "ICCA_10_DEATHWHISPER_VALITHRIA";
    ScenarioId[ScenarioId["LOOTA_DUNGEON"] = 2663] = "LOOTA_DUNGEON";
    ScenarioId[ScenarioId["AUTOMATIONSCENARIO_DEVONLY"] = 2682] = "AUTOMATIONSCENARIO_DEVONLY";
    ScenarioId[ScenarioId["GILA_DUNGEON"] = 2706] = "GILA_DUNGEON";
    ScenarioId[ScenarioId["TEST_CHESS_TIMER"] = 2796] = "TEST_CHESS_TIMER";
    ScenarioId[ScenarioId["GILA_61_HAGATHA"] = 2821] = "GILA_61_HAGATHA";
    ScenarioId[ScenarioId["TRLA_DUNGEON"] = 2890] = "TRLA_DUNGEON";
    ScenarioId[ScenarioId["BOTA_MIRROR_PUZZLE_1"] = 2902] = "BOTA_MIRROR_PUZZLE_1";
    ScenarioId[ScenarioId["BOTA_SURVIVAL_PUZZLE_1"] = 2909] = "BOTA_SURVIVAL_PUZZLE_1";
    ScenarioId[ScenarioId["BOTA_MIRROR_PUZZLE_2"] = 2911] = "BOTA_MIRROR_PUZZLE_2";
    ScenarioId[ScenarioId["BOTA_MIRROR_PUZZLE_3"] = 2913] = "BOTA_MIRROR_PUZZLE_3";
    ScenarioId[ScenarioId["BOTA_MIRROR_PUZZLE_4"] = 2914] = "BOTA_MIRROR_PUZZLE_4";
    ScenarioId[ScenarioId["BOTA_SURVIVAL_PUZZLE_2"] = 2917] = "BOTA_SURVIVAL_PUZZLE_2";
    ScenarioId[ScenarioId["BOTA_SURVIVAL_PUZZLE_3"] = 2918] = "BOTA_SURVIVAL_PUZZLE_3";
    ScenarioId[ScenarioId["BOTA_SURVIVAL_PUZZLE_4"] = 2919] = "BOTA_SURVIVAL_PUZZLE_4";
    ScenarioId[ScenarioId["BOTA_MIRROR_PUZZLE_BOOM"] = 2920] = "BOTA_MIRROR_PUZZLE_BOOM";
    ScenarioId[ScenarioId["BOTA_LETHAL_PUZZLE_1"] = 2941] = "BOTA_LETHAL_PUZZLE_1";
    ScenarioId[ScenarioId["BOTA_LETHAL_PUZZLE_2"] = 2942] = "BOTA_LETHAL_PUZZLE_2";
    ScenarioId[ScenarioId["BOTA_LETHAL_PUZZLE_3"] = 2943] = "BOTA_LETHAL_PUZZLE_3";
    ScenarioId[ScenarioId["BOTA_LETHAL_PUZZLE_4"] = 2944] = "BOTA_LETHAL_PUZZLE_4";
    ScenarioId[ScenarioId["BOTA_CLEAR_PUZZLE_1"] = 2956] = "BOTA_CLEAR_PUZZLE_1";
    ScenarioId[ScenarioId["BOTA_CLEAR_PUZZLE_2"] = 2957] = "BOTA_CLEAR_PUZZLE_2";
    ScenarioId[ScenarioId["BOTA_CLEAR_PUZZLE_3"] = 2958] = "BOTA_CLEAR_PUZZLE_3";
    ScenarioId[ScenarioId["BOTA_CLEAR_PUZZLE_4"] = 2959] = "BOTA_CLEAR_PUZZLE_4";
    ScenarioId[ScenarioId["BOTA_LETHAL_PUZZLE_BOOM"] = 2977] = "BOTA_LETHAL_PUZZLE_BOOM";
    ScenarioId[ScenarioId["BOTA_SURVIVAL_PUZZLE_BOOM"] = 2978] = "BOTA_SURVIVAL_PUZZLE_BOOM";
    ScenarioId[ScenarioId["BOTA_CLEAR_PUZZLE_BOOM"] = 2979] = "BOTA_CLEAR_PUZZLE_BOOM";
    ScenarioId[ScenarioId["DALA_BANK"] = 3005] = "DALA_BANK";
    ScenarioId[ScenarioId["AIDECK___EVEN_SHAMAN"] = 3149] = "AIDECK___EVEN_SHAMAN";
    ScenarioId[ScenarioId["DALA_VIOLETHOLD"] = 3188] = "DALA_VIOLETHOLD";
    ScenarioId[ScenarioId["DALA_STREETS"] = 3189] = "DALA_STREETS";
    ScenarioId[ScenarioId["DALA_UNDERBELLY"] = 3190] = "DALA_UNDERBELLY";
    ScenarioId[ScenarioId["DALA_CITADEL"] = 3191] = "DALA_CITADEL";
    ScenarioId[ScenarioId["DALA_TAVERN"] = 3236] = "DALA_TAVERN";
    ScenarioId[ScenarioId["_140_ARENA_SEASON"] = 3280] = "_140_ARENA_SEASON";
    ScenarioId[ScenarioId["DALA_BANK_HEROIC"] = 3328] = "DALA_BANK_HEROIC";
    ScenarioId[ScenarioId["DALA_VIOLETHOLD_HEROIC"] = 3329] = "DALA_VIOLETHOLD_HEROIC";
    ScenarioId[ScenarioId["DALA_STREETS_HEROIC"] = 3330] = "DALA_STREETS_HEROIC";
    ScenarioId[ScenarioId["DALA_UNDERBELLY_HEROIC"] = 3331] = "DALA_UNDERBELLY_HEROIC";
    ScenarioId[ScenarioId["DALA_CITADEL_HEROIC"] = 3332] = "DALA_CITADEL_HEROIC";
    ScenarioId[ScenarioId["_144_ARENA_SEASON"] = 3349] = "_144_ARENA_SEASON";
    ScenarioId[ScenarioId["_150_ARENA_SEASON"] = 3355] = "_150_ARENA_SEASON";
    ScenarioId[ScenarioId["AIDECK___ARENAMAGETEST"] = 3356] = "AIDECK___ARENAMAGETEST";
    ScenarioId[ScenarioId["DALA_TAVERN_HEROIC"] = 3359] = "DALA_TAVERN_HEROIC";
    ScenarioId[ScenarioId["ULDA_CITY"] = 3428] = "ULDA_CITY";
    ScenarioId[ScenarioId["ULDA_DESERT"] = 3429] = "ULDA_DESERT";
    ScenarioId[ScenarioId["ULDA_TOMB"] = 3430] = "ULDA_TOMB";
    ScenarioId[ScenarioId["ULDA_HALLS"] = 3431] = "ULDA_HALLS";
    ScenarioId[ScenarioId["ULDA_SANCTUM"] = 3432] = "ULDA_SANCTUM";
    ScenarioId[ScenarioId["ULDA_CITY_HEROIC"] = 3433] = "ULDA_CITY_HEROIC";
    ScenarioId[ScenarioId["ULDA_DESERT_HEROIC"] = 3434] = "ULDA_DESERT_HEROIC";
    ScenarioId[ScenarioId["ULDA_TOMB_HEROIC"] = 3435] = "ULDA_TOMB_HEROIC";
    ScenarioId[ScenarioId["ULDA_HALLS_HEROIC"] = 3436] = "ULDA_HALLS_HEROIC";
    ScenarioId[ScenarioId["ULDA_SANCTUM_HEROIC"] = 3437] = "ULDA_SANCTUM_HEROIC";
    ScenarioId[ScenarioId["ULDA_TAVERN"] = 3438] = "ULDA_TAVERN";
    ScenarioId[ScenarioId["ULDA_TAVERN_HEROIC"] = 3439] = "ULDA_TAVERN_HEROIC";
    ScenarioId[ScenarioId["BATTLEGROUNDS_8P"] = 3459] = "BATTLEGROUNDS_8P";
    ScenarioId[ScenarioId["WIZARD_DUELS"] = 3466] = "WIZARD_DUELS";
    ScenarioId[ScenarioId["DRGA_GOOD_1_1"] = 3469] = "DRGA_GOOD_1_1";
    ScenarioId[ScenarioId["DRGA_GOOD_1_2"] = 3470] = "DRGA_GOOD_1_2";
    ScenarioId[ScenarioId["DRGA_GOOD_1_3"] = 3471] = "DRGA_GOOD_1_3";
    ScenarioId[ScenarioId["DRGA_GOOD_2_1"] = 3472] = "DRGA_GOOD_2_1";
    ScenarioId[ScenarioId["DRGA_GOOD_2_2"] = 3473] = "DRGA_GOOD_2_2";
    ScenarioId[ScenarioId["DRGA_GOOD_2_3"] = 3475] = "DRGA_GOOD_2_3";
    ScenarioId[ScenarioId["DRGA_GOOD_3_1"] = 3477] = "DRGA_GOOD_3_1";
    ScenarioId[ScenarioId["DRGA_GOOD_3_2"] = 3478] = "DRGA_GOOD_3_2";
    ScenarioId[ScenarioId["DRGA_GOOD_3_3"] = 3479] = "DRGA_GOOD_3_3";
    ScenarioId[ScenarioId["DRGA_GOOD_4_1"] = 3480] = "DRGA_GOOD_4_1";
    ScenarioId[ScenarioId["DRGA_GOOD_4_2"] = 3481] = "DRGA_GOOD_4_2";
    ScenarioId[ScenarioId["DRGA_GOOD_4_3"] = 3483] = "DRGA_GOOD_4_3";
    ScenarioId[ScenarioId["DRGA_EVIL_1_1"] = 3484] = "DRGA_EVIL_1_1";
    ScenarioId[ScenarioId["_154_ARENA_SEASON"] = 3487] = "_154_ARENA_SEASON";
    ScenarioId[ScenarioId["DRGA_EVIL_1_2"] = 3488] = "DRGA_EVIL_1_2";
    ScenarioId[ScenarioId["DRGA_EVIL_1_3"] = 3489] = "DRGA_EVIL_1_3";
    ScenarioId[ScenarioId["DRGA_EVIL_2_1"] = 3490] = "DRGA_EVIL_2_1";
    ScenarioId[ScenarioId["DRGA_EVIL_2_2"] = 3491] = "DRGA_EVIL_2_2";
    ScenarioId[ScenarioId["_160_ARENA_SEASON"] = 3492] = "_160_ARENA_SEASON";
    ScenarioId[ScenarioId["DRGA_EVIL_2_3"] = 3493] = "DRGA_EVIL_2_3";
    ScenarioId[ScenarioId["DRGA_EVIL_3_1"] = 3494] = "DRGA_EVIL_3_1";
    ScenarioId[ScenarioId["DRGA_EVIL_3_2"] = 3495] = "DRGA_EVIL_3_2";
    ScenarioId[ScenarioId["DRGA_EVIL_3_3"] = 3497] = "DRGA_EVIL_3_3";
    ScenarioId[ScenarioId["DRGA_EVIL_4_1"] = 3498] = "DRGA_EVIL_4_1";
    ScenarioId[ScenarioId["DRGA_EVIL_4_2"] = 3499] = "DRGA_EVIL_4_2";
    ScenarioId[ScenarioId["DRGA_EVIL_4_3"] = 3500] = "DRGA_EVIL_4_3";
    ScenarioId[ScenarioId["_164_ARENA_SEASON"] = 3502] = "_164_ARENA_SEASON";
    ScenarioId[ScenarioId["BATTLEGROUNDS_TUTORIAL_1P"] = 3539] = "BATTLEGROUNDS_TUTORIAL_1P";
    ScenarioId[ScenarioId["RETURNINGPLAYER_CHALLENGE_1"] = 3540] = "RETURNINGPLAYER_CHALLENGE_1";
    ScenarioId[ScenarioId["RETURNINGPLAYER_CHALLENGE_2"] = 3541] = "RETURNINGPLAYER_CHALLENGE_2";
    ScenarioId[ScenarioId["RETURNINGPLAYER_CHALLENGE_3"] = 3543] = "RETURNINGPLAYER_CHALLENGE_3";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_ORGRIMMAR"] = 3553] = "TB_ROADTONORTHREND_ORGRIMMAR";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_TAVERN"] = 3555] = "TB_ROADTONORTHREND_TAVERN";
    ScenarioId[ScenarioId["DRGA_GOOD_1_1_H"] = 3556] = "DRGA_GOOD_1_1_H";
    ScenarioId[ScenarioId["BTA_FIGHT_01"] = 3557] = "BTA_FIGHT_01";
    ScenarioId[ScenarioId["BTA_FIGHT_02"] = 3558] = "BTA_FIGHT_02";
    ScenarioId[ScenarioId["BTA_FIGHT_03"] = 3559] = "BTA_FIGHT_03";
    ScenarioId[ScenarioId["BTA_FIGHT_04"] = 3560] = "BTA_FIGHT_04";
    ScenarioId[ScenarioId["BTA_FIGHT_05"] = 3561] = "BTA_FIGHT_05";
    ScenarioId[ScenarioId["BTA_FIGHT_06"] = 3562] = "BTA_FIGHT_06";
    ScenarioId[ScenarioId["BTA_FIGHT_07"] = 3563] = "BTA_FIGHT_07";
    ScenarioId[ScenarioId["BTA_FIGHT_08"] = 3564] = "BTA_FIGHT_08";
    ScenarioId[ScenarioId["BTA_FIGHT_09"] = 3565] = "BTA_FIGHT_09";
    ScenarioId[ScenarioId["BTA_FIGHT_10"] = 3566] = "BTA_FIGHT_10";
    ScenarioId[ScenarioId["BTA_FIGHT_11"] = 3567] = "BTA_FIGHT_11";
    ScenarioId[ScenarioId["BTA_FIGHT_12"] = 3568] = "BTA_FIGHT_12";
    ScenarioId[ScenarioId["BTA_FIGHT_13"] = 3569] = "BTA_FIGHT_13";
    ScenarioId[ScenarioId["BTA_FIGHT_14"] = 3570] = "BTA_FIGHT_14";
    ScenarioId[ScenarioId["BTA_FIGHT_15"] = 3571] = "BTA_FIGHT_15";
    ScenarioId[ScenarioId["BTA_FIGHT_16"] = 3572] = "BTA_FIGHT_16";
    ScenarioId[ScenarioId["BTA_FIGHT_17"] = 3573] = "BTA_FIGHT_17";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_01"] = 3574] = "BTA_HEROIC_FIGHT_01";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_02"] = 3575] = "BTA_HEROIC_FIGHT_02";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_03"] = 3576] = "BTA_HEROIC_FIGHT_03";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_04"] = 3577] = "BTA_HEROIC_FIGHT_04";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_05"] = 3578] = "BTA_HEROIC_FIGHT_05";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_06"] = 3579] = "BTA_HEROIC_FIGHT_06";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_07"] = 3580] = "BTA_HEROIC_FIGHT_07";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_08"] = 3581] = "BTA_HEROIC_FIGHT_08";
    ScenarioId[ScenarioId["BTA_HEROIC_FIGHT_09"] = 3582] = "BTA_HEROIC_FIGHT_09";
    ScenarioId[ScenarioId["DRGA_GOOD_1_2_H"] = 3583] = "DRGA_GOOD_1_2_H";
    ScenarioId[ScenarioId["DRGA_GOOD_1_3_H"] = 3584] = "DRGA_GOOD_1_3_H";
    ScenarioId[ScenarioId["DRGA_GOOD_2_1_H"] = 3585] = "DRGA_GOOD_2_1_H";
    ScenarioId[ScenarioId["DRGA_GOOD_2_2_H"] = 3586] = "DRGA_GOOD_2_2_H";
    ScenarioId[ScenarioId["DRGA_GOOD_2_3_H"] = 3587] = "DRGA_GOOD_2_3_H";
    ScenarioId[ScenarioId["DRGA_GOOD_3_1_H"] = 3588] = "DRGA_GOOD_3_1_H";
    ScenarioId[ScenarioId["DRGA_GOOD_3_2_H"] = 3589] = "DRGA_GOOD_3_2_H";
    ScenarioId[ScenarioId["DRGA_GOOD_3_3_H"] = 3590] = "DRGA_GOOD_3_3_H";
    ScenarioId[ScenarioId["DRGA_GOOD_4_1_H"] = 3591] = "DRGA_GOOD_4_1_H";
    ScenarioId[ScenarioId["DRGA_GOOD_4_2_H"] = 3592] = "DRGA_GOOD_4_2_H";
    ScenarioId[ScenarioId["DRGA_GOOD_4_3_H"] = 3593] = "DRGA_GOOD_4_3_H";
    ScenarioId[ScenarioId["DRGA_EVIL_1_1_H"] = 3594] = "DRGA_EVIL_1_1_H";
    ScenarioId[ScenarioId["DRGA_EVIL_1_2_H"] = 3595] = "DRGA_EVIL_1_2_H";
    ScenarioId[ScenarioId["DRGA_EVIL_1_3_H"] = 3596] = "DRGA_EVIL_1_3_H";
    ScenarioId[ScenarioId["DRGA_EVIL_2_1_H"] = 3597] = "DRGA_EVIL_2_1_H";
    ScenarioId[ScenarioId["DRGA_EVIL_2_2_H"] = 3598] = "DRGA_EVIL_2_2_H";
    ScenarioId[ScenarioId["DRGA_EVIL_2_3_H"] = 3599] = "DRGA_EVIL_2_3_H";
    ScenarioId[ScenarioId["DRGA_EVIL_3_1_H"] = 3600] = "DRGA_EVIL_3_1_H";
    ScenarioId[ScenarioId["DRGA_EVIL_3_2_H"] = 3601] = "DRGA_EVIL_3_2_H";
    ScenarioId[ScenarioId["DRGA_EVIL_3_3_H"] = 3602] = "DRGA_EVIL_3_3_H";
    ScenarioId[ScenarioId["DRGA_EVIL_4_1_H"] = 3603] = "DRGA_EVIL_4_1_H";
    ScenarioId[ScenarioId["DRGA_EVIL_4_2_H"] = 3604] = "DRGA_EVIL_4_2_H";
    ScenarioId[ScenarioId["DRGA_EVIL_4_3_H"] = 3605] = "DRGA_EVIL_4_3_H";
    ScenarioId[ScenarioId["AIDECK___NORMAL_DEMON_HUNTER"] = 3608] = "AIDECK___NORMAL_DEMON_HUNTER";
    ScenarioId[ScenarioId["AIDECK___EXPERT_DEMON_HUNTER"] = 3609] = "AIDECK___EXPERT_DEMON_HUNTER";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_UNGORO"] = 3614] = "TB_ROADTONORTHREND_UNGORO";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_KARAZHAN"] = 3615] = "TB_ROADTONORTHREND_KARAZHAN";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_MUSEUM"] = 3616] = "TB_ROADTONORTHREND_MUSEUM";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_BRM"] = 3617] = "TB_ROADTONORTHREND_BRM";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_WITCHWOOD"] = 3618] = "TB_ROADTONORTHREND_WITCHWOOD";
    ScenarioId[ScenarioId["TB_ROADTONORTHREND_GURUBASHI"] = 3619] = "TB_ROADTONORTHREND_GURUBASHI";
    ScenarioId[ScenarioId["DEMON_HUNTER_PROLOGUE_1"] = 3624] = "DEMON_HUNTER_PROLOGUE_1";
    ScenarioId[ScenarioId["AIDECK___DEV_DEMONHUNTER"] = 3626] = "AIDECK___DEV_DEMONHUNTER";
    ScenarioId[ScenarioId["_170_ARENA_SEASON"] = 3632] = "_170_ARENA_SEASON";
    ScenarioId[ScenarioId["_174_ARENA_SEASON"] = 3634] = "_174_ARENA_SEASON";
    ScenarioId[ScenarioId["DEMON_HUNTER_PROLOGUE_2"] = 3648] = "DEMON_HUNTER_PROLOGUE_2";
    ScenarioId[ScenarioId["DEMON_HUNTER_PROLOGUE_3"] = 3649] = "DEMON_HUNTER_PROLOGUE_3";
    ScenarioId[ScenarioId["DEMON_HUNTER_PROLOGUE_4"] = 3650] = "DEMON_HUNTER_PROLOGUE_4";
    ScenarioId[ScenarioId["STANDARD_GAME_VS_SPECIFIED_AI_DECK"] = 3680] = "STANDARD_GAME_VS_SPECIFIED_AI_DECK";
    ScenarioId[ScenarioId["AIDECK___GALAKROND_PRIEST"] = 3685] = "AIDECK___GALAKROND_PRIEST";
    ScenarioId[ScenarioId["TEST_BLANKSTATE"] = 3692] = "TEST_BLANKSTATE";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_01"] = 3724] = "BOH_JAINA_STORY_01";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_02"] = 3725] = "BOH_JAINA_STORY_02";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_03"] = 3726] = "BOH_JAINA_STORY_03";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_04"] = 3727] = "BOH_JAINA_STORY_04";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_05"] = 3728] = "BOH_JAINA_STORY_05";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_06"] = 3729] = "BOH_JAINA_STORY_06";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_07"] = 3730] = "BOH_JAINA_STORY_07";
    ScenarioId[ScenarioId["BOH_JAINA_STORY_08"] = 3731] = "BOH_JAINA_STORY_08";
    ScenarioId[ScenarioId["USERRESEARCH___200_HEALING_PRIEST"] = 3742] = "USERRESEARCH___200_HEALING_PRIEST";
    ScenarioId[ScenarioId["LETTUCE_PVP"] = 3743] = "LETTUCE_PVP";
    ScenarioId[ScenarioId["LETTUCE_PVE_DEV_TEST"] = 3744] = "LETTUCE_PVE_DEV_TEST";
    ScenarioId[ScenarioId["PVPDR_TAVERN"] = 3745] = "PVPDR_TAVERN";
    ScenarioId[ScenarioId["_184_ARENA_SEASON"] = 3753] = "_184_ARENA_SEASON";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_01"] = 3766] = "BOH_REXXAR_STORY_01";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_02"] = 3767] = "BOH_REXXAR_STORY_02";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_03"] = 3768] = "BOH_REXXAR_STORY_03";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_04"] = 3769] = "BOH_REXXAR_STORY_04";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_05"] = 3770] = "BOH_REXXAR_STORY_05";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_06"] = 3771] = "BOH_REXXAR_STORY_06";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_07"] = 3772] = "BOH_REXXAR_STORY_07";
    ScenarioId[ScenarioId["BOH_REXXAR_STORY_08"] = 3773] = "BOH_REXXAR_STORY_08";
    ScenarioId[ScenarioId["LETTUCE_TUTORIAL_1"] = 3778] = "LETTUCE_TUTORIAL_1";
    ScenarioId[ScenarioId["LETTUCE_TUTORIAL_BOSS"] = 3779] = "LETTUCE_TUTORIAL_BOSS";
    ScenarioId[ScenarioId["_190_ARENA_SEASON"] = 3781] = "_190_ARENA_SEASON";
    ScenarioId[ScenarioId["_194_ARENA_SEASON"] = 3782] = "_194_ARENA_SEASON";
    ScenarioId[ScenarioId["LETTUCE_MAP_PVE"] = 3790] = "LETTUCE_MAP_PVE";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_01"] = 3793] = "BOH_GARROSH_STORY_01";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_02"] = 3794] = "BOH_GARROSH_STORY_02";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_03"] = 3795] = "BOH_GARROSH_STORY_03";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_04"] = 3796] = "BOH_GARROSH_STORY_04";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_05"] = 3797] = "BOH_GARROSH_STORY_05";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_06"] = 3798] = "BOH_GARROSH_STORY_06";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_07"] = 3799] = "BOH_GARROSH_STORY_07";
    ScenarioId[ScenarioId["BOH_GARROSH_STORY_08"] = 3800] = "BOH_GARROSH_STORY_08";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_01"] = 3810] = "BOH_UTHER_STORY_01";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_02"] = 3811] = "BOH_UTHER_STORY_02";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_03"] = 3812] = "BOH_UTHER_STORY_03";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_04"] = 3813] = "BOH_UTHER_STORY_04";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_05"] = 3814] = "BOH_UTHER_STORY_05";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_06"] = 3815] = "BOH_UTHER_STORY_06";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_07"] = 3816] = "BOH_UTHER_STORY_07";
    ScenarioId[ScenarioId["BOH_UTHER_STORY_08"] = 3817] = "BOH_UTHER_STORY_08";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_01"] = 3825] = "BOH_ANDUIN_STORY_01";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_02"] = 3826] = "BOH_ANDUIN_STORY_02";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_03"] = 3827] = "BOH_ANDUIN_STORY_03";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_04"] = 3828] = "BOH_ANDUIN_STORY_04";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_05"] = 3829] = "BOH_ANDUIN_STORY_05";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_06"] = 3830] = "BOH_ANDUIN_STORY_06";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_07"] = 3831] = "BOH_ANDUIN_STORY_07";
    ScenarioId[ScenarioId["BOH_ANDUIN_STORY_08"] = 3832] = "BOH_ANDUIN_STORY_08";
    ScenarioId[ScenarioId["_200_ARENA_SEASON"] = 3835] = "_200_ARENA_SEASON";
    ScenarioId[ScenarioId["BOM_01_ROKARA_01"] = 3839] = "BOM_01_ROKARA_01";
    ScenarioId[ScenarioId["BOM_01_ROKARA_02"] = 3840] = "BOM_01_ROKARA_02";
    ScenarioId[ScenarioId["BOM_01_ROKARA_03"] = 3841] = "BOM_01_ROKARA_03";
    ScenarioId[ScenarioId["BOM_01_ROKARA_04"] = 3842] = "BOM_01_ROKARA_04";
    ScenarioId[ScenarioId["BOM_01_ROKARA_05"] = 3843] = "BOM_01_ROKARA_05";
    ScenarioId[ScenarioId["BOM_01_ROKARA_06"] = 3844] = "BOM_01_ROKARA_06";
    ScenarioId[ScenarioId["BOM_01_ROKARA_07"] = 3845] = "BOM_01_ROKARA_07";
    ScenarioId[ScenarioId["BOM_01_ROKARA_08"] = 3846] = "BOM_01_ROKARA_08";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_01"] = 3851] = "BOH_VALEERA_STORY_01";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_02"] = 3852] = "BOH_VALEERA_STORY_02";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_03"] = 3853] = "BOH_VALEERA_STORY_03";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_04"] = 3854] = "BOH_VALEERA_STORY_04";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_05"] = 3855] = "BOH_VALEERA_STORY_05";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_06"] = 3856] = "BOH_VALEERA_STORY_06";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_07"] = 3857] = "BOH_VALEERA_STORY_07";
    ScenarioId[ScenarioId["BOH_VALEERA_STORY_08"] = 3858] = "BOH_VALEERA_STORY_08";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_01"] = 3891] = "BOH_THRALL_STORY_01";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_02"] = 3892] = "BOH_THRALL_STORY_02";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_03"] = 3893] = "BOH_THRALL_STORY_03";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_04"] = 3894] = "BOH_THRALL_STORY_04";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_05"] = 3895] = "BOH_THRALL_STORY_05";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_06"] = 3896] = "BOH_THRALL_STORY_06";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_07"] = 3897] = "BOH_THRALL_STORY_07";
    ScenarioId[ScenarioId["BOH_THRALL_STORY_08"] = 3898] = "BOH_THRALL_STORY_08";
    ScenarioId[ScenarioId["LETTUCE_TUTORIAL_2"] = 3900] = "LETTUCE_TUTORIAL_2";
    ScenarioId[ScenarioId["LETTUCE_TUTORIAL_3"] = 3901] = "LETTUCE_TUTORIAL_3";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_01"] = 3923] = "BOH_MALFURION_STORY_01";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_02"] = 3925] = "BOH_MALFURION_STORY_02";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_03"] = 3926] = "BOH_MALFURION_STORY_03";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_04"] = 3927] = "BOH_MALFURION_STORY_04";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_05"] = 3928] = "BOH_MALFURION_STORY_05";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_06"] = 3929] = "BOH_MALFURION_STORY_06";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_07"] = 3932] = "BOH_MALFURION_STORY_07";
    ScenarioId[ScenarioId["BOH_MALFURION_STORY_08"] = 3933] = "BOH_MALFURION_STORY_08";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_01"] = 3991] = "BOM_02_XYRELLA_01";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_02"] = 3992] = "BOM_02_XYRELLA_02";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_03"] = 3993] = "BOM_02_XYRELLA_03";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_04"] = 3994] = "BOM_02_XYRELLA_04";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_05"] = 3995] = "BOM_02_XYRELLA_05";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_06"] = 3996] = "BOM_02_XYRELLA_06";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_07"] = 3997] = "BOM_02_XYRELLA_07";
    ScenarioId[ScenarioId["BOM_02_XYRELLA_08"] = 3998] = "BOM_02_XYRELLA_08";
    ScenarioId[ScenarioId["_204_ARENA_SEASON"] = 4063] = "_204_ARENA_SEASON";
    ScenarioId[ScenarioId["LETTUCE_TUTORIAL_4"] = 4067] = "LETTUCE_TUTORIAL_4";
    ScenarioId[ScenarioId["BOM_03_GUFF_01"] = 4074] = "BOM_03_GUFF_01";
    ScenarioId[ScenarioId["BOM_03_GUFF_02"] = 4075] = "BOM_03_GUFF_02";
    ScenarioId[ScenarioId["BOM_03_GUFF_03"] = 4076] = "BOM_03_GUFF_03";
    ScenarioId[ScenarioId["BOM_03_GUFF_04"] = 4077] = "BOM_03_GUFF_04";
    ScenarioId[ScenarioId["BOM_03_GUFF_05"] = 4078] = "BOM_03_GUFF_05";
    ScenarioId[ScenarioId["BOM_03_GUFF_06"] = 4079] = "BOM_03_GUFF_06";
    ScenarioId[ScenarioId["BOM_03_GUFF_07"] = 4080] = "BOM_03_GUFF_07";
    ScenarioId[ScenarioId["BOM_03_GUFF_08"] = 4081] = "BOM_03_GUFF_08";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_01"] = 4085] = "BOH_GULDAN_STORY_01";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_02"] = 4086] = "BOH_GULDAN_STORY_02";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_03"] = 4087] = "BOH_GULDAN_STORY_03";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_04"] = 4088] = "BOH_GULDAN_STORY_04";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_05"] = 4089] = "BOH_GULDAN_STORY_05";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_06"] = 4090] = "BOH_GULDAN_STORY_06";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_07"] = 4091] = "BOH_GULDAN_STORY_07";
    ScenarioId[ScenarioId["BOH_GULDAN_STORY_08"] = 4092] = "BOH_GULDAN_STORY_08";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_01"] = 4105] = "BOM_04_KURTRUS_01";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_02"] = 4113] = "BOM_04_KURTRUS_02";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_03"] = 4114] = "BOM_04_KURTRUS_03";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_04"] = 4115] = "BOM_04_KURTRUS_04";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_05"] = 4116] = "BOM_04_KURTRUS_05";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_06"] = 4117] = "BOM_04_KURTRUS_06";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_07"] = 4118] = "BOM_04_KURTRUS_07";
    ScenarioId[ScenarioId["BOM_04_KURTRUS_08"] = 4119] = "BOM_04_KURTRUS_08";
    ScenarioId[ScenarioId["_210_ARENA_SEASON_"] = 4120] = "_210_ARENA_SEASON_";
    ScenarioId[ScenarioId["HUAWEI_PROMOTION_FAKE_ENTRY"] = 4122] = "HUAWEI_PROMOTION_FAKE_ENTRY";
    ScenarioId[ScenarioId["LETTUCE_PVP_VS_AI"] = 4123] = "LETTUCE_PVP_VS_AI";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_01"] = 4137] = "BOH_ILLIDAN_STORY_01";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_02"] = 4138] = "BOH_ILLIDAN_STORY_02";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_03"] = 4139] = "BOH_ILLIDAN_STORY_03";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_04"] = 4140] = "BOH_ILLIDAN_STORY_04";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_05"] = 4141] = "BOH_ILLIDAN_STORY_05";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_06"] = 4142] = "BOH_ILLIDAN_STORY_06";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_07"] = 4143] = "BOH_ILLIDAN_STORY_07";
    ScenarioId[ScenarioId["BOH_ILLIDAN_STORY_08"] = 4144] = "BOH_ILLIDAN_STORY_08";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_01"] = 4160] = "BOM_05_TAMSIN_01";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_02"] = 4161] = "BOM_05_TAMSIN_02";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_03"] = 4162] = "BOM_05_TAMSIN_03";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_04"] = 4163] = "BOM_05_TAMSIN_04";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_05"] = 4164] = "BOM_05_TAMSIN_05";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_06"] = 4165] = "BOM_05_TAMSIN_06";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_07"] = 4166] = "BOM_05_TAMSIN_07";
    ScenarioId[ScenarioId["BOM_05_TAMSIN_08"] = 4167] = "BOM_05_TAMSIN_08";
    ScenarioId[ScenarioId["BOM_06_CARIEL_01"] = 4307] = "BOM_06_CARIEL_01";
    ScenarioId[ScenarioId["BOM_06_CARIEL_02"] = 4308] = "BOM_06_CARIEL_02";
    ScenarioId[ScenarioId["BOM_06_CARIEL_03"] = 4309] = "BOM_06_CARIEL_03";
    ScenarioId[ScenarioId["BOM_06_CARIEL_04"] = 4310] = "BOM_06_CARIEL_04";
    ScenarioId[ScenarioId["BOM_06_CARIEL_05"] = 4311] = "BOM_06_CARIEL_05";
    ScenarioId[ScenarioId["BOM_06_CARIEL_06"] = 4312] = "BOM_06_CARIEL_06";
    ScenarioId[ScenarioId["BOM_06_CARIEL_07"] = 4313] = "BOM_06_CARIEL_07";
    ScenarioId[ScenarioId["BOM_06_CARIEL_08"] = 4314] = "BOM_06_CARIEL_08";
    ScenarioId[ScenarioId["AIDECK___NORMAL_DEATH_KNIGHT"] = 4316] = "AIDECK___NORMAL_DEATH_KNIGHT";
    ScenarioId[ScenarioId["AIDECK___EXPERT_DEATH_KNIGHT"] = 4317] = "AIDECK___EXPERT_DEATH_KNIGHT";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_PROLOGUE"] = 4391] = "BOH_FAELIN_STORY_PROLOGUE";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_01"] = 4392] = "BOH_FAELIN_STORY_01";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_02"] = 4393] = "BOH_FAELIN_STORY_02";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_03"] = 4394] = "BOH_FAELIN_STORY_03";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_04"] = 4395] = "BOH_FAELIN_STORY_04";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_05A"] = 4396] = "BOH_FAELIN_STORY_05A";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_05B"] = 4397] = "BOH_FAELIN_STORY_05B";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_06"] = 4398] = "BOH_FAELIN_STORY_06";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_07"] = 4399] = "BOH_FAELIN_STORY_07";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_08"] = 4400] = "BOH_FAELIN_STORY_08";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_09A"] = 4401] = "BOH_FAELIN_STORY_09A";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_09B"] = 4402] = "BOH_FAELIN_STORY_09B";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_10A"] = 4403] = "BOH_FAELIN_STORY_10A";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_10B"] = 4404] = "BOH_FAELIN_STORY_10B";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_11"] = 4405] = "BOH_FAELIN_STORY_11";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_12"] = 4406] = "BOH_FAELIN_STORY_12";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_13"] = 4407] = "BOH_FAELIN_STORY_13";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_14"] = 4408] = "BOH_FAELIN_STORY_14";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_15"] = 4409] = "BOH_FAELIN_STORY_15";
    ScenarioId[ScenarioId["BOH_FAELIN_STORY_16"] = 4410] = "BOH_FAELIN_STORY_16";
    ScenarioId[ScenarioId["WIZARD_DUELS___ALTERAC_VALLEY"] = 4429] = "WIZARD_DUELS___ALTERAC_VALLEY";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_001"] = 4478] = "BOM_07_SCABBS_FIGHT_001";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_006"] = 4479] = "BOM_07_SCABBS_FIGHT_006";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_007"] = 4481] = "BOM_07_SCABBS_FIGHT_007";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_008"] = 4482] = "BOM_07_SCABBS_FIGHT_008";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_005"] = 4483] = "BOM_07_SCABBS_FIGHT_005";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_002"] = 4484] = "BOM_07_SCABBS_FIGHT_002";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_003"] = 4485] = "BOM_07_SCABBS_FIGHT_003";
    ScenarioId[ScenarioId["BOM_07_SCABBS_FIGHT_004"] = 4486] = "BOM_07_SCABBS_FIGHT_004";
    ScenarioId[ScenarioId["_220_ARENA_SEASON"] = 4689] = "_220_ARENA_SEASON";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_001"] = 4731] = "BOM_08_TAVISH_FIGHT_001";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_002"] = 4732] = "BOM_08_TAVISH_FIGHT_002";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_003"] = 4733] = "BOM_08_TAVISH_FIGHT_003";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_004"] = 4734] = "BOM_08_TAVISH_FIGHT_004";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_005"] = 4735] = "BOM_08_TAVISH_FIGHT_005";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_006"] = 4736] = "BOM_08_TAVISH_FIGHT_006";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_007"] = 4737] = "BOM_08_TAVISH_FIGHT_007";
    ScenarioId[ScenarioId["BOM_08_TAVISH_FIGHT_008"] = 4738] = "BOM_08_TAVISH_FIGHT_008";
    ScenarioId[ScenarioId["WIZARD_DUELS___ALTERAC_VALLEY_HEROES"] = 4818] = "WIZARD_DUELS___ALTERAC_VALLEY_HEROES";
    ScenarioId[ScenarioId["AV_PUZZLE_01"] = 4826] = "AV_PUZZLE_01";
    ScenarioId[ScenarioId["AV_PUZZLE_02"] = 4827] = "AV_PUZZLE_02";
    ScenarioId[ScenarioId["BATTLEGROUNDS_8P_PLAYER_VS_AI"] = 4837] = "BATTLEGROUNDS_8P_PLAYER_VS_AI";
    ScenarioId[ScenarioId["AV_PUZZLE_04"] = 4839] = "AV_PUZZLE_04";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_001"] = 4844] = "BOM_09_BRUKAN_FIGHT_001";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_002"] = 4845] = "BOM_09_BRUKAN_FIGHT_002";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_003"] = 4846] = "BOM_09_BRUKAN_FIGHT_003";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_004"] = 4847] = "BOM_09_BRUKAN_FIGHT_004";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_005"] = 4848] = "BOM_09_BRUKAN_FIGHT_005";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_006"] = 4849] = "BOM_09_BRUKAN_FIGHT_006";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_007"] = 4850] = "BOM_09_BRUKAN_FIGHT_007";
    ScenarioId[ScenarioId["BOM_09_BRUKAN_FIGHT_008"] = 4851] = "BOM_09_BRUKAN_FIGHT_008";
    ScenarioId[ScenarioId["_224_ARENA_SEASON"] = 4860] = "_224_ARENA_SEASON";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_001"] = 4877] = "BOM_10_DAWNGRASP_FIGHT_001";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_002"] = 4878] = "BOM_10_DAWNGRASP_FIGHT_002";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_003"] = 4879] = "BOM_10_DAWNGRASP_FIGHT_003";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_004"] = 4880] = "BOM_10_DAWNGRASP_FIGHT_004";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_005"] = 4881] = "BOM_10_DAWNGRASP_FIGHT_005";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_006"] = 4882] = "BOM_10_DAWNGRASP_FIGHT_006";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_007"] = 4883] = "BOM_10_DAWNGRASP_FIGHT_007";
    ScenarioId[ScenarioId["BOM_10_DAWNGRASP_FIGHT_008"] = 4884] = "BOM_10_DAWNGRASP_FIGHT_008";
    ScenarioId[ScenarioId["WIZARD_DUELS___THE_SUNKEN_CITY"] = 4891] = "WIZARD_DUELS___THE_SUNKEN_CITY";
    ScenarioId[ScenarioId["_230_ARENA_SEASON"] = 4892] = "_230_ARENA_SEASON";
    ScenarioId[ScenarioId["DEATH_KNIGHT_PROLOGUE"] = 4909] = "DEATH_KNIGHT_PROLOGUE";
    ScenarioId[ScenarioId["DEATH_KNIGHT_PROLOGUE_2"] = 4910] = "DEATH_KNIGHT_PROLOGUE_2";
    ScenarioId[ScenarioId["DEATH_KNIGHT_PROLOGUE_3"] = 4911] = "DEATH_KNIGHT_PROLOGUE_3";
    ScenarioId[ScenarioId["DEATH_KNIGHT_PROLOGUE_4"] = 4912] = "DEATH_KNIGHT_PROLOGUE_4";
    ScenarioId[ScenarioId["WIZARD_DUELS___REVENDRETH"] = 4914] = "WIZARD_DUELS___REVENDRETH";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_REVENDRETH_03"] = 4955] = "AIBOT_DEMONHUNTER_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_DRUID_REVENDRETH_03_STANDARDGAMEENTITY"] = 4956] = "AIBOT_DRUID_REVENDRETH_03_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_HUNTER_REVENDRETH_03"] = 4957] = "AIBOT_HUNTER_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_MAGE_REVENDRETH_03_STANDARDGAMEENTITY"] = 4958] = "AIBOT_MAGE_REVENDRETH_03_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_PALADIN_REVENDRETH_03_STANDARDGAMEENTITY"] = 4959] = "AIBOT_PALADIN_REVENDRETH_03_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_PRIEST_REVENDRETH_03_TRAINING"] = 4960] = "AIBOT_PRIEST_REVENDRETH_03_TRAINING";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_03_TRAINING"] = 4961] = "AIBOT_ROGUE_REVENDRETH_03_TRAINING";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_REVENDRETH_03_TRAINING"] = 4962] = "AIBOT_SHAMAN_REVENDRETH_03_TRAINING";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_REVENDRETH_03_TRAINING"] = 4963] = "AIBOT_WARLOCK_REVENDRETH_03_TRAINING";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_REVENDRETH_03_TRAINING"] = 4964] = "AIBOT_WARRIOR_REVENDRETH_03_TRAINING";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_REVENDRETH_03_1"] = 4966] = "AIBOT_DEMONHUNTER_REVENDRETH_03_1";
    ScenarioId[ScenarioId["AIBOT_DRUID_REVENDRETH_03"] = 4967] = "AIBOT_DRUID_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_HUNTER_REVENDRETH_03_"] = 4968] = "AIBOT_HUNTER_REVENDRETH_03_";
    ScenarioId[ScenarioId["AIBOT_MAGE_REVENDRETH_03"] = 4969] = "AIBOT_MAGE_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_PALADIN_REVENDRETH_03"] = 4970] = "AIBOT_PALADIN_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_PRIEST_REVENDRETH_03"] = 4971] = "AIBOT_PRIEST_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_03"] = 4972] = "AIBOT_ROGUE_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_03_1"] = 4973] = "AIBOT_ROGUE_REVENDRETH_03_1";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_REVENDRETH_03"] = 4974] = "AIBOT_WARLOCK_REVENDRETH_03";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_REVENDRETH_03"] = 4975] = "AIBOT_WARRIOR_REVENDRETH_03";
    ScenarioId[ScenarioId["_234_ARENA_SEASON"] = 4978] = "_234_ARENA_SEASON";
    ScenarioId[ScenarioId["WIZARD_DUELS___DEATH_KNIGHT"] = 4997] = "WIZARD_DUELS___DEATH_KNIGHT";
    ScenarioId[ScenarioId["_236_TAVERNS_OF_TIME_ARENA_SEASON"] = 5003] = "_236_TAVERNS_OF_TIME_ARENA_SEASON";
    ScenarioId[ScenarioId["_240_ARENA_SEASON"] = 5022] = "_240_ARENA_SEASON";
    ScenarioId[ScenarioId["WHODUNNIT_QUESTION"] = 5026] = "WHODUNNIT_QUESTION";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_REVENDRETH_02"] = 5028] = "AIBOT_DEMONHUNTER_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_DRUID_REVENDRETH_02"] = 5029] = "AIBOT_DRUID_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_HUNTER_REVENDRETH_02"] = 5030] = "AIBOT_HUNTER_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_MAGE_REVENDRETH_02"] = 5031] = "AIBOT_MAGE_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_PALADIN_REVENDRETH_02"] = 5032] = "AIBOT_PALADIN_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_PRIEST_REVENDRETH_02"] = 5033] = "AIBOT_PRIEST_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_02"] = 5034] = "AIBOT_ROGUE_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_02_1"] = 5035] = "AIBOT_ROGUE_REVENDRETH_02_1";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_REVENDRETH_02"] = 5036] = "AIBOT_WARLOCK_REVENDRETH_02";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_REVENDRETH_02"] = 5037] = "AIBOT_WARRIOR_REVENDRETH_02";
    ScenarioId[ScenarioId["_244_ARENA_SEASON"] = 5038] = "_244_ARENA_SEASON";
    ScenarioId[ScenarioId["AIBOT_PATRON_FIGHT_PLAYTESTING_RANDOMCLASS"] = 5040] = "AIBOT_PATRON_FIGHT_PLAYTESTING_RANDOMCLASS";
    ScenarioId[ScenarioId["_250_ARENA_SEASON"] = 5041] = "_250_ARENA_SEASON";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_CORE"] = 5042] = "AIBOT_DEMONHUNTER_CORE";
    ScenarioId[ScenarioId["AIBOT_DRUID_CORE"] = 5043] = "AIBOT_DRUID_CORE";
    ScenarioId[ScenarioId["AIBOT_HUNTER_CORE"] = 5044] = "AIBOT_HUNTER_CORE";
    ScenarioId[ScenarioId["AIBOT_MAGE_CORE"] = 5045] = "AIBOT_MAGE_CORE";
    ScenarioId[ScenarioId["AIBOT_PALADIN_CORE"] = 5046] = "AIBOT_PALADIN_CORE";
    ScenarioId[ScenarioId["AIBOT_PRIEST_CORE"] = 5047] = "AIBOT_PRIEST_CORE";
    ScenarioId[ScenarioId["AIBOT_ROGUE_CORE"] = 5048] = "AIBOT_ROGUE_CORE";
    ScenarioId[ScenarioId["AIBOT_ROGUE_CORE_1"] = 5049] = "AIBOT_ROGUE_CORE_1";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_CORE"] = 5050] = "AIBOT_WARLOCK_CORE";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_CORE"] = 5051] = "AIBOT_WARRIOR_CORE";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_REVENDRETH_01"] = 5056] = "AIBOT_DEMONHUNTER_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_DRUID_REVENDRETH_01"] = 5057] = "AIBOT_DRUID_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_HUNTER_REVENDRETH_01"] = 5058] = "AIBOT_HUNTER_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_MAGE_REVENDRETH_01"] = 5059] = "AIBOT_MAGE_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_PALADIN_REVENDRETH_01"] = 5060] = "AIBOT_PALADIN_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_PRIEST_REVENDRETH_01"] = 5061] = "AIBOT_PRIEST_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_01"] = 5062] = "AIBOT_ROGUE_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_ROGUE_REVENDRETH_01_1"] = 5063] = "AIBOT_ROGUE_REVENDRETH_01_1";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_REVENDRETH_01"] = 5064] = "AIBOT_WARLOCK_REVENDRETH_01";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_REVENDRETH_01_1"] = 5065] = "AIBOT_WARLOCK_REVENDRETH_01_1";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_CORE_STANDARD"] = 5066] = "AIBOT_DEMONHUNTER_CORE_STANDARD";
    ScenarioId[ScenarioId["AIBOT_DRUID_CORE_STANDARDGAMEENTITY"] = 5067] = "AIBOT_DRUID_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_HUNTER_CORE_STANDARDGAMEENTITY"] = 5068] = "AIBOT_HUNTER_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_MAGE_CORE_STANDARDGAMEENTITY"] = 5069] = "AIBOT_MAGE_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_PALADIN_CORE_STANDARDGAMEENTITY"] = 5070] = "AIBOT_PALADIN_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_PRIEST_CORE_STANDARD"] = 5071] = "AIBOT_PRIEST_CORE_STANDARD";
    ScenarioId[ScenarioId["AIBOT_ROGUE_CORE_STANDARDGAMEENTITY"] = 5072] = "AIBOT_ROGUE_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_ROGUE_CORE_STANDARDGAMEENTITY_1"] = 5073] = "AIBOT_ROGUE_CORE_STANDARDGAMEENTITY_1";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_CORE_STANDARDGAMEENTITY"] = 5074] = "AIBOT_WARLOCK_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_CORE_STANDARDGAMEENTITY"] = 5075] = "AIBOT_WARRIOR_CORE_STANDARDGAMEENTITY";
    ScenarioId[ScenarioId["TB_BOTB_MUKLA"] = 5088] = "TB_BOTB_MUKLA";
    ScenarioId[ScenarioId["TB_BOTB_ZOKFOGSNOUT"] = 5089] = "TB_BOTB_ZOKFOGSNOUT";
    ScenarioId[ScenarioId["TB_BOTB_INZAH"] = 5090] = "TB_BOTB_INZAH";
    ScenarioId[ScenarioId["TB_BOTB_MANASTORM"] = 5091] = "TB_BOTB_MANASTORM";
    ScenarioId[ScenarioId["TB_BOTB_HEDANIS"] = 5092] = "TB_BOTB_HEDANIS";
    ScenarioId[ScenarioId["TB_BOTB_BLINGTRON"] = 5093] = "TB_BOTB_BLINGTRON";
    ScenarioId[ScenarioId["TB_BOTB_KANGOR"] = 5094] = "TB_BOTB_KANGOR";
    ScenarioId[ScenarioId["TB_BOTB_RIN"] = 5095] = "TB_BOTB_RIN";
    ScenarioId[ScenarioId["TB_BOTB_CAGEHEAD"] = 5096] = "TB_BOTB_CAGEHEAD";
    ScenarioId[ScenarioId["TB_BOTB_VOONE"] = 5097] = "TB_BOTB_VOONE";
    ScenarioId[ScenarioId["TB_BOTB_HALVERIA"] = 5098] = "TB_BOTB_HALVERIA";
    ScenarioId[ScenarioId["TB_BOTB_ETC"] = 5099] = "TB_BOTB_ETC";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_BASIC"] = 5157] = "AIBOT_DEMONHUNTER_BASIC";
    ScenarioId[ScenarioId["AIBOT_DRUID_BASIC"] = 5158] = "AIBOT_DRUID_BASIC";
    ScenarioId[ScenarioId["AIBOT_HUNTER_BASIC"] = 5159] = "AIBOT_HUNTER_BASIC";
    ScenarioId[ScenarioId["AIBOT_MAGE_BASIC"] = 5160] = "AIBOT_MAGE_BASIC";
    ScenarioId[ScenarioId["AIBOT_PALADIN_BASIC"] = 5161] = "AIBOT_PALADIN_BASIC";
    ScenarioId[ScenarioId["AIBOT_PRIEST_BASIC"] = 5162] = "AIBOT_PRIEST_BASIC";
    ScenarioId[ScenarioId["AIBOT_ROGUE_BASIC"] = 5163] = "AIBOT_ROGUE_BASIC";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_BASIC"] = 5164] = "AIBOT_SHAMAN_BASIC";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_BASIC"] = 5165] = "AIBOT_WARLOCK_BASIC";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_BASIC"] = 5166] = "AIBOT_WARRIOR_BASIC";
    ScenarioId[ScenarioId["_254_ARENA_SEASON"] = 5170] = "_254_ARENA_SEASON";
    ScenarioId[ScenarioId["BATTLEGROUNDS_DUO_VS_AI"] = 5172] = "BATTLEGROUNDS_DUO_VS_AI";
    ScenarioId[ScenarioId["BATTLEGROUNDS_DUO"] = 5173] = "BATTLEGROUNDS_DUO";
    ScenarioId[ScenarioId["_260_ARENA_SEASON"] = 5175] = "_260_ARENA_SEASON";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_FESTIVAL_03"] = 5179] = "AIBOT_DEMONHUNTER_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_DRUID_FESTIVAL_03"] = 5180] = "AIBOT_DRUID_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_HUNTER_FESTIVAL_03"] = 5181] = "AIBOT_HUNTER_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_MAGE_FESTIVAL_03"] = 5182] = "AIBOT_MAGE_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_PALADIN_FESTIVAL_03"] = 5183] = "AIBOT_PALADIN_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_PRIEST_FESTIVAL_03"] = 5184] = "AIBOT_PRIEST_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_ROGUE_FESTIVAL_03"] = 5185] = "AIBOT_ROGUE_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_FESTIVAL_03"] = 5186] = "AIBOT_SHAMAN_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_FESTIVAL_03"] = 5187] = "AIBOT_WARLOCK_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_FESTIVAL_03"] = 5188] = "AIBOT_WARRIOR_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_FESTIVAL_02"] = 5189] = "AIBOT_DEMONHUNTER_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_DRUID_FESTIVAL_02"] = 5190] = "AIBOT_DRUID_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_HUNTER_FESTIVAL_02"] = 5191] = "AIBOT_HUNTER_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_MAGE_FESTIVAL_02"] = 5192] = "AIBOT_MAGE_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_PALADIN_FESTIVAL_02"] = 5193] = "AIBOT_PALADIN_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_PRIEST_FESTIVAL_02"] = 5194] = "AIBOT_PRIEST_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_FESTIVAL_02"] = 5195] = "AIBOT_SHAMAN_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_ROGUE_FESTIVAL_02"] = 5196] = "AIBOT_ROGUE_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_FESTIVAL_02"] = 5197] = "AIBOT_WARLOCK_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_FESTIVAL_02"] = 5198] = "AIBOT_WARRIOR_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_FESTIVAL_03"] = 5199] = "AIBOT_DEATHKNIGHT_FESTIVAL_03";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_FESTIVAL_02"] = 5200] = "AIBOT_DEATHKNIGHT_FESTIVAL_02";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_CORE_2023"] = 5201] = "AIBOT_DEMONHUNTER_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_DRUID_CORE_2023"] = 5202] = "AIBOT_DRUID_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_HUNTER_CORE_2023"] = 5203] = "AIBOT_HUNTER_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_MAGE_CORE_2023"] = 5204] = "AIBOT_MAGE_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_PALADIN_CORE_2023"] = 5205] = "AIBOT_PALADIN_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_PRIEST_CORE_2023"] = 5206] = "AIBOT_PRIEST_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_ROGUE_CORE_2023"] = 5207] = "AIBOT_ROGUE_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_CORE_2023"] = 5208] = "AIBOT_SHAMAN_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_CORE_2023"] = 5209] = "AIBOT_WARLOCK_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_CORE_2023"] = 5210] = "AIBOT_WARRIOR_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_CORE_2023"] = 5211] = "AIBOT_DEATHKNIGHT_CORE_2023";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_BASIC_2023"] = 5212] = "AIBOT_DEMONHUNTER_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_DRUID_BASIC_2023"] = 5213] = "AIBOT_DRUID_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_HUNTER_BASIC_2023"] = 5214] = "AIBOT_HUNTER_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_MAGE_BASIC_2023"] = 5215] = "AIBOT_MAGE_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_PALADIN_BASIC_2023"] = 5216] = "AIBOT_PALADIN_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_PRIEST_BASIC_2023"] = 5217] = "AIBOT_PRIEST_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_ROGUE_BASIC_2023"] = 5218] = "AIBOT_ROGUE_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_BASIC_2023"] = 5219] = "AIBOT_SHAMAN_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_BASIC_2023"] = 5220] = "AIBOT_WARLOCK_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_BASIC_2023"] = 5221] = "AIBOT_WARRIOR_BASIC_2023";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_BASIC_2023"] = 5222] = "AIBOT_DEATHKNIGHT_BASIC_2023";
    ScenarioId[ScenarioId["TWIST___NEW_AGE"] = 5234] = "TWIST___NEW_AGE";
    ScenarioId[ScenarioId["_264_ARENA_SEASON"] = 5237] = "_264_ARENA_SEASON";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_FESTIVAL_03_BOTJAIL"] = 5252] = "AIBOT_DEMONHUNTER_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DRUID_FESTIVAL_03_BOTJAIL"] = 5253] = "AIBOT_DRUID_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_HUNTER_FESTIVAL_03_BOTJAIL"] = 5254] = "AIBOT_HUNTER_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_MAGE_FESTIVAL_03_BOTJAIL"] = 5255] = "AIBOT_MAGE_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_PALADIN_FESTIVAL_03_BOTJAIL"] = 5256] = "AIBOT_PALADIN_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_PRIEST_FESTIVAL_03_BOTJAIL"] = 5257] = "AIBOT_PRIEST_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_ROGUE_FESTIVAL_03_BOTJAIL"] = 5258] = "AIBOT_ROGUE_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_FESTIVAL_03_BOTJAIL"] = 5259] = "AIBOT_SHAMAN_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_FESTIVAL_03_BOTJAIL"] = 5260] = "AIBOT_WARLOCK_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_FESTIVAL_03_BOTJAIL"] = 5261] = "AIBOT_WARRIOR_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_FESTIVAL_02_BOTJAIL"] = 5262] = "AIBOT_DEMONHUNTER_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DRUID_FESTIVAL_02_BOTJAIL"] = 5263] = "AIBOT_DRUID_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_HUNTER_FESTIVAL_02_BOTJAIL"] = 5264] = "AIBOT_HUNTER_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_MAGE_FESTIVAL_02_BOTJAIL"] = 5265] = "AIBOT_MAGE_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_PALADIN_FESTIVAL_02_BOTJAIL"] = 5266] = "AIBOT_PALADIN_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_PRIEST_FESTIVAL_02_BOTJAIL"] = 5267] = "AIBOT_PRIEST_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_FESTIVAL_02_BOTJAIL"] = 5268] = "AIBOT_SHAMAN_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_ROGUE_FESTIVAL_02_BOTJAIL"] = 5269] = "AIBOT_ROGUE_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_FESTIVAL_02_BOTJAIL"] = 5270] = "AIBOT_WARLOCK_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_FESTIVAL_02_BOTJAIL"] = 5271] = "AIBOT_WARRIOR_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_FESTIVAL_03_BOTJAIL"] = 5272] = "AIBOT_DEATHKNIGHT_FESTIVAL_03_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_FESTIVAL_02_BOTJAIL"] = 5273] = "AIBOT_DEATHKNIGHT_FESTIVAL_02_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_CORE_2023_BOTJAIL"] = 5274] = "AIBOT_DEMONHUNTER_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DRUID_CORE_2023_BOTJAIL"] = 5275] = "AIBOT_DRUID_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_HUNTER_CORE_2023_BOTJAIL"] = 5276] = "AIBOT_HUNTER_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_MAGE_CORE_2023_BOTJAIL"] = 5277] = "AIBOT_MAGE_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_PALADIN_CORE_2023_BOTJAIL"] = 5278] = "AIBOT_PALADIN_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_PRIEST_CORE_2023_BOTJAIL"] = 5279] = "AIBOT_PRIEST_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_ROGUE_CORE_2023_BOTJAIL"] = 5280] = "AIBOT_ROGUE_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_CORE_2023__BOTJAIL"] = 5281] = "AIBOT_SHAMAN_CORE_2023__BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_CORE_2023_BOTJAIL"] = 5282] = "AIBOT_WARLOCK_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_CORE_2023_BOTJAIL"] = 5283] = "AIBOT_WARRIOR_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_CORE_2023_BOTJAIL"] = 5284] = "AIBOT_DEATHKNIGHT_CORE_2023_BOTJAIL";
    ScenarioId[ScenarioId["TUTORIAL_REVISED_FIGHT_01"] = 5287] = "TUTORIAL_REVISED_FIGHT_01";
    ScenarioId[ScenarioId["TUTORIAL_REVISED_FIGHT_02"] = 5289] = "TUTORIAL_REVISED_FIGHT_02";
    ScenarioId[ScenarioId["TUTORIAL_REVISED_FIGHT_03"] = 5290] = "TUTORIAL_REVISED_FIGHT_03";
    ScenarioId[ScenarioId["_270_ARENA_SEASON"] = 5292] = "_270_ARENA_SEASON";
    ScenarioId[ScenarioId["TWIST___WONDERS"] = 5294] = "TWIST___WONDERS";
    ScenarioId[ScenarioId["TWIST___WONDERS_XL"] = 5295] = "TWIST___WONDERS_XL";
    ScenarioId[ScenarioId["WIZARD_DUELS___TITANS_MINISET"] = 5300] = "WIZARD_DUELS___TITANS_MINISET";
    ScenarioId[ScenarioId["TWIST___UNGORO"] = 5301] = "TWIST___UNGORO";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_TITANS_03"] = 5302] = "AIBOT_DEMONHUNTER_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_DRUID_TITANS_03"] = 5303] = "AIBOT_DRUID_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_HUNTER_TITANS_03"] = 5304] = "AIBOT_HUNTER_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_MAGE_TITANS_03"] = 5305] = "AIBOT_MAGE_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_PALADIN_TITANS_03"] = 5306] = "AIBOT_PALADIN_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_PRIEST_TITANS_03"] = 5307] = "AIBOT_PRIEST_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_ROGUE_TITANS_03"] = 5308] = "AIBOT_ROGUE_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_TITANS_03"] = 5309] = "AIBOT_SHAMAN_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_TITANS_03"] = 5310] = "AIBOT_WARLOCK_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_TITANS_03"] = 5311] = "AIBOT_WARRIOR_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_TITANS_02"] = 5312] = "AIBOT_DEMONHUNTER_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_DRUID_TITANS_02"] = 5313] = "AIBOT_DRUID_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_HUNTER_TITANS_02"] = 5314] = "AIBOT_HUNTER_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_MAGE_TITANS_02"] = 5315] = "AIBOT_MAGE_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_PALADIN_TITANS_02"] = 5316] = "AIBOT_PALADIN_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_PRIEST_TITANS_02"] = 5317] = "AIBOT_PRIEST_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_TITANS_02"] = 5318] = "AIBOT_SHAMAN_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_ROGUE_TITANS_02"] = 5319] = "AIBOT_ROGUE_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_TITANS_02"] = 5320] = "AIBOT_WARLOCK_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_TITANS_02"] = 5321] = "AIBOT_WARRIOR_TITANS_02";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_TITANS_03"] = 5322] = "AIBOT_DEATHKNIGHT_TITANS_03";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_TITANS_02"] = 5323] = "AIBOT_DEATHKNIGHT_TITANS_02";
    ScenarioId[ScenarioId["TWIST___CLASSIC"] = 5325] = "TWIST___CLASSIC";
    ScenarioId[ScenarioId["_274_ARENA_SEASON"] = 5337] = "_274_ARENA_SEASON";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_BASIC_2023_1"] = 5339] = "AIBOT_DEMONHUNTER_BASIC_2023_1";
    ScenarioId[ScenarioId["AIBOT_DRUID_BASICPOLYMORPH_2023"] = 5340] = "AIBOT_DRUID_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_HUNTER_BASICPOLYMORPH_2023"] = 5341] = "AIBOT_HUNTER_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_MAGE_BASICPOLYMORPH_2023"] = 5342] = "AIBOT_MAGE_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_PALADIN_BASICPOLYMORPH_2023"] = 5343] = "AIBOT_PALADIN_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_PRIEST_BASICPOLYMORPH_2023"] = 5344] = "AIBOT_PRIEST_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_ROGUE_BASICPOLYMORPH_2023"] = 5345] = "AIBOT_ROGUE_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_SHAMAN_BASICPOLYMORPH_2023"] = 5346] = "AIBOT_SHAMAN_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_WARLOCK_BASICPOLYMORPH_2023"] = 5347] = "AIBOT_WARLOCK_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_WARRIOR_BASICPOLYMORPH_2023"] = 5348] = "AIBOT_WARRIOR_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_DEATHKNIGHT_BASICPOLYMORPH_2023"] = 5349] = "AIBOT_DEATHKNIGHT_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_DEMONHUNTER_BASICPOLYMORPH_2023"] = 5351] = "AIBOT_DEMONHUNTER_BASICPOLYMORPH_2023";
    ScenarioId[ScenarioId["AIBOT_BASICPOLYMORPH_2023_INTRODUCTION_RANDOMCLASS"] = 5353] = "AIBOT_BASICPOLYMORPH_2023_INTRODUCTION_RANDOMCLASS";
    ScenarioId[ScenarioId["WIZARD_DUELS___SHOWDOWN_IN_THE_BADLANDS"] = 5354] = "WIZARD_DUELS___SHOWDOWN_IN_THE_BADLANDS";
    ScenarioId[ScenarioId["_280_ARENA_SEASON"] = 5357] = "_280_ARENA_SEASON";
    ScenarioId[ScenarioId["TWIST___COMMONS"] = 5358] = "TWIST___COMMONS";
    ScenarioId[ScenarioId["_284_ARENA_SEASON"] = 5361] = "_284_ARENA_SEASON";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_1"] = 5362] = "TWIST___FAST_FORWARD___DAY_1";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_2"] = 5363] = "TWIST___FAST_FORWARD___DAY_2";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_3"] = 5364] = "TWIST___FAST_FORWARD___DAY_3";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_4"] = 5365] = "TWIST___FAST_FORWARD___DAY_4";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_5"] = 5366] = "TWIST___FAST_FORWARD___DAY_5";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_6"] = 5367] = "TWIST___FAST_FORWARD___DAY_6";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_7"] = 5368] = "TWIST___FAST_FORWARD___DAY_7";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_8"] = 5369] = "TWIST___FAST_FORWARD___DAY_8";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_9"] = 5370] = "TWIST___FAST_FORWARD___DAY_9";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_10"] = 5371] = "TWIST___FAST_FORWARD___DAY_10";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_11"] = 5372] = "TWIST___FAST_FORWARD___DAY_11";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_12"] = 5373] = "TWIST___FAST_FORWARD___DAY_12";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_13"] = 5374] = "TWIST___FAST_FORWARD___DAY_13";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_14"] = 5375] = "TWIST___FAST_FORWARD___DAY_14";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_15"] = 5376] = "TWIST___FAST_FORWARD___DAY_15";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_16"] = 5377] = "TWIST___FAST_FORWARD___DAY_16";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_17"] = 5378] = "TWIST___FAST_FORWARD___DAY_17";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_18"] = 5379] = "TWIST___FAST_FORWARD___DAY_18";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_19"] = 5380] = "TWIST___FAST_FORWARD___DAY_19";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_20"] = 5381] = "TWIST___FAST_FORWARD___DAY_20";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_21"] = 5382] = "TWIST___FAST_FORWARD___DAY_21";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_22"] = 5383] = "TWIST___FAST_FORWARD___DAY_22";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_23"] = 5384] = "TWIST___FAST_FORWARD___DAY_23";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_24"] = 5385] = "TWIST___FAST_FORWARD___DAY_24";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_25"] = 5386] = "TWIST___FAST_FORWARD___DAY_25";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_26"] = 5387] = "TWIST___FAST_FORWARD___DAY_26";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_27"] = 5388] = "TWIST___FAST_FORWARD___DAY_27";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_28"] = 5389] = "TWIST___FAST_FORWARD___DAY_28";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_29"] = 5390] = "TWIST___FAST_FORWARD___DAY_29";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_30"] = 5391] = "TWIST___FAST_FORWARD___DAY_30";
    ScenarioId[ScenarioId["TWIST___FAST_FORWARD___DAY_31"] = 5392] = "TWIST___FAST_FORWARD___DAY_31";
    ScenarioId[ScenarioId["HM_PUZZLE_02"] = 5394] = "HM_PUZZLE_02";
    ScenarioId[ScenarioId["HM_PUZZLE_04"] = 5396] = "HM_PUZZLE_04";
    ScenarioId[ScenarioId["TWIST___HEROIC_DECK___MONTH_1"] = 5399] = "TWIST___HEROIC_DECK___MONTH_1";
    ScenarioId[ScenarioId["_290_ARENA_SEASON"] = 5401] = "_290_ARENA_SEASON";
    ScenarioId[ScenarioId["_292_ARENA_SEASON"] = 5403] = "_292_ARENA_SEASON";
    ScenarioId[ScenarioId["_294_ARENA_SEASON"] = 5407] = "_294_ARENA_SEASON";
    ScenarioId[ScenarioId["TWIST___HEROIC_DECK___MONTH_2"] = 5409] = "TWIST___HEROIC_DECK___MONTH_2";
    ScenarioId[ScenarioId["BATTLEGROUNDS_DUO_1_PLAYER_VS_AI"] = 5427] = "BATTLEGROUNDS_DUO_1_PLAYER_VS_AI";
    ScenarioId[ScenarioId["_310_ARENA_SEASON"] = 5431] = "_310_ARENA_SEASON";
    ScenarioId[ScenarioId["_312_ARENA_SEASON"] = 5436] = "_312_ARENA_SEASON";
    ScenarioId[ScenarioId["_314_ARENA_SEASON"] = 5437] = "_314_ARENA_SEASON";
    ScenarioId[ScenarioId["_316_ARENA_SEASON"] = 5438] = "_316_ARENA_SEASON";
    ScenarioId[ScenarioId["_320_ARENA_SEASON"] = 5455] = "_320_ARENA_SEASON";
    ScenarioId[ScenarioId["_322_ARENA_SEASON"] = 5457] = "_322_ARENA_SEASON";
    ScenarioId[ScenarioId["_324_ARENA_SEASON"] = 5464] = "_324_ARENA_SEASON";
    ScenarioId[ScenarioId["_330_ARENA_SEASON"] = 5470] = "_330_ARENA_SEASON";
    ScenarioId[ScenarioId["CATAA_01_DEATHWING"] = 5473] = "CATAA_01_DEATHWING";
    ScenarioId[ScenarioId["_334_ARENA_SEASON"] = 5476] = "_334_ARENA_SEASON";
    ScenarioId[ScenarioId["_340_ARENA_SEASON"] = 5481] = "_340_ARENA_SEASON";
    ScenarioId[ScenarioId["_344_ARENA_SEASON"] = 5485] = "_344_ARENA_SEASON";
    ScenarioId[ScenarioId["_350_ARENA_SEASON"] = 5501] = "_350_ARENA_SEASON";
})(ScenarioId = exports.ScenarioId || (exports.ScenarioId = {}));
