// @firestone-hs/reference-data/dist/enums/scene.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SceneMode = void 0;
var SceneMode;
(function (SceneMode) {
    SceneMode[SceneMode["INVALID"] = 0] = "INVALID";
    SceneMode[SceneMode["STARTUP"] = 1] = "STARTUP";
    SceneMode[SceneMode["LOGIN"] = 2] = "LOGIN";
    SceneMode[SceneMode["HUB"] = 3] = "HUB";
    SceneMode[SceneMode["GAMEPLAY"] = 4] = "GAMEPLAY";
    SceneMode[SceneMode["COLLECTIONMANAGER"] = 5] = "COLLECTIONMANAGER";
    SceneMode[SceneMode["PACKOPENING"] = 6] = "PACKOPENING";
    SceneMode[SceneMode["TOURNAMENT"] = 7] = "TOURNAMENT";
    SceneMode[SceneMode["FRIENDLY"] = 8] = "FRIENDLY";
    SceneMode[SceneMode["FATAL_ERROR"] = 9] = "FATAL_ERROR";
    SceneMode[SceneMode["DRAFT"] = 10] = "DRAFT";
    SceneMode[SceneMode["CREDITS"] = 11] = "CREDITS";
    SceneMode[SceneMode["RESET"] = 12] = "RESET";
    SceneMode[SceneMode["ADVENTURE"] = 13] = "ADVENTURE";
    SceneMode[SceneMode["TAVERN_BRAWL"] = 14] = "TAVERN_BRAWL";
    SceneMode[SceneMode["BACON"] = 15] = "BACON";
    SceneMode[SceneMode["GAME_MODE"] = 16] = "GAME_MODE";
    SceneMode[SceneMode["PVP_DUNGEON_RUN"] = 17] = "PVP_DUNGEON_RUN";
    SceneMode[SceneMode["BACON_COLLECTION"] = 18] = "BACON_COLLECTION";
    SceneMode[SceneMode["LETTUCE_VILLAGE"] = 19] = "LETTUCE_VILLAGE";
    SceneMode[SceneMode["LETTUCE_BOUNTY_BOARD"] = 20] = "LETTUCE_BOUNTY_BOARD";
    SceneMode[SceneMode["LETTUCE_MAP"] = 21] = "LETTUCE_MAP";
    SceneMode[SceneMode["LETTUCE_PLAY"] = 22] = "LETTUCE_PLAY";
    SceneMode[SceneMode["LETTUCE_COLLECTION"] = 23] = "LETTUCE_COLLECTION";
    SceneMode[SceneMode["LETTUCE_COOP"] = 24] = "LETTUCE_COOP";
    SceneMode[SceneMode["LETTUCE_FRIENDLY"] = 25] = "LETTUCE_FRIENDLY";
    SceneMode[SceneMode["LETTUCE_BOUNTY_TEAM_SELECT"] = 26] = "LETTUCE_BOUNTY_TEAM_SELECT";
    SceneMode[SceneMode["LETTUCE_PACK_OPENING"] = 27] = "LETTUCE_PACK_OPENING";
    SceneMode[SceneMode["LUCKY_DRAW"] = 28] = "LUCKY_DRAW";
})(SceneMode = exports.SceneMode || (exports.SceneMode = {}));
