// @firestone-hs/reference-data/dist/enums/scenario-groups.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SOLO_SCENARIO_WITH_LOGGED_DECKLIST = exports.SCENARIO_WITHOUT_RESTART = exports.PRACTICE_ALL = exports.GALAKROND_ALL = exports.GALAKROND_EVIL = exports.GALAKROND_EXPLORER = exports.TOMBS_OF_TERROR_ALL = exports.DALARAN_HEIST_ALL = exports.ARENAS = void 0;
const scenario_id_1 = require("./scenario-id");
exports.ARENAS = [
    scenario_id_1.ScenarioId._140_ARENA_SEASON,
    scenario_id_1.ScenarioId._144_ARENA_SEASON,
    scenario_id_1.ScenarioId._150_ARENA_SEASON,
    scenario_id_1.ScenarioId._154_ARENA_SEASON,
    scenario_id_1.ScenarioId._160_ARENA_SEASON,
    scenario_id_1.ScenarioId._164_ARENA_SEASON,
    scenario_id_1.ScenarioId._170_ARENA_SEASON,
    scenario_id_1.ScenarioId._174_ARENA_SEASON,
    scenario_id_1.ScenarioId._184_ARENA_SEASON,
    scenario_id_1.ScenarioId._190_ARENA_SEASON,
    scenario_id_1.ScenarioId._194_ARENA_SEASON,
    scenario_id_1.ScenarioId._200_ARENA_SEASON,
    scenario_id_1.ScenarioId._204_ARENA_SEASON,
    scenario_id_1.ScenarioId._210_ARENA_SEASON_,
    scenario_id_1.ScenarioId._220_ARENA_SEASON,
    scenario_id_1.ScenarioId._224_ARENA_SEASON,
    scenario_id_1.ScenarioId._230_ARENA_SEASON,
    scenario_id_1.ScenarioId._234_ARENA_SEASON,
    scenario_id_1.ScenarioId._236_TAVERNS_OF_TIME_ARENA_SEASON,
    scenario_id_1.ScenarioId._240_ARENA_SEASON,
    scenario_id_1.ScenarioId._244_ARENA_SEASON,
    scenario_id_1.ScenarioId._254_ARENA_SEASON,
    scenario_id_1.ScenarioId._260_ARENA_SEASON,
    scenario_id_1.ScenarioId._264_ARENA_SEASON,
    scenario_id_1.ScenarioId._270_ARENA_SEASON,
    scenario_id_1.ScenarioId._280_ARENA_SEASON,
    scenario_id_1.ScenarioId._284_ARENA_SEASON,
    scenario_id_1.ScenarioId._290_ARENA_SEASON,
    scenario_id_1.ScenarioId._292_ARENA_SEASON,
    scenario_id_1.ScenarioId._294_ARENA_SEASON,
    scenario_id_1.ScenarioId._310_ARENA_SEASON,
    scenario_id_1.ScenarioId._312_ARENA_SEASON,
    scenario_id_1.ScenarioId._314_ARENA_SEASON,
    scenario_id_1.ScenarioId._316_ARENA_SEASON,
    scenario_id_1.ScenarioId._320_ARENA_SEASON,
    scenario_id_1.ScenarioId._322_ARENA_SEASON,
    scenario_id_1.ScenarioId._324_ARENA_SEASON,
    scenario_id_1.ScenarioId._330_ARENA_SEASON,
    scenario_id_1.ScenarioId._334_ARENA_SEASON,
    scenario_id_1.ScenarioId._340_ARENA_SEASON,
    scenario_id_1.ScenarioId._344_ARENA_SEASON,
    scenario_id_1.ScenarioId._350_ARENA_SEASON,
];
exports.DALARAN_HEIST_ALL = [
    scenario_id_1.ScenarioId.DALA_BANK,
    scenario_id_1.ScenarioId.DALA_VIOLETHOLD,
    scenario_id_1.ScenarioId.DALA_STREETS,
    scenario_id_1.ScenarioId.DALA_UNDERBELLY,
    scenario_id_1.ScenarioId.DALA_CITADEL,
    scenario_id_1.ScenarioId.DALA_TAVERN,
    scenario_id_1.ScenarioId.DALA_BANK_HEROIC,
    scenario_id_1.ScenarioId.DALA_VIOLETHOLD_HEROIC,
    scenario_id_1.ScenarioId.DALA_STREETS_HEROIC,
    scenario_id_1.ScenarioId.DALA_UNDERBELLY_HEROIC,
    scenario_id_1.ScenarioId.DALA_CITADEL_HEROIC,
    scenario_id_1.ScenarioId.DALA_TAVERN_HEROIC,
];
exports.TOMBS_OF_TERROR_ALL = [
    scenario_id_1.ScenarioId.ULDA_CITY,
    scenario_id_1.ScenarioId.ULDA_DESERT,
    scenario_id_1.ScenarioId.ULDA_TOMB,
    scenario_id_1.ScenarioId.ULDA_HALLS,
    scenario_id_1.ScenarioId.ULDA_SANCTUM,
    scenario_id_1.ScenarioId.ULDA_CITY_HEROIC,
    scenario_id_1.ScenarioId.ULDA_DESERT_HEROIC,
    scenario_id_1.ScenarioId.ULDA_TOMB_HEROIC,
    scenario_id_1.ScenarioId.ULDA_HALLS_HEROIC,
    scenario_id_1.ScenarioId.ULDA_SANCTUM_HEROIC,
    scenario_id_1.ScenarioId.ULDA_TAVERN,
    scenario_id_1.ScenarioId.ULDA_TAVERN_HEROIC,
];
exports.GALAKROND_EXPLORER = [
    scenario_id_1.ScenarioId.DRGA_GOOD_1_1,
    scenario_id_1.ScenarioId.DRGA_GOOD_1_2,
    scenario_id_1.ScenarioId.DRGA_GOOD_1_3,
    scenario_id_1.ScenarioId.DRGA_GOOD_2_1,
    scenario_id_1.ScenarioId.DRGA_GOOD_2_2,
    scenario_id_1.ScenarioId.DRGA_GOOD_2_3,
    scenario_id_1.ScenarioId.DRGA_GOOD_3_1,
    scenario_id_1.ScenarioId.DRGA_GOOD_3_2,
    scenario_id_1.ScenarioId.DRGA_GOOD_3_3,
    scenario_id_1.ScenarioId.DRGA_GOOD_4_1,
    scenario_id_1.ScenarioId.DRGA_GOOD_4_2,
    scenario_id_1.ScenarioId.DRGA_GOOD_4_3,
    scenario_id_1.ScenarioId.DRGA_GOOD_1_1_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_1_2_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_1_3_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_2_1_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_2_2_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_2_3_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_3_1_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_3_2_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_3_3_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_4_1_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_4_2_H,
    scenario_id_1.ScenarioId.DRGA_GOOD_4_3_H,
];
exports.GALAKROND_EVIL = [
    scenario_id_1.ScenarioId.DRGA_EVIL_1_1,
    scenario_id_1.ScenarioId.DRGA_EVIL_1_2,
    scenario_id_1.ScenarioId.DRGA_EVIL_1_3,
    scenario_id_1.ScenarioId.DRGA_EVIL_2_1,
    scenario_id_1.ScenarioId.DRGA_EVIL_2_2,
    scenario_id_1.ScenarioId.DRGA_EVIL_2_3,
    scenario_id_1.ScenarioId.DRGA_EVIL_3_1,
    scenario_id_1.ScenarioId.DRGA_EVIL_3_2,
    scenario_id_1.ScenarioId.DRGA_EVIL_3_3,
    scenario_id_1.ScenarioId.DRGA_EVIL_4_1,
    scenario_id_1.ScenarioId.DRGA_EVIL_4_2,
    scenario_id_1.ScenarioId.DRGA_EVIL_4_3,
    scenario_id_1.ScenarioId.DRGA_EVIL_1_1_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_1_2_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_1_3_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_2_1_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_2_2_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_2_3_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_3_1_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_3_2_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_3_3_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_4_1_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_4_2_H,
    scenario_id_1.ScenarioId.DRGA_EVIL_4_3_H,
];
exports.GALAKROND_ALL = [...exports.GALAKROND_EXPLORER, ...exports.GALAKROND_EVIL];
exports.PRACTICE_ALL = [
    scenario_id_1.ScenarioId.AIDECK___NORMAL_MAGE,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_WARLOCK,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_HUNTER,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_ROGUE,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_PRIEST,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_WARRIOR,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_MAGE,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_DRUID,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_PALADIN,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_SHAMAN,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_WARRIOR,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_PRIEST,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_WARLOCK,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_DRUID,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_ROGUE,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_HUNTER,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_PALADIN,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_SHAMAN,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_DEMON_HUNTER,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_DEMON_HUNTER,
    scenario_id_1.ScenarioId.AIDECK___NORMAL_DEATH_KNIGHT,
    scenario_id_1.ScenarioId.AIDECK___EXPERT_DEATH_KNIGHT,
];
exports.SCENARIO_WITHOUT_RESTART = [
    scenario_id_1.ScenarioId.LOOTA_DUNGEON,
    scenario_id_1.ScenarioId.GILA_DUNGEON,
    scenario_id_1.ScenarioId.GILA_61_HAGATHA,
    scenario_id_1.ScenarioId.TRLA_DUNGEON,
    ...exports.DALARAN_HEIST_ALL,
    ...exports.TOMBS_OF_TERROR_ALL,
];
exports.SOLO_SCENARIO_WITH_LOGGED_DECKLIST = [
    scenario_id_1.ScenarioId.ICCA_01_LICHKINGPROLOGUE,
    scenario_id_1.ScenarioId._ICCA_06_MARROWGAR,
    scenario_id_1.ScenarioId.ICCA_09_DEATHBRINGER_SAURFANG,
    scenario_id_1.ScenarioId.ICCA_10_DEATHWHISPER_VALITHRIA,
    scenario_id_1.ScenarioId.ICCA_05_LANATHEL,
    scenario_id_1.ScenarioId.ICCA_07_PUTRICIDE,
    scenario_id_1.ScenarioId.ICCA_04_SINDRAGOSA,
    scenario_id_1.ScenarioId.ICCA_08_LICHKING,
    scenario_id_1.ScenarioId.KARA_02_PANTRY,
    scenario_id_1.ScenarioId.KARA_02_PANTRY_HEROIC,
    scenario_id_1.ScenarioId.KARA_01_MIRROR,
    scenario_id_1.ScenarioId.KARA_01_MIRROR_HEROIC,
    scenario_id_1.ScenarioId.KARA_04_JULIANNE,
    scenario_id_1.ScenarioId.KARA_04_HEROIC,
    scenario_id_1.ScenarioId.KARA_05_WOLF,
    scenario_id_1.ScenarioId.KARA_05_HEROIC,
    scenario_id_1.ScenarioId.KARA_06_CRONE,
    scenario_id_1.ScenarioId.KARA_06_HEROIC,
    scenario_id_1.ScenarioId.KARA_07_CURATOR,
    scenario_id_1.ScenarioId.KARA_07_HEROIC,
    scenario_id_1.ScenarioId.KARA_08_NIGHTBANE,
    scenario_id_1.ScenarioId.KARA_08_HEROIC,
    scenario_id_1.ScenarioId.KARA_09_ILLHOOF,
    scenario_id_1.ScenarioId.KARA_09_HEROIC,
    scenario_id_1.ScenarioId.KARA_12_SHADE,
    scenario_id_1.ScenarioId.KARA_11_ARAN_HEROIC,
    scenario_id_1.ScenarioId.KARA_10_NETHERSPITE,
    scenario_id_1.ScenarioId.KARA_12_NETHERSPITE_HEROIC,
    scenario_id_1.ScenarioId.KARA_13_MEDIVH,
    scenario_id_1.ScenarioId.KARA_13_MEDIVH_2,
    scenario_id_1.ScenarioId.KARA_13_MEDIVH_HEROIC,
    scenario_id_1.ScenarioId.KARA_13_FINAL_HEROIC,
    ...exports.GALAKROND_ALL,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_01,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_02,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_03,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_04,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_05,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_06,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_07,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_08,
    scenario_id_1.ScenarioId.BTA_HEROIC_FIGHT_09,
    scenario_id_1.ScenarioId.LOE_13_DEVILSAUR,
    scenario_id_1.ScenarioId.LOEA13_SKELESAURUS_HEROIC,
    scenario_id_1.ScenarioId.LOE_14_STEEL_SENTINEL,
    scenario_id_1.ScenarioId.LOEA14_STEEL_SENTINEL_HEROIC,
    scenario_id_1.ScenarioId.LOE_15_RAFAAM,
    scenario_id_1.ScenarioId.LOEA15_RAFAAM_HEROIC,
    scenario_id_1.ScenarioId.LOE_16_RAFAAM2,
    scenario_id_1.ScenarioId.LOEA16_RAFAAM_HEROIC,
    scenario_id_1.ScenarioId.LOE_09_LORDSLITHERSPEAR,
    scenario_id_1.ScenarioId.LOE_09_LORDSLITHERSPEAR_HEROIC,
    scenario_id_1.ScenarioId.LOE_10_GIANTFIN,
    scenario_id_1.ScenarioId.LOE_10_GIANTFIN_HEROIC,
    scenario_id_1.ScenarioId.LOE_12_NAZJAR,
    scenario_id_1.ScenarioId.LOEA12_NAZJAR_HEROIC,
    scenario_id_1.ScenarioId.LOE_05_SCARVASH,
    scenario_id_1.ScenarioId.LOE_05_SCARVASH_HEROIC,
    scenario_id_1.ScenarioId.LOE_08_ARCHAEDAS,
    scenario_id_1.ScenarioId.LOE_08_ARCHAEDAS_HEROIC,
    scenario_id_1.ScenarioId.LOE_02_ZINAAR,
    scenario_id_1.ScenarioId.LOE_02_ZINAAR_HEROIC,
    scenario_id_1.ScenarioId.LOE_01_SUN_RAIDER_PHAERIX,
    scenario_id_1.ScenarioId.LOE_01_SUN_RAIDER_PHAERIX_HEROIC,
    scenario_id_1.ScenarioId.BRM_14_OMNOTRON,
    scenario_id_1.ScenarioId.BRM_14_OMNOTRON_HEROIC,
    scenario_id_1.ScenarioId.BRM_15_MALORIAK,
    scenario_id_1.ScenarioId.BRM_15_MALORIAK_HEROIC,
    scenario_id_1.ScenarioId.BRM_16_ATRAMEDES,
    scenario_id_1.ScenarioId.BRM_16_ATRAMEDES_HEROIC,
    scenario_id_1.ScenarioId.BRM_17_SUPERNEFARIAN,
    scenario_id_1.ScenarioId.BRM_17_SUPERNEFARIAN_HEROIC,
    scenario_id_1.ScenarioId.BRM_10_RAZORGORE,
    scenario_id_1.ScenarioId.BRM_10_RAZORGORE_HEROIC,
    scenario_id_1.ScenarioId.BRM_11_VAELASTRASZ,
    scenario_id_1.ScenarioId.BRM_11_VAELASTRASZ_HEROIC,
    scenario_id_1.ScenarioId.BRM_12_CHROMAGGUS,
    scenario_id_1.ScenarioId.BRM_12_CHROMAGGUS_HEROIC,
    scenario_id_1.ScenarioId.BRM_13_NEFARIAN,
    scenario_id_1.ScenarioId.BRM_13_NEFARIAN_HEROIC,
    scenario_id_1.ScenarioId.BRM_07_OMOKK,
    scenario_id_1.ScenarioId.BRM_07_OMOKK_HEROIC,
    scenario_id_1.ScenarioId.BRM_08_DRAKKISATH,
    scenario_id_1.ScenarioId.BRM_08_DRAKKISATH_HEROIC,
    scenario_id_1.ScenarioId.BRM_09_REND,
    scenario_id_1.ScenarioId.BRM_09_REND_HEROIC,
    scenario_id_1.ScenarioId.BRM_04_GARR,
    scenario_id_1.ScenarioId.BRM_04_GARR_HEROIC,
    scenario_id_1.ScenarioId.BRM_05_GEDDON,
    scenario_id_1.ScenarioId.BRM_05_GEDDON_HEROIC,
    scenario_id_1.ScenarioId.BRM_06_MAJORDOMO,
    scenario_id_1.ScenarioId.BRM_06_MAJORDOMO_HEROIC,
    scenario_id_1.ScenarioId.BRM_01_DIREBREW,
    scenario_id_1.ScenarioId.BRM_01_DIREBREW_HEROIC,
    scenario_id_1.ScenarioId.BRM_02_GRIMSTONE,
    scenario_id_1.ScenarioId.BRM_02_GRIMSTONE_HEROIC,
    scenario_id_1.ScenarioId.BRM_03_THAURISSAN,
    scenario_id_1.ScenarioId.BRM_03_THAURISSAN_HEROIC,
    scenario_id_1.ScenarioId.NAX_01_ANUB,
    scenario_id_1.ScenarioId.NAX_01_ANUB_HEROIC,
    scenario_id_1.ScenarioId.NAX_02_FAERLINA,
    scenario_id_1.ScenarioId.NAX_02_FAERLINA_HEROIC,
    scenario_id_1.ScenarioId.NAX_03_MAEXXNA,
    scenario_id_1.ScenarioId.NAX_03_MAEXXNA_HEROIC,
    scenario_id_1.ScenarioId.NAX_04_NOTH,
    scenario_id_1.ScenarioId.NAX_04_NOTH_HEROIC,
    scenario_id_1.ScenarioId.NAX_05_HEIGAN,
    scenario_id_1.ScenarioId.NAX_05_HEIGAN_HEROIC,
    scenario_id_1.ScenarioId.NAX_06_LOATHEB,
    scenario_id_1.ScenarioId.NAX_06_LOATHEB_HEROIC,
    scenario_id_1.ScenarioId.NAX_07_RAZUVIOUS,
    scenario_id_1.ScenarioId.NAX_07_RAZUVIOUS_HEROIC,
    scenario_id_1.ScenarioId.NAX_08_GOTHIK,
    scenario_id_1.ScenarioId.NAX_08_GOTHIK_HEROIC,
    scenario_id_1.ScenarioId.NAX_09_HORSEMEN,
    scenario_id_1.ScenarioId.NAX_09_HORSEMEN_HEROIC,
    scenario_id_1.ScenarioId.NAX_10_PATCHWERK,
    scenario_id_1.ScenarioId.NAX_10_PATCHWERK_HEROIC,
    scenario_id_1.ScenarioId.NAX_11_GROBBULUS,
    scenario_id_1.ScenarioId.NAX_11_GROBBULUS_HEROIC,
    scenario_id_1.ScenarioId.NAX_12_GLUTH,
    scenario_id_1.ScenarioId.NAX_12_GLUTH_HEROIC,
    scenario_id_1.ScenarioId.NAX_13_THADDIUS,
    scenario_id_1.ScenarioId.NAX_13_THADDIUS_HEROIC,
    scenario_id_1.ScenarioId.NAX_14_SAPPHIRON,
    scenario_id_1.ScenarioId.NAX_14_SAPPHIRON_HEROIC,
    scenario_id_1.ScenarioId.NAX_15_KELTHUZAD,
    scenario_id_1.ScenarioId.NAX_15_KELTHUZAD_HEROIC,
];
