// @firestone-hs/reference-data/dist/enums/spell-school.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpellSchool = void 0;
var SpellSchool;
(function (SpellSchool) {
    SpellSchool[SpellSchool["NONE"] = 0] = "NONE";
    SpellSchool[SpellSchool["ARCANE"] = 1] = "ARCANE";
    SpellSchool[SpellSchool["FIRE"] = 2] = "FIRE";
    SpellSchool[SpellSchool["FROST"] = 3] = "FROST";
    SpellSchool[SpellSchool["NATURE"] = 4] = "NATURE";
    SpellSchool[SpellSchool["HOLY"] = 5] = "HOLY";
    SpellSchool[SpellSchool["SHADOW"] = 6] = "SHADOW";
    SpellSchool[SpellSchool["FEL"] = 7] = "FEL";
    SpellSchool[SpellSchool["PHYSICAL_COMBAT"] = 8] = "PHYSICAL_COMBAT";
    SpellSchool[SpellSchool["TAVERN"] = 9] = "TAVERN";
    SpellSchool[SpellSchool["SPELLCRAFT"] = 10] = "SPELLCRAFT";
    SpellSchool[SpellSchool["LESSER_TRINKET"] = 11] = "LESSER_TRINKET";
    SpellSchool[SpellSchool["GREATER_TRINKET"] = 12] = "GREATER_TRINKET";
    SpellSchool[SpellSchool["UPGRADE"] = 13] = "UPGRADE";
})(SpellSchool = exports.SpellSchool || (exports.SpellSchool = {}));
