// @firestone-hs/reference-data/dist/enums/step.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Step = void 0;
var Step;
(function (Step) {
    Step[Step["INVALID"] = 0] = "INVALID";
    Step[Step["BEGIN_FIRST"] = 1] = "BEGIN_FIRST";
    Step[Step["BEGIN_SHUFFLE"] = 2] = "BEGIN_SHUFFLE";
    Step[Step["BEGIN_DRAW"] = 3] = "BEGIN_DRAW";
    Step[Step["BEGIN_MULLIGAN"] = 4] = "BEGIN_MULLIGAN";
    Step[Step["MAIN_BEGIN"] = 5] = "MAIN_BEGIN";
    Step[Step["MAIN_READY"] = 6] = "MAIN_READY";
    Step[Step["MAIN_RESOURCE"] = 7] = "MAIN_RESOURCE";
    Step[Step["MAIN_DRAW"] = 8] = "MAIN_DRAW";
    Step[Step["MAIN_START"] = 9] = "MAIN_START";
    Step[Step["MAIN_ACTION"] = 10] = "MAIN_ACTION";
    Step[Step["MAIN_COMBAT"] = 11] = "MAIN_COMBAT";
    Step[Step["MAIN_END"] = 12] = "MAIN_END";
    Step[Step["MAIN_NEXT"] = 13] = "MAIN_NEXT";
    Step[Step["FINAL_WRAPUP"] = 14] = "FINAL_WRAPUP";
    Step[Step["FINAL_GAMEOVER"] = 15] = "FINAL_GAMEOVER";
    Step[Step["MAIN_CLEANUP"] = 16] = "MAIN_CLEANUP";
    Step[Step["MAIN_START_TRIGGERS"] = 17] = "MAIN_START_TRIGGERS";
    Step[Step["MAIN_SET_ACTION_STEP_TYPE"] = 18] = "MAIN_SET_ACTION_STEP_TYPE";
    Step[Step["MAIN_PRE_ACTION"] = 19] = "MAIN_PRE_ACTION";
    Step[Step["MAIN_POST_ACTION"] = 20] = "MAIN_POST_ACTION";
})(Step = exports.Step || (exports.Step = {}));
