// @firestone-hs/simulate-bgs-battle/dist/cards/impl/hero-power/glorious-gloop.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GloriousGloop = void 0;
const utils_1 = require("../../../services/utils");
const utils_2 = require("../../../utils");
exports.GloriousGloop = {
    cardIds: ["BGDUO_HERO_101p"],
    startOfCombatTiming: 'pre-combat',
    startOfCombat: (trinket, input) => {
        var _a, _b;
        for (const heroPower of input.playerEntity.heroPowers) {
            if (exports.GloriousGloop.cardIds.includes(heroPower.cardId) && heroPower.used) {
                if (!((_a = input.playerBoard) === null || _a === void 0 ? void 0 : _a.length)) {
                    return false;
                }
                const target = input.playerBoard.find((m) => m.enchantments.some((e) => e.cardId === "BGDUO_HERO_101pe2"));
                if (!target) {
                    return false;
                }
                const teammateState = (0, utils_2.getTeammateInitialState)(input.gameState.gameState, input.playerEntity);
                if (!((_b = teammateState === null || teammateState === void 0 ? void 0 : teammateState.board) === null || _b === void 0 ? void 0 : _b.length)) {
                    return false;
                }
                const highestTier = Math.max(...teammateState.board.map((entity) => input.gameState.allCards.getCard(entity.cardId).techLevel));
                const candidates = teammateState.board.filter((entity) => input.gameState.allCards.getCard(entity.cardId).techLevel === highestTier);
                if (!candidates.length) {
                    return false;
                }
                const highestTierMinion = (0, utils_1.pickRandom)(candidates);
                const clone = (0, utils_2.addImpliedMechanics)({
                    ...highestTierMinion,
                    entityId: input.gameState.sharedState.currentEntityId++,
                    lastAffectedByEntity: null,
                    definitelyDead: false,
                    attackImmediately: false,
                }, input.gameState.cardsData);
                input.gameState.spectator.registerPowerTarget(input.playerEntity, target, input.playerBoard, input.playerEntity, input.opponentEntity);
                const index = input.playerBoard.indexOf(target);
                input.playerBoard.splice(index, 1, clone);
                return true;
            }
        }
    },
};
