// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/apexis-guardian.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApexisGuardian = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const utils_1 = require("../../../services/utils");
const magnetize_1 = require("../../../simulation/magnetize");
const utils_2 = require("../../../utils");
exports.ApexisGuardian = {
    cardIds: ["BG34_173", "BG34_173_G"],
    deathrattleSpawn: (minion, input) => {
        handleEffect(minion, {
            board: input.boardWithDeadEntity,
            hero: input.boardWithDeadEntityHero,
            gameState: input.gameState,
            otherBoard: input.otherBoard,
            otherHero: input.otherBoardHero,
        });
        return [];
    },
    rally: (minion, input) => {
        handleEffect(minion, {
            board: input.attackingBoard,
            hero: input.attackingHero,
            gameState: input.gameState,
            otherBoard: input.defendingBoard,
            otherHero: input.defendingHero,
        });
        return { dmgDoneByAttacker: 0, dmgDoneByDefender: 0 };
    },
};
const handleEffect = (minion, input) => {
    const mult = minion.cardId === "BG34_173_G" ? 2 : 1;
    const cardsToMagnetize = [];
    const possibleTargets = input.board.filter((e) => e.health > 0 &&
        !e.definitelyDead &&
        e !== minion &&
        (0, utils_2.hasCorrectTribe)(e, input.otherHero, reference_data_1.Race.MECH, input.gameState.anomalies, input.gameState.allCards));
    const targets = [];
    for (let i = 0; i < mult; i++) {
        const remaining = possibleTargets.filter((t) => !targets.includes(t));
        const picked = (0, utils_1.pickRandom)(remaining, minion);
        if (picked)
            targets.push(picked);
    }
    for (const target of targets) {
        const cardIdToMagnetize = input.gameState.cardsData.getRandomMagneticVolumizer(input.otherHero, input.gameState.anomalies, 6, minion, target);
        const cardToMagnetize = {
            ...input.gameState.allCards.getCard(cardIdToMagnetize),
        };
        cardsToMagnetize.push(cardToMagnetize);
        (0, magnetize_1.magnetizeToTarget)([target], minion, cardsToMagnetize, input.board, input.hero, input.otherBoard, input.otherHero, input.gameState);
    }
    return [];
};
