// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/carrier.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Carrier = void 0;
const deathrattle_spawns_1 = require("../../../simulation/deathrattle-spawns");
const stats_1 = require("../../../simulation/stats");
exports.Carrier = {
    cardIds: ["BG31_HERO_802pt1", "BG31_HERO_802pt1_G"],
    baseAvengeValue: (cardId) => 4,
    avenge: (minion, input) => {
        var _a;
        const spawnInput = {
            boardWithDeadEntity: input.board,
            boardWithDeadEntityHero: input.hero,
            gameState: input.gameState,
            deadEntity: minion,
            otherBoard: input.otherBoard,
            otherBoardHero: input.otherHero,
        };
        const singleStatBuff = minion.cardId === "BG31_HERO_802pt1_G" ? 14 : 7;
        const statBuff = (_a = minion.scriptDataNum1) !== null && _a !== void 0 ? _a : 0;
        const cardId = minion.cardId === "BG31_HERO_802pt1_G"
            ? "BG31_HERO_802pt1t_G"
            : "BG31_HERO_802pt1t";
        const spawned = (0, deathrattle_spawns_1.simplifiedSpawnEntities)(cardId, 1, spawnInput);
        spawned.forEach((e) => {
            (0, stats_1.setEntityStats)(e, singleStatBuff, singleStatBuff, input.board, input.hero, input.gameState);
            (0, stats_1.modifyStats)(e, minion, statBuff, statBuff, input.board, input.hero, input.gameState, false);
        });
        minion.scriptDataNum1 = statBuff + singleStatBuff;
        return spawned;
    },
};
