// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/expert-aviator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpertAviator = void 0;
const deathrattle_spawns_1 = require("../../../simulation/deathrattle-spawns");
const spawns_1 = require("../../../simulation/spawns");
const stats_1 = require("../../../simulation/stats");
const utils_1 = require("../../../utils");
exports.ExpertAviator = {
    cardIds: ["BG34_140", "BG34_140_G"],
    rally: (minion, input) => {
        const mult = minion.cardId === "BG34_140_G" ? 2 : 1;
        const target = input.attackingHero.hand.filter((e) => !!e.maxHealth && e.cardId)[0];
        if (!target) {
            return { dmgDoneByAttacker: 0, dmgDoneByDefender: 0 };
        }
        (0, stats_1.modifyStats)(target, minion, 1 * mult, 1 * mult, input.attackingBoard, input.attackingHero, input.gameState);
        if (target.locked) {
            return { dmgDoneByAttacker: 0, dmgDoneByDefender: 0 };
        }
        target.locked = true;
        const clone = (0, utils_1.copyEntity)(target);
        const spawnInput = {
            boardWithDeadEntity: input.attackingBoard,
            boardWithDeadEntityHero: input.attackingHero,
            gameState: input.gameState,
            deadEntity: minion,
            otherBoard: input.defendingBoard,
            otherBoardHero: input.defendingHero,
        };
        const spawns = (0, deathrattle_spawns_1.simplifiedSpawnEntities)(clone.cardId, 1, spawnInput, clone);
        for (const s of spawns) {
            s.onCanceledSummon = () => (target.locked = false);
        }
        (0, spawns_1.performEntitySpawns)(spawns, spawnInput.boardWithDeadEntity, spawnInput.boardWithDeadEntityHero, minion, input.attackingBoard.length - input.attackingBoard.indexOf(minion) - 1, spawnInput.otherBoard, spawnInput.otherBoardHero, input.gameState);
        return {
            dmgDoneByAttacker: 0,
            dmgDoneByDefender: 0,
        };
    },
};
