// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/fauna-whisperer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FaunaWhisperer = void 0;
const cast_tavern_spell_1 = require("../../../mechanics/cast-tavern-spell");
const attack_1 = require("../../../simulation/attack");
exports.FaunaWhisperer = {
    cardIds: ["BG32_837", "BG32_837_G"],
    endOfTurn: (minion, input) => {
        const targets = (0, attack_1.getNeighbours)(input.board, minion);
        const mult = minion.cardId === "BG32_837_G" ? 2 : 1;
        for (let i = 0; i < mult; i++) {
            for (const target of targets) {
                (0, cast_tavern_spell_1.castTavernSpell)("BG28_845", {
                    spellCardId: "BG28_845",
                    source: minion,
                    target: target,
                    board: input.board,
                    hero: input.hero,
                    otherBoard: input.otherBoard,
                    otherHero: input.otherHero,
                    gameState: input.gameState,
                });
            }
        }
    },
};
