// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/friendly-bouncer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FriendlyBouncer = void 0;
const cards_in_hand_1 = require("../../../simulation/cards-in-hand");
const deathrattle_spawns_1 = require("../../../simulation/deathrattle-spawns");
exports.FriendlyBouncer = {
    cardIds: ["BG33_700", "BG33_700_G"],
    rally: (minion, input) => {
        const loops = minion.cardId === "BG33_700_G" ? 2 : 1;
        for (let i = 0; i < loops; i++) {
            const spawnId = input.gameState.cardsData.getRandomRally(input.attackingHero.tavernTier);
            if (!!spawnId) {
                (0, deathrattle_spawns_1.simplifiedSpawnEntitiesWithAddToBoard)(spawnId, 1, {
                    boardWithDeadEntity: input.attackingBoard,
                    boardWithDeadEntityHero: input.attackingHero,
                    gameState: input.gameState,
                    deadEntity: input.attacker,
                    otherBoard: input.defendingBoard,
                    otherBoardHero: input.defendingHero,
                }, input.attacker, input.attackingBoard.length - input.attackingBoard.indexOf(input.attacker) - 1);
                (0, cards_in_hand_1.addCardsInHand)(input.attackingHero, input.attackingBoard, [spawnId], input.gameState);
            }
        }
        return { dmgDoneByAttacker: 0, dmgDoneByDefender: 0 };
    },
};
