// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/implant-subject.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImplantSubject = void 0;
const stats_1 = require("../../../simulation/stats");
exports.ImplantSubject = {
    cardIds: ["BG31_147", "BG31_147_G"],
    onTauntUpdated: (entity, impactedEntity, previousValue, input) => {
        if (previousValue && !impactedEntity.taunt) {
            updateEntity(entity, input.board, input.hero, input.gameState);
        }
    },
    onDivineShieldUpdated: (entity, input) => {
        if (input.previousValue && !input.target.divineShield) {
            updateEntity(entity, input.board, input.hero, input.gameState);
        }
    },
    onRebornUpdated: (entity, impactedEntity, previousValue, input) => {
        if (previousValue && !impactedEntity.reborn) {
            updateEntity(entity, input.board, input.hero, input.gameState);
        }
    },
    rebornEffect: (entity, input) => {
        updateEntity(entity, input.boardWithKilledMinion, input.boardWithKilledMinionHero, input.gameState);
    },
    onStealthUpdated: (entity, impactedEntity, previousValue, input) => {
        if (previousValue && !impactedEntity.stealth) {
            updateEntity(entity, input.board, input.hero, input.gameState);
        }
    },
    onVenomousUpdated: (entity, impactedEntity, previousValue, input) => {
        if (previousValue && !impactedEntity.venomous) {
            updateEntity(entity, input.board, input.hero, input.gameState);
        }
    },
    onWindfuryUpdated: (entity, impactedEntity, previousValue, input) => {
        if (previousValue && !impactedEntity.windfury) {
            updateEntity(entity, input.board, input.hero, input.gameState);
        }
    },
};
const updateEntity = (entity, board, hero, gameState) => {
    const mult = entity.cardId === "BG31_147_G" ? 2 : 1;
    (0, stats_1.modifyStats)(entity, entity, 1 * mult, 1 * mult, board, hero, gameState);
};
