// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/profound-thinker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfoundThinker = void 0;
const cast_tavern_spell_1 = require("../../../mechanics/cast-tavern-spell");
const cards_in_hand_1 = require("../../../simulation/cards-in-hand");
exports.ProfoundThinker = {
    cardIds: ["BG34_929", "BG34_929_G"],
    rally: (minion, input) => {
        var _a;
        const mult = minion.cardId === "BG34_929_G" ? 2 : 1;
        for (let i = 0; i < mult; i++) {
            const spell = input.gameState.cardsData.getRandomSpellcraft({
                maxTavernTier: (_a = input.attackingHero.tavernTier) !== null && _a !== void 0 ? _a : 3,
            });
            (0, cast_tavern_spell_1.castTavernSpell)(spell, {
                spellCardId: spell,
                source: minion,
                target: minion,
                board: input.attackingBoard,
                hero: input.attackingHero,
                otherBoard: input.defendingBoard,
                otherHero: input.defendingHero,
                gameState: input.gameState,
            });
            (0, cards_in_hand_1.addCardsInHand)(input.attackingHero, input.attackingBoard, [spell], input.gameState);
        }
        return { dmgDoneByAttacker: 0, dmgDoneByDefender: 0 };
    },
};
