// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/stoneshell-guardian.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoneshellGuardian = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const utils_1 = require("../../../utils");
exports.StoneshellGuardian = {
    cardIds: [
        "BG33_HERO_000_Buddy",
        "BG33_HERO_000_Buddy_G",
        "BG34_Giant_601",
        "BG34_Giant_601_G",
    ],
    startOfCombat: (minion, input) => {
        const numberOfCopies = minion.cardId === "BG33_HERO_000_Buddy_G" ||
            minion.cardId === "BG34_Giant_601_G"
            ? 2
            : 1;
        for (let i = 0; i < numberOfCopies; i++) {
            const rallyMinions = input.playerBoard.filter((e) => !exports.StoneshellGuardian.cardIds.includes(e.cardId) &&
                (0, utils_1.hasEntityMechanic)(e, reference_data_1.GameTag.BACON_RALLY, input.gameState.allCards));
            for (const target of rallyMinions) {
                const rallyEnchantments = target.enchantments.filter((e) => (0, utils_1.hasEntityMechanic)(e, reference_data_1.GameTag.BACON_RALLY, input.gameState.allCards));
                minion.enchantments.push({
                    cardId: target.cardId,
                    originEntityId: target.entityId,
                    timing: input.gameState.sharedState.currentEntityId++,
                });
                for (const rallyEnchantment of rallyEnchantments) {
                    minion.enchantments.push({
                        cardId: rallyEnchantment.cardId,
                        originEntityId: rallyEnchantment.originEntityId,
                        timing: input.gameState.sharedState.currentEntityId++,
                    });
                }
                input.gameState.spectator.registerPowerTarget(minion, target, input.playerBoard, input.playerEntity, input.opponentEntity);
            }
        }
        return true;
    },
};
