// @firestone-hs/simulate-bgs-battle/dist/cards/impl/minion/timewarped-alleycat.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimewarpedAlleycat = void 0;
const deathrattle_spawns_1 = require("../../../simulation/deathrattle-spawns");
const spawns_1 = require("../../../simulation/spawns");
exports.TimewarpedAlleycat = {
    cardIds: ["BG34_Giant_009", "BG34_Giant_009_G"],
    endOfTurn: (minion, input) => {
        const mult = minion.cardId === "BG34_Giant_009_G" ? 2 : 1;
        const spawnInput = {
            boardWithDeadEntity: input.board,
            boardWithDeadEntityHero: input.hero,
            gameState: input.gameState,
            deadEntity: minion,
            otherBoard: input.otherBoard,
            otherBoardHero: input.otherHero,
        };
        const spawns = (0, deathrattle_spawns_1.simplifiedSpawnEntities)("BG_CFM_315t", mult, spawnInput);
        spawns.forEach((e) => {
            e.attack = minion.attack;
            e.health = minion.health;
            e.maxHealth = minion.maxHealth;
            e.maxAttack = minion.maxAttack;
        });
        (0, spawns_1.performEntitySpawns)(spawns, input.board, input.hero, minion, input.board.indexOf(minion), input.otherBoard, input.otherHero, input.gameState);
        return true;
    },
};
