// @firestone-hs/simulate-bgs-battle/dist/cards/impl/trinket/jarred-frostling.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.JarredFrostling = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const utils_1 = require("../../../services/utils");
const utils_2 = require("../../../utils");
exports.JarredFrostling = {
    startOfCombat: (trinket, input) => {
        const elementals = (0, utils_1.shuffleArray)(input.playerBoard.filter((e) => (0, utils_2.hasCorrectTribe)(e, input.playerEntity, reference_data_1.Race.ELEMENTAL, input.gameState.anomalies, input.gameState.allCards)));
        if (!!(elementals === null || elementals === void 0 ? void 0 : elementals.length)) {
            const targets = elementals.slice(0, 2);
            targets.forEach((e) => {
                e.enchantments.push({
                    cardId: "BG30_MagicItem_952e",
                    originEntityId: trinket.entityId,
                    repeats: 1,
                    timing: input.gameState.sharedState.currentEntityId++,
                });
                input.gameState.spectator.registerPowerTarget(input.playerEntity, e, input.playerBoard, null, null);
            });
            return true;
        }
    },
};
