// @firestone-hs/simulate-bgs-battle/dist/cards/impl/trinket/sthara-sticker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StharaSticker = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const deathrattle_spawns_1 = require("../../../simulation/deathrattle-spawns");
const spawns_1 = require("../../../simulation/spawns");
const utils_1 = require("../../../utils");
exports.StharaSticker = {
    cardIds: ["BG32_MagicItem_907"],
    defaultScriptDataNum: (cardId) => 1,
    onAfterDeath: (trinket, input) => {
        if (input.board.length > 0 || trinket.scriptDataNum1 <= 0) {
            return;
        }
        const candidate = input.gameState.sharedState.deaths
            .filter((entity) => entity.friendly === input.hero.friendly)
            .filter((entity) => (0, utils_1.hasCorrectTribe)(entity, input.hero, reference_data_1.Race.DEMON, input.gameState.anomalies, input.gameState.allCards))[0];
        if (!candidate) {
            return;
        }
        const spawns = (0, deathrattle_spawns_1.spawnEntities)(candidate.cardId, 1, input.board, input.hero, input.otherBoard, input.otherHero, input.gameState, input.hero.friendly, false).map((e) => ({
            ...e,
            attack: candidate.maxAttack,
            health: candidate.maxHealth,
            maxHealth: candidate.maxHealth,
            maxAttack: candidate.maxAttack,
        }));
        (0, spawns_1.performEntitySpawns)(spawns, input.board, input.hero, input.hero, 0, input.otherBoard, input.otherHero, input.gameState);
        trinket.scriptDataNum1--;
    },
};
