// @firestone-hs/simulate-bgs-battle/dist/cards/impl/trinket/summoning-sphere.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummoningSphere = void 0;
const utils_1 = require("../../../services/utils");
const deathrattle_spawns_1 = require("../../../simulation/deathrattle-spawns");
const spawns_1 = require("../../../simulation/spawns");
const utils_2 = require("../../../utils");
exports.SummoningSphere = {
    startOfCombat: (trinket, input) => {
        var _a;
        const teammateState = (0, utils_2.getTeammateInitialState)(input.gameState.gameState, input.playerEntity);
        if (!((_a = teammateState === null || teammateState === void 0 ? void 0 : teammateState.board) === null || _a === void 0 ? void 0 : _a.length)) {
            return;
        }
        const highestHealthMinion = (0, utils_1.shuffleArray)([...teammateState.board]).sort((a, b) => b.health - a.health)[0];
        const clone = (0, utils_2.addImpliedMechanics)({
            ...highestHealthMinion,
            entityId: input.gameState.sharedState.currentEntityId++,
            lastAffectedByEntity: null,
            definitelyDead: false,
            attackImmediately: false,
        }, input.gameState.cardsData);
        const newMinions = (0, deathrattle_spawns_1.spawnEntities)(clone.cardId, 1, input.playerBoard, input.playerEntity, input.opponentBoard, input.opponentEntity, input.gameState, highestHealthMinion.friendly, true, false, false, clone);
        const indexFromRight = 0;
        const spawned = (0, spawns_1.performEntitySpawns)(newMinions, input.playerBoard, input.playerEntity, highestHealthMinion, indexFromRight, input.opponentBoard, input.opponentEntity, input.gameState);
        if (spawned.length > 0) {
            input.gameState.spectator.registerPowerTarget(input.playerEntity, clone, input.playerBoard, input.playerEntity, input.opponentEntity);
            return { hasTriggered: true, shouldRecomputeCurrentAttacker: true };
        }
    },
};
