// @firestone-hs/simulate-bgs-battle/dist/simulation/start-of-combat/soc-anomalies.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleStartOfCombatAnomalies = void 0;
const summon_when_space_1 = require("../summon-when-space");
const soc_action_processor_1 = require("./soc-action-processor");
const handleStartOfCombatAnomalies = (playerEntity, playerBoard, opponentEntity, opponentBoard, currentAttacker, gameState) => {
    currentAttacker = handleStartOfCombatAnomaliesForPlayer({
        playerEntity: playerEntity,
        playerBoard: playerBoard,
        opponentEntity: opponentEntity,
        opponentBoard: opponentBoard,
        currentAttacker,
        gameState,
        playerIsFriendly: true,
    });
    currentAttacker = handleStartOfCombatAnomaliesForPlayer({
        playerEntity: opponentEntity,
        playerBoard: opponentBoard,
        opponentEntity: playerEntity,
        opponentBoard: playerBoard,
        currentAttacker,
        gameState,
        playerIsFriendly: false,
    });
    const shouldRecomputeCurrentAttacker = (0, summon_when_space_1.handleSummonsWhenSpace)(playerBoard, playerEntity, opponentBoard, opponentEntity, gameState);
    if (shouldRecomputeCurrentAttacker) {
        currentAttacker =
            playerBoard.length > opponentBoard.length
                ? 0
                : opponentBoard.length > playerBoard.length
                    ? 1
                    : Math.round(Math.random());
    }
    return currentAttacker;
};
exports.handleStartOfCombatAnomalies = handleStartOfCombatAnomalies;
const handleStartOfCombatAnomaliesForPlayer = (input) => {
    var _a, _b;
    const loops = ((_a = input.playerEntity.trinkets) === null || _a === void 0 ? void 0 : _a.some((t) => t.cardId === "BG32_MagicItem_365"))
        ? 2
        : 1;
    for (let i = 0; i < loops; i++) {
        if (!((_b = input.gameState.anomalies) === null || _b === void 0 ? void 0 : _b.length) || input.playerEntity.startOfCombatDone) {
            return input.currentAttacker;
        }
        for (const anomaly of input.gameState.anomalies) {
            (0, soc_action_processor_1.performStartOfCombatAction)(anomaly, input.playerEntity, input, false);
        }
    }
    return input.currentAttacker;
};
