// @firestone-hs/simulate-bgs-battle/dist/simulation/start-of-combat/soc-minion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.performStartOfCombatMinionsForPlayer = exports.handleStartOfCombatMinions = void 0;
const card_interface_1 = require("../../cards/card.interface");
const _card_mappings_1 = require("../../cards/impl/_card-mappings");
const summon_when_space_1 = require("../summon-when-space");
const soc_action_processor_1 = require("./soc-action-processor");
const handleStartOfCombatMinions = (playerEntity, playerBoard, opponentEntity, opponentBoard, currentAttacker, playerBoardBefore, opponentBoardBefore, gameState) => {
    let attackerForStart = Math.random() < 0.5 ? 0 : 1;
    const playerAttackers = [
        ...playerEntity.hand.filter((e) => (0, card_interface_1.hasStartOfCombatFromHand)(_card_mappings_1.cardMappings[e.cardId])),
        ...playerBoard,
    ];
    const opponentAttackers = [
        ...opponentEntity.hand.filter((e) => (0, card_interface_1.hasStartOfCombatFromHand)(_card_mappings_1.cardMappings[e.cardId])),
        ...opponentBoard,
    ];
    while (playerAttackers.length > 0 || opponentAttackers.length > 0) {
        let shouldUpdateNextPlayer = false;
        if (attackerForStart === 0 && playerAttackers.length > 0) {
            const attacker = playerAttackers.splice(0, 1)[0];
            if (attacker.health <= 0 || attacker.definitelyDead) {
                continue;
            }
            const input = {
                playerBoard: playerBoard,
                playerEntity: playerEntity,
                opponentBoard: opponentBoard,
                opponentEntity: opponentEntity,
                playerBoardBefore: playerBoardBefore,
                opponentBoardBefore: opponentBoardBefore,
                playerIsFriendly: true,
                currentAttacker: currentAttacker,
                gameState,
            };
            const newShouldUpdateNextPlayer = (0, exports.performStartOfCombatMinionsForPlayer)(attacker, input);
            currentAttacker = input.currentAttacker;
        }
        else if (attackerForStart === 0 && playerAttackers.length === 0) {
            shouldUpdateNextPlayer = true;
        }
        else if (attackerForStart === 1 && opponentAttackers.length > 0) {
            const attacker = opponentAttackers.splice(0, 1)[0];
            if (attacker.health <= 0 || attacker.definitelyDead) {
                continue;
            }
            const input = {
                playerBoard: opponentBoard,
                playerEntity: opponentEntity,
                opponentBoard: playerBoard,
                opponentEntity: playerEntity,
                playerBoardBefore: opponentBoardBefore,
                opponentBoardBefore: playerBoardBefore,
                playerIsFriendly: false,
                currentAttacker: currentAttacker,
                gameState,
            };
            const newShouldUpdateNextPlayer = (0, exports.performStartOfCombatMinionsForPlayer)(attacker, input);
            currentAttacker = input.currentAttacker;
        }
        else if (attackerForStart === 1 && opponentAttackers.length === 0) {
            shouldUpdateNextPlayer = true;
        }
        if (shouldUpdateNextPlayer) {
            attackerForStart = (attackerForStart + 1) % 2;
        }
    }
    const shouldRecomputeCurrentAttacker = (0, summon_when_space_1.handleSummonsWhenSpace)(playerBoard, playerEntity, opponentBoard, opponentEntity, gameState);
    if (shouldRecomputeCurrentAttacker) {
        currentAttacker =
            playerBoard.length > opponentBoard.length
                ? 0
                : opponentBoard.length > playerBoard.length
                    ? 1
                    : Math.round(Math.random());
    }
    return currentAttacker;
};
exports.handleStartOfCombatMinions = handleStartOfCombatMinions;
const performStartOfCombatMinionsForPlayer = (minion, input) => {
    var _a;
    const loops = ((_a = input.playerEntity.trinkets) === null || _a === void 0 ? void 0 : _a.some((t) => t.cardId === "BG32_MagicItem_365"))
        ? 2
        : 1;
    let hasProcessed = false;
    for (let i = 0; i < loops; i++) {
        if (input.playerEntity.startOfCombatDone) {
            return false;
        }
        hasProcessed = (0, soc_action_processor_1.performStartOfCombatAction)(minion.cardId, minion, input, true) || hasProcessed;
    }
    return hasProcessed;
};
exports.performStartOfCombatMinionsForPlayer = performStartOfCombatMinionsForPlayer;
