// @firestone-hs/reference-data/dist/models/reference-cards/reference-data.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RELIC_IDS = exports.LIBRAM_IDS = exports.WATCH_POST_IDS = exports.EXCAVATE_TREASURE_3_IDS = exports.EXCAVATE_TREASURE_2_IDS = exports.EXCAVATE_TREASURE_1_IDS = exports.isCoin = void 0;
const card_ids_1 = require("../../card-ids");
const isCoin = (cardId, allCards) => {
    var _a;
    return (_a = allCards.getCard(cardId)) === null || _a === void 0 ? void 0 : _a.isCoin;
};
exports.isCoin = isCoin;
exports.EXCAVATE_TREASURE_1_IDS = [
    card_ids_1.CardIds.KoboldMiner_EscapingTroggToken_WW_001t4,
    card_ids_1.CardIds.KoboldMiner_FoolsAzeriteToken_WW_001t3,
    card_ids_1.CardIds.KoboldMiner_PouchOfCoinsToken_WW_001t18,
    card_ids_1.CardIds.KoboldMiner_RockToken_WW_001t,
    card_ids_1.CardIds.KoboldMiner_WaterSourceToken_WW_001t2,
    card_ids_1.CardIds.HeartblossomToken_DEEP_999t1,
];
exports.EXCAVATE_TREASURE_2_IDS = [
    card_ids_1.CardIds.KoboldMiner_AzeriteChunkToken_WW_001t9,
    card_ids_1.CardIds.KoboldMiner_CanaryToken_WW_001t7,
    card_ids_1.CardIds.KoboldMiner_FallingStalactiteToken_WW_001t5,
    card_ids_1.CardIds.KoboldMiner_GlowingGlyphToken_WW_001t8,
    card_ids_1.CardIds.KoboldMiner_LivingStoneToken_WW_001t16,
    card_ids_1.CardIds.DeepholmGeodeToken_DEEP_999t2,
];
exports.EXCAVATE_TREASURE_3_IDS = [
    card_ids_1.CardIds.KoboldMiner_AzeriteGemToken_WW_001t14,
    card_ids_1.CardIds.KoboldMiner_CollapseToken_WW_001t12,
    card_ids_1.CardIds.KoboldMiner_MotherlodeDrakeToken_WW_001t17,
    card_ids_1.CardIds.KoboldMiner_OgrefistBoulderToken_WW_001t11,
    card_ids_1.CardIds.KoboldMiner_SteelhideMoleToken_WW_001t13,
    card_ids_1.CardIds.WorldPillarFragmentToken_DEEP_999t3,
];
exports.WATCH_POST_IDS = [card_ids_1.CardIds.FarWatchPost, card_ids_1.CardIds.MorshanWatchPost, card_ids_1.CardIds.CrossroadsWatchPost];
exports.LIBRAM_IDS = [
    card_ids_1.CardIds.LibramOfJustice_BT_011,
    card_ids_1.CardIds.LibramOfJustice_Story_04_LibramofJustice,
    card_ids_1.CardIds.LibramOfHope,
    card_ids_1.CardIds.LibramOfWisdom_BT_025,
    card_ids_1.CardIds.LibramOfWisdom_Story_01_LibramofWisdom,
    card_ids_1.CardIds.LibramOfJudgment,
    card_ids_1.CardIds.LibramOfJudgment_LibramOfJudgmentToken,
    card_ids_1.CardIds.LibramOfHoliness,
    card_ids_1.CardIds.LibramOfCompassion,
    card_ids_1.CardIds.LibramOfClarity_GDB_137,
    card_ids_1.CardIds.LibramOfFaith_GDB_139,
    card_ids_1.CardIds.LibramOfDivinity_GDB_138,
];
exports.RELIC_IDS = [card_ids_1.CardIds.RelicOfDimensions, card_ids_1.CardIds.RelicOfExtinction, card_ids_1.CardIds.RelicOfPhantasms];
