// @firestone-hs/simulate-bgs-battle/dist/mechanics/rally.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.triggerRally = void 0;
const reference_data_1 = require("@firestone-hs/reference-data");
const card_interface_1 = require("../cards/card.interface");
const _card_mappings_1 = require("../cards/impl/_card-mappings");
const enchantments_1 = require("../simulation/enchantments");
const utils_1 = require("../utils");
const DOUBLE_RALLY_CARD_IDS = ["BG34_Giant_376"];
const TRIPLE_RALLY_CARD_IDS = ["BG34_Giant_376_G"];
const triggerRally = (attackingBoard, attackingBoardHero, attacker, defendingBoard, defendingBoardHero, defendingEntity, gameState) => {
    var _a, _b, _c, _d;
    let damageDoneByAttacker = 0;
    let damageDoneByDefender = 0;
    const isAttackerRallying = (0, utils_1.hasEntityMechanic)(attacker, reference_data_1.GameTag.BACON_RALLY, gameState.allCards);
    const cardIds = [
        ...attackingBoard.map((e) => e.cardId),
        ...((_b = (_a = attackingBoardHero.secrets) === null || _a === void 0 ? void 0 : _a.map((e) => e.cardId)) !== null && _b !== void 0 ? _b : []),
    ];
    const triggerMult = cardIds.some((cardId) => TRIPLE_RALLY_CARD_IDS.includes(cardId))
        ? 3
        : cardIds.some((cardId) => DOUBLE_RALLY_CARD_IDS.includes(cardId))
            ? 2
            : 1;
    const numberOfRallyingCries = (_d = (_c = attackingBoardHero.questRewardEntities) === null || _c === void 0 ? void 0 : _c.filter((r) => r.cardId === "BG33_Reward_021").length) !== null && _d !== void 0 ? _d : 0;
    const multiplier = triggerMult + numberOfRallyingCries;
    const rallyLoops = !isAttackerRallying ? 1 : multiplier;
    for (let i = 0; i < rallyLoops; i++) {
        const onAttackImpl = _card_mappings_1.cardMappings[attacker.cardId];
        if ((0, card_interface_1.hasRally)(onAttackImpl)) {
            const { dmgDoneByAttacker, dmgDoneByDefender } = onAttackImpl.rally(attacker, {
                attacker: attacker,
                attackingHero: attackingBoardHero,
                attackingBoard: attackingBoard,
                defendingEntity: defendingEntity,
                defendingHero: defendingBoardHero,
                defendingBoard: defendingBoard,
                gameState,
                playerIsFriendly: attackingBoardHero.friendly,
            });
            damageDoneByAttacker += dmgDoneByAttacker;
            damageDoneByDefender += dmgDoneByDefender;
        }
        const enchantments = attacker.enchantments;
        for (const enchantment of enchantments) {
            const onAttackImpl = _card_mappings_1.cardMappings[enchantment.cardId];
            if ((0, card_interface_1.hasRally)(onAttackImpl)) {
                let enchantmentToMinion = {
                    ...enchantment,
                    entityId: attacker.entityId,
                    attack: attacker.attack,
                    health: attacker.health,
                    maxHealth: attacker.maxHealth,
                    maxAttack: attacker.maxAttack,
                    abiityChargesLeft: attacker.abiityChargesLeft,
                };
                enchantmentToMinion = (0, enchantments_1.fixEnchantments)(enchantmentToMinion, gameState.allCards);
                const { dmgDoneByAttacker, dmgDoneByDefender } = onAttackImpl.rally(enchantmentToMinion, {
                    attacker: attacker,
                    attackingHero: attackingBoardHero,
                    attackingBoard: attackingBoard,
                    defendingEntity: defendingEntity,
                    defendingHero: defendingBoardHero,
                    defendingBoard: defendingBoard,
                    gameState,
                    playerIsFriendly: attackingBoardHero.friendly,
                });
                damageDoneByAttacker += dmgDoneByAttacker;
                damageDoneByDefender += dmgDoneByDefender;
            }
        }
    }
    return { dmgDoneByAttacker: damageDoneByAttacker, dmgDoneByDefender: damageDoneByDefender };
};
exports.triggerRally = triggerRally;
