// @firestone-hs/reference-data/dist/services/duels-utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isSignatureTreasure = exports.duelsTreasureRank = exports.isPassive = exports.allDuelsHeroesExtended = exports.allDuelsHeroes = exports.allDuelsSignatureTreasures = exports.duelsHeroConfigs = exports.normalizeDuelsHeroCardId = exports.allDuelsTreasureCardIds = exports.allDuelsPassiveTreasures = exports.duelsActivePool2UltraRare = exports.duelsActivePool2 = exports.duelsActivePool1 = exports.duelsPassivePool2UltraRare = exports.duelsPassivePool2 = exports.duelsPassivePool1 = void 0;
const card_ids_1 = require("../card-ids");
const public_api_1 = require("../public-api");
exports.duelsPassivePool1 = [
    card_ids_1.CardIds.AllShallServeTavernBrawl,
    card_ids_1.CardIds.ArcaneFluxTavernBrawl,
    card_ids_1.CardIds.ArcaniteCrystalTavernBrawl,
    card_ids_1.CardIds.ArcticArmorTavernBrawl,
    card_ids_1.CardIds.BeckoningBicornTavernBrawl,
    card_ids_1.CardIds.BitterColdTavernBrawl,
    card_ids_1.CardIds.BloodShieldsTavernBrawl,
    card_ids_1.CardIds.BookOfWonders,
    card_ids_1.CardIds.CadaverCollectorTavernBrawl,
    card_ids_1.CardIds.CannibalismTavernBrawl,
    card_ids_1.CardIds.CoilCastingTavernBrawl,
    card_ids_1.CardIds.CrystalGemTavernBrawl,
    card_ids_1.CardIds.DivineIlluminationTavernBrawl,
    card_ids_1.CardIds.DragonbloodTavernBrawl,
    card_ids_1.CardIds.DragonAffinityTavernBrawl,
    card_ids_1.CardIds.EdgeOfDredgeTavernBrawl,
    card_ids_1.CardIds.FirekeepersIdolTavernBrawl,
    card_ids_1.CardIds.ForgottenDepthsTavernBrawl,
    card_ids_1.CardIds.FreezeSolidTavernBrawl,
    card_ids_1.CardIds.FromTheSwampTavernBrawl,
    card_ids_1.CardIds.GrommashsArmguardsTavernBrawl,
    card_ids_1.CardIds.HoldTheLineTavernBrawl,
    card_ids_1.CardIds.InspiringPresenceTavernBrawl,
    card_ids_1.CardIds.InvigoratingLightTavernBrawl,
    card_ids_1.CardIds.KindlingFlameTavernBrawl,
    card_ids_1.CardIds.MendingPoolsTavernBrawl,
    card_ids_1.CardIds.MysteriousTomeTavernBrawl,
    card_ids_1.CardIds.NaturalForceTavernBrawl,
    card_ids_1.CardIds.OpenTheDoorwaysTavernBrawl,
    card_ids_1.CardIds.OptimizedPolarityTavernBrawl,
    card_ids_1.CardIds.PillageTheFallenTavernBrawl,
    card_ids_1.CardIds.PotionOfSparkingTavernBrawl,
    card_ids_1.CardIds.RecyclingTavernBrawl,
    card_ids_1.CardIds.RighteousReservesTavernBrawl,
    card_ids_1.CardIds.RingOfPhaseshiftingTavernBrawl,
    card_ids_1.CardIds.RobeOfTheApprenticeTavernBrawl,
    card_ids_1.CardIds.RobesOfShrinkingTavernBrawl,
    card_ids_1.CardIds.RocketBackpacksTavernBrawl,
    card_ids_1.CardIds.SmallBackpacksTavernBrawl,
    card_ids_1.CardIds.StaffOfPainTavernBrawl,
    card_ids_1.CardIds.StakingAClaimTavernBrawl,
    card_ids_1.CardIds.StarvingTavernBrawl,
    card_ids_1.CardIds.StickyFingersTavernBrawl,
    card_ids_1.CardIds.TheFloorIsLavaTavernBrawl,
    card_ids_1.CardIds.UnholyGiftTavernBrawl,
    card_ids_1.CardIds.WitherTheWeakTavernBrawl,
];
exports.duelsPassivePool2 = [
    card_ids_1.CardIds.AllTogetherNowTavernBrawl,
    card_ids_1.CardIds.AvengingArmamentsTavernBrawl,
    card_ids_1.CardIds.BattleStanceTavernBrawl,
    card_ids_1.CardIds.BrittleBonesTavernBrawl,
    card_ids_1.CardIds.BronzeSignetTavernBrawl,
    card_ids_1.CardIds.CloakOfEmeraldDreamsTavernBrawl,
    card_ids_1.CardIds.ColdFeetPactTavernBrawl,
    card_ids_1.CardIds.CookiesLadleTavernBrawl,
    card_ids_1.CardIds.CorruptedFelstoneTavernBrawl,
    card_ids_1.CardIds.DeathlyDeathTavernBrawl,
    card_ids_1.CardIds.DoubleTime,
    card_ids_1.CardIds.DraconicDreamTavernBrawl,
    card_ids_1.CardIds.DragonboneRitualTavernBrawl,
    card_ids_1.CardIds.EerieStoneTavernBrawl,
    card_ids_1.CardIds.ElixirOfVigorTavernBrawl,
    card_ids_1.CardIds.EmeraldGogglesTavernBrawl,
    card_ids_1.CardIds.EnduranceTrainingTavernBrawl,
    card_ids_1.CardIds.FlamesOfTheKirinTorTavernBrawl,
    card_ids_1.CardIds.FlameWavesTavernBrawl,
    card_ids_1.CardIds.GhoulsRushInTavernBrawl,
    card_ids_1.CardIds.GlacialDownpourTavernBrawl,
    card_ids_1.CardIds.GreedyGainsTavernBrawl,
    card_ids_1.CardIds.GuardianLightTavernBrawl,
    card_ids_1.CardIds.HagathasEmbraceTavernBrawl,
    card_ids_1.CardIds.IdolsOfEluneTavernBrawl,
    card_ids_1.CardIds.ImpCredibleTrousersTavernBrawl,
    card_ids_1.CardIds.IronRootsTavernBrawl,
    card_ids_1.CardIds.KhadgarsScryingOrb,
    card_ids_1.CardIds.LocationLocationLocationTavernBrawl,
    card_ids_1.CardIds.MeekMasteryTavernBrawl,
    card_ids_1.CardIds.MulchMadnessTavernBrawl,
    card_ids_1.CardIds.MummyMagic,
    card_ids_1.CardIds.OopsAllSpellsTavernBrawl,
    card_ids_1.CardIds.OrbOfRevelationTavernBrawl,
    card_ids_1.CardIds.PartyReplacementTavernBrawl,
    card_ids_1.CardIds.RallyTheTroopsTavernBrawl,
    card_ids_1.CardIds.RingOfBlackIceTavernBrawl,
    card_ids_1.CardIds.RingOfRefreshmentTavernBrawl,
    card_ids_1.CardIds.RunicHelmTavernBrawl,
    card_ids_1.CardIds.SpreadingSaplingsTavernBrawl,
    card_ids_1.CardIds.SunstridersCrownTavernBrawl,
];
exports.duelsPassivePool2UltraRare = [
    card_ids_1.CardIds.BandOfBeesTavernBrawl,
    card_ids_1.CardIds.BattleTotem_LOOTA_846,
    card_ids_1.CardIds.CapturedFlag,
    card_ids_1.CardIds.DisksOfLegendTavernBrawl,
    card_ids_1.CardIds.ExpeditedBurialTavernBrawl,
    card_ids_1.CardIds.LegendaryLootTavernBrawl,
    card_ids_1.CardIds.ManastormTavernBrawl,
    card_ids_1.CardIds.MantleOfIgnitionTavernBrawl,
    card_ids_1.CardIds.RingOfHasteTavernBrawl,
    card_ids_1.CardIds.RobeOfTheMagi,
    card_ids_1.CardIds.ScepterOfSummoning,
    card_ids_1.CardIds.Shadowcasting101TavernBrawl,
    card_ids_1.CardIds.SpecialDeliveryTavernBrawl,
    card_ids_1.CardIds.Stargazing,
    card_ids_1.CardIds.TotemOfTheDead_LOOTA_845,
    card_ids_1.CardIds.UnlockedPotential,
    card_ids_1.CardIds.UnstableMagicTavernBrawl,
];
exports.duelsActivePool1 = [
    card_ids_1.CardIds.AmalgamateTavernBrawl,
    card_ids_1.CardIds.AstralPortalTavernBrawl,
    card_ids_1.CardIds.BeastlyBeautyTavernBrawl,
    card_ids_1.CardIds.BubbaTavernBrawl,
    card_ids_1.CardIds.ButchTavernBrawl,
    card_ids_1.CardIds.BladeOfQueldelarTavernBrawl,
    card_ids_1.CardIds.BloodMoonTavernBrawl,
    card_ids_1.CardIds.ClockworkAssistantTavernBrawl_PVPDR_SCH_Active48,
    card_ids_1.CardIds.CoinPouchTavernBrawl,
    card_ids_1.CardIds.CreepyCurioTavernBrawl,
    card_ids_1.CardIds.CrustyTheCrustaceanTavernBrawl,
    card_ids_1.CardIds.Queldelar_HiltOfQueldelar,
    card_ids_1.CardIds.Deathcharger,
    card_ids_1.CardIds.Enrage,
    card_ids_1.CardIds.GentlemansTopHat,
    card_ids_1.CardIds.GoldenKobold,
    card_ids_1.CardIds.GreedyPickaxeTavernBrawl,
    card_ids_1.CardIds.HolyBookTavernBrawl,
    card_ids_1.CardIds.HuntersInsightTavernBrawl,
    card_ids_1.CardIds.JawsTavernBrawl,
    card_ids_1.CardIds.LoyalHenchmanTavernBrawl,
    card_ids_1.CardIds.LoyalSidekickTavernBrawl,
    card_ids_1.CardIds.LoomingPresence_LOEA_01,
    card_ids_1.CardIds.MageArmorTavernBrawl,
    card_ids_1.CardIds.MindpocalypseTavernBrawl,
    card_ids_1.CardIds.DetectiveMurlocHolmesTavernBrawl,
    card_ids_1.CardIds.MutatingInjection_NAX11_04,
    card_ids_1.CardIds.NecroticPoisonTavernBrawl,
    card_ids_1.CardIds.OldMilitiaHornTavernBrawl,
    card_ids_1.CardIds.PartyPortalTavernBrawl_PVPDR_SCH_Active08,
    card_ids_1.CardIds.PrincessTavernBrawl,
    card_ids_1.CardIds.PureColdTavernBrawl,
    card_ids_1.CardIds.PuzzleBoxTavernBrawl_PVPDR_SCH_Active02,
    card_ids_1.CardIds.SowTheSeedsTavernBrawl,
    card_ids_1.CardIds.SpyglassTavernBrawl,
    card_ids_1.CardIds.SuperchargeTavernBrawl,
    card_ids_1.CardIds.SurlyMobTavernBrawl,
    card_ids_1.CardIds.TheExorcisorTavernBrawl,
    card_ids_1.CardIds.TheStoneOfJordanTavernBrawlToken,
    card_ids_1.CardIds.VampiricFangsToken,
    card_ids_1.CardIds.WorshipperHeroic,
];
exports.duelsActivePool2 = [
    card_ids_1.CardIds.ArchmageStaffTavernBrawl,
    card_ids_1.CardIds.BagOfCoinsTavernBrawl,
    card_ids_1.CardIds.BookOfTheDeadTavernBrawl,
    card_ids_1.CardIds.GrimmerPatron_DALA_705,
    card_ids_1.CardIds.HyperblasterTavernBrawl,
    card_ids_1.CardIds.BagOfStuffingTavernBrawl,
    card_ids_1.CardIds.BlackSoulstoneTavernBrawl,
    card_ids_1.CardIds.BladeOfTheBurningSun,
    card_ids_1.CardIds.GnomishArmyKnife_DALA_702,
    card_ids_1.CardIds.BananaSplitTavernBrawl,
    card_ids_1.CardIds.CanopicJarsTavernBrawl,
    card_ids_1.CardIds.PhaorisBlade_ULDA_043,
    card_ids_1.CardIds.RoyalGift,
    card_ids_1.CardIds.StaffOfScales_ULDA_008,
    card_ids_1.CardIds.TinyThimbleTavernBrawl,
    card_ids_1.CardIds.WandOfDisintegrationTavernBrawl,
];
exports.duelsActivePool2UltraRare = [
    card_ids_1.CardIds.ChaosTheoryTavernBrawl_PVPDR_SCH_Active49,
    card_ids_1.CardIds.OverpoweredTavernBrawl,
    card_ids_1.CardIds.DreamgroveRing,
    card_ids_1.CardIds.EmbersOfRagnarosTavernBrawl,
    card_ids_1.CardIds.WaxRager_LOOTA_840,
    card_ids_1.CardIds.WishTavernBrawl,
];
exports.allDuelsPassiveTreasures = [...exports.duelsPassivePool1, ...exports.duelsPassivePool2, ...exports.duelsPassivePool2UltraRare];
exports.allDuelsTreasureCardIds = [
    ...exports.allDuelsPassiveTreasures,
    ...exports.duelsActivePool1,
    ...exports.duelsActivePool2,
    ...exports.duelsActivePool2UltraRare,
];
const normalizeDuelsHeroCardId = (heroCardId) => {
    switch (heroCardId) {
        case card_ids_1.CardIds.VanndarStormpikeTavernBrawl:
        case card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv1:
        case card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv2:
        case card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv3:
        case card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv4:
        case card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv5:
            return card_ids_1.CardIds.VanndarStormpikeTavernBrawl;
        case card_ids_1.CardIds.DrektharTavernBrawl:
        case card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv1:
        case card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv2:
        case card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv3:
        case card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv4:
        case card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv5:
            return card_ids_1.CardIds.DrektharTavernBrawl;
        default:
            return heroCardId;
    }
};
exports.normalizeDuelsHeroCardId = normalizeDuelsHeroCardId;
exports.duelsHeroConfigs = [
    {
        hero: card_ids_1.CardIds.StarStudentStelinaTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.InfernalStrikeTavernBrawl,
            card_ids_1.CardIds.OutlanderTavernBrawl,
            card_ids_1.CardIds.IllidariStrikeTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.SummoningRitualTavernBrawlToken,
            card_ids_1.CardIds.MoargOutcastTavernBrawlToken,
            card_ids_1.CardIds.GiftOfTheLegionTavernBrawl,
            card_ids_1.CardIds.BindingChainsTavernBrawlToken,
            card_ids_1.CardIds.BladeOfQuickeningTavernBrawlToken,
            card_ids_1.CardIds.TraktamerAelessaTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.ForestWardenOmuTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.NaturesGiftsTavernBrawl,
            card_ids_1.CardIds.InvigoratingBloomTavernBrawl,
            card_ids_1.CardIds.HarvestTimeTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.WardensInsightTavernBrawlToken,
            card_ids_1.CardIds.MoonbeastTavernBrawlToken,
            card_ids_1.CardIds.HerdingHornTavernBrawlToken,
            card_ids_1.CardIds.MarvelousMyceliumTavernBrawlToken,
            card_ids_1.CardIds.AwakenedAncientTavernBrawl,
            card_ids_1.CardIds.ZukaraTheWildTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.ProfessorSlateTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.SurvivalTrainingTavernBrawl,
            card_ids_1.CardIds.DeathGamesTavernBrawl,
            card_ids_1.CardIds.SavageSecretsTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.SlatesSyringeTavernBrawl,
            card_ids_1.CardIds.BonecrusherTavernBrawlToken,
            card_ids_1.CardIds.StalkingPrideTavernBrawlToken,
            card_ids_1.CardIds.DeathstriderTavernBrawl,
            card_ids_1.CardIds.CarrotOnAStickTavernBrawl,
            card_ids_1.CardIds.TraktamerAelessaTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.MozakiMasterDuelistTavernBrawl,
        heroPowers: [card_ids_1.CardIds.WyrmBoltTavernBrawl, card_ids_1.CardIds.SecretStudiesTavernBrawl, card_ids_1.CardIds.FrostShardsTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.WandOfDuelingTavernBrawlToken,
            card_ids_1.CardIds.ElementalLearningTavernBrawl,
            card_ids_1.CardIds.InfiniteArcaneTavernBrawlToken,
            card_ids_1.CardIds.EmbercasterTavernBrawl,
            card_ids_1.CardIds.YoggTasticTastiesTavernBrawl,
            card_ids_1.CardIds.BritzBlazebucketTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.TuralyonTheTenuredTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.FromGoldenLightTavernBrawl,
            card_ids_1.CardIds.BringOnRecruitsTavernBrawl,
            card_ids_1.CardIds.ModestAspirationsTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.RoyalGreatswordTavernBrawlToken,
            card_ids_1.CardIds.MenAtArmsTavernBrawlToken,
            card_ids_1.CardIds.DrocomurchanicasTavernBrawlToken,
            card_ids_1.CardIds.FavoredRacerTavernBrawl,
            card_ids_1.CardIds.HumbleBlessingsTavernBrawl,
            card_ids_1.CardIds.SuLeadfootTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.MindrenderIlluciaTavernBrawl,
        heroPowers: [card_ids_1.CardIds.ShadowMendTavernBrawl, card_ids_1.CardIds.MindTetherTavernBrawl, card_ids_1.CardIds.CallOfMadnessTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.FracturedSpiritsTavernBrawlToken,
            card_ids_1.CardIds.DropletOfInsanityTavernBrawlToken,
            card_ids_1.CardIds.ScionOfTheDeepTavernBrawlToken,
            card_ids_1.CardIds.ShadowWordVoidTavernBrawl,
            card_ids_1.CardIds.HeraldOfTheScaledOnesTavernBrawl,
            card_ids_1.CardIds.JorasThuldoomTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.InfiltratorLilianTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.RoguishManeuversTavernBrawl,
            card_ids_1.CardIds.ConnectionsTavernBrawl,
            card_ids_1.CardIds.VileConcoctionTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.DeadlyWeapons101TavernBrawlToken,
            card_ids_1.CardIds.NerubianPeddlerTavernBrawlToken,
            card_ids_1.CardIds.AcquiredAlliesTavernBrawlToken,
            card_ids_1.CardIds.AceInTheHoleTavernBrawlToken,
            card_ids_1.CardIds.GreenTortollanShellTavernBrawl,
            card_ids_1.CardIds.BritzBlazebucketTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.InstructorFireheartTavernBrawlToken,
        heroPowers: [
            card_ids_1.CardIds.StormcatcherTavernBrawl,
            card_ids_1.CardIds.TotemicPowerTavernBrawl,
            card_ids_1.CardIds.FerociousFlurryTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.ChaosStormTavernBrawl,
            card_ids_1.CardIds.FluctuatingTotemTavernBrawlToken,
            card_ids_1.CardIds.InvokeTheVoidTavernBrawlToken,
            card_ids_1.CardIds.PayloadTotemSpecialistTavernBrawl,
            card_ids_1.CardIds.TempestsFuryTavernBrawlToken,
            card_ids_1.CardIds.ZukaraTheWildTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.ArchwitchWillowTavernBrawl,
        heroPowers: [card_ids_1.CardIds.DarkArtsTavernBrawl, card_ids_1.CardIds.DemonBloodTavernBrawl, card_ids_1.CardIds.SoulcialStudiesTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.ImpishAidTavernBrawl,
            card_ids_1.CardIds.GiftOfTheOldGodsTavernBrawlToken,
            card_ids_1.CardIds.KillmoxTheBanishedOneTavernBrawlToken,
            card_ids_1.CardIds.Demonology101TavernBrawlToken,
            card_ids_1.CardIds.DemonizerTavernBrawlToken,
            card_ids_1.CardIds.JorasThuldoomTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.RattlegoreTavernBrawl,
        heroPowers: [card_ids_1.CardIds.NoGutsNoGloryTavernBrawl, card_ids_1.CardIds.BruisingTavernBrawl, card_ids_1.CardIds.MagneticMinesTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.AutoArmamentsTavernBrawlToken,
            card_ids_1.CardIds.CollectorsIreTavernBrawlToken,
            card_ids_1.CardIds.BrewsterTheBrutalTavernBrawlToken,
            card_ids_1.CardIds.ScrapmetalDemolitionistTavernBrawl,
            card_ids_1.CardIds.SpikedArmsTavernBrawlToken,
            card_ids_1.CardIds.SuLeadfootTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.DrektharTavernBrawl,
        alternateHeroes: [
            card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv1,
            card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv2,
            card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv3,
            card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv4,
            card_ids_1.CardIds.Drekthar_DrektharTavernBrawl_PVPDR_Hero_DrekTharv5,
        ],
        heroPowers: [
            card_ids_1.CardIds.WarmastersFrenzyTavernBrawl,
            card_ids_1.CardIds.HarnessTheElementsTavernBrawl,
            card_ids_1.CardIds.WarCommandsTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.ValorousDisplayTavernBrawlToken,
            card_ids_1.CardIds.ElementalChaosTavernBrawlToken,
            card_ids_1.CardIds.PactOfTheLichTavernBrawlToken,
            card_ids_1.CardIds.MagisterUnchainedTavernBrawlToken,
            card_ids_1.CardIds.GiftOfTheHeartTavernBrawlToken,
            card_ids_1.CardIds.ForTheHordeTavernBrawlToken,
        ],
    },
    {
        hero: card_ids_1.CardIds.VanndarStormpikeTavernBrawl,
        alternateHeroes: [
            card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv1,
            card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv2,
            card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv3,
            card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv4,
            card_ids_1.CardIds.VanndarStormpike_VanndarStormpikeTavernBrawl_PVPDR_Hero_Vanndarv5,
        ],
        heroPowers: [card_ids_1.CardIds.PromoteTavernBrawl, card_ids_1.CardIds.SendInTheScoutTavernBrawl, card_ids_1.CardIds.BattleTacticsTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.ForgeInLightTavernBrawlToken,
            card_ids_1.CardIds.SpymastersGambitTavernBrawlToken,
            card_ids_1.CardIds.StalkersSuppliesTavernBrawlToken,
            card_ids_1.CardIds.RendingAmbushTavernBrawlToken,
            card_ids_1.CardIds.DevoutBlessingsTavernBrawlToken,
            card_ids_1.CardIds.ForTheAllianceTavernBrawlToken,
        ],
    },
    {
        hero: card_ids_1.CardIds.RenoJacksonTavernBrawl_PVPDR_Hero_Reno,
        heroPowers: [
            card_ids_1.CardIds.AmateurMageTavernBrawl,
            card_ids_1.CardIds.RelicologistTavernBrawl,
            card_ids_1.CardIds.ArcaneCraftinessTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.RenosLuckyHatTavernBrawl,
            card_ids_1.CardIds.RenosCraftyLassoTavernBrawl,
            card_ids_1.CardIds.RenosMagicalTorchTavernBrawl,
            card_ids_1.CardIds.SrTombDiverTavernBrawl,
            card_ids_1.CardIds.LeiFlamepawTavernBrawl,
            card_ids_1.CardIds.TheGatlingWandTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.SirFinleyTavernBrawl_PVPDR_Hero_Finley,
        heroPowers: [
            card_ids_1.CardIds.NewRecruitsTavernBrawl,
            card_ids_1.CardIds.BubbleBlowerTavernBrawl,
            card_ids_1.CardIds.PowerUpTavernBrawl_PVPDR_Finley_HP3,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.FinleysPithHelmetTavernBrawl,
            card_ids_1.CardIds.MaxwellMightySteedTavernBrawl,
            card_ids_1.CardIds.TruesilverLanceTavernBrawl,
            card_ids_1.CardIds.SrScoutTavernBrawl,
            card_ids_1.CardIds.KarlTheLostTavernBrawl,
            card_ids_1.CardIds.ScalesOfJusticeTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.EliseStarseekerTavernBrawl_PVPDR_Hero_Elise,
        heroPowers: [card_ids_1.CardIds.ElisesMightTavernBrawl, card_ids_1.CardIds.DruidicTeachingTavernBrawl, card_ids_1.CardIds.StarseekerTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.StarseekersToolsTavernBrawl,
            card_ids_1.CardIds.ElisesMacheteTavernBrawl,
            card_ids_1.CardIds.AddarahTavernBrawl,
            card_ids_1.CardIds.SrNavigatorTavernBrawl,
            card_ids_1.CardIds.AcademicResearchTavernBrawl,
            card_ids_1.CardIds.StaffOfRenewalTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.BrannBronzebeardTavernBrawl_PVPDR_Hero_Brann,
        heroPowers: [
            card_ids_1.CardIds.SpreadShotTavernBrawl,
            card_ids_1.CardIds.WellEquippedTavernBrawl_PVPDR_Brann_HP2,
            card_ids_1.CardIds.DinoTrackingTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.OlFaithfulTavernBrawl,
            card_ids_1.CardIds.BrannsSaddleTavernBrawl,
            card_ids_1.CardIds.TitanForgedGrapnelTavernBrawl,
            card_ids_1.CardIds.SrExcavatorTavernBrawl,
            card_ids_1.CardIds.FloSlatebrandTavernBrawl,
            card_ids_1.CardIds.BrannsEpicEggTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.DariusCrowleyTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.FireTavernBrawl_PVPDR_Darius_P1,
            card_ids_1.CardIds.FireAwayTavernBrawl,
            card_ids_1.CardIds.FireAtTheeTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.SafeHarborTavernBrawl,
            card_ids_1.CardIds.GrizzledReinforcementTavernBrawl,
            card_ids_1.CardIds.TuskarrRaiderTavernBrawl,
            card_ids_1.CardIds.SeabreakerGoliathTavernBrawl,
            card_ids_1.CardIds.DeckSwabbieTavernBrawl,
            card_ids_1.CardIds.DraconicMunitionTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.SaiShadestormTavernBrawl,
        heroPowers: [
            card_ids_1.CardIds.GhoulBlitzTavernBrawl,
            card_ids_1.CardIds.GatheringStormTavernBrawl,
            card_ids_1.CardIds.LichborneMightTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.GluthTavernBrawl_PVPDR_Sai_T1,
            card_ids_1.CardIds.GlaciaxeTavernBrawl,
            card_ids_1.CardIds.RemembranceOfIceTavernBrawl,
            card_ids_1.CardIds.UnholyEmbraceTavernBrawl,
            card_ids_1.CardIds.GluthSicleTavernBrawl,
            card_ids_1.CardIds.InexorableGhoulTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.ScarletLeafdancerTavernBrawlToken,
        heroPowers: [card_ids_1.CardIds.BloodParasiteTavernBrawl, card_ids_1.CardIds.HematologyTavernBrawl, card_ids_1.CardIds.ScourgingTavernBrawl],
        signatureTreasures: [
            card_ids_1.CardIds.RunedSoulbladeTavernBrawl,
            card_ids_1.CardIds.ScourgeStrikeTavernBrawl,
            card_ids_1.CardIds.CrimsonTavernBrawl,
            card_ids_1.CardIds.IronweaveBloodletterTavernBrawl,
            card_ids_1.CardIds.GreatswordOfTheEbonBladeTavernBrawl,
            card_ids_1.CardIds.ContagionConcoctionTavernBrawl,
        ],
    },
    {
        hero: card_ids_1.CardIds.DiabloTavernBrawl_PVPDR_Hero_Diablo,
        heroPowers: [
            card_ids_1.CardIds.DoomChargeTavernBrawl,
            card_ids_1.CardIds.PrimalPowerTavernBrawl,
            card_ids_1.CardIds.DemonicTransformationTavernBrawl,
        ],
        signatureTreasures: [
            card_ids_1.CardIds.ClawsOfTerrorTavernBrawlToken,
            card_ids_1.CardIds.FireStompTavernBrawlToken,
            card_ids_1.CardIds.ApocalypseTavernBrawlToken,
            card_ids_1.CardIds.SoulstoneTrapTavernBrawlToken,
            card_ids_1.CardIds.HornsOfFlameTavernBrawlToken,
            card_ids_1.CardIds.JourneyToTheEastTavernBrawlToken,
        ],
    },
];
exports.allDuelsSignatureTreasures = exports.duelsHeroConfigs.flatMap((info) => info.signatureTreasures);
exports.allDuelsHeroes = exports.duelsHeroConfigs.map((conf) => conf.hero);
exports.allDuelsHeroesExtended = exports.duelsHeroConfigs.flatMap((conf) => { var _a; return [conf.hero, ...((_a = conf.alternateHeroes) !== null && _a !== void 0 ? _a : [])]; });
const isPassive = (cardId, allCards) => {
    var _a, _b;
    return (exports.allDuelsPassiveTreasures.includes(cardId) ||
        ((_b = (_a = allCards.getCard(cardId)) === null || _a === void 0 ? void 0 : _a.mechanics) === null || _b === void 0 ? void 0 : _b.includes(public_api_1.GameTag[public_api_1.GameTag.DUNGEON_PASSIVE_BUFF])));
};
exports.isPassive = isPassive;
const duelsTreasureRank = (cardId) => {
    if (!cardId) {
        return 0;
    }
    if (!exports.allDuelsTreasureCardIds.includes(cardId)) {
        console.error('Incorrect config for duels card IDs?', cardId);
        return 0;
    }
    if (exports.duelsPassivePool2.includes(cardId) || exports.duelsActivePool2.includes(cardId)) {
        return 2;
    }
    else if (exports.duelsPassivePool2UltraRare.includes(cardId) ||
        exports.duelsActivePool2UltraRare.includes(cardId)) {
        return 3;
    }
    return 1;
};
exports.duelsTreasureRank = duelsTreasureRank;
const isSignatureTreasure = (cardId) => {
    return exports.allDuelsSignatureTreasures.includes(cardId);
};
exports.isSignatureTreasure = isSignatureTreasure;
